---
title: "Evaluation of external PRS or external GWAS"
author: "Panagiotis Moulos"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{Evaluation of external PRS or external GWAS}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

# Introduction

Although the main PRISMA functionality has to do with *de novo* PRS derivation
and evaluation, one can easily evaluate an externally derived PRS, for example
a PRS retrieved from the [EBI PGS Catalog](https://www.pgscatalog.org/). This
can be done in two ways:

1. Subsetting the `GWASExperiment` according to the SNPs in the retrieved 
PRS/PGS and running a PRISMA pipeline (`prisma()` or `prismaPipeline()`). This
is the better way as new effects should be calculated suitable for our 
population. This pipeline should run quite fast, if the number of SNPs included
in the external PRS are few.
2. Provided the retireved PRS directly in the `prsCrossValidate()` function.
This will take the object as is (with own effects) and return some evaluation
scores when the PRS is tested with our population and genotypes. The PRS
retrieval functions included in the package (`getPGSScores()`) will fetch data 
frames ready for usage with `prsCrossValidate()`. Quality control of the initial
dataset may or may not be performed.

```{r ext-0, eval=TRUE, echo=FALSE, tidy=FALSE, message=FALSE, warning=FALSE}
library(prisma)
```

# Testing an externally retrieved PRS

For the purposes of this vignette we are going to use the score file reported
for [PGS000034](https://www.pgscatalog.org/score/PGS000034/) with *Adult Body
Mass Index* reported as trait. This score included with the package. We have
also created a smaller toy dataset called `pgd` from the same source as the
toy dataset but containing the common SNPs between PGS000034 and the source
data that was used to create the toy dataset.

```{r ext-1, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
base <- file.path(system.file(package="prisma"),"extdata","scores")
pgsId <- "PGS000034"
exPgs <- getPGSScores(pgsId=pgsId,base=base)
rownames(exPgs) <- exPgs$variant_id

head(exPgs)
```

Load the **pgd** dataset and use the common SNPs with `exPgs`. They are already
the same but we state the commands for demo purposes.

```{r ext-2, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
data(pgd,package="prisma")
gdsfile(pgd) <- system.file(package="prisma","extdata","pgd.gds")

pgd

exPgs <- exPgs[rownames(pgd),]
```

Run a PRISMA full pipeline, only with GLM:

```{r ext-3, eval=FALSE, echo=TRUE, tidy=FALSE, message=FALSE, warning=FALSE, collapse=TRUE}
outPath <- tempfile()
d1 <- tempfile()
wspace1 <- file.path(outPath,d1)

# Traits
response <- "BMI"
covariates <- c("Case_Control","Gender","Age")

# Lose filters for the toy dataset
filts <- getDefaults("filters")
filts$IBD <- NA
filts$hwe <- 1e-3
filts$pcaOut <- FALSE
filts$inbreed <- NA

prismaPipeline(
    gwe=pgd,
    phenotype=response,
    covariates=covariates,
    pcs=FALSE,
    trainSize=0.8,
    niter=4,
    resolution="frequency",
    minSnps=2,
    filters=filts,
    pcaMethod="snprel",
    imputeMissing=FALSE,
    gwaMethods="glm",
    prsiceOpts=list(clump_p=0.75,perm=100),
    prsWorkspace=wspace1,
    cleanup="intermediate",
    logging="file",
    output="normal",
    rc=0.25,
    outPath=outPath
)
```

A raw matrix of evaluation metrics:

```{r ext-4, eval=FALSE, echo=TRUE, tidy=FALSE, message=FALSE, warning=FALSE, collapse=TRUE}
m <- prsCrossValidate(exPgs,pgd,response=response,
    covariates=covariates,leaveOut=0.2,times=5)
```

# Testing a derived PRS with independent genotypic data

Often (actually always) we would like to test our derived PRS candidates to an
independent population which could share similar characteristics, for example
living in the same country and having similar origins. Sometimes it is also
beneficial to test random PRSs with this population to see whether the original
PRS does indeed have added value and it is not a random effect. The `applyPRS()`
function can be used to accomplish the aforementioned processes.

In the example below, we do not run the PRS randomization process

```{r ext-5, eval=FALSE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
data(toy,package="prisma")
gdsfile(toy) <- system.file(package="prisma","extdata","toy.gds")

# Traits
response <- "BMI"
covariates <- c("Case_Control","Gender","Age")

# Lose filters for the toy dataset
filts <- getDefaults("filters")
filts$IBD <- NA
filts$hwe <- 1e-3
filts$pcaOut <- FALSE
filts$inbreed <- NA

wspace2 <- tempfile()
prismaOut <- prisma(
    gwe=toy,
    phenotype=response,
    covariates=covariates,
    pcs=FALSE,
    trainSize=0.8,
    niter=4,
    resolution="frequency",
    minSnps=2,
    filters=filts,
    pcaMethod="snprel",
    imputeMissing=FALSE,
    gwaMethods="statgen",
    prsiceOpts=list(clump_p=0.75,perm=100),
    prsWorkspace=wspace2,
    cleanup="intermediate",
    logging="file",
    output="normal",
    rc=0.5
)

snpSelection <- getPrsCandidates(prismaOut,"statgen",1)

metrics <- applyPRS(snpSelection,toy,response=response,covariates=covariates)

metrics$metrics

head(metrics$values)
```

# R session info

```{r session-0, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
sessionInfo()
```
