---
title: "Reading and writing files and objects"
author: "Panagiotis Moulos"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{Reading and writing files and objects}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

# Reading PLINK files to GWASExperiment objects

PRISMA provides facilities to create `GWASExperiment` objects directly from
PLINK `bed/bim/fam` triplets (not from PED files for the time being) through
the usage of the `read.plink` function from the package snpStats along with
other functionalities to attach phenotypes, metadata etc. For example:

```{r io-0, eval=TRUE, echo=FALSE, tidy=FALSE, message=FALSE, warning=FALSE}
library(prisma)
```

```{r io-1, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
library(prisma)

fam <- system.file("extdata/sample.fam",package="snpStats")
bim <- system.file("extdata/sample.bim",package="snpStats")
bed <- system.file("extdata/sample.bed",package="snpStats")
input <- list(fam=fam,bim=bim,bed=bed)

gwe <- importGWAS(input,backend="snpStats")
gwe
```

We can also attach a data frame or a file (text-tab delimited) of phenotypes:

```{r io-2, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
set.seed(42)

fam <- system.file("extdata/sample.fam",package="snpStats")
bim <- system.file("extdata/sample.bim",package="snpStats")
bed <- system.file("extdata/sample.bed",package="snpStats")
input <- list(fam=fam,bim=bim,bed=bed)
sample <- snpStats::read.plink(input$bed,input$bim,input$fam)

pseudopheno <- data.frame(
    case_control=sample(c(0,1),nrow(sample$fam),replace=TRUE),
    other_pheno=sample(c("drug","nodrug"),nrow(sample$fam),replace=TRUE),
    cont=round(runif(nrow(sample$fam)),3),
    row.names=rownames(sample$fam)
)

gwe <- importGWAS(input,phenos=pseudopheno,backend="snpStats")

gwe
```

Other functions of `importGWAS` include the selection of a subset of SNPs to
import (`selection` argument), whether or not to reverse alleles (`alleleOrder`
argument, useful some times although it generally should be left to the default)
and whether a GDS file should be written or not (generally yes, especially if
you intend to perform further filtering or PCA). For example, if you do not
intend to do any of the previous and you know that the SNPs locations are mapped
to the hg19 human genome version:

```{r io-3, eval=FALSE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
library(prisma)

fam <- system.file("extdata/sample.fam",package="snpStats")
bim <- system.file("extdata/sample.bim",package="snpStats")
bed <- system.file("extdata/sample.bed",package="snpStats")
input <- list(fam=fam,bim=bim,bed=bed)

gwe <- importGWAS(input,backend="snpStats",writeGds=FALSE,genome="hg19")
gwe
```

