---
title: "Introduction to the prisma package"
author: "Panagiotis Moulos"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{Introduction to the prisma package}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

# Introduction

The rise of modern genotyping technologies such as Next Generation Sequencing
(Whole Exome and Whole Genome Sequencing) as well as the continuous thriving of
older genotyping technologies such as genotyping DNA microarrays have allowed an
unprecedented collection of massive, large-cohort and population-level genetic
data spanning a variety of application, from medical research to consumer
genomics. At the same time, several perspective and sometimes country-level
studies are reaching maturity levels regarding the amount and quality of 
data collected at various levels such as anthropometric, phenotypic, biomarker
and genetic levels, and some times other molecular -omics layers such as gene 
expression. The aforementioned facts have allowed not only for more accurate
and versatile Genome-Wide Association Studies to identify the potential genetic
cause for many simple or more complex genetic traits but also combine phenotypic
and genotypic data in unique ways to derive multi-way genetic signatures that
characterize individuals with respect to several anthropometric features as well
as these traits and biomarker levels. These genetic signatures have a seamingly
endless portfolio of applications, spanning the characterization of the 
predisposition of an individual to general healthstyle issues such as obesity
and up to estimating the risk to several diseases like cancer and other 
potential life-threatening conditions including cardiovascular disease and
kidney malfunction. They are also used to characterize less life-threatening
conditions but which deteriorate the general well-being such as psychological
disease and anomalies of the central nervous system.

The aforementioned biotechnological thriving as well as the increased complete
data sources availability (coupled genotypic and phenotypic records for large
cohorts) are gradually allowing the creation and exploration of Polygenic Risk 
Scores as risk predictors and evaluators for a variety of disease and 
healthstyle conditions. Polygenic Risk Scores (PRS) are (usually) simple 
genotype scoring schemes which take into account many variants of an individual
and based on the calculated score, attempt to assess a heritable risk of 
developing/expressing a particular phenotype or combination of phenotypes. These
phenotypes can be disease-related phenotypes such as cancer, cardiovascular
disease and obesity, or plain lifestyle traits such as endurance in sports or
effects of caffeine. They are different from the Genetic Risk Scores (GRS) in 
the sense that while GRSs are compiled from a (usually) limited number of well
characterized variants related to a particular phenotype across the literature
with strong statistical evidence, PRSs are more relaxed regarding statistical
power and comprise a usually much number of variants accounting for the additive
genetic risk of an individual given his/hers total genetic profile.

PRS and GRS are essentially variables constructed based on the (weighted) sum of
genotypes (usually coded as 0, 1, 2 according to the alleles of a genotype) and
comprise covariates in regression models that attempt to explain observed 
genotypic/phenotypic relationships. While this approach seems over-simplified,
nevertheless it has been shown in numerous studies that genetic scores are able
to explain the aforementioned variabilities and most times even outperform 
complex Machine Learning approaches. Then, after a succesful model construction,
a distribution of PRS is constructed based on the ppopulation used to extract
and calculate the PRS, and based on this distribution, the risk of new samples
can be assessed according to the position of the new PRS in the reference
distribution. The process of extracting a PRS is not very complex itself and
includes the main following steps:

* Derivation of summary statistics (regression coefficients/Odds Ratios, 
p-values etc.) for each variant from a major and representative population 
study (often a GWAS) recording also the phenotypes of interest.
* Filtering of these variants according to their potential informative 
redundancy (Linkage Disequilbrium), content with respect to the population
(Hardy-Weinberg Equilibrium) and statistical significance/power to contribute
to the phenotype under investigation (effects as assessed from regression 
coefficients, p-values). This process is often refered as *Clumping + 
Thresholding* and although popular is not the sole method to reduce noise while
extracting a PRS. Another often complementary process is *penalized regression*
or *lasso*.
* Construction of the PRS: in this step, the variants (usually SNPs) having 
passed the filtering procedure are used to calculate the PRS. There are various
ways to calculate a PRS, the simplest one being just the sum of the genotypes
(0, 1, 2) but the most popular seem to be the *weighted* PRS where the genotype
of each marker (SNP, variant) participating in the PRS is mutliplied by its
inferred effect given summary statistics e.g. from a GWAS. Multiple PRSs can be
constructed and tested given different thresholds of p-values and/or other
scores. The PRSs are constructed based on the genotypes of samples other than
the ones used to derive the summary statistics, otherwise, strong overfitting
effects are observed.
* Construction and evaluation of multiple regression models: in this step many
regression models are fitted and evaluated with the PRS candidates as covariates
with the aim to select the PRS providing the largest explanatory power in terms
of some metric such as the R<sup>2</sup> of the model and the significance of
the PRS as a covariate (p-value). This is the strategy followed e.g. by 
[PRSice-2](https://www.prsice.info/). Other strategies exist such as selecting
one PRS without repetitions or using more SNPs in the PRS by effect-shrinking
methods (this is what [lassosum](https://github.com/tshmak/lassosum) does) using
penalized regression and LASSO techniques.
* Selection and evaluation of the final PRS candidates with the goal of
selecting a final PRS for the phenotype under investigation. This evaluation can
be done based on split-and-test cross-validations on the initital dataset. In
any case, the final PRS is subjected to slight changes if the initial dataset
is changed (e.g. new samples added).

# The PRISMA package

## General introduction

The PRISMA package is designed to provide a one-stop shop for all the PRS 
extraction and evaluation steps described above. On top of that, it implements
a linear optimization methodology which combines the regression effects from
multiple statistical tests for GWAS-like association tests and offers a variety
of PRS evaluation metrics and scores. Furthermore, PRISMA implements a semi-
automated methodology for selecting and suggesting the best performing PRS 
candidates which are extracted based on an iterative procedure where the input
dataset is split in training (for summary statistics) and test (for PRS 
extraction) subsets and the resulting PRS candidates are integrated and 
evaluated resulting in the best candidate sets. The findings are finally 
collected and described in an interactive report which also includes lookup for
the selected SNPs in the EBI GWAS Catalog. Finally, PRISMA offers several side
functions such as facilities for merging datasets (e.g. two datasets that
complement each other or datasets coming from different array platforms),
input/output functions and also SNP imputation based on reference panels by
wrapping [IMPUTE2](https://mathgen.stats.ox.ac.uk/impute/impute_v2.html).

Although it is generally assumed that any PRS extraction and evaluation is and
should be based on summary statistics calculalated from GWAS and based on
sufficiently large and as homogenous as possible cohorts this may not be always
possible. Examples would include smaller cohorts that are used to study a
rare disease (e.g. pediatric nephropathies) or to examine traits in smaller 
groups such as isolated populations or athletic groups. In such cases, any PRS
extraction process would result in rather unstable PRSs in terms of SNP content.
In fact, such instability may also show up in cases where sufficiently large
populations have been used to estimate summary statistics. PRISMA tries to 
provide a solution in such cases by offering a resampling framework where PRS
candidates are partly stabilized based on an ensemble technique where markers
are gathered to PRSs based on aggregation from multiple PRS extraction runs
and smart usage of state-of-the art algorithms such as PRSice2 and lassosum,
coupled by a semi-automated selection of best candidates based on metrics such
as the R<sup>2</sup> of regression models including the PRS as covariate.

## Summary statistics extraction and GWAS analysis

PRISMA supports five methods for estimating SNP effects for single-trait 
associations. More specifically:

1. General Linear Models (GLM), through the respective R functions. Each SNP is 
regressed against the phenotype and the selected covariates. The SNP effects and 
p-value are reported.
2. Ridge regression through the R package 
[rrBLUP](https://cran.r-project.org/web/packages/rrBLUP/index.html). This 
package deploys penalized regression for shrinking the effects of less 
informative SNPs in relation with the phenotype.
3. Single-trait association tests with the 
[statgenGWAS](https://biometris.github.io/statgenGWAS/) R package. The 
association is based on linear regression and the incorporation of an internally
calculated kinship matrix.
4. Frequentist association tests with 
[SNPTEST](https://www.well.ox.ac.uk/~gav/snptest/) (Oxford). Each SNP is 
regressed against the phenotype and the selected covariates using an additive 
model (all other SNPTEST options are also supported).
5. Association tests with [PLINK](https://www.cog-genomics.org/plink/) (linear 
regression with covariates or simple X<sup>2</sup> association tests if no
covariates are present).

## De *novo* PRS extraction
PRISMA currently supports two methods for extracting PRSs. More specifically:

1. [PRSice2](https://www.prsice.info/), which operates based on the PLINK 
framework and automates a PRS extraction process based on clumping and 
thresholding followed by weighted PRS construction and evaluation with linear 
regression.
2. [lassosum](https://github.com/tshmak/lassosum), which utilizes penalized 
(LASSO) regression to shrink the effects of  less informative variants without 
completely removing them from the PRS.

By default, PRISMA uses both PRSice2 and lassosum and offers two ways of 
integrating their results: 
i. by intersecting the PRS candidates from each algorithm 
ii. by considering the respective union. 

In all cases, the effects from GWA tests are considered as the weights of each 
SNP in the constructed PRS.

## PRISMA pipelines for PRS extraction

Starting from a set of phenotypic and genotypic data assembled in a PRISMA
`GWASExperiment` object and/or a PRS from an external resource (e.g. PGS 
Catalog) which can be used to subset the former, PRISMA supports four pipelines 
outlined below. Prior to each pipeline, a basic filtering of the genotypic data 
is performed across SNPs and samples, and missing genotype imputation is 
performed followed by PCA (optionally) to extract covariates characterizing the 
whole available population. These are stored in the original object, to be used 
towards the final steps of model evaluation.

### De novo PRS extraction pipeline

This is the main PRS candidate extraction pipeline and performs the following 
steps:
1. Split of the input data object to a usually larger training set and a smaller 
independent test set.
2. De novo quality control on the training and test sets as well as optional PCA 
in the training set to capture local population structure to be used as 
covariates in the subsequent GWA tests.
3. Execution of GWA tests against a phenotype using one or more of the supported 
methods to extract summary statistics (SNP effects and p-values).
4. Execution of one or more PRS extraction algorithms to derive PRS candidate 
SNPs.

Steps (1) – (4) are executed N times with different training and test sets to 
capture as much variation in the selection process as possible. Subsequently:

5. The candidate SNPs from PRSs extracted at each iteration are aggregated and 
the frequency of their appearance is recorded. At the same time, new effects are 
calculated either by averaging the SNP effects from each iteration, a 
[statistically valid process](https://www.biorxiv.org/content/10.1101/133785v8.full.pdf) 
which generally corrects for overfitting, or by weighted averaging of the 
effects based on the performance of PRSs constructed at each iteration (as 
measured by the adjusted PRS R<sup>2</sup> returned by PRSice2).

### De novo PRS evaluation pipeline

This is an evaluation of a de novo extracted PRS pipeline and performs the 
following steps:

1. Create sets of PRS candidates from step (5) of the de novo PRS extraction 
pipeline based on:
  a. Fixed quantiles of the frequency distribution (e.g. 10%, 20%, 30%, 40%, 
  50%, 60%, 75%, 80%, 90%, 95%, 99%).
  b. A stepwise set creation process based on frequency of appearance, i.e. each 
  set contains SNPs that appear 5, 6, … up to the maximum number of appearances.
2. For each set created at (1), split de novo the initial data object in a 
training and test set. Using the training test, reperform GWA tests to extract 
summary statistics (the averaged effects are not used) and use PRSice2 in 
*PRS application* mode to assess the PRS sets of (1) on the test set with the 
new summary statistics.
3. Evaluate each PRS application using the adjusted PRS R2 coefficient of 
PRSice2.

Steps (2) and (3) are repeated M times (ideally M=N iterations). This pipeline 
includes heavy input/output in the form of PLINK files and is quite slow. In 
some cases it may be more accurate though, although it does not provide a 
straightforward way of calculating the final SNP effects for the PRS. Finally, 
this pipeline may be more appropriate for evaluating external PGS scores, like 
scores retrieved from PGS Catalog.

### PRS evaluation pipeline based on the original train/test split with PRSice2

This pipeline is similar to the de novo PRS evaluation pipeline but it uses the 
initial train/test split described above for PRS evaluation. The steps are 
similar to those described above and the evaluation is performed with the 
PRSice2 framework, limiting somehow the available model evaluation metrics. 
Also, averaged SNP effects are used.

### PRS evaluation pipeline based on the original train/test split with R

Again, this pipeline is similar to the one described above but this time 
evaluation is done with native R functions (manual PRS construction outside 
PRSice2 followed by GLM regressions). This is the fastest evaluation pipeline 
and allows for the collection of many additional metrics.


## PRS selection process
The final PRS selection out of the several candidates is performed in a two-step 
semi-automatic process which involves graphical inspection of evaluation plots 
and also suggestion of the best PRSs using simple mathematical functions on 
the evaluation plots.

### The PRISMA adjusted PRS R<sup>2</sup>
Although many cases in related work are based on the R2 coefficient (or 
adjusted versions such as the one suggested in PRSice2) to explain the 
phenotypic variance percentages explained by the PRS, it is often desired that
a PRS may not include huge number of SNPs for its construction, for several
reasons, a few including:

i. Potential overfitting by including many SNPs which may be heavily dependent 
on specific populations used to derive summary statistics, thus presenting 
problematic generalization when applied to slightly different populations.
ii. The ability to use other non-high-throughput techniques to derive 
genotypes, such as qPCR, which would benefit by a smaller number of markers 
for genotyping.
iii. Easier downstream investigation and tracking of the SNP markers included 
in the PRS.
iv. Potential robustness when further tuning a predictive regression model, 
for example including or excluding covariates.

For these reasons, we introduce the PRISMA adjusted PRS R2 calculated using 
the formula:

$$R_{P}^{2} = \sqrt[2]{\frac{R_{PRS}^{2}}{log(N)}} $$

where $R_{PRS}^{2}$ is the adjusted R<sup>2</sup> explaining the PRS 
contribution (R<sup>2</sup> of the full model including the PRS – R<sup>2</sup> 
of the reduced model including only the covariates against the phenotype) and
*N* is the number of SNPs included in the PRS.

Essentially, $R_{P}^{2}$ penalizes $R_{PRS}^{2}$ according to the number of
SNPs *N*, while maintaining scale and monotonic pattern by using the square
root. In practice, it does not seriously differ from using only the 
$R_{PRS}^{2}$ for selection of the optimal number of SNPs, but it is very
useful when controlling the number of SNPs while maintaining accuracy is 
desirable.

### Graphical evaluations

After executing a possible evaluation pipeline, the best PRS is firstly 
evaluated graphically by:

i. A bar plot of average adjusted PRS R2 ± standard deviations for each SNP 
frequency cutoff where it can be checked if there is a cutoff increasing the 
R2 as compared to the baseline (R2 from the de novo PRS extraction iterations).
ii. A dot plot of the PRISMA adjusted PRS R2 R_P^2, where it can be inspected 
where the adjusted R2 is potentially maximized.
iii. A dot plot of SNP frequencies.

### Automatic PRS selection

The number of SNPs that will finally be used to build one or more candidate 
PRS is determined with two possible stepwise SNP inclusion methods:
    
i. Using the Elbow method, which essentially detects a cutoff in the number of
SNPs in the PRS where the R2 metrics do not significantly increase after
including more SNPs in the PRS. This method can be applied when the 
R<sup>2</sup> distribution is mostly monotonic.
ii. By detecting local maxima in the R2 distribution, corresponding to local 
maximizations of the latter at specific number of SNPs. The main PRS includes 
the SNPs which maximize the $R_{PRS}^{2}$ or the $R_{P}^{2}$ so as to 
compensate for the number of SNPs in the PRS (default behavior). This option
returns also SNP sets at other local maxima of the R<sup>2</sup> distribution.  

A similar approach like (ii) above regarding the maximization of R2 is being 
used in a recent work presented in
[this](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02479-9) study.

## PRS evaluation metrics

The following metrics are used for the evaluation of PRS candidates and during 
their iterative selection:

* Full regression model R<sup>2</sup>: the R<sup>2</sup> of a model including
the PRS.
* Reduced model R<sup>2</sup>: the R<sup>2</sup> of a model without the PRS
(only covariates against the phenotype under interrogation).
* Adjusted PRS R<sup>2</sup>: the PRSice2 adjusted PRS R<sup>2</sup>.
* Full and reduced model Root Mean Squared Error (RMSE).
* Full and reduced model Mean Absolute Error (MAE).
* Statistical significance of the average differences of the above using the
Student’s t-test, the Wilcoxon rank test and an empirical bootstrap p-value.
* All of these metrics are collected during the evaluation pipeline with 
vanilla R and also in cross-validation runs based on regression models with
the selected PRSs and the total available dataset.

## PRS retrieval from external resources

PRISMA supports the retrieval of candidate PRSs or candidate PRS components 
(SNPs) for a trait from the following resources:

* The [EBI PRS Catalog](https://www.pgscatalog.org/)
* The [EBI GWAS Catalog](https://www.ebi.ac.uk/gwas/)

Retrieval of data regarding specific PRS also from 
[impute.me](https://impute.me/) is being currently tested. The data retrieved 
from online resources are being harmonized into a common format including the 
recording of existing PRS coefficients, statistical significance, related loci 
and GWAS association scores (p-values). They are also enriched with the 
following data:

* Location assignment when only dbSNP (rs) ids are only provided.
* dbSNP id assignment when only location is provided.
* Locus (gene) assignment when not provided and if possible (i.e. SNP is not 
intronic).

The resulting structure can be used with one of the PRISMA pipelines to 
evaluate the retrieved PRS (i.e. check if all markers are needed with local 
data) and reconstruct a PRS.

# Quick start

The commands below can be executed within R's command line interface and use the
toy data accompanying the package. For many more details regarding installation,
usage and package presentation, you should also look at the other vignettes.
Before running these examples, you should have at least installed 
[PRSice2](https://www.prsice.info/) (have also a look at the Installation
vignette) which should be visible by R (i.e. included in the system's `PATH`
variable).

```{r intro-0, eval=TRUE, echo=FALSE, tidy=FALSE, message=FALSE, warning=FALSE}
library(prisma)
```

### Load some toy data

The toy dataset contains genotypes for chromosomes 18 and 21 from a the 
case/control described in 
[Kalafati et al.](https://doi.org/10.1016/j.nut.2018.10.032). The trait for 
which PRS candidates are sought is the Body-Mass Index (continuous variable).

```{r intro-1, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
library(prisma)

data(toy,package="prisma")
gdsfile(toy) <- system.file(package="prisma","extdata","toy.gds")
```

### Apply some filters

```{r intro-2, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
# Lose filters for the toy dataset
filts <- getDefaults("filters")
filts$IBD <- NA
filts$hwe <- 1e-3
filts$pcaOut <- FALSE
filts$inbreed <- NA
```

### Full pipeline - create PRS candidates

```{r intro-3, eval=FALSE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
# Workspace
d1 <- tempfile()
wspace1 <- file.path(tempdir(),d1)

# Traits
response <- "BMI"
covariates <- c("Case_Control","Gender","Age")

# Complete 4-iteration PRS pipeline - ~1 min to run with 4 cores - no PCA
if (Sys.which("PRSice_linux") != "") {
    prismaOut <- prisma(
        gwe=toy,
        phenotype=response,
        covariates=covariates,
        pcs=FALSE,
        trainSize=0.8,
        niter=4,
        resolution="frequency",
        minSnps=2,
        filters=filts,
        pcaMethod="snprel",
        imputeMissing=FALSE,
        gwaMethods=c("glm","statgen"),
        prsiceOpts=list(clump_p=0.75,perm=100),
        prsWorkspace=wspace1,
        cleanup="intermediate",
        logging="file",
        output="normal",
        rc=0.25
    )
}
```

### Step-by-step pipeline

Below, we execute step-by-step the individual steps implemented in the `prisma`
wrapper only for the `statgenGWAS` GWA testing.

#### 1. Iterate PRS extraction over the dataset

```{r intro-4, eval=FALSE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
d2 <- tempfile()
wspace2 <- file.path(tempdir(),d2)
if (Sys.which("PRSice_linux") != "") {
    dnList <- prsPipeline(
        gwe=toy,
        phenotype=response,
        covariates=covariates,
        pcs=FALSE,
        trainSize=0.8,
        niter=4,
        filters=filts,
        pcaMethod="snprel",
        imputeMissing=FALSE,
        gwaMethods="statgen",
        prsiceOpts=list(clump_p=0.75,perm=100),
        prsWorkspace=wspace2,
        cleanup="intermediate",
        logging="file",
        rc=0.25
    )
}
```

#### 2. Calculate the evaluation metrics that lead to PRS selection

```{r intro-5, eval=FALSE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
d3 <- tempfile()
wspace3 <- file.path(tempdir(),d3)
evalMetrics <- prsSelection(
    dnList=dnList,
    gwe=toy,
    phenotype=response,
    covariates=covariates,
    pcs=FALSE,
    trainSize=0.8,
    niter=4,
    resolution="frequency",
    minSnps=2,
    filters=filts,
    pcaMethod="snprel",
    imputeMissing=FALSE,
    gwaMethods="statgen",
    prsiceOpts=list(clump_p=0.75,perm=100),
    prsWorkspace=wspace3,
    cleanup="intermediate",
    logging="file",
    output="summary",
    useDenovoWorkspace=dnList,
    rc=0.25
)
```

#### 3. Based on the metrics, find the best PRS candidates

```{r intro-6, eval=FALSE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
candidates <- selectPrs(
    metrics=evalMetrics$metrics,
    snpSelection=evalMetrics$pgs,
    gwe=toy,
    base=evalMetrics$baseline
)
```

The `prisma` wrapper implements additional steps to assemble the results and
create some plots to be displayed in the final report.

### Complete PRISMA pipeline with report

The `prismaPipeline` wrapper servers as a complete "wrapper of wrapper" or a
"meta-wrapper" for the end-to-end PRS extraction pipeline, from SNP effect
estimation given a population with genotypes and phenotypes. The pipeline
runs the `prisma` wrapper making sure that any crashes are resumed (up to 10
times by default), followed by cross-validation for the PRS candidates, lookup
in the EBI GWAS Catalog and the creation of an HTML report.

```{r intro-7, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE, collapse=TRUE}
outPath <- file.path(tempdir(),tempfile())
d4 <- tempfile()
wspace4 <- file.path(outPath,d4)

prismaPipeline(
    gwe=toy,
    phenotype=response,
    covariates=covariates,
    pcs=FALSE,
    trainSize=0.8,
    niter=4,
    resolution="frequency",
    minSnps=2,
    filters=filts,
    pcaMethod="snprel",
    imputeMissing=FALSE,
    gwaMethods=c("glm","statgen"),
    prsiceOpts=list(clump_p=0.75,perm=100),
    prsWorkspace=wspace4,
    cleanup="intermediate",
    logging="file",
    output="normal",
    rc=0.25,
    outPath=outPath
)
```

# R session info

```{r session-0, eval=TRUE, echo=TRUE, tidy=FALSE, message=TRUE, warning=FALSE}
sessionInfo()
```
