---
title: PRISMA report
subtitle: "`r cat('A project name here')`"
date: "`r format(Sys.time(), '%d/%m/%Y')`"
output:
  rmdformats::robobook:
    fig_width: 7
    fig_height: 7
    fig_caption: true
    highlight: kate
    lightbox: true
    #thumbnails: true
    #gallery: true
    cards: true
    use_bookdown: false
    mathjax: null
    self_contained: false
---

<style>
body {
    text-align: justify;
}

.lightbold {
    font-weight: 500;
}

.figure-hint {
    margin: 24px 48px;
    text-align: justify;
    font-size: 0.9em;
}

.selector-label {
    color: #0D47A1;
    font-size: 0.9em;
    font-weight: 600;
}

.prs_selector_helper {
    display: none;
}

.link-box {
    background-color: #E1E4FF; 
    border-style: solid; 
    border-color: #000C88; 
    border-width: 1px; 
    width: 80%; 
    padding: 10px;
    margin-top: 20px;
    -moz-border-radius: 5px;
    -webkit-border-radius: 5px;
    -khtml-border-radius: 5px;
    border-radius: 5px;
}

.dataTables_paginate .paginate_button:hover {
    background-color: #333333 !important;
}

#opts_container {
    font-size: 0.9em;
    height: 400px;
    overflow-y: scroll;
    overflow-x: hidden;
    overflow-wrap: normal;
}
</style>

```{r knitr_init, echo=FALSE, cache=FALSE}
library(knitr)
library(rmdformats)

# Global options
options(max.print="75")
opts_knit$set(progress=FALSE,verbose=FALSE)
opts_chunk$set(
    echo=FALSE,
    cache=FALSE,
    prompt=FALSE,
    tidy=TRUE,
    comment=NA,
    message=FALSE,
    warning=FALSE,
    results="asis",
    eval=TRUE,
    fig.align="center"
)
opts_knit$set(width=75)
```

```{r report_init}
library(DT)
library(ggplot2)
library(htmltools)
library(magrittr)
library(pander)
library(grid)
library(DT)
library(htmltools)

# Functions onatop
subchunkify <- function(g,fig_height=7,fig_width=5) {
    g_deparsed <- paste0(deparse(
        function() {g}
    ), collapse = '')

    sub_chunk <- paste0("
        `","``{r sub_chunk_",floor(runif(1)*10000), ", fig.height=",
        fig_height, ", fig.width=",fig_width, ", echo=FALSE}",
        "\n(",g_deparsed, ")()","\n`","``
    ")
    cat(trimws(knitr::knit(text=knitr::knit_expand(text=sub_chunk),quiet=TRUE)))
}

nosci <- function(x,d) {
    format(round(x,d),scientific=FALSE)
}

args <- prismaOut$params$args
funCall <- prismaOut$params$call

msgs <- .makeReportMessages("en")
#msgs <- prisma:::.makeReportMessages("en")

theCvMetrics <- list(
    `PRS R<sup>2</sup>`="r2",
    `RMSE`="rmse",
    `MAE`="mae",
    `Predicted R<sup>2</sup>`="pr2",
    `Correlation`="crl"
)

cvMetExpl <- list(
    `PRS R<sup>2</sup>`=paste0(
        "The plots depict the following R<sup>2</sup> measurements:",
        "<ul>",
        "<li>Full regression model R<sup>2</sup>: the R<sup>2</sup> of a ",
            "model including the PRS.</li>",
        "<li>Reduced model R<sup>2</sup>: the R<sup>2</sup> of a model ",
            "without  the PRS (only covariates against the phenotype under ",
            "interrogation).</li>",
        "<li>Adjusted PRS R<sup>2</sup>: Full R<sup>2</sup> – Reduced ",
            "R<sup>2</sup>.</li>",
        "</ul>"
    ),
    `RMSE`=paste0(
        "The plots depict the following RMSE measurements:",
        "<ul>",
        "<li>Full model RMSE (including the covariates and the PRS).</li>",
        "<li>Reduced model RMSE (including only the covariates).</li>",
        "</ul>"
    ),
    `MAE`=paste0(
        "The plots depict the following MAE measurements:",
        "<ul>",
        "<li>Full model MAE (including the covariates and the PRS).</li>",
        "<li>Reduced model MAE (including only the covariates).</li>",
        "</ul>"
    ),
    `Predicted R<sup>2</sup>`=paste0(
        "The plots depict the following R2 measurements:",
        "<ul>",
        "<li>R<sup>2</sup> from prediction using the full model (including ",
            "the PRS).</li>",
        "<li>R<sup>2</sup> from prediction using the reduced model (excluding ",
            "the PRS).</li>",
        "<li>Adjusted R<sup>2</sup>: Full R<sup>2</sup> – Reduced ",
            "R<sup>2</sup>.</li>",
        "</ul>"
    ),
    `Correlation`=paste0(
        "The plots depict the following R measurements:",
        "<ul>",
        "<li>R from prediction using the full model (including the PRS).</li>",
        "<li>R from prediction using the reduced model (excluding the PRS).",
            "</li>",
        "</ul>"
    )
)

metDir <- list(
    `PRS R<sup>2</sup>`="higher",
    `RMSE`="lower",
    `MAE`="lower",
    `Predicted R<sup>2</sup>`="higher",
    `Correlation`="higher"
)

# Prepare the CV metrics matrices
cvDetMatInd <- list(
    `PRS R<sup>2</sup>`=c("mean_full_r2","sd_full_r2","mean_reduced_r2",
        "sd_reduced_r2","mean_prs_r2","sd_prs_r2","max_prs_pvalue",
        "ttest_full_reduced_r2","wilcox_full_reduced_r2",
        "empirical_full_reduced_r2"),
    `RMSE`=c("mean_full_rmse","sd_full_rmse","mean_reduced_rmse",
        "sd_reduced_rmse","ttest_full_reduced_rmse","wilcox_full_reduced_rmse",
        "empirical_full_reduced_rmse"),
    `MAE`=c("mean_full_mae","sd_full_mae","mean_reduced_mae","sd_reduced_mae",    
        "ttest_full_reduced_mae","wilcox_full_reduced_mae",
        "empirical_full_reduced_mae"),
    `Predicted R<sup>2</sup>`=c("mean_full_pred_r2","sd_full_pred_r2",
        "mean_reduced_pred_r2","sd_reduced_pred_r2","mean_prs_pred_r2",
        "sd_prs_pred_r2","ttest_full_reduced_pred_r2",
        "wilcox_full_reduced_pred_r2","empirical_full_reduced_pred_r2"),
    `Correlation`=c("mean_full_pred_cor","sd_full_pred_cor",
        "mean_reduced_pred_cor","sd_reduced_pred_cor",
        "ttest_full_reduced_pred_cor","wilcox_full_reduced_pred_cor",
        "empirical_full_reduced_pred_cor")
)

cvDetMatNam <- list(
    `PRS R<sup>2</sup>`=c("Full R<sup>2</sup>","Reduced R<sup>2</sup>",
        "PRS R<sup>2</sup>","PRS p_value","Student","Wilcoxon","Empirical"),
    `RMSE`=c("Full RMSE","Reduced RMSE","Student","Wilcoxon","Empirical"),
    `MAE`=c("Full MAE","Reduced MAE","Student","Wilcoxon","Empirical"),
    `Predicted R<sup>2</sup>`=c("Full Predicted R<sup>2</sup>",
        "Reduced Predicted R<sup>2</sup>","PRS Predicted R<sup>2</sup>",
        "Student","Wilcoxon","Empirical"),
    `Correlation`=c("Full Correlation","Reduced Correlation","Student",
        "Wilcoxon","Empirical")
)

# The CV metric summary matrices is a list of data frames for each GWA method,
# for each metric and for each PRS candidate
cvSumDf <- vector("list",length(prismaOut$results))
names(cvSumDf) <- names(prismaOut$results)
for (m in names(prismaOut$results)) {
    if (!is.null(cvMetricsOut[[m]])) {
        # Calculate current cv metrics
        curSum <- summarizeCvMetrics(cvMetricsOut[[m]])
        
        # Compose the data frames
        cvSumDf[[m]] <- vector("list",length(cvDetMatNam))
        names(cvSumDf[[m]]) <- names(cvDetMatNam)
        for (e in names(cvDetMatNam)) {
            cvSumDf[[m]][[e]] <- vector("list",length(curSum))
            names(cvSumDf[[m]][[e]]) <- names(curSum)
            
            for (n in names(curSum)) {
                if (e == "PRS R<sup>2</sup>") {
                    f <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(1,2)]],4)
                    r <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(3,4)]],4)
                    p <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(5,6)]],4)
                    z <- formatC(curSum[[n]][,cvDetMatInd[[e]][7],drop=FALSE],
                        format="e",digits=4)
                    s <- formatC(curSum[[n]][,cvDetMatInd[[e]][c(8,9)]],
                        format="e",digits=4)
                    v <- nosci(curSum[[n]][,cvDetMatInd[[e]][10],drop=FALSE],3)
                    cvSumDf[[m]][[e]][[n]] <- data.frame(
                        paste(f[,1],f[,2],sep=" &plusmn; "),
                        paste(r[,1],r[,2],sep=" &plusmn; "),
                        paste(p[,1],p[,2],sep=" &plusmn; "),
                        z,s,v
                    )
                }
                else if (e %in% c("RMSE","MAE","Correlation")) {
                    f <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(1,2)]],4)
                    r <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(3,4)]],4)
                    s <- formatC(curSum[[n]][,cvDetMatInd[[e]][c(5,6)]],
                        format="e",digits=4)
                    v <- nosci(curSum[[n]][,cvDetMatInd[[e]][7],drop=FALSE],3)
                    cvSumDf[[m]][[e]][[n]] <- data.frame(
                        paste(f[,1],f[,2],sep=" &plusmn; "),
                        paste(r[,1],r[,2],sep=" &plusmn; "),s,v
                    )
                }
                else if (e == "Predicted R<sup>2</sup>") {
                    f <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(1,2)]],4)
                    r <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(3,4)]],4)
                    p <- nosci(curSum[[n]][,cvDetMatInd[[e]][c(5,6)]],4)
                    s <- formatC(curSum[[n]][,cvDetMatInd[[e]][c(7,8)]],
                        format="e",digits=4)
                    v <- nosci(curSum[[n]][,cvDetMatInd[[e]][9],drop=FALSE],3)
                    cvSumDf[[m]][[e]][[n]] <- data.frame(
                        paste(f[,1],f[,2],sep=" &plusmn; "),
                        paste(r[,1],r[,2],sep=" &plusmn; "),
                        paste(p[,1],p[,2],sep=" &plusmn; "),s,v
                    )
                }
                
                names(cvSumDf[[m]][[e]][[n]]) <- cvDetMatNam[[e]]
            }
        }
    }
}
```

# Summary {.tabset .tabset-pills}

## General

### PRS candidates summary {.tabset .tabset-pills}

```{r summary general_1}
cat("<p>The queried phenotype for PRS extraction was <strong>",args$phenotype,
    "</strong>. ",sep="")
if (!is.null(args$covariates))
    cat("<strong>",paste(args$covariates,collapse=", "),"</strong> were used ",
        "as regression covariates. ",sep="")
if (args$pcs)
    cat("A number of principal components was also included in the covariates",
        "set to capture potential population-based genotypic structure. ")
cat("The following GWA tests were used to extract summary statistics for SNP ",
    "markers included in PRS candidates: <strong>",
    paste(msgs$stat,collapse=", "),"</strong>. Each method yielded the ",
    "following candidates, when compared to the baseline evaluation metric ", 
    "over ",args$niter," PRS extraction iterations:</p>",sep="")
for (m in names(prismaOut$results)) {
    ns <- names(prismaOut$results[[m]]$candidates)
    fm <- prismaOut$results[[m]]$reportData$evalMetrics$metrics[ns,c(2,3,6,31),
        drop=FALSE]
    
    pandoc.header(msgs$stat[[m]],level=4)
    
    cat("The optimal number of markers for PRS based on adjusted PRS ",
        "R<sup>2</sup> is <strong>",ns[1],"</strong> at adjusted PRS <strong>",
        "R<sup>2</sup> = ",nosci(fm[ns[1],2],5)," &plusmn; ",
        nosci(fm[ns[1],3],5)," (p = ",nosci(fm[ns[1],4],6),") </strong> with ",
        "SNPs appearing  at least <strong>",fm[ns[1],1],"</strong> times over ",
        args$niter," iterations.",sep="")
        cat("\n") # Otherwise breaks headers
    if (length(ns) > 1) {
        cat("Other possible options include:")
        cat("<ul>")
        for (i in 2:length(ns))
            cat("<li><strong>",ns[i],"</strong> markers for PRS with adjusted ",
                "PRS <strong>R<sup>2</sup> = ",nosci(fm[ns[i],2],5)," &plusmn;",
                " ",nosci(fm[ns[i],3],5)," (p = ",nosci(fm[ns[i],4],6),
                ")</strong> with SNPs appearing at least <strong>",fm[ns[i],1],
                "</strong> over <strong>",args$niter,"</strong> iterations.",
                "</li>",sep="")
        cat("</ul>")
    }
}
```

## Input options

### Input options

<div id="opts_container">

#### Genotype input

**Input dataset: **
```{r summary_input_options_1}
cat("a GWASExperiment object with ",nrow(gwe),
        " markers (SNPs) and ",ncol(gwe)," samples")
```

#### Regression parameters

**Phenotypic response:** 
```{r summary_input_options_2}
cat(args$phenotype)
```

**Phenotypic covariates:** 
```{r summary_input_options_3}
cat(paste(args$covariates,collapse=", "))
```

**Use population structure covariates (PCA):** 
```{r summary_input_options_4}
cat(ifelse(args$pcs,"Yes","No"))
```

**Number of Principal Components:** 
```{r summary_input_options_5}
cat(ifelse(args$npcs==0,"Automatic (Tracy-Widom test)",as.character(args$npcs)))
```

**General PCA method:** 
```{r summary_input_options_6}
cat(args$pcaMethod)
```

#### Preprocessing options

**Genotype quality controls:** 
```{r summary_input_options_7}
filts <- eval(args$filters)
cat("<ul>")
cat("<li><em class=lightbold>SNP call rate:</em> ",
    filts$snpCallRate,"</li>")
cat("<li><em class=lightbold>Sample call rate:</em> ",
    filts$sampleCallRate,"</li>")
cat("<li><em class=lightbold>Minor allele frequency:</em> ",
    filts$maf,"</li>")
cat("<li><em class=lightbold>Hardy-Weinberg equilibrium:</em> ",
    filts$hwe,"</li>")
cat("<li><em class=lightbold>Heterozygosity cutoff statistic:</em> ",
    filts$heteroStat,"</li>")
cat("<li><em class=lightbold>Heterozygosity cutoff scale factor:</em> ",
    filts$heteroFac,"</li>")
cat("<li><em class=lightbold>Heterozygosity hard cutoff:</em> ",
    filts$heteroHard,"</li>")
cat("<li><em class=lightbold>Automatic sample removal with robust PCA:</em> ",
    ifelse(filts$pcaOut,"Yes","No"),"</li>")
cat("<li><em class=lightbold>Robust PCA method:</em> ",
    filts$pcaRobust,"</li>")
cat("<li><em class=lightbold>Number of PCs for robust PCA:</em> ",
    ifelse(is.na(filts$nPC),"auto",filts$nPC),"</li>")
cat("<li><em class=lightbold>Linkage Disequilibrium (LD) cutoff:</em> ",
    ifelse(is.na(filts$LD),"none",filts$LD),"</li>")
cat("<li><em class=lightbold>Identity by descent (IBD) cutoff:</em> ",
    ifelse(is.na(filts$IBD),"none",filts$IBD),"</li>")
cat("<li><em class=lightbold>Inbreed coefficient cutoff:</em> ",
    ifelse(is.na(filts$IBD),"none",filts$LD),"</li>")
cat("</ul>")
```

**General PCA method:** 
```{r summary_input_options_8}
cat(args$pcaMethod)
```

**Impute missing genotypes:** 
```{r summary_input_options_9}
cat(ifelse(args$imputeMissing,"Yes","No"))
```

**Missing genotype imputation strategy:** 
```{r summary_input_options_10}
cat(args$imputeMethod)
```

#### GWA analysis options

**GWA test(s) for summary statistics:** 
```{r summary_input_options_11}
cat(paste(args$gwaMethods,collapse=", "))
```

**GWA test(s) p-value combination method:** 
```{r summary_input_options_12}
cat(args$gwaCombine)
```

**General Linear Model options:** 
```{r summary_input_options_13}
cat("<ul>")
cat("<li><em class=lightbold>Model family:</em> ",
    ifelse(is.null(args$glmOpts$family),"automatic selection",
    args$glmOpts$family),"</li>")
cat("<li><em class=lightbold>SNP regression chunk size:</em> ",
    args$glmOpts$size,"</li>")
cat("</ul>")
```

**rrBLUP (ridge regression) options:** 
```{r summary_input_options_14}
cat("<ul>")
cat("<li><em class=lightbold>Principal Components method:</em> ",
    args$rrblupOpts$pcblup,"</li>")
cat("<li><em class=lightbold>Number of PCs:</em> ",
    ifelse(is.null(args$rrblupOpts$npcs),"auto estimate",args$rrblupOpts$npcs),
    "</li>")
cat("</ul>")
```

**statgenGWAS options:** 
```{r summary_input_options_15}
cat("<ul>")
cat("<li><em class=lightbold>Kinship estimation method:</em> ",
    args$statgenOpts$kinship,"</li>")
cat("<li><em class=lightbold>Restricted Maximum Likelihood (REML) method:</em>",
    args$statgenOpts$reml,"</li>")
cat("<li><em class=lightbold>Effect estimate kinship method:</em> ",
    args$statgenOpts$gls,"</li>")
cat("</ul>")
```

**SNPTEST options:** 
```{r summary_input_options_16}
cat("<ul>")
cat("<li><em class=lightbold>Basic association test type:</em> ",
    args$snptestOpts$test,"</li>")
cat("<li><em class=lightbold>Genotype model type:</em> ",args$snptestOpts$model,
    "</li>")
cat("<li><em class=lightbold>SNPTEST workspace:</em> ",
    ifelse(is.null(args$snptestOpts$workspace),"default",
    args$snptestOpts$workspace),"</li>")
cat("</ul>")
```

**PLINK options:** 
```{r summary_input_options_17}
cat("<ul>")
cat("<li><em class=lightbold>Effect estimation type:</em> ",
    args$plinkOpts$effect,"</li>")
cat("<li><em class=lightbold>Random seed:</em> ",args$plinkOpts$seed,"</li>")
cat("<li><em class=lightbold>PLINK workspace:</em> ",
    ifelse(is.null(args$plinkOpts$workspace),"default",
    args$plinkOpts$workspace),"</li>")
cat("</ul>")
```

#### PRS candidate extraction options

**PRS extraction method(s):** 
```{r summary_input_options_18}
cat(paste(args$prsMethods,collapse=", "))
```

**lassosum options:** 
```{r summary_input_options_19}
cat("<ul>")
cat("<li><em class=lightbold>Ancestry for LD regions:</em> ",
    args$lassosumOpts$anc,"</li>")
cat("<li><em class=lightbold>Validation method:</em> ",args$lassosumOpts$valid,
    "</li>")
cat("</ul>")
```

**PRSice-2 options:** 
```{r summary_input_options_20}
cat("<ul>")
cat("<li><em class=lightbold>Clumping window size (kb):</em> ",
    args$prsiceOpts$clump_kb,"</li>")
cat("<li><em class=lightbold>Clumping R<sup>2</sup> threshold:</em> ",
    args$prsiceOpts$clump_r2,"</li>")
cat("<li><em class=lightbold>Clumping p-value threshold:</em> ",
    args$prsiceOpts$clump_p,"</li>")
cat("<li><em class=lightbold>Clumping window size (kb):</em> ",
    args$prsiceOpts$clump_kb,"</li>")
cat("<li><em class=lightbold>PRS calculation method:</em> ",
    args$prsiceOpts$score,"</li>")
cat("<li><em class=lightbold>Number of permutations:</em> ",
    args$prsiceOpts$perm,"</li>")
cat("<li><em class=lightbold>Random seed:</em> ",args$prsiceOpts$seed,"</li>")
cat("</ul>")
```

#### PRS construction and evaluation options

**Training set split size:** 
```{r summary_input_options_21}
cat(paste0(100*args$trainSize,"%"))
```

**Number of PRS extraction iterations:** 
```{r summary_input_options_22}
cat(args$niter)
```

**Best PRS resolution method:** 
```{r summary_input_options_23}
cat(args$resolution)
```

**Number of PRS resolution step:** 
```{r summary_input_options_24}
cat(paste(args$step,collapse=", "))
```

**Minimum SNP frequency for PRS inclusion:** 
```{r summary_input_options_25}
cat(args$minFreq)
```

**Minimum number of SNPs in PRS:** 
```{r summary_input_options_26}
cat(args$minSnps)
```

**Drop quantiles with same SNPs:** 
```{r summary_input_options_27}
cat(ifelse(args$dropSameQuantiles,"Yes","No"))
```

**PRS marker (SNP) aggregation method:** 
```{r summary_input_options_28}
cat(args$aggregation)
```

**SNP effect weighting method:** 
```{r summary_input_options_29}
cat(args$effectWeight)
```

**Evaluation framework:** 
```{r summary_input_options_30}
cat(args$evalWith)
```

#### PRS selection options

**PRS selection method:** 
```{r summary_input_options_31}
cat(args$prsSelectMethod)
```

**PRS selection criterion:** 
```{r summary_input_options_32}
cat(args$prsSelectCrit)
```

**PRS selection averaging method:** 
```{r summary_input_options_33}
cat(args$prsSelectStat)
```

**PRS selection adjusted R<sup>2</sup> type:** 
```{r summary_input_options_34}
cat(args$prsSelectR2)
```

#### Miscellaneous options

**Main workspace:** 
```{r summary_input_options_35}
cat(ifelse(is.null(args$prsWorkspace),"auto",args$prsWorkspace))
```

**Cleanup level:** 
```{r summary_input_options_36}
cat(args$cleanup)
```

**Logging device:** 
```{r summary_input_options_37}
cat(args$logging)
```

**De novo run output type:** 
```{r summary_input_options_38}
cat(args$dnOutput)
```

**Final output type:** 
```{r summary_input_options_39}
cat(args$output)
```

**Pipeline continuation:** 
```{r summary_input_options_40}
cat(ifelse(args$continue,"Yes","No"))
```

**De novo workspace for validation:** 
```{r summary_input_options_41}
cat(ifelse(is.null(args$useDenovoWorkspace),"none",
    ifelse(is.list(args$useDenovoWorkspace),"previous run",
    args$useDenovoWorkspace)))
```

**Run ID:** 
```{r summary_input_options_42}
cat(ifelse(is.null(args$runId),"auto",args$runId))
```

**Dataset split for evaluation:** 
```{r summary_input_options_43}
cat(args$evalOnSplit)
```

**Cores percentage for parallel run:** 
```{r summary_input_options_44}
cat(ifelse(is.null(args$rc),"none",paste0(100*args$rc,"%")))
```

</div>

## Basic QC Results

### Basic quality contol results

The following tables summarize how many SNPs and samples were excluded from
further analysis after the application of QC filters:

```{r qc_summary}
qct <- filterRecord(gwe)[,c(4,2,1,3,5),drop=FALSE]
rownames(qct) <- NULL
qct %>% 
    kbl() %>%
    kable_styling(bootstrap_options=c("striped","hover","condensed",
        "responsive"))
```

## Command

### Input command

The Polygenic Risk Score extraction and evaluation analysis and this report were 
generated using the following command:

```{r fun_call}
cat("<pre><code>")
cat(paste(funCall,collapse="<br/>"))
cat("</code></pre>")
```

# Selection {.tabset .tabset-pills}

The **Evaluation** section summarizes the performance of one or more performed
GWA tests with respect to the associative power of the derived PRS as extracted 
through the `prisma` iterative procedure. The aforementioned summary comprises
several assessment graphics which aid the automatic selection of the *best*
performing PRS but also allow the user to adjust this selection based on other
possible solutions and an iterative cross-validation procedure using the whole 
input dataset. The section comprises the following:

* **PRS selection plots**: These plots can be used to inspect the `PRSice` 
adjusted R<sup>2</sup> that explains the phenotypic variability introduced by 
the PRS, or its `prisma` adjusted version that penalizes the R<sup>2</sup>
according to the number of SNP markers included in the PRS candidates.
* **Model cross-validation plots** for various evaluation metrics.

Each plot is accompanied by a small explanation text and interpretation 
instructions.


```{r multi_r2_plot, fig.width=7, fig.height=5}
if (length(prismaOut$results) > 1) { # Plot of baselines
    cat("<h3>Evaluation of performed GWA tests</h3>")
    
    cat("<div class=\"figure-hint\">")
    cat("The following figure depicts the mean PRSice-2 adusted R<sup>2</sup>",
        "+/- its standard deviation, for each performed GWA test across",
        args$niter," PRS extraction iterations. Apart from bar height, higher",
        "adjusted R<sup>2</sup>s are depicted by lighter color. In principle,",
        "the method presenting the higher adjusted R<sup>2</sup> has yielded",
        "the best results over ",args$niter," iterations.")
    cat("</div>")
    
    R2 <- unlist(lapply(names(prismaOut$results),function(n,D) {
        return(mean(D[[n]]$reportData$evalMetrics$baseline))
    },prismaOut$results),use.names=FALSE)
    SD <- unlist(lapply(names(prismaOut$results),function(n,D) {
        return(sd(D[[n]]$reportData$evalMetrics$baseline))
    },prismaOut$results),use.names=FALSE)
    testNames <- unlist(msgs$stat[names(prismaOut$results)],use.names=FALSE)
    
    ggdata <- data.frame(
        R2=R2,
        SD=SD,
        Test=factor(testNames,levels=testNames)
    )
    g <- ggplot(ggdata,aes(x=Test,y=R2)) + 
        geom_bar(aes(fill=R2),stat="identity",width=0.7) +
        geom_errorbar(aes(ymin=R2-SD,ymax=R2+SD),width=0.2) +
        xlab("\nGWA test") +
        ylab(expression("Mean PRSice adjusted R"^"2")) +
        theme_bw() + 
        theme(
            axis.title.x=element_text(size=14),
            axis.title.y=element_text(size=14,face="bold"),
            axis.text.x=element_text(size=12,face="bold"),
            axis.text.y=element_text(size=12,face="bold")
        )
    g
}
```

The following tabs contain evaluation plots for each performed GWA test.

```{r selection_basic, fig.width=14, fig.height=6, eval=TRUE}
cat("<h3>Evaluation of <em>de novo</em> extracted PRSs</h3>")
for (m in names(prismaOut$results)) {
    pandoc.header(paste0(msgs$stat[[m]]," {.tabset .tabset-pills}"),level=2)
    
    # De novo extraction part
    cat("<div class='figure-hint'>")
    cat("The following figure depicts the mean PRS R<sup>2</sup> +/- its",
        "standard deviation (barplots and errorbars), for PRS candidates",
        "assembled from SNPs at different frequencies of appearance in the",
        "PRS candidates across ",args$niter," PRISMA PRS extraction",
        "iterations. The vertical axis depicts the mean adjusted PRS",
        "R<sup>2</sup> while the horizotnal axis depicts the number of SNPs in",
        "each PRS candidate. The number inside the parentheses, next to the",
        "number of SNPs in the horizontal axis depicts the SNP frequency of",
        "appearance in the PRS. For example, **245 (8)** means that the PRS at",
        "that particular R<sup>2</sup> consists of 245 SNPs that appear at",
        "least 8 times over ",args$niter," iterations. The colorscale denotes",
        "the statistical significance (p-value in -log<sub>10</sub> scale) of",
        "the adjusted R<sup>2</sup> distribution over ",args$niter," *de novo*",
        "PRS extraction iterations (baseline R<sup>2</sup>) as compared to the", 
        "adjusted R<sup>2</sup> distribution of each PRISMA assembled PRS",
        "candidate in the horizontal axis. The mean baseline R<sup>2</sup> is",
        "depicted with the dashed grey horizontal line, and the dotted grey",
        "horizontal lines depict the standard deviation of the former. The",
        "best performing PRSs should be among the ones showing higher",
        "R<sup>2</sup> as compared to the baseline along with statistical",
        "significance.")
    cat("</div>")
    cat("<div>")
    print(prismaOut$results[[m]]$reportData$plots$r2snp)
    cat("</div>")
       
    cat("<div class='figure-hint'>")
    cat("The following figure depicts the mean PRISMA PRS R<sup>2</sup> in the",
        "vertical axis, which penalizes the PRS R<sup>2</sup> according to the",
        "number of SNPs that are included in the PRS. The best performing PRSs", 
        "should be among the ones showing higher PRISMA adjusted R<sup>2</sup>", 
        "values along with statistical significance.")
    cat("</div>")
    cat("<div>")
    print(prismaOut$results[[m]]$reportData$plots$ar2snp)
    cat("</div>")
    
    cat("<div class='figure-hint'>")
    cat("The following figure depicts the mean PRS R<sup>2</sup> in",
        "relationship with the number of SNPs in the PRS. Lower frequencies",
        "mean that SNPs appear in the PRS a few times over ",args$niter,
        " PRISMA iterations. The horizontal axis depicts the number of SNPs in",
        "log<sub>10</sub> scale.")
    cat("</div>")
    cat("<div>")
    print(prismaOut$results[[m]]$reportData$plots$r2fr)
    cat("</div>")
}
```

```{r cv_basic_pct, fig.width=14, fig.height=8, eval=TRUE}
if (!is.null(cvMetricsOut)) {
    pandoc.header("Cross validation {.tabset .tabset-pills}",level=1)
    
    cat("The cross-validation section summarizes the performance and",
        "potential validity of the PRS by iteratively fitting regression",
        "models using the examined phenotype as response variable and the",
        "rest of the covariates (other phenotypes and/or principal components)",
        "with (<strong>full</strong>) and without (<strong>reduced</strong>)",
        "the PRS as covariate. The models are iteratively fitted by excluding",
        "each time a percentage of the samples and using these samples to",
        "then assess the performance of the model with various metrics. These",
        "metrics are:")
    cat("<ul>")
    cat("<li>Model fit as depicted by the R<sup>2</sup> coefficient",
        "(<strong>PRS R<sup>2</sup></strong>).</li>")
    cat("<li>Root Mean Square Error (<strong>RMSE</strong>). The RMSE is the",
        "standard deviation of the residuals (prediction errors). The",
        "predictions are performed by applying the model to the leave-out",
        "samples and RMSE measures how concentrated the data are around the",
        "line of best fit and is robust to outliers.</li>")
    cat("<li>Mean Absolute Error (<strong>MAE</strong>). The MAE is a measure",
        "of errors between paired observations (here original values and",
        "predictions) and is simply the mean difference between observed",
        "and predicted phenotype values.</li>")
    cat("<li>Squared correlation coefficient (R<sup>2</sup>) of the observed",
        "versus the predicted phenotype values.</li>")
    cat("<li>Correlation coefficient (R<sup>2</sup>) of the observed versus",
        "the predicted phenotype values.</li>")
    cat("</ul>")
    
    for (m in names(prismaOut$results)) {
        if (!is.null(cvMetricsOut[[m]])) {
            pandoc.header(paste0(msgs$stat[[m]]," {.tabset .tabset-pills}"),
                level=2)
            
            cat(
                "<div>Each of the following tabs contains evaluation plots ",
                "produced after model cross-validations with (<strong>full",
                "</strong>) or without (<strong>reduced</strong>) the PRS ",
                "across",nrow(cvMetricsOut[[m]][[1]][[1]]),"cross-validation ",
                "iterations. The cross-validation iterations are performed ",
                "with different percentages of withheld samples, and namely, ",
                "leaving out ",
                paste(paste(100*as.numeric(names(cvMetricsOut[[m]][[1]])),"%",
                    sep=""),collapse=", "),
                " of the samples. Each time, the rest of the samples are used ",
                "to build the regression model including or excluding the PRS.",
                " The following model evaluation metrics are collected:</div>",
                sep=""
            )
            
            # Init plots list
            cvPlots <- vector("list",length(theCvMetrics))
            names(cvPlots) <- names(theCvMetrics)
            
            for (e in names(theCvMetrics)) {
                # Create CV plots
                cvPlots <- plotCvMetrics(cvMetricsOut[[m]],theCvMetrics[[e]],
                    silent=TRUE)
                
                pandoc.header(paste0(e," {.tabset .tabset-pills}"),level=3)
                
                # What we see
                cat(cvMetExpl[[e]])
                
                # Create the selectors
                cat("<hr style='height: 2px;'>")
                cat("<div class='selector-label' style='margin-bottom:20px;'>",
                "Select a PRS candidate to display CV metrics for ",sep="")
                # Create a selector for each PRS candidate
                lens <- names(cvSumDf[[m]][[e]])
                itemsLen <- paste("<option value=",
                    paste(names(cvSumDf[[m]][[e]]),m,theCvMetrics[[e]],sep="_"),
                    ">",paste(names(cvSumDf[[m]][[e]]),"SNPs"),"</option>",
                    sep="")
                cat("<select class=\"prscv_table_selector\" id=\"",
                    paste("select",m,theCvMetrics[[e]],sep="_"),"\">",itemsLen,
                        "</select></div>",sep="")
                
                # Summary tables
                for (nn in names(cvSumDf[[m]][[e]])) {
                    M <- cvSumDf[[m]][[e]][[nn]]
                    M <- data.frame('LO%'=paste0(100*as.numeric(rownames(M)),
                        "%"),M,check.names=FALSE)
                    rownames(M) <- NULL
                    cat("<div class='cvtable-wrapper-",m,"_",theCvMetrics[[e]],
                    "' id='cvtable_",nn,"_",m,"_",theCvMetrics[[e]],"' ",
                    "style='display:none;'>",sep="")
                    M %>% 
                        kbl(escape=FALSE) %>%
                        kable_styling(bootstrap_options=c("striped","hover",
                            "condensed","responsive"),font_size=11) %>% 
                        htmltools::HTML() %>% 
                        print()
                    cat("</div>")
                }
                
                # A plot across leave-out percentages, if performed
                if (length(cvMetricsOut[[m]][[1]]) > 1) {
                    cat("<div class='figure-hint'>")
                    cat("The following plot depicts the average values of",e,
                        "and their standard deviations across",
                        length(cvMetricsOut[[m]][[1]]),
                        "different leave-out sample percentages. Its purpose",
                        "is to evaluate the stability of the model based on", 
                        "different number of samples used for fit and",
                        "prediction, with or without the PRS. Each panel",
                        "corresponds to a PRS candidate with its number of",
                        "SNPs displayed on the top. A good PRS should also", 
                        "result in a stable model.")
                    cat("</div>")
                    cat("<div>")
                    print(cvPlots$cvBased)
                    #subchunkify(cvPlots$cvBased,14,14)
                    cat("</div>")
                }
                
                # Detailed plots for each CV percentage
                for (pct in names(cvPlots$prsBased)) {
                    pctp <- paste0(100*as.numeric(pct),"%")
                    pandoc.header(pctp,level=4)
                    
                    cat("<div class='figure-hint'>")
                    cat("The following plot depicts the average values of",e,
                        "and their standard deviations across the PRS",
                        "candidates while leaving out",pctp,"samples. The",
                        "best candidate should show",metDir[[e]],e)
                    cat("</div>")
                    
                    cat("<div>")
                    #subchunkify(cvPlots$prsBased[[pct]],8,6)
                    print(cvPlots$prsBased[[pct]])
                    cat("</div>")
                }
            }
            
        }
    }
}

```

# Results


```{r dt_hack, include=FALSE}
DT::datatable(NULL)
```

```{r results_basic_prs}
# Create the selectors
cat("<hr style='height: 2px;'>")
cat("<p>The following tables present a summary of the respective PRS",
    "cadidates. Use the selectors below to inspect the tables and the links",
    "below the tables to download the respective PRS candidate(s) as well as", 
    "evidence for the markers derived from GWAS catalog. The files are in", 
    "Excel format.</p>")
cat("The PRS candidate tables contain the following columns:")
cat("<div style='font-size: 0.9em'><ul>")
cat("<li><em>chrom</em>: The human chromosome the SNP is located</li>")
cat("<li><em>position</em>: The chromosomal start position of the SNP</li>")
cat("<li><em>variant_id</em>: The dbSNP ID of the marker where available</li>")
cat("<li><em>risk_allele</em>: The risk (minor) allele as defined in the",
    "input PLINK fileset</li>")
cat("<li><em>ref_allele</em>: The major allele as defined in the input PLINK",
    "fileset</li>")
cat("<li><em>effect_weight</em>: The effect of each SNP as derived by the",
    "respective GWA test and summarized from the",args$niter,"iterations</li>")
cat("<li><em>freq</em>: The number of times the marker appears in the PRS over",
    args$niter,"iterations</li>")
cat("</ul></div>")

cat("The GWAS Catalog association tables contain the following columns:")
cat("<div style='font-size: 0.9em'><ul>")
cat("<li><em>variant_id</em>: The dbSNP ID of the marker where available</li>")
cat("<li><em>risk_allele</em>: The risk (minor) allele as recorded in GWAS",
    "Catalog</li>")
cat("<li><em>association_id</em>: The GWAS Catalog association id</li>")
cat("<li><em>gene_name</em>: The associated gene name as recorded in GWAS",
    "Catalog</li>")
cat("<li><em>trait</em>: The associated trait as recorded in GWAS Catalog</li>")
cat("<li><em>freq</em>: The number of times the marker appears in the PRS over",
    args$niter,"iterations</li>")
cat("</ul></div>")

cat("<hr style='height: 2px;'>")
cat("<div class=\"row\"><div class=\"col-sm-6\">")
cat("<div class=\"selector-label\">Select a GWA test to display ",
    "PRS tables for </div>",sep="")
# Create a selector for each GWA algorithm
algs <- unlist(msgs$stat[names(prismaOut$results)],use.names=FALSE)
itemsGwa <- paste("<option value=",names(prismaOut$results),">",algs,
    "</option>",sep="")
cat("<select id=\"prsgwa_table_selector\">",itemsGwa,"</select>",sep="")
cat("</div><div class=\"col-sm-6\">")
# Create the sub-selectors for each PRS candidate - start as hidden
for (m in names(prismaOut$results)) {
    items <- paste("<option value=",
        paste(m,names(prismaOut$results[[m]]$candidates),sep="_"),">",
        paste(names(prismaOut$results[[m]]$candidates),"SNPs"),"</option>",
        sep="")
    cat("<div id=\"prstable_",m,"_label\" class=\"selector-label",
        " prs_selector_helper\">Select a PRS candidate to display</div>",sep="")
    cat("<select id=\"prstable_",m,"_selector\" class=\"prs_selector_helper ",
        "main_prs_selector\">",items,"</select>",sep="")
}
cat("</div></div><br/>")

# The tables?
for (m in names(prismaOut$results)) {
    for (n in names(prismaOut$results[[m]]$candidates)) {
        snps <- prismaOut$results[[m]]$candidates[[n]]
        snps <- snps[,-c(8,9)]
        snps[,"effect_weight"] <- round(snps[,"effect_weight"],5)
        snps[,"variant_id"] <- paste("<a href='https://www.ncbi.nlm.nih.gov",
            "/snp/",snps[,"variant_id"],"' target='_blank'>",
            snps[,"variant_id"],"</a>",sep="")
        na <- is.na(snps$locus_name)
        snps[!na,"locus_name"] <- paste("<a href='https://www.genenames.org/",
            "data/gene-symbol-report/#!/symbol/",snps[!na,"locus_name"],
            "' target='_blank'>",snps[!na,"locus_name"],"</a>",sep="")
        colnames(snps)[1] <- "chrom"
        colnames(snps)[5] <- "ref_allele"
        
        # Display less SNPs, avoid annoying warning messages, since all SNPs are
        # available for download
        topSnps <- FALSE
        if (nrow(snps) > 1000) {
            snps <- snps[seq_len(1000),]
            topSnps <- TRUE
        }
        
        cat("<div class='prstable-helper-wrapper' id='prstable_",m,"_",n,"'>",
            sep="")
        
        cat("<h4>PRS with ",n," SNPs derived with ",msgs$stat[[m]]," effect ",
            " estimations</h4>",sep="")
        
        cat("<div class='row'><div class='col-sm-6'>")
        cat("<span style='text-align: center'><h5>Correlation of PRS with ",
            "phenotype</h5></span>")
        print(prismaOut$results[[m]]$reportData$plots$prsScatter[[n]])
        cat("</div><div class='col-sm-6'>")
        cat("<span style='text-align: center'><h5>PRS distribution</h5></span>")
        print(prismaOut$results[[m]]$reportData$plots$prsHist[[n]])
        cat("</div></div>")
        
        cat("<h4>SNPs and effect weights in PRS</h4>")
        
        if (topSnps)
            cat("<div style='font-weight:600;font-size:0.8em;color:#E30101;'>",
                "The following table presents only the first 1000 SNPs in the ",
                "PRS. Use the download link below to retrieve the full ",
                "PRS.</div>")
        
        cat("<div style='overflow-y: auto;'>")
        thePrsTable <- DT::datatable(
            snps,
            rownames=FALSE,
            width="95%",
            height=460,
            style="bootstrap",
            class="table-condensed table-striped table-responsive",
            elementId=paste("prstable_",m,"_",n,"_",sep=""),
            escape=FALSE,
            options=list(
                scrollX=TRUE,
                scrollY=TRUE,
                dom='lfrtip'
            )
        )
        print(htmltools::tagList(tags$div(class="prstable-container",
            thePrsTable)))
        cat("</div>")
        
        cat("<div class='figure-hint link-box'>")
        cat("<strong><a href='candidates/",m,"_candidates.xlsx'",
            " download>Download</a> the PRS candidates list.</strong><br/>",
            sep="")
        cat("</div></div>")
        
        if (!is.null(lookupOut[[m]])) {
            lkp <- lookupOut[[m]][[n]]
            if (nrow(lkp) > 0) {
                lkp[,"variant_id"] <- paste("<a href='https://www.ncbi.nlm.nih",
                    ".gov/snp/",lkp[,"variant_id"],"' target='_blank'>",
                    lkp[,"variant_id"],"</a>",sep="")
                lkp[,"trait"] <- paste("<a href='http://www.ebi.ac.uk/efo/",
                    lkp[,"efo_id"],"' target='_blank'>",lkp[,"trait"],"</a>",
                    sep="")
            }
            lkp <- lkp[,-5]
            
            cat("<div class='prstable-helper-wrapper' id='efotable_",m,"_",n,
                "' style='margin-top: 50px;'>",sep="")
            cat("<h4>GWAS Catalog associations with SNPs in PRS</h4>")
            cat("<div style='overflow-y: auto;'>")
            theEfoTable <- DT::datatable(
                lkp,
                rownames=FALSE,
                width="95%",
                height=460,
                style="bootstrap",
                class="table-condensed table-striped table-responsive",
                elementId=paste("efotable_",m,"_",n,"_",sep=""),
                escape=FALSE,
                options=list(
                    scrollX=TRUE,
                    scrollY=TRUE,
                    dom='lfrtip'
                )
            )        
            print(htmltools::tagList(tags$div(class="efotable-container",
                theEfoTable)))
            cat("</div>")
            
            cat("<div class='figure-hint link-box'>")
            cat("<strong><a href='candidates/",m,"_evidence.xlsx'",
                " download>Download</a> the evidence list.</strong><br/>",
                sep="")
            cat("</div>")
            
            cat("</div>")
        }
        
    }
}
```

```{js detable_render}
var pids = $('.prstable-helper-wrapper').map(function(index) {
    return(this.id);
});
$('.prstable-helper-wrapper').hide(0);
$('#'+pids[0]).show();
$('#'+pids[1]).show();

// The only way to align columns...
var timer = setInterval(function() {
    if ($('#'+pids[0]+"_").find("table").length > 0) {
        if ($('#'+pids[0]+"_").find("table")[1].id !== undefined) {
            var pid = $('#'+pids[0]+"_").find("table")[1].id;
            var eid = $('#'+pids[1]+"_").find("table")[1].id;
            $('#'+pid).DataTable().columns.adjust();
            $('#'+eid).DataTable().columns.adjust();
            clearInterval(timer);
        }
    }
},100);
```

```{js bind_event_changes}
$("#prsgwa_table_selector").on('change',function() {
    var name = this.value;
    $(".prs_selector_helper").hide(0);
    $("#prstable_"+name+"_label").show();
    $("#prstable_"+name+"_selector").show();
    
    // Trigger the change of the other selector too
    var parts = this.value.split("_");
    
    $("#prstable_"+parts[0]+"_selector").trigger("change");
    //$(".main_prs_selector").trigger("change");
});

$(".main_prs_selector").on('change',function() {
    var name = this.value;
    console.log(name);
    
    $(".prstable-helper-wrapper").hide(0);
    $("#prstable_"+name).show();
    $("#efotable_"+name).show();
    
    // Hack to make the table visible (htmlwidget known issue)
    window.dispatchEvent(new Event('resize'));
    // Hack to align the column headers
    //var id = $('#prstable_'+name+"_").find("table")[1].id;
    //$('#'+id).DataTable().columns.adjust();
    //var id = $('#efotable_'+name+"_").find("table")[1].id;
    //$('#'+id).DataTable().columns.adjust();
});

$(".prscv_table_selector").on('change',function() {
    var parts = this.value.split("_");
    var part12 = parts[1] + "_" + parts[2];
    $(".cvtable-wrapper-"+part12).hide(0);
    $("#cvtable_"+this.value).show();
});

$("#prsgwa_table_selector").trigger('change');
```

```{r js_init_cv_tables}
cat("<script>\n")
for (m in names(prismaOut$results)) {
    if (!is.null(cvMetricsOut[[m]])) {
        for (e in names(theCvMetrics)) {
            nn <- names(cvSumDf[[m]][[e]][1])
            sid <- paste(nn,m,theCvMetrics[[e]],sep="_")
            cat('$("#cvtable_',sid,'").show()\n',sep="");
        }
    }
}
cat("</script>")
```
