<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Introduction to the prisma package • prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the prisma package">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to the prisma package</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/01-introduction.Rmd" class="external-link"><code>vignettes/01-introduction.Rmd</code></a></small>
      <div class="d-none name"><code>01-introduction.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The rise of modern genotyping technologies such as Next Generation Sequencing (Whole Exome and Whole Genome Sequencing) as well as the continuous thriving of older genotyping technologies such as genotyping DNA microarrays have allowed an unprecedented collection of massive, large-cohort and population-level genetic data spanning a variety of application, from medical research to consumer genomics. At the same time, several perspective and sometimes country-level studies are reaching maturity levels regarding the amount and quality of data collected at various levels such as anthropometric, phenotypic, biomarker and genetic levels, and some times other molecular -omics layers such as gene expression. The aforementioned facts have allowed not only for more accurate and versatile Genome-Wide Association Studies to identify the potential genetic cause for many simple or more complex genetic traits but also combine phenotypic and genotypic data in unique ways to derive multi-way genetic signatures that characterize individuals with respect to several anthropometric features as well as these traits and biomarker levels. These genetic signatures have a seamingly endless portfolio of applications, spanning the characterization of the predisposition of an individual to general healthstyle issues such as obesity and up to estimating the risk to several diseases like cancer and other potential life-threatening conditions including cardiovascular disease and kidney malfunction. They are also used to characterize less life-threatening conditions but which deteriorate the general well-being such as psychological disease and anomalies of the central nervous system.</p>
<p>The aforementioned biotechnological thriving as well as the increased complete data sources availability (coupled genotypic and phenotypic records for large cohorts) are gradually allowing the creation and exploration of Polygenic Risk Scores as risk predictors and evaluators for a variety of disease and healthstyle conditions. Polygenic Risk Scores (PRS) are (usually) simple genotype scoring schemes which take into account many variants of an individual and based on the calculated score, attempt to assess a heritable risk of developing/expressing a particular phenotype or combination of phenotypes. These phenotypes can be disease-related phenotypes such as cancer, cardiovascular disease and obesity, or plain lifestyle traits such as endurance in sports or effects of caffeine. They are different from the Genetic Risk Scores (GRS) in the sense that while GRSs are compiled from a (usually) limited number of well characterized variants related to a particular phenotype across the literature with strong statistical evidence, PRSs are more relaxed regarding statistical power and comprise a usually much number of variants accounting for the additive genetic risk of an individual given his/hers total genetic profile.</p>
<p>PRS and GRS are essentially variables constructed based on the (weighted) sum of genotypes (usually coded as 0, 1, 2 according to the alleles of a genotype) and comprise covariates in regression models that attempt to explain observed genotypic/phenotypic relationships. While this approach seems over-simplified, nevertheless it has been shown in numerous studies that genetic scores are able to explain the aforementioned variabilities and most times even outperform complex Machine Learning approaches. Then, after a succesful model construction, a distribution of PRS is constructed based on the ppopulation used to extract and calculate the PRS, and based on this distribution, the risk of new samples can be assessed according to the position of the new PRS in the reference distribution. The process of extracting a PRS is not very complex itself and includes the main following steps:</p>
<ul>
<li>Derivation of summary statistics (regression coefficients/Odds Ratios, p-values etc.) for each variant from a major and representative population study (often a GWAS) recording also the phenotypes of interest.</li>
<li>Filtering of these variants according to their potential informative redundancy (Linkage Disequilbrium), content with respect to the population (Hardy-Weinberg Equilibrium) and statistical significance/power to contribute to the phenotype under investigation (effects as assessed from regression coefficients, p-values). This process is often refered as <em>Clumping + Thresholding</em> and although popular is not the sole method to reduce noise while extracting a PRS. Another often complementary process is <em>penalized regression</em> or <em>lasso</em>.</li>
<li>Construction of the PRS: in this step, the variants (usually SNPs) having passed the filtering procedure are used to calculate the PRS. There are various ways to calculate a PRS, the simplest one being just the sum of the genotypes (0, 1, 2) but the most popular seem to be the <em>weighted</em> PRS where the genotype of each marker (SNP, variant) participating in the PRS is mutliplied by its inferred effect given summary statistics e.g. from a GWAS. Multiple PRSs can be constructed and tested given different thresholds of p-values and/or other scores. The PRSs are constructed based on the genotypes of samples other than the ones used to derive the summary statistics, otherwise, strong overfitting effects are observed.</li>
<li>Construction and evaluation of multiple regression models: in this step many regression models are fitted and evaluated with the PRS candidates as covariates with the aim to select the PRS providing the largest explanatory power in terms of some metric such as the R<sup>2</sup> of the model and the significance of the PRS as a covariate (p-value). This is the strategy followed e.g. by <a href="https://www.prsice.info/" class="external-link">PRSice-2</a>. Other strategies exist such as selecting one PRS without repetitions or using more SNPs in the PRS by effect-shrinking methods (this is what <a href="https://github.com/tshmak/lassosum" class="external-link">lassosum</a> does) using penalized regression and LASSO techniques.</li>
<li>Selection and evaluation of the final PRS candidates with the goal of selecting a final PRS for the phenotype under investigation. This evaluation can be done based on split-and-test cross-validations on the initital dataset. In any case, the final PRS is subjected to slight changes if the initial dataset is changed (e.g. new samples added).</li>
</ul>
</div>
<div class="section level2">
<h2 id="the-prisma-package">The PRISMA package<a class="anchor" aria-label="anchor" href="#the-prisma-package"></a>
</h2>
<div class="section level3">
<h3 id="general-introduction">General introduction<a class="anchor" aria-label="anchor" href="#general-introduction"></a>
</h3>
<p>The PRISMA package is designed to provide a one-stop shop for all the PRS extraction and evaluation steps described above. On top of that, it implements a linear optimization methodology which combines the regression effects from multiple statistical tests for GWAS-like association tests and offers a variety of PRS evaluation metrics and scores. Furthermore, PRISMA implements a semi- automated methodology for selecting and suggesting the best performing PRS candidates which are extracted based on an iterative procedure where the input dataset is split in training (for summary statistics) and test (for PRS extraction) subsets and the resulting PRS candidates are integrated and evaluated resulting in the best candidate sets. The findings are finally collected and described in an interactive report which also includes lookup for the selected SNPs in the EBI GWAS Catalog. Finally, PRISMA offers several side functions such as facilities for merging datasets (e.g. two datasets that complement each other or datasets coming from different array platforms), input/output functions and also SNP imputation based on reference panels by wrapping <a href="https://mathgen.stats.ox.ac.uk/impute/impute_v2.html" class="external-link">IMPUTE2</a>.</p>
<p>Although it is generally assumed that any PRS extraction and evaluation is and should be based on summary statistics calculalated from GWAS and based on sufficiently large and as homogenous as possible cohorts this may not be always possible. Examples would include smaller cohorts that are used to study a rare disease (e.g. pediatric nephropathies) or to examine traits in smaller groups such as isolated populations or athletic groups. In such cases, any PRS extraction process would result in rather unstable PRSs in terms of SNP content. In fact, such instability may also show up in cases where sufficiently large populations have been used to estimate summary statistics. PRISMA tries to provide a solution in such cases by offering a resampling framework where PRS candidates are partly stabilized based on an ensemble technique where markers are gathered to PRSs based on aggregation from multiple PRS extraction runs and smart usage of state-of-the art algorithms such as PRSice2 and lassosum, coupled by a semi-automated selection of best candidates based on metrics such as the R<sup>2</sup> of regression models including the PRS as covariate.</p>
</div>
<div class="section level3">
<h3 id="summary-statistics-extraction-and-gwas-analysis">Summary statistics extraction and GWAS analysis<a class="anchor" aria-label="anchor" href="#summary-statistics-extraction-and-gwas-analysis"></a>
</h3>
<p>PRISMA supports five methods for estimating SNP effects for single-trait associations. More specifically:</p>
<ol style="list-style-type: decimal">
<li>General Linear Models (GLM), through the respective R functions. Each SNP is regressed against the phenotype and the selected covariates. The SNP effects and p-value are reported.</li>
<li>Ridge regression through the R package <a href="https://cran.r-project.org/web/packages/rrBLUP/index.html" class="external-link">rrBLUP</a>. This package deploys penalized regression for shrinking the effects of less informative SNPs in relation with the phenotype.</li>
<li>Single-trait association tests with the <a href="https://biometris.github.io/statgenGWAS/" class="external-link">statgenGWAS</a> R package. The association is based on linear regression and the incorporation of an internally calculated kinship matrix.</li>
<li>Frequentist association tests with <a href="https://www.well.ox.ac.uk/~gav/snptest/" class="external-link">SNPTEST</a> (Oxford). Each SNP is regressed against the phenotype and the selected covariates using an additive model (all other SNPTEST options are also supported).</li>
<li>Association tests with <a href="https://www.cog-genomics.org/plink/" class="external-link">PLINK</a> (linear regression with covariates or simple X<sup>2</sup> association tests if no covariates are present).</li>
</ol>
</div>
<div class="section level3">
<h3 id="de-novo-prs-extraction">De <em>novo</em> PRS extraction<a class="anchor" aria-label="anchor" href="#de-novo-prs-extraction"></a>
</h3>
<p>PRISMA currently supports two methods for extracting PRSs. More specifically:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://www.prsice.info/" class="external-link">PRSice2</a>, which operates based on the PLINK framework and automates a PRS extraction process based on clumping and thresholding followed by weighted PRS construction and evaluation with linear regression.</li>
<li>
<a href="https://github.com/tshmak/lassosum" class="external-link">lassosum</a>, which utilizes penalized (LASSO) regression to shrink the effects of less informative variants without completely removing them from the PRS.</li>
</ol>
<p>By default, PRISMA uses both PRSice2 and lassosum and offers two ways of integrating their results: i. by intersecting the PRS candidates from each algorithm ii. by considering the respective union.</p>
<p>In all cases, the effects from GWA tests are considered as the weights of each SNP in the constructed PRS.</p>
</div>
<div class="section level3">
<h3 id="prisma-pipelines-for-prs-extraction">PRISMA pipelines for PRS extraction<a class="anchor" aria-label="anchor" href="#prisma-pipelines-for-prs-extraction"></a>
</h3>
<p>Starting from a set of phenotypic and genotypic data assembled in a PRISMA <code>GWASExperiment</code> object and/or a PRS from an external resource (e.g. PGS Catalog) which can be used to subset the former, PRISMA supports four pipelines outlined below. Prior to each pipeline, a basic filtering of the genotypic data is performed across SNPs and samples, and missing genotype imputation is performed followed by PCA (optionally) to extract covariates characterizing the whole available population. These are stored in the original object, to be used towards the final steps of model evaluation.</p>
<div class="section level4">
<h4 id="de-novo-prs-extraction-pipeline">De novo PRS extraction pipeline<a class="anchor" aria-label="anchor" href="#de-novo-prs-extraction-pipeline"></a>
</h4>
<p>This is the main PRS candidate extraction pipeline and performs the following steps: 1. Split of the input data object to a usually larger training set and a smaller independent test set. 2. De novo quality control on the training and test sets as well as optional PCA in the training set to capture local population structure to be used as covariates in the subsequent GWA tests. 3. Execution of GWA tests against a phenotype using one or more of the supported methods to extract summary statistics (SNP effects and p-values). 4. Execution of one or more PRS extraction algorithms to derive PRS candidate SNPs.</p>
<p>Steps (1) – (4) are executed N times with different training and test sets to capture as much variation in the selection process as possible. Subsequently:</p>
<ol start="5" style="list-style-type: decimal">
<li>The candidate SNPs from PRSs extracted at each iteration are aggregated and the frequency of their appearance is recorded. At the same time, new effects are calculated either by averaging the SNP effects from each iteration, a <a href="https://www.biorxiv.org/content/10.1101/133785v8.full.pdf" class="external-link">statistically valid process</a> which generally corrects for overfitting, or by weighted averaging of the effects based on the performance of PRSs constructed at each iteration (as measured by the adjusted PRS R<sup>2</sup> returned by PRSice2).</li>
</ol>
</div>
<div class="section level4">
<h4 id="de-novo-prs-evaluation-pipeline">De novo PRS evaluation pipeline<a class="anchor" aria-label="anchor" href="#de-novo-prs-evaluation-pipeline"></a>
</h4>
<p>This is an evaluation of a de novo extracted PRS pipeline and performs the following steps:</p>
<ol style="list-style-type: decimal">
<li>Create sets of PRS candidates from step (5) of the de novo PRS extraction pipeline based on:</li>
</ol>
<ol style="list-style-type: lower-alpha">
<li>Fixed quantiles of the frequency distribution (e.g. 10%, 20%, 30%, 40%, 50%, 60%, 75%, 80%, 90%, 95%, 99%).</li>
<li>A stepwise set creation process based on frequency of appearance, i.e. each set contains SNPs that appear 5, 6, … up to the maximum number of appearances.</li>
</ol>
<ol start="2" style="list-style-type: decimal">
<li>For each set created at (1), split de novo the initial data object in a training and test set. Using the training test, reperform GWA tests to extract summary statistics (the averaged effects are not used) and use PRSice2 in <em>PRS application</em> mode to assess the PRS sets of (1) on the test set with the new summary statistics.</li>
<li>Evaluate each PRS application using the adjusted PRS R2 coefficient of PRSice2.</li>
</ol>
<p>Steps (2) and (3) are repeated M times (ideally M=N iterations). This pipeline includes heavy input/output in the form of PLINK files and is quite slow. In some cases it may be more accurate though, although it does not provide a straightforward way of calculating the final SNP effects for the PRS. Finally, this pipeline may be more appropriate for evaluating external PGS scores, like scores retrieved from PGS Catalog.</p>
</div>
<div class="section level4">
<h4 id="prs-evaluation-pipeline-based-on-the-original-traintest-split-with-prsice2">PRS evaluation pipeline based on the original train/test split with PRSice2<a class="anchor" aria-label="anchor" href="#prs-evaluation-pipeline-based-on-the-original-traintest-split-with-prsice2"></a>
</h4>
<p>This pipeline is similar to the de novo PRS evaluation pipeline but it uses the initial train/test split described above for PRS evaluation. The steps are similar to those described above and the evaluation is performed with the PRSice2 framework, limiting somehow the available model evaluation metrics. Also, averaged SNP effects are used.</p>
</div>
<div class="section level4">
<h4 id="prs-evaluation-pipeline-based-on-the-original-traintest-split-with-r">PRS evaluation pipeline based on the original train/test split with R<a class="anchor" aria-label="anchor" href="#prs-evaluation-pipeline-based-on-the-original-traintest-split-with-r"></a>
</h4>
<p>Again, this pipeline is similar to the one described above but this time evaluation is done with native R functions (manual PRS construction outside PRSice2 followed by GLM regressions). This is the fastest evaluation pipeline and allows for the collection of many additional metrics.</p>
</div>
</div>
<div class="section level3">
<h3 id="prs-selection-process">PRS selection process<a class="anchor" aria-label="anchor" href="#prs-selection-process"></a>
</h3>
<p>The final PRS selection out of the several candidates is performed in a two-step semi-automatic process which involves graphical inspection of evaluation plots and also suggestion of the best PRSs using simple mathematical functions on the evaluation plots.</p>
<div class="section level4">
<h4 id="the-prisma-adjusted-prs-r2">The PRISMA adjusted PRS R<sup>2</sup><a class="anchor" aria-label="anchor" href="#the-prisma-adjusted-prs-r2"></a>
</h4>
<p>Although many cases in related work are based on the R2 coefficient (or adjusted versions such as the one suggested in PRSice2) to explain the phenotypic variance percentages explained by the PRS, it is often desired that a PRS may not include huge number of SNPs for its construction, for several reasons, a few including:</p>
<ol style="list-style-type: lower-roman">
<li>Potential overfitting by including many SNPs which may be heavily dependent on specific populations used to derive summary statistics, thus presenting problematic generalization when applied to slightly different populations.</li>
<li>The ability to use other non-high-throughput techniques to derive genotypes, such as qPCR, which would benefit by a smaller number of markers for genotyping.</li>
<li>Easier downstream investigation and tracking of the SNP markers included in the PRS.</li>
<li>Potential robustness when further tuning a predictive regression model, for example including or excluding covariates.</li>
</ol>
<p>For these reasons, we introduce the PRISMA adjusted PRS R2 calculated using the formula:</p>
<p><span class="math display">\[R_{P}^{2} = \sqrt[2]{\frac{R_{PRS}^{2}}{log(N)}} \]</span></p>
<p>where <span class="math inline">\(R_{PRS}^{2}\)</span> is the adjusted R<sup>2</sup> explaining the PRS contribution (R<sup>2</sup> of the full model including the PRS – R<sup>2</sup> of the reduced model including only the covariates against the phenotype) and <em>N</em> is the number of SNPs included in the PRS.</p>
<p>Essentially, <span class="math inline">\(R_{P}^{2}\)</span> penalizes <span class="math inline">\(R_{PRS}^{2}\)</span> according to the number of SNPs <em>N</em>, while maintaining scale and monotonic pattern by using the square root. In practice, it does not seriously differ from using only the <span class="math inline">\(R_{PRS}^{2}\)</span> for selection of the optimal number of SNPs, but it is very useful when controlling the number of SNPs while maintaining accuracy is desirable.</p>
</div>
<div class="section level4">
<h4 id="graphical-evaluations">Graphical evaluations<a class="anchor" aria-label="anchor" href="#graphical-evaluations"></a>
</h4>
<p>After executing a possible evaluation pipeline, the best PRS is firstly evaluated graphically by:</p>
<ol style="list-style-type: lower-roman">
<li>A bar plot of average adjusted PRS R2 ± standard deviations for each SNP frequency cutoff where it can be checked if there is a cutoff increasing the R2 as compared to the baseline (R2 from the de novo PRS extraction iterations).</li>
<li>A dot plot of the PRISMA adjusted PRS R2 R_P^2, where it can be inspected where the adjusted R2 is potentially maximized.</li>
<li>A dot plot of SNP frequencies.</li>
</ol>
</div>
<div class="section level4">
<h4 id="automatic-prs-selection">Automatic PRS selection<a class="anchor" aria-label="anchor" href="#automatic-prs-selection"></a>
</h4>
<p>The number of SNPs that will finally be used to build one or more candidate PRS is determined with two possible stepwise SNP inclusion methods:</p>
<ol style="list-style-type: lower-roman">
<li>Using the Elbow method, which essentially detects a cutoff in the number of SNPs in the PRS where the R2 metrics do not significantly increase after including more SNPs in the PRS. This method can be applied when the R<sup>2</sup> distribution is mostly monotonic.</li>
<li>By detecting local maxima in the R2 distribution, corresponding to local maximizations of the latter at specific number of SNPs. The main PRS includes the SNPs which maximize the <span class="math inline">\(R_{PRS}^{2}\)</span> or the <span class="math inline">\(R_{P}^{2}\)</span> so as to compensate for the number of SNPs in the PRS (default behavior). This option returns also SNP sets at other local maxima of the R<sup>2</sup> distribution.</li>
</ol>
<p>A similar approach like (ii) above regarding the maximization of R2 is being used in a recent work presented in <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02479-9" class="external-link">this</a> study.</p>
</div>
</div>
<div class="section level3">
<h3 id="prs-evaluation-metrics">PRS evaluation metrics<a class="anchor" aria-label="anchor" href="#prs-evaluation-metrics"></a>
</h3>
<p>The following metrics are used for the evaluation of PRS candidates and during their iterative selection:</p>
<ul>
<li>Full regression model R<sup>2</sup>: the R<sup>2</sup> of a model including the PRS.</li>
<li>Reduced model R<sup>2</sup>: the R<sup>2</sup> of a model without the PRS (only covariates against the phenotype under interrogation).</li>
<li>Adjusted PRS R<sup>2</sup>: the PRSice2 adjusted PRS R<sup>2</sup>.</li>
<li>Full and reduced model Root Mean Squared Error (RMSE).</li>
<li>Full and reduced model Mean Absolute Error (MAE).</li>
<li>Statistical significance of the average differences of the above using the Student’s t-test, the Wilcoxon rank test and an empirical bootstrap p-value.</li>
<li>All of these metrics are collected during the evaluation pipeline with vanilla R and also in cross-validation runs based on regression models with the selected PRSs and the total available dataset.</li>
</ul>
</div>
<div class="section level3">
<h3 id="prs-retrieval-from-external-resources">PRS retrieval from external resources<a class="anchor" aria-label="anchor" href="#prs-retrieval-from-external-resources"></a>
</h3>
<p>PRISMA supports the retrieval of candidate PRSs or candidate PRS components (SNPs) for a trait from the following resources:</p>
<ul>
<li>The <a href="https://www.pgscatalog.org/" class="external-link">EBI PRS Catalog</a>
</li>
<li>The <a href="https://www.ebi.ac.uk/gwas/" class="external-link">EBI GWAS Catalog</a>
</li>
</ul>
<p>Retrieval of data regarding specific PRS also from <a href="https://impute.me/" class="external-link">impute.me</a> is being currently tested. The data retrieved from online resources are being harmonized into a common format including the recording of existing PRS coefficients, statistical significance, related loci and GWAS association scores (p-values). They are also enriched with the following data:</p>
<ul>
<li>Location assignment when only dbSNP (rs) ids are only provided.</li>
<li>dbSNP id assignment when only location is provided.</li>
<li>Locus (gene) assignment when not provided and if possible (i.e. SNP is not intronic).</li>
</ul>
<p>The resulting structure can be used with one of the PRISMA pipelines to evaluate the retrieved PRS (i.e. check if all markers are needed with local data) and reconstruct a PRS.</p>
</div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>The commands below can be executed within R's command line interface and use the toy data accompanying the package. For many more details regarding installation, usage and package presentation, you should also look at the other vignettes. Before running these examples, you should have at least installed <a href="https://www.prsice.info/" class="external-link">PRSice2</a> (have also a look at the Installation vignette) which should be visible by R (i.e. included in the system's <code>PATH</code> variable).</p>
<div class="section level4">
<h4 id="load-some-toy-data">Load some toy data<a class="anchor" aria-label="anchor" href="#load-some-toy-data"></a>
</h4>
<p>The toy dataset contains genotypes for chromosomes 18 and 21 from a the case/control described in <a href="https://doi.org/10.1016/j.nut.2018.10.032" class="external-link">Kalafati et al.</a>. The trait for which PRS candidates are sought is the Body-Mass Index (continuous variable).</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmoulos/prisma" class="external-link">prisma</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="apply-some-filters">Apply some filters<a class="anchor" aria-label="anchor" href="#apply-some-filters"></a>
</h4>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="full-pipeline---create-prs-candidates">Full pipeline - create PRS candidates<a class="anchor" aria-label="anchor" href="#full-pipeline---create-prs-candidates"></a>
</h4>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Workspace</span>
<span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d1</span><span class="op">)</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="co"># Complete 4-iteration PRS pipeline - ~1 min to run with 4 cores - no PCA</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">prismaOut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prisma</a></span><span class="op">(</span>
        gwe<span class="op">=</span><span class="va">toy</span>,
        phenotype<span class="op">=</span><span class="va">response</span>,
        covariates<span class="op">=</span><span class="va">covariates</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,
        trainSize<span class="op">=</span><span class="fl">0.8</span>,
        niter<span class="op">=</span><span class="fl">4</span>,
        resolution<span class="op">=</span><span class="st">"frequency"</span>,
        minSnps<span class="op">=</span><span class="fl">2</span>,
        filters<span class="op">=</span><span class="va">filts</span>,
        pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
        imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
        gwaMethods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span>,
        prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        prsWorkspace<span class="op">=</span><span class="va">wspace1</span>,
        cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
        logging<span class="op">=</span><span class="st">"file"</span>,
        output<span class="op">=</span><span class="st">"normal"</span>,
        rc<span class="op">=</span><span class="fl">0.25</span>
    <span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-by-step-pipeline">Step-by-step pipeline<a class="anchor" aria-label="anchor" href="#step-by-step-pipeline"></a>
</h4>
<p>Below, we execute step-by-step the individual steps implemented in the <code>prisma</code> wrapper only for the <code>statgenGWAS</code> GWA testing.</p>
<div class="section level5">
<h5 id="iterate-prs-extraction-over-the-dataset">1. Iterate PRS extraction over the dataset<a class="anchor" aria-label="anchor" href="#iterate-prs-extraction-over-the-dataset"></a>
</h5>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d2</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">dnList</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsPipeline</a></span><span class="op">(</span>
        gwe<span class="op">=</span><span class="va">toy</span>,
        phenotype<span class="op">=</span><span class="va">response</span>,
        covariates<span class="op">=</span><span class="va">covariates</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,
        trainSize<span class="op">=</span><span class="fl">0.8</span>,
        niter<span class="op">=</span><span class="fl">4</span>,
        filters<span class="op">=</span><span class="va">filts</span>,
        pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
        imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
        gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
        prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        prsWorkspace<span class="op">=</span><span class="va">wspace2</span>,
        cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
        logging<span class="op">=</span><span class="st">"file"</span>,
        rc<span class="op">=</span><span class="fl">0.25</span>
    <span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="calculate-the-evaluation-metrics-that-lead-to-prs-selection">2. Calculate the evaluation metrics that lead to PRS selection<a class="anchor" aria-label="anchor" href="#calculate-the-evaluation-metrics-that-lead-to-prs-selection"></a>
</h5>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d3</span><span class="op">)</span>
<span class="va">evalMetrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsSelection</a></span><span class="op">(</span>
    dnList<span class="op">=</span><span class="va">dnList</span>,
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace3</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"summary"</span>,
    useDenovoWorkspace<span class="op">=</span><span class="va">dnList</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="based-on-the-metrics-find-the-best-prs-candidates">3. Based on the metrics, find the best PRS candidates<a class="anchor" aria-label="anchor" href="#based-on-the-metrics-find-the-best-prs-candidates"></a>
</h5>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">candidates</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectPrs.html">selectPrs</a></span><span class="op">(</span>
    metrics<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">metrics</span>,
    snpSelection<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">pgs</span>,
    gwe<span class="op">=</span><span class="va">toy</span>,
    base<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">baseline</span>
<span class="op">)</span></code></pre></div>
<p>The <code>prisma</code> wrapper implements additional steps to assemble the results and create some plots to be displayed in the final report.</p>
</div>
</div>
<div class="section level4">
<h4 id="complete-prisma-pipeline-with-report">Complete PRISMA pipeline with report<a class="anchor" aria-label="anchor" href="#complete-prisma-pipeline-with-report"></a>
</h4>
<p>The <code>prismaPipeline</code> wrapper servers as a complete "wrapper of wrapper" or a "meta-wrapper" for the end-to-end PRS extraction pipeline, from SNP effect estimation given a population with genotypes and phenotypes. The pipeline runs the <code>prisma</code> wrapper making sure that any crashes are resumed (up to 10 times by default), followed by cross-validation for the PRS candidates, lookup in the EBI GWAS Catalog and the creation of an HTML report.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">d4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,<span class="va">d4</span><span class="op">)</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="fu"><a href="../reference/prismaFuns.html">prismaPipeline</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace4</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"normal"</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>,
    outPath<span class="op">=</span><span class="va">outPath</span>
<span class="op">)</span>
<span class="co">## </span>
<span class="co">## $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$</span>
<span class="co">## This is PRISMA 0.0.2 PRS builder</span>
<span class="co">## $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$</span>
<span class="co">## </span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## 2022-06-06 20:21:49 - Try 1</span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## </span>
<span class="co">## 2022-06-06 20:21:49: Data processing started...</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : singlesplit</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : glm, statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : fishersimesmaxminharmonicwhitlockpandora</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   glm :</span>
<span class="co">##     family : -</span>
<span class="co">##     size   : 1</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 0.75</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 100</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)                               : 80%</span>
<span class="co">## Number of PRS selection iterations (niter)                  : 4</span>
<span class="co">## PRS selection resolution method (resolution)                : frequency</span>
<span class="co">## PRS selection resolution step (step)                        : 1</span>
<span class="co">## Minimum SNP frequency in PRS (minFreq)                      : 2</span>
<span class="co">## Minimum number of SNPs in PRS (minSnps)                     : 2</span>
<span class="co">## Drop quantiles with same number of SNPs (dropSameQuantiles) : Yes</span>
<span class="co">## PRS aggregation method (aggregation)                        : intersectionunion</span>
<span class="co">## SNP effect summarization method (effectWeight)              : meanmedianweight</span>
<span class="co">## PRS evaluation framework (evalWith)                         : vanillaprscice</span>
<span class="co">## </span>
<span class="co">## PRS selection options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS selection method (prsSelectMethod)           : maxima</span>
<span class="co">## PRS selection criterion (prsSelectCrit)          : prs_r2</span>
<span class="co">## PRS criterion summarization (prsSelectStat)      : mean</span>
<span class="co">## R^2 type if criterion includes R^2 (prsSelectR2) : adjusted</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## PRS pipeline output type (dnOutput)      : normal</span>
<span class="co">## Output level (output)                    : normal</span>
<span class="co">## Run ID (runId)                           : auto</span>
<span class="co">## Available cores fraction (rc)            : 25%</span>
<span class="co">## </span>
<span class="co">## ****************************************************************</span>
<span class="co">## Executing PRISMA pipeline with glm</span>
<span class="co">## ****************************************************************</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 1. Discovery of de novo PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline</span>
<span class="co">## </span>
<span class="co">## Executing denovo pipeline iteration 1, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 120</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 124</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 13 associations found out of 562 markers</span>
<span class="co">##   Part 21: 13 associations found out of 317 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 26 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0299419330259365</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1/base_ASEYL8061W \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1/pheno_ASEYL8061W \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1/plink_prsice_ASEYL8061W \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1/covar_ASEYL8061W \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_ASEYL8061W \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.128117</span>
<span class="co">## Full model R^2 is: 0.460406</span>
<span class="co">## Null model R^2 is: 0.332289</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_1/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 2, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 116</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 142</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 28 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 28 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 28 associations found out of 569 markers</span>
<span class="co">##   Part 21: 13 associations found out of 315 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 41 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.000260334133183589</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2/base_RPITG5899K \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2/pheno_RPITG5899K \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2/plink_prsice_RPITG5899K \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2/covar_RPITG5899K \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_RPITG5899K \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0462982</span>
<span class="co">## Full model R^2 is: 0.454096</span>
<span class="co">## Null model R^2 is: 0.407798</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_2/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 3, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 120</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 28 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 28 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 15 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 15 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 28 associations found out of 566 markers</span>
<span class="co">##   Part 21: 15 associations found out of 314 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 43 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.00649079901227329</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3/base_IURIS9128L \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3/pheno_IURIS9128L \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3/plink_prsice_IURIS9128L \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3/covar_IURIS9128L \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_IURIS9128L \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0704192</span>
<span class="co">## Full model R^2 is: 0.620807</span>
<span class="co">## Null model R^2 is: 0.550387</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_3/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 4, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 123</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 124</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 19 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 14 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 19 associations found out of 566 markers</span>
<span class="co">##   Part 21: 14 associations found out of 311 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 33 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0745972421160187</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4/base_FPGMY8235M \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4/pheno_FPGMY8235M \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4/plink_prsice_FPGMY8235M \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4/covar_FPGMY8235M \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_FPGMY8235M \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0861154</span>
<span class="co">## Full model R^2 is: 0.604763</span>
<span class="co">## Null model R^2 is: 0.518648</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline/TAQGFLMFRTW7949V_denovo_4/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/baseline</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 2. Calculation of PRS evaluation metrics</span>
<span class="co">## ################################################################</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 8 markers</span>
<span class="co">##   iteration 2 yields 10 markers</span>
<span class="co">##   iteration 3 yields 187 markers</span>
<span class="co">##   iteration 4 yields 80 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 2 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## 2 SNPs not found in target data! Excluding...</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/2</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 3 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/3</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 4 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4/4_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/glm/selection/4</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 3. Selection of best PRS candidate markers</span>
<span class="co">## ################################################################</span>
<span class="co">## The optimal number of markers for PRS based on prs_r2 is 3 at prs_r2=0.12735</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 4. Wrap-up and graphics</span>
<span class="co">## ################################################################</span>
<span class="co">## sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="co">## Getting latest annotation on the fly for hg19 from refseq</span>
<span class="co">## Retrieving genome information for hg19 from refseq</span>
<span class="co">## Retrieving latest gene annotation for hg19 from refseq</span>
<span class="co">## Loading required namespace: RMySQL</span>
<span class="co">## Converting annotation to GenomicRanges object...</span>
<span class="co">## </span>
<span class="co">## Attaching package: 'Biostrings'</span>
<span class="co">## The following object is masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     strsplit</span>
<span class="co">## Getting DNA sequences...</span>
<span class="co">## Getting GC content...</span>
<span class="co">## </span>
<span class="co">## ****************************************************************</span>
<span class="co">## Executing PRISMA pipeline with statgen</span>
<span class="co">## ****************************************************************</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 1. Discovery of de novo PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline</span>
<span class="co">## </span>
<span class="co">## Executing denovo pipeline iteration 1, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 120</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 118</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 28 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 28 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 28 associations found out of 565 markers</span>
<span class="co">##   Part 21: 17 associations found out of 315 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 45 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0290419547627459</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1/base_WDLAA4420N \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1/pheno_WDLAA4420N \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1/plink_prsice_WDLAA4420N \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1/covar_WDLAA4420N \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_WDLAA4420N \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0367345</span>
<span class="co">## Full model R^2 is: 0.678118</span>
<span class="co">## Null model R^2 is: 0.641384</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_1/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 2, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 120</span>
<span class="co">##   Hardy-Weinberg equilibrium: 4</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 128</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 20 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 20 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 19 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 20 associations found out of 565 markers</span>
<span class="co">##   Part 21: 19 associations found out of 313 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 39 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.00454142411810441</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2/base_HKQMJ5481Y \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2/pheno_HKQMJ5481Y \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2/plink_prsice_HKQMJ5481Y \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2/covar_HKQMJ5481Y \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_HKQMJ5481Y \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.102703</span>
<span class="co">## Full model R^2 is: 0.770589</span>
<span class="co">## Null model R^2 is: 0.667886</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_2/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 3, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 124</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 118</span>
<span class="co">##   Hardy-Weinberg equilibrium: 0</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 22 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 22 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 22 associations found out of 563 markers</span>
<span class="co">##   Part 21: 17 associations found out of 313 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 39 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0085994593263661</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3/base_MHDVL4655S \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3/pheno_MHDVL4655S \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3/plink_prsice_MHDVL4655S \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3/covar_MHDVL4655S \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_MHDVL4655S \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.163973</span>
<span class="co">## Full model R^2 is: 0.581528</span>
<span class="co">## Null model R^2 is: 0.417555</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_3/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 4, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 119</span>
<span class="co">##   Hardy-Weinberg equilibrium: 4</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 120</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 21 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 21 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 21 associations found out of 563 markers</span>
<span class="co">##   Part 21: 13 associations found out of 316 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 34 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0180435952879536</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4/base_VSKNM0230S \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4/pheno_VSKNM0230S \</span>
<span class="co">##   --target /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4/plink_prsice_VSKNM0230S \</span>
<span class="co">##   --cov /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4/covar_VSKNM0230S \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_VSKNM0230S \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0654275</span>
<span class="co">## Full model R^2 is: 0.554396</span>
<span class="co">## Null model R^2 is: 0.488968</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline/TAQGFLMFRTW7949V_denovo_4/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/baseline</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 2. Calculation of PRS evaluation metrics</span>
<span class="co">## ################################################################</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 1 markers</span>
<span class="co">##   iteration 2 yields 5 markers</span>
<span class="co">##   iteration 3 yields 23 markers</span>
<span class="co">##   iteration 4 yields 2 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 2 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpfOBGMb/file9f86ff09697//tmp/RtmpfOBGMb/file9f87292a3d2/statgen/selection/2</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 3. Selection of best PRS candidate markers</span>
<span class="co">## ################################################################</span>
<span class="co">## The optimal number of markers for PRS based on prs_r2 is 4 at prs_r2=0.16732</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 4. Wrap-up and graphics</span>
<span class="co">## ################################################################</span>
<span class="co">## sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="co">## Getting latest annotation on the fly for hg19 from refseq</span>
<span class="co">## Retrieving genome information for hg19 from refseq</span>
<span class="co">## Retrieving latest gene annotation for hg19 from refseq</span>
<span class="co">## Converting annotation to GenomicRanges object...</span>
<span class="co">## Getting DNA sequences...</span>
<span class="co">## Getting GC content...</span>
<span class="co">## </span>
<span class="co">## 2022-06-06 20:25:41: Data processing finished!</span>
<span class="co">## Total processing time: 03 minutes 51 seconds</span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## 2022-06-06 20:25:41 - Cross-validation</span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## Running cross-validations for glm with 10 leave-out cases across 1 PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Cross-validating PRS candidate with 3 markers</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 5% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 10% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 15% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 20% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 25% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 30% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 35% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 40% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 45% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 50% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## Running cross-validations for statgen with 10 leave-out cases across 1 PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Cross-validating PRS candidate with 4 markers</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 5% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 10% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 15% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 20% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 25% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 30% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 35% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 40% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 45% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 50% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## Done! Use summarizeCvMetrics() and plotCvMetrics() to summarize and inspect the results.</span>
<span class="co">## ################################################################</span>
<span class="co">## </span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## 2022-06-06 20:26:05 - GWAS Catalog lookup</span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## -----&gt; Querying for glm</span>
<span class="co">## Querying GWAS Catalog with the largest PRS</span>
<span class="co">## -----&gt; Querying for statgen</span>
<span class="co">## Querying GWAS Catalog with the largest PRS</span>
<span class="co">## </span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## 2022-06-06 20:26:08 - Report</span>
<span class="co">## @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="co">## processing file: prisma_report.Rmd</span>
<span class="co">## </span>
<span class="co">## Attaching package: 'pander'</span>
<span class="co">## The following object is masked from 'package:htmltools':</span>
<span class="co">## </span>
<span class="co">##     p</span>
<span class="co">## </span>
<span class="co">## Attaching package: 'grid'</span>
<span class="co">## The following object is masked from 'package:Biostrings':</span>
<span class="co">## </span>
<span class="co">##     pattern</span>
<span class="co">## Summarizing cross-validation statistics for PRS with 3 SNPs</span>
<span class="co">## Summarizing cross-validation statistics for PRS with 4 SNPs</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## A project name here</span>
<span class="co">## output file: prisma_report.knit.md</span>
<span class="co">## /usr/bin/pandoc +RTS -K512m -RTS prisma_report.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output /tmp/RtmpfOBGMb/file9f86ff09697/index.html --variable 'robobook:true' --variable 'lightbox:true' --variable 'gallery:false' --variable 'cards:true' --variable bs3=TRUE --standalone --section-divs --table-of-contents --toc-depth 2 --template /home/panos/R/x86_64-pc-linux-gnu-library/4.1/rmdformats/templates/template.html --highlight-style kate --variable theme=bootstrap --include-in-header /tmp/RtmpfOBGMb/rmarkdown-str9f855104b09.html</span>
<span class="co">## </span>
<span class="co">## Output created: /tmp/RtmpfOBGMb/file9f86ff09697/index.html</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">## [8] methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] pander_0.6.5                      magrittr_2.0.3                   </span>
<span class="co">##  [3] htmltools_0.5.2                   ggplot2_3.3.5                    </span>
<span class="co">##  [5] DT_0.22                           kableExtra_1.3.4                 </span>
<span class="co">##  [7] rmdformats_1.0.3                  knitr_1.39                       </span>
<span class="co">##  [9] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.62.0                  </span>
<span class="co">## [11] rtracklayer_1.54.0                Biostrings_2.62.0                </span>
<span class="co">## [13] XVector_0.34.0                    prisma_0.0.2                     </span>
<span class="co">## [15] SNPRelate_1.28.0                  gdsfmt_1.30.0                    </span>
<span class="co">## [17] snpStats_1.44.0                   Matrix_1.4-1                     </span>
<span class="co">## [19] survival_3.3-1                    SummarizedExperiment_1.24.0      </span>
<span class="co">## [21] Biobase_2.54.0                    GenomicRanges_1.46.1             </span>
<span class="co">## [23] GenomeInfoDb_1.30.1               IRanges_2.28.0                   </span>
<span class="co">## [25] S4Vectors_0.32.4                  BiocGenerics_0.40.0              </span>
<span class="co">## [27] MatrixGenerics_1.6.0              matrixStats_0.62.0               </span>
<span class="co">## [29] lassosum_0.4.5                   </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] utf8_1.2.2             R.utils_2.11.0         tidyselect_1.1.2      </span>
<span class="co">##  [4] gwascat_2.26.0         htmlwidgets_1.5.4      RSQLite_2.2.14        </span>
<span class="co">##  [7] AnnotationDbi_1.56.2   BiocParallel_1.28.3    munsell_0.5.0         </span>
<span class="co">## [10] codetools_0.2-18       ragg_1.2.2             future_1.25.0         </span>
<span class="co">## [13] withr_2.5.0            quincunx_0.1.4         islasso_1.4.3         </span>
<span class="co">## [16] colorspace_2.0-3       filelock_1.0.2         OrganismDbi_1.36.0    </span>
<span class="co">## [19] highr_0.9              rstudioapi_0.13        robustbase_0.95-0     </span>
<span class="co">## [22] TTR_0.24.3             listenv_0.8.0          labeling_0.4.2        </span>
<span class="co">## [25] urltools_1.7.3         optparse_1.7.1         GenomeInfoDbData_1.2.7</span>
<span class="co">## [28] farver_2.1.0           bit64_4.0.5            rprojroot_2.0.3       </span>
<span class="co">## [31] parallelly_1.31.1      vctrs_0.4.1            generics_0.1.2        </span>
<span class="co">## [34] xfun_0.31              BiocFileCache_2.2.1    R6_2.5.1              </span>
<span class="co">## [37] bitops_1.0-7           cachem_1.0.6           DelayedArray_0.20.0   </span>
<span class="co">## [40] assertthat_0.2.1       gwasrapidd_0.99.12     BiocIO_1.4.0          </span>
<span class="co">## [43] Homo.sapiens_1.3.1     scales_1.2.0           gtable_0.3.0          </span>
<span class="co">## [46] globals_0.15.0         rlang_1.0.2            systemfonts_1.0.4     </span>
<span class="co">## [49] splines_4.1.3          rsnps_0.5.0.0          sitadela_1.3.1        </span>
<span class="co">## [52] BiocManager_1.30.16    yaml_2.3.5             reshape2_1.4.4        </span>
<span class="co">## [55] crosstalk_1.2.0        GenomicFeatures_1.46.5 quantmod_0.4.20       </span>
<span class="co">## [58] RMySQL_0.10.23         RBGL_1.70.0            tools_4.1.3           </span>
<span class="co">## [61] lava_1.6.10            bookdown_0.26          liftOver_1.18.0       </span>
<span class="co">## [64] ellipsis_0.3.2         jquerylib_0.1.4        Rcpp_1.0.8.3          </span>
<span class="co">## [67] rrBLUP_4.6.1           plyr_1.8.7             progress_1.2.2        </span>
<span class="co">## [70] zlibbioc_1.40.0        purrr_0.3.4            RCurl_1.98-1.6        </span>
<span class="co">## [73] prettyunits_1.1.1      cowplot_1.1.1          zoo_1.8-10            </span>
<span class="co">##  [ reached getOption("max.print") -- omitted 92 entries ]</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
