<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>The GWASExperiment class â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The GWASExperiment class">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>The GWASExperiment class</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/03-gwasexperiment.Rmd" class="external-link"><code>vignettes/03-gwasexperiment.Rmd</code></a></small>
      <div class="d-none name"><code>03-gwasexperiment.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="the-gwasexperiment-class">The GWASExperiment class<a class="anchor" aria-label="anchor" href="#the-gwasexperiment-class"></a>
</h2>
<p>The following subsections outline the main data object used throughout the package for various data interchange and package functions.</p>
<div class="section level3">
<h3 id="description-and-structure">Description and structure<a class="anchor" aria-label="anchor" href="#description-and-structure"></a>
</h3>
<p>The main data structure of PRISMA is the GWASExperiment class which briefly provides facilities for:</p>
<ul>
<li>Data import using facilities from the Bioconductor package <a href="https://www.bioconductor.org/packages/release/bioc/html/snpStats.html" class="external-link">snpStats</a>.</li>
<li>Data integrity checks, data selections as well as recording of metadata gathered during analysis steps, such as filtering result tracking and Principal Component Analysis.</li>
<li>Robust Principal Component Analysis for automatic exclusion of outlier samples (through the R package <a href="https://cran.r-project.org/web/packages/rrcov/index.html" class="external-link">rrcov</a>.</li>
<li>Automatic selection of informative PCs to be used as population stratification covariates in subsequent GWAS and PRS analyses using the Tracy-Widom statistic.</li>
<li>Quality Control and filtering of GWAS data at the genotype level based on:
<ul>
<li>SNP call rate.</li>
<li>Minor Allele Frequency and monomorphic markers (MAF).</li>
<li>Hardy-Weinberg equilibrium (HWE).</li>
<li>Linkage Disequilibrium pruning (LD).</li>
</ul>
</li>
<li>Quality Control and filtering of GWAS data at the sample level (using genotypic data) based on:
<ul>
<li>Sample call rate.</li>
<li>Heterozygosity.</li>
<li>Identity By Descent (IBD, preceded by LD pruning).</li>
<li>Outlier filtering based on robust PCA preceded by LD pruning.</li>
</ul>
</li>
<li>Missing value imputation (not to be confused with imputation of genotypes based on external panels such as 1000 genomes) based on a multi-step process comprising:
<ul>
<li>Iterative imputation using facilities from snpStats package (regression of each SNP with nearby ones and extraction of imputation rules).</li>
<li>For the remaining missing values, imputation using the R package scrime, specialized on missing value imputation for genotypic data.</li>
</ul>
</li>
<li>Slots which hold marker calculated effects and statistical significance (p-values) from the application of one or multiple GWA tests, for any number of association tests (e.g. using different covariates).</li>
<li>Facilities for combining the outcome of multiple GWA tests (p-value combination methods).</li>
<li>Slots which hold the outcome of the PRS extraction outcomes (effects or adjusted betas).</li>
</ul>
<p>Apart from the GWASExperiment class, PRISMA deploys additional simpler structures with the goals of</p>
<ol style="list-style-type: lower-roman">
<li>Grouping results in a way useful for reporting.</li>
<li>Harmonizing the PRS candidate SNP format and accompanying details in a format similar to the scores format of the PGS Catalog.</li>
</ol>
<p>Finally, the aforementioned missing value imputation process was found to be between 88% and 92% accurate in imputing known genotypes (data not shown in this report).</p>
</div>
<div class="section level3">
<h3 id="object-design">Object design<a class="anchor" aria-label="anchor" href="#object-design"></a>
</h3>
<p>The GWASExperiment class is a direct extension of the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a> class, with additional slots to host phenotypes, statistical outcomes, effects used to weight the PRS candidates and certain metadata. Below, the main extensions are summarized (for more information please see the related man page):</p>
<ul>
<li>The <code>rowData</code> slot/accessor of the <code>SummarizedExperiment</code> is renamed to <code>gfeatures</code> (<code>rowData</code> can still be used but no formatting checks are performed which may cause crashes) and is a <code>DataFrame</code> with SNP data similar to the contents of the PLINK <code>bim</code> files, imported with <code>snpStats</code>.</li>
<li>The <code>colData</code> slot/accessor of the <code>SummarizedExperiment</code> is renamed to <code>gsamples</code> (<code>colData</code> can still be used as above) and is a <code>DataFrame</code> with SNP data similar to the contents of the PLINK <code>fam</code> files, imported with <code>snpStats</code>.</li>
<li>The <code>assays</code> and <code>assay</code> slot/accessor of the <code>SummarizedExperiment</code> is extended to <code>genotypes</code> (<code>assay</code>, <code>assays</code> can still be used) and checks are performed with respect to the genotype format.</li>
<li>The <code>metadata</code> slot/accessor of the <code>SummarizedExperiment</code> does not have completely free content and must at least contain some information about the data such as genome version and applied filters.</li>
<li>It hosts additional data parallel to <code>gfeatures</code> and <code>gsamples</code>:
<ul>
<li>The <code>phenotypes</code> slot/accessor hosts phenotypic or other measurements for each sample and is parallel to <code>gsamples</code>.</li>
<li>The <code>pvalues</code>, <code>effects</code>, <code>prsbetas</code> slots/accessors hosts statistical scores from GWA tests (<code>pvalues</code>, <code>effects</code>) or PRS extraction algorithms (<code>prsbetas</code>) for each SNP/feature and is parallel to <code>gfeatures</code>. These slots are lists (<code>SimpleList</code>) with as many members as the number of different models tested (e.g. with different covariates, number of PCs). In this way, the whole <code>GWASExperiment</code> object is not replicated while maintaining information on many models.</li>
<li>
<code>allPvalues</code>, <code>allEffects</code>, <code>allPrsbetas</code> can be used to retrieve the totality of the respective scores for all models associated with a <code>GWASExperiment</code> object.</li>
</ul>
</li>
<li>Other accessors and purpose-related functions are <code>genome</code> to get/set the genome version of the <code>gfeatures</code>, <code>pcaCovariates</code> to get/set the PCs calculated over the <code>GWASExperiment</code> object data, <code>gsplit</code> to split the object feature- or sample-wise according to variables present to features or samples and others mentioned in the respective man page.</li>
<li>Combining functions (<code>cbind</code>, <code>rbind</code>) are also supported.</li>
</ul>
<p>The following figure summarizes the <code>GWASExperiment</code> object structure:</p>
<p><img src="gwasexperiment.png" width="800px"></p>
</div>
<div class="section level3">
<h3 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h3>
<p>The following sections show some examples of usage of the <code>GWASExperiment</code> objects including their creation and function.</p>
<div class="section level4">
<h4 id="gwasexperiment-from-simulated-data">GWASExperiment from simulated data<a class="anchor" aria-label="anchor" href="#gwasexperiment-from-simulated-data"></a>
</h4>
<p>We create some simulated data with 100 SNPs, 10 samples and 10 causative SNPs for speed and simplicity. We also add pseudo-effects and pseudo-pvalues from two virtual GWA tests to demonstrate the slots of the object. The genome is unknown.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSimData.html">makeSimData</a></span><span class="op">(</span>nsnp<span class="op">=</span><span class="fl">100</span>,nsam<span class="op">=</span><span class="fl">10</span>,nphe<span class="op">=</span><span class="fl">3</span>,csnp<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
<span class="co">## coercing object of mode  numeric  to SnpMatrix</span>

<span class="co"># Simple GWASExperiment object</span>
<span class="va">gwe1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">GWASExperiment</a></span><span class="op">(</span>
    genotypes<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">snp</span>,
    features<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">feature</span>,
    samples<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">sample</span>,
    phenotypes<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">pheno</span>
<span class="op">)</span>

<span class="co"># With pseudo-effects and pseudo-pvalues</span>
<span class="va">pspval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">200</span><span class="op">)</span>,<span class="fl">100</span>,<span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pspval</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">snp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pspval</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"test1"</span>,<span class="st">"test2"</span><span class="op">)</span>

<span class="co"># Pseudo-effects and directions</span>
<span class="va">pseff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">200</span>,min<span class="op">=</span><span class="fl">0</span>,max<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,<span class="fl">100</span>,<span class="fl">2</span><span class="op">)</span>
<span class="va">dire</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="cn">FALSE</span>,<span class="fl">100</span>,<span class="fl">2</span><span class="op">)</span>
<span class="va">dire</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">dire</span><span class="op">)</span>,<span class="fl">50</span><span class="op">)</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
<span class="va">dire</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">dire</span><span class="op">)</span>,<span class="fl">50</span><span class="op">)</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
<span class="va">pseff</span><span class="op">[</span><span class="va">dire</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="va">pseff</span><span class="op">[</span><span class="va">dire</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,<span class="fl">1</span><span class="op">]</span>
<span class="va">pseff</span><span class="op">[</span><span class="va">dire</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="va">pseff</span><span class="op">[</span><span class="va">dire</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,<span class="fl">2</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pseff</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">snp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pseff</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"test1"</span>,<span class="st">"test2"</span><span class="op">)</span>

<span class="co"># Assign effects and p-values to the object, assuming a model</span>
<span class="co"># with 'Trait_1' as response and 'Trait_2', 'Trait_3' as</span>
<span class="co"># covariates has been fit (no PCs)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">pvalues</a></span><span class="op">(</span><span class="va">gwe1</span>,response<span class="op">=</span><span class="st">"Trait_1"</span>,
    covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Trait_2"</span>,<span class="st">"Trait_3"</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pspval</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">effects</a></span><span class="op">(</span><span class="va">gwe1</span>,response<span class="op">=</span><span class="st">"Trait_1"</span>,
    covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Trait_2"</span>,<span class="st">"Trait_3"</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pseff</span>

<span class="va">gwe1</span>
<span class="co">## class: GWASExperiment </span>
<span class="co">## dim: 100 10 </span>
<span class="co">## metadata(3): genome backend filters</span>
<span class="co">## assays class: SnpMatrix with 100 rows (SNPs) and 10 columns (samples)</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(100): snp_1 snp_2 ... snp_99 snp_100</span>
<span class="co">## feature names(5): chromosome snp.name position allele.1 allele.2</span>
<span class="co">## colnames(10): sample_1 sample_2 ... sample_9 sample_10</span>
<span class="co">## sample names(6): pedigree member ... sex affected</span>
<span class="co">## phenotype names(3): Trait_1 Trait_2 Trait_3</span>
<span class="co">## performed tests(2): test1 test2</span>
<span class="co">## associated PRS(0):</span></code></pre></div>
<p>We create an additional simulated dataset to demonstrate joining and splitting operations.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Add SNPs (rbind)</span>
<span class="va">dat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSimData.html">makeSimData</a></span><span class="op">(</span>nsnp<span class="op">=</span><span class="fl">100</span>,nsam<span class="op">=</span><span class="fl">10</span>,nphe<span class="op">=</span><span class="fl">3</span>,csnp<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
<span class="co">## coercing object of mode  numeric  to SnpMatrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">snp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">feature</span><span class="op">)</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"snp"</span>,<span class="fl">101</span><span class="op">:</span><span class="fl">200</span>,sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>
<span class="va">gwe2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">GWASExperiment</a></span><span class="op">(</span>
    genotypes<span class="op">=</span><span class="va">dat2</span><span class="op">$</span><span class="va">snp</span>,
    features<span class="op">=</span><span class="va">dat2</span><span class="op">$</span><span class="va">feature</span>,
    samples<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">sample</span>,
    phenotypes<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">pheno</span>
<span class="op">)</span>
<span class="co"># p-values and effects will be dropped with warning</span>
<span class="va">gwe12</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">gwe1</span>,<span class="va">gwe2</span><span class="op">)</span><span class="op">)</span>

<span class="va">gwe12</span>
<span class="co">## class: GWASExperiment </span>
<span class="co">## dim: 200 10 </span>
<span class="co">## metadata(5): genome backend genome backend filters</span>
<span class="co">## assays class: SnpMatrix with 200 rows (SNPs) and 10 columns (samples)</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(200): snp_1 snp_2 ... snp_199 snp_200</span>
<span class="co">## feature names(5): chromosome snp.name position allele.1 allele.2</span>
<span class="co">## colnames(10): sample_1 sample_2 ... sample_9 sample_10</span>
<span class="co">## sample names(6): pedigree member ... sex affected</span>
<span class="co">## phenotype names(3): Trait_1 Trait_2 Trait_3</span>
<span class="co">## performed tests(0):</span>
<span class="co">## associated PRS(0):</span>

<span class="co"># Add samples (cbind)</span>
<span class="va">dat3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSimData.html">makeSimData</a></span><span class="op">(</span>nsnp<span class="op">=</span><span class="fl">100</span>,nsam<span class="op">=</span><span class="fl">10</span>,nphe<span class="op">=</span><span class="fl">3</span>,csnp<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
<span class="co">## coercing object of mode  numeric  to SnpMatrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat3</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat3</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dat3</span><span class="op">$</span><span class="va">snp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"sample"</span>,<span class="fl">11</span><span class="op">:</span><span class="fl">20</span>,sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>
<span class="va">gwe3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">GWASExperiment</a></span><span class="op">(</span>
    genotypes<span class="op">=</span><span class="va">dat3</span><span class="op">$</span><span class="va">snp</span>,
    features<span class="op">=</span><span class="va">dat1</span><span class="op">$</span><span class="va">feature</span>,
    samples<span class="op">=</span><span class="va">dat3</span><span class="op">$</span><span class="va">sample</span>,
    phenotypes<span class="op">=</span><span class="va">dat3</span><span class="op">$</span><span class="va">pheno</span>
<span class="op">)</span>
<span class="co"># p-values and effects will be dropped with warning</span>
<span class="va">gwe13</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">gwe1</span>,<span class="va">gwe3</span><span class="op">)</span><span class="op">)</span>

<span class="va">gwe13</span>
<span class="co">## class: GWASExperiment </span>
<span class="co">## dim: 100 20 </span>
<span class="co">## metadata(5): genome backend genome backend filters</span>
<span class="co">## assays class: SnpMatrix with 100 rows (SNPs) and 20 columns (samples)</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(100): snp_1 snp_2 ... snp_99 snp_100</span>
<span class="co">## feature names(5): chromosome snp.name position allele.1 allele.2</span>
<span class="co">## colnames(20): sample_1 sample_2 ... sample_19 sample_20</span>
<span class="co">## sample names(6): pedigree member ... sex affected</span>
<span class="co">## phenotype names(3): Trait_1 Trait_2 Trait_3</span>
<span class="co">## performed tests(0):</span>
<span class="co">## associated PRS(0):</span></code></pre></div>
<p>It is important to note that when merging datasets via <code>cbind</code> or <code>rbind</code> the following changes are made with respect to the final object:</p>
<ul>
<li>Any statistical GWA/PRS operations and scores are dropped. p-values, effects and PRS betas do no longer have meaning as either the tested SNPs or the number of samples changes.</li>
<li>In the case of change of sample number (<code>cbind</code>), principal components are also dropped. We do not drop them in <code>rbind</code> as they still might be of usage but they can be recalculated.</li>
<li>The drop of PCs also applies in subsetting.</li>
<li>
<code>metadata</code> are merged. If of importance, you should check the outcome and adjust accordingly.</li>
<li>The genome versions of the combining objects must be identical.</li>
</ul>
</div>
<div class="section level4">
<h4 id="gwasexperiment-from-plink-files">GWASExperiment from PLINK files<a class="anchor" aria-label="anchor" href="#gwasexperiment-from-plink-files"></a>
</h4>
<p>We use the example files shipping with the <code>snpStats</code> package:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.fam"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span>
<span class="va">bim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.bim"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span>
<span class="va">bed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.bed"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span>
<span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>fam<span class="op">=</span><span class="va">fam</span>,bim<span class="op">=</span><span class="va">bim</span>,bed<span class="op">=</span><span class="va">bed</span><span class="op">)</span></code></pre></div>
<p>Without phenotypes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gwe &lt;-<span class="st"> </span><span class="kw">importGWAS</span>(input,<span class="dt">backend=</span><span class="st">"snpStats"</span>)
## Reading PLINK files with snpStats framework
## Reading PLINK files with SNPRelate framework and storing output tp /tmp/RtmpNz2YkH/filed6823cbf419
## Start file conversion from PLINK BED to SNP GDS ...
##     BED file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bed'
##         SNP-major mode (Sample X SNP), 603 bytes
##     FAM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.fam'
##     BIM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bim'
## Mon Jun  6 20:29:57 2022     (store sample id, snp id, position, and chromosome)
##     start writing: 120 samples, 20 SNPs ...
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:29:57 2022     Done.
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file '/tmp/RtmpNz2YkH/filed6823cbf419' (5.1K)
##     # of fragments: 50
##     save to '/tmp/RtmpNz2YkH/filed6823cbf419.tmp'
##     rename '/tmp/RtmpNz2YkH/filed6823cbf419.tmp' (4.8K, reduced: 312B)
##     # of fragments: 24

gwe
## class: GWASExperiment 
## dim: 20 120 
## metadata(5): genome backend filters gdsfile alleleOrder
## assays class: SnpMatrix with 20 rows (SNPs) and 120 columns (samples)
## assays(1): ''
## rownames(20): IGR1118a_1 IGR1119a_1 ... IGR2016a_1 IGR2020a_1
## feature names(6): chromosome snp.name ... allele.1 allele.2
## colnames(120): 430 412 ... 1991 17702
## sample names(6): pedigree member ... sex affected
## phenotype names(0):
## performed tests(0):
## associated PRS(0):</code></pre></div>
<p>With phenotypes</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">42</span>)

fam &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.fam"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
bim &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.bim"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
bed &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.bed"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
input &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">fam=</span>fam,<span class="dt">bim=</span>bim,<span class="dt">bed=</span>bed)
sample &lt;-<span class="st"> </span>snpStats<span class="op">::</span><span class="kw">read.plink</span>(input<span class="op">$</span>bed,input<span class="op">$</span>bim,input<span class="op">$</span>fam)

pseudopheno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
    <span class="dt">case_control=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="kw">nrow</span>(sample<span class="op">$</span>fam),<span class="dt">replace=</span><span class="ot">TRUE</span>),
    <span class="dt">other_pheno=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">"drug"</span>,<span class="st">"nodrug"</span>),<span class="kw">nrow</span>(sample<span class="op">$</span>fam),<span class="dt">replace=</span><span class="ot">TRUE</span>),
    <span class="dt">cont=</span><span class="kw">round</span>(<span class="kw">runif</span>(<span class="kw">nrow</span>(sample<span class="op">$</span>fam)),<span class="dv">3</span>),
    <span class="dt">row.names=</span><span class="kw">rownames</span>(sample<span class="op">$</span>fam)
)
gwe &lt;-<span class="st"> </span><span class="kw">importGWAS</span>(input,<span class="dt">phenos=</span>pseudopheno,<span class="dt">backend=</span><span class="st">"snpStats"</span>)
## Reading PLINK files with snpStats framework
## Reading PLINK files with SNPRelate framework and storing output tp /tmp/RtmpNz2YkH/filed684e7dcc74
## Start file conversion from PLINK BED to SNP GDS ...
##     BED file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bed'
##         SNP-major mode (Sample X SNP), 603 bytes
##     FAM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.fam'
##     BIM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bim'
## Mon Jun  6 20:29:58 2022     (store sample id, snp id, position, and chromosome)
##     start writing: 120 samples, 20 SNPs ...
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:29:58 2022     Done.
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file '/tmp/RtmpNz2YkH/filed684e7dcc74' (5.1K)
##     # of fragments: 50
##     save to '/tmp/RtmpNz2YkH/filed684e7dcc74.tmp'
##     rename '/tmp/RtmpNz2YkH/filed684e7dcc74.tmp' (4.8K, reduced: 312B)
##     # of fragments: 24

gwe
## class: GWASExperiment 
## dim: 20 120 
## metadata(5): genome backend filters gdsfile alleleOrder
## assays class: SnpMatrix with 20 rows (SNPs) and 120 columns (samples)
## assays(1): ''
## rownames(20): IGR1118a_1 IGR1119a_1 ... IGR2016a_1 IGR2020a_1
## feature names(6): chromosome snp.name ... allele.1 allele.2
## colnames(120): 430 412 ... 1991 17702
## sample names(6): pedigree member ... sex affected
## phenotype names(3): case_control other_pheno cont
## performed tests(0):
## associated PRS(0):</code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] prisma_0.0.2                SNPRelate_1.28.0           </span>
<span class="co">##  [3] gdsfmt_1.30.0               snpStats_1.44.0            </span>
<span class="co">##  [5] Matrix_1.4-1                survival_3.3-1             </span>
<span class="co">##  [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">## [11] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [15] matrixStats_0.62.0          lassosum_0.4.5             </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] highr_0.9                               </span>
<span class="co">##  [19] knitr_1.39                              </span>
<span class="co">##  [20] rstudioapi_0.13                         </span>
<span class="co">##  [21] robustbase_0.95-0                       </span>
<span class="co">##  [22] TTR_0.24.3                              </span>
<span class="co">##  [23] listenv_0.8.0                           </span>
<span class="co">##  [24] optparse_1.7.1                          </span>
<span class="co">##  [25] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [26] bit64_4.0.5                             </span>
<span class="co">##  [27] rprojroot_2.0.3                         </span>
<span class="co">##  [28] parallelly_1.31.1                       </span>
<span class="co">##  [29] vctrs_0.4.1                             </span>
<span class="co">##  [30] generics_0.1.2                          </span>
<span class="co">##  [31] xfun_0.31                               </span>
<span class="co">##  [32] BiocFileCache_2.2.1                     </span>
<span class="co">##  [33] R6_2.5.1                                </span>
<span class="co">##  [34] rmdformats_1.0.3                        </span>
<span class="co">##  [35] bitops_1.0-7                            </span>
<span class="co">##  [36] cachem_1.0.6                            </span>
<span class="co">##  [37] DelayedArray_0.20.0                     </span>
<span class="co">##  [38] assertthat_0.2.1                        </span>
<span class="co">##  [39] gwasrapidd_0.99.12                      </span>
<span class="co">##  [40] BiocIO_1.4.0                            </span>
<span class="co">##  [41] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [42] scales_1.2.0                            </span>
<span class="co">##  [43] gtable_0.3.0                            </span>
<span class="co">##  [44] globals_0.15.0                          </span>
<span class="co">##  [45] rlang_1.0.2                             </span>
<span class="co">##  [46] systemfonts_1.0.4                       </span>
<span class="co">##  [47] splines_4.1.3                           </span>
<span class="co">##  [48] rtracklayer_1.54.0                      </span>
<span class="co">##  [49] rsnps_0.5.0.0                           </span>
<span class="co">##  [50] sitadela_1.3.1                          </span>
<span class="co">##  [51] BiocManager_1.30.16                     </span>
<span class="co">##  [52] yaml_2.3.5                              </span>
<span class="co">##  [53] reshape2_1.4.4                          </span>
<span class="co">##  [54] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [55] quantmod_0.4.20                         </span>
<span class="co">##  [56] RBGL_1.70.0                             </span>
<span class="co">##  [57] tools_4.1.3                             </span>
<span class="co">##  [58] lava_1.6.10                             </span>
<span class="co">##  [59] bookdown_0.26                           </span>
<span class="co">##  [60] liftOver_1.18.0                         </span>
<span class="co">##  [61] ggplot2_3.3.5                           </span>
<span class="co">##  [62] ellipsis_0.3.2                          </span>
<span class="co">##  [63] kableExtra_1.3.4                        </span>
<span class="co">##  [64] jquerylib_0.1.4                         </span>
<span class="co">##  [65] Rcpp_1.0.8.3                            </span>
<span class="co">##  [66] rrBLUP_4.6.1                            </span>
<span class="co">##  [67] plyr_1.8.7                              </span>
<span class="co">##  [68] progress_1.2.2                          </span>
<span class="co">##  [69] zlibbioc_1.40.0                         </span>
<span class="co">##  [70] purrr_0.3.4                             </span>
<span class="co">##  [71] RCurl_1.98-1.6                          </span>
<span class="co">##  [72] prettyunits_1.1.1                       </span>
<span class="co">##  [73] cowplot_1.1.1                           </span>
<span class="co">##  [74] zoo_1.8-10                              </span>
<span class="co">##  [75] fs_1.5.2                                </span>
<span class="co">##  [76] crul_1.2.0                              </span>
<span class="co">##  [77] magrittr_2.0.3                          </span>
<span class="co">##  [78] data.table_1.14.2                       </span>
<span class="co">##  [79] openxlsx_4.2.5                          </span>
<span class="co">##  [80] mvtnorm_1.1-3                           </span>
<span class="co">##  [81] survcomp_1.44.1                         </span>
<span class="co">##  [82] hms_1.1.1                               </span>
<span class="co">##  [83] evaluate_0.15                           </span>
<span class="co">##  [84] XML_3.99-0.9                            </span>
<span class="co">##  [85] RMTstat_0.3.1                           </span>
<span class="co">##  [86] shape_1.4.6                             </span>
<span class="co">##  [87] compiler_4.1.3                          </span>
<span class="co">##  [88] biomaRt_2.50.3                          </span>
<span class="co">##  [89] tibble_3.1.7                            </span>
<span class="co">##  [90] KernSmooth_2.23-20                      </span>
<span class="co">##  [91] crayon_1.5.1                            </span>
<span class="co">##  [92] R.oo_1.24.0                             </span>
<span class="co">##  [93] htmltools_0.5.2                         </span>
<span class="co">##  [94] pcaPP_2.0-1                             </span>
<span class="co">##  [95] tzdb_0.3.0                              </span>
<span class="co">##  [96] rrcov_1.7-0                             </span>
<span class="co">##  [97] DBI_1.1.2                               </span>
<span class="co">##  [98] SuppDists_1.1-9.7                       </span>
<span class="co">##  [99] dbplyr_2.1.1                            </span>
<span class="co">## [100] MASS_7.3-56                             </span>
<span class="co">## [101] rappdirs_0.3.3                          </span>
<span class="co">## [102] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [103] getopt_1.20.3                           </span>
<span class="co">## [104] readr_2.1.2                             </span>
<span class="co">## [105] rmeta_3.0                               </span>
<span class="co">## [106] cli_3.3.0                               </span>
<span class="co">## [107] quadprog_1.5-8                          </span>
<span class="co">## [108] R.methodsS3_1.8.1                       </span>
<span class="co">## [109] parallel_4.1.3                          </span>
<span class="co">## [110] pkgconfig_2.0.3                         </span>
<span class="co">## [111] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [112] pkgdown_2.0.3.9000                      </span>
<span class="co">## [113] GenomicAlignments_1.30.0                </span>
<span class="co">## [114] signal_0.7-7                            </span>
<span class="co">## [115] xml2_1.3.3                              </span>
<span class="co">## [116] foreach_1.5.2                           </span>
<span class="co">## [117] svglite_2.1.0                           </span>
<span class="co">## [118] bslib_0.3.1                             </span>
<span class="co">## [119] webshot_0.5.3                           </span>
<span class="co">## [120] XVector_0.34.0                          </span>
<span class="co">## [121] prodlim_2019.11.13                      </span>
<span class="co">## [122] rvest_1.0.2                             </span>
<span class="co">## [123] stringr_1.4.0                           </span>
<span class="co">## [124] VariantAnnotation_1.40.0                </span>
<span class="co">## [125] digest_0.6.29                           </span>
<span class="co">## [126] graph_1.72.0                            </span>
<span class="co">## [127] httpcode_0.3.0                          </span>
<span class="co">## [128] Biostrings_2.62.0                       </span>
<span class="co">## [129] rmarkdown_2.14                          </span>
<span class="co">## [130] harmonicmeanp_3.0                       </span>
<span class="co">## [131] restfulr_0.0.13                         </span>
<span class="co">## [132] curl_4.3.2                              </span>
<span class="co">## [133] Rsamtools_2.10.0                        </span>
<span class="co">## [134] rjson_0.2.21                            </span>
<span class="co">## [135] tseries_0.10-51                         </span>
<span class="co">## [136] lifecycle_1.0.1                         </span>
<span class="co">## [137] jsonlite_1.8.0                          </span>
<span class="co">## [138] survivalROC_1.0.3                       </span>
<span class="co">## [139] desc_1.4.1                              </span>
<span class="co">## [140] viridisLite_0.4.0                       </span>
<span class="co">## [141] BSgenome_1.62.0                         </span>
<span class="co">## [142] fansi_1.0.3                             </span>
<span class="co">## [143] pillar_1.7.0                            </span>
<span class="co">## [144] lattice_0.20-45                         </span>
<span class="co">## [145] DEoptimR_1.0-11                         </span>
<span class="co">## [146] KEGGREST_1.34.0                         </span>
<span class="co">## [147] fastmap_1.1.0                           </span>
<span class="co">## [148] httr_1.4.3                              </span>
<span class="co">## [149] GO.db_3.14.0                            </span>
<span class="co">## [150] xts_0.12.1                              </span>
<span class="co">## [151] glue_1.6.2                              </span>
<span class="co">## [152] zip_2.2.0                               </span>
<span class="co">## [153] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [154] png_0.1-7                               </span>
<span class="co">## [155] iterators_1.0.14                        </span>
<span class="co">## [156] pander_0.6.5                            </span>
<span class="co">## [157] glmnet_4.1-4                            </span>
<span class="co">## [158] bit_4.0.4                               </span>
<span class="co">## [159] stringi_1.7.6                           </span>
<span class="co">## [160] sass_0.4.1                              </span>
<span class="co">## [161] bootstrap_2019.6                        </span>
<span class="co">## [162] blob_1.2.3                              </span>
<span class="co">## [163] statgenGWAS_1.0.8                       </span>
<span class="co">## [164] textshaping_0.3.6                       </span>
<span class="co">## [165] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [166] memoise_2.0.1                           </span>
<span class="co">## [167] FMStable_0.1-2                          </span>
<span class="co">## [168] dplyr_1.0.8                             </span>
<span class="co">## [169] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
