<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Reading, writing and filtering files and objects • prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Reading, writing and filtering files and objects">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Reading, writing and filtering files and objects</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/04-inoutfilter.Rmd" class="external-link"><code>vignettes/04-inoutfilter.Rmd</code></a></small>
      <div class="d-none name"><code>04-inoutfilter.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="reading-plink-files-to-gwasexperiment-objects">Reading PLINK files to GWASExperiment objects<a class="anchor" aria-label="anchor" href="#reading-plink-files-to-gwasexperiment-objects"></a>
</h2>
<p>PRISMA provides facilities to create <code>GWASExperiment</code> objects directly from PLINK <code>bed/bim/fam</code> triplets (not from PED files for the time being) through the usage of the <code>read.plink</code> function from the package snpStats along with other functionalities to attach phenotypes, metadata etc. For example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(prisma)

fam &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.fam"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
bim &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.bim"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
bed &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.bed"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
input &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">fam=</span>fam,<span class="dt">bim=</span>bim,<span class="dt">bed=</span>bed)

gwe &lt;-<span class="st"> </span><span class="kw">importGWAS</span>(input,<span class="dt">backend=</span><span class="st">"snpStats"</span>)
## Reading PLINK files with snpStats framework
## Reading PLINK files with SNPRelate framework and storing output tp /tmp/RtmpeiVKQg/file26136fd2f695
## Start file conversion from PLINK BED to SNP GDS ...
##     BED file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bed'
##         SNP-major mode (Sample X SNP), 603 bytes
##     FAM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.fam'
##     BIM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bim'
## Fri Jun  3 14:51:41 2022     (store sample id, snp id, position, and chromosome)
##     start writing: 120 samples, 20 SNPs ...
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Fri Jun  3 14:51:41 2022     Done.
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file '/tmp/RtmpeiVKQg/file26136fd2f695' (5.1K)
##     # of fragments: 50
##     save to '/tmp/RtmpeiVKQg/file26136fd2f695.tmp'
##     rename '/tmp/RtmpeiVKQg/file26136fd2f695.tmp' (4.8K, reduced: 312B)
##     # of fragments: 24
gwe
## class: GWASExperiment 
## dim: 20 120 
## metadata(5): genome backend filters gdsfile alleleOrder
## assays class: SnpMatrix with 20 rows (SNPs) and 120 columns (samples)
## assays(1): ''
## rownames(20): IGR1118a_1 IGR1119a_1 ... IGR2016a_1 IGR2020a_1
## feature names(6): chromosome snp.name ... allele.1 allele.2
## colnames(120): 430 412 ... 1991 17702
## sample names(6): pedigree member ... sex affected
## phenotype names(0):
## performed tests(0):
## associated PRS(0):</code></pre></div>
<p>We can also attach a data frame or a file (text-tab delimited) of phenotypes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">42</span>)

fam &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.fam"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
bim &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.bim"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
bed &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/sample.bed"</span>,<span class="dt">package=</span><span class="st">"snpStats"</span>)
input &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">fam=</span>fam,<span class="dt">bim=</span>bim,<span class="dt">bed=</span>bed)
sample &lt;-<span class="st"> </span>snpStats<span class="op">::</span><span class="kw">read.plink</span>(input<span class="op">$</span>bed,input<span class="op">$</span>bim,input<span class="op">$</span>fam)

pseudopheno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
    <span class="dt">case_control=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="kw">nrow</span>(sample<span class="op">$</span>fam),<span class="dt">replace=</span><span class="ot">TRUE</span>),
    <span class="dt">other_pheno=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">"drug"</span>,<span class="st">"nodrug"</span>),<span class="kw">nrow</span>(sample<span class="op">$</span>fam),<span class="dt">replace=</span><span class="ot">TRUE</span>),
    <span class="dt">cont=</span><span class="kw">round</span>(<span class="kw">runif</span>(<span class="kw">nrow</span>(sample<span class="op">$</span>fam)),<span class="dv">3</span>),
    <span class="dt">row.names=</span><span class="kw">rownames</span>(sample<span class="op">$</span>fam)
)

gwe &lt;-<span class="st"> </span><span class="kw">importGWAS</span>(input,<span class="dt">phenos=</span>pseudopheno,<span class="dt">backend=</span><span class="st">"snpStats"</span>)
## Reading PLINK files with snpStats framework
## Reading PLINK files with SNPRelate framework and storing output tp /tmp/RtmpeiVKQg/file26134be5c5f3
## Start file conversion from PLINK BED to SNP GDS ...
##     BED file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bed'
##         SNP-major mode (Sample X SNP), 603 bytes
##     FAM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.fam'
##     BIM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bim'
## Fri Jun  3 14:51:41 2022     (store sample id, snp id, position, and chromosome)
##     start writing: 120 samples, 20 SNPs ...
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Fri Jun  3 14:51:42 2022     Done.
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file '/tmp/RtmpeiVKQg/file26134be5c5f3' (5.1K)
##     # of fragments: 50
##     save to '/tmp/RtmpeiVKQg/file26134be5c5f3.tmp'
##     rename '/tmp/RtmpeiVKQg/file26134be5c5f3.tmp' (4.8K, reduced: 312B)
##     # of fragments: 24

gwe
## class: GWASExperiment 
## dim: 20 120 
## metadata(5): genome backend filters gdsfile alleleOrder
## assays class: SnpMatrix with 20 rows (SNPs) and 120 columns (samples)
## assays(1): ''
## rownames(20): IGR1118a_1 IGR1119a_1 ... IGR2016a_1 IGR2020a_1
## feature names(6): chromosome snp.name ... allele.1 allele.2
## colnames(120): 430 412 ... 1991 17702
## sample names(6): pedigree member ... sex affected
## phenotype names(3): case_control other_pheno cont
## performed tests(0):
## associated PRS(0):</code></pre></div>
<p>Other functions of <code>importGWAS</code> include the selection of a subset of SNPs to import (<code>selection</code> argument), whether or not to reverse alleles (<code>alleleOrder</code> argument, useful some times although it generally should be left to the default) and whether a GDS file should be written or not (generally yes, especially if you intend to perform further filtering or PCA). For example, if you do not intend to do any of the previous and you know that the SNPs locations are mapped to the hg19 human genome version:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmoulos/prisma" class="external-link">prisma</a></span><span class="op">)</span>

<span class="va">fam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.fam"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span>
<span class="va">bim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.bim"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span>
<span class="va">bed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.bed"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span>
<span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>fam<span class="op">=</span><span class="va">fam</span>,bim<span class="op">=</span><span class="va">bim</span>,bed<span class="op">=</span><span class="va">bed</span><span class="op">)</span>

<span class="va">gwe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importGWAS.html">importGWAS</a></span><span class="op">(</span><span class="va">input</span>,backend<span class="op">=</span><span class="st">"snpStats"</span>,writeGds<span class="op">=</span><span class="cn">FALSE</span>,genome<span class="op">=</span><span class="st">"hg19"</span><span class="op">)</span>
<span class="va">gwe</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exporting-gwasexperiment-objects-to-plink-files">Exporting GWASExperiment objects to PLINK files<a class="anchor" aria-label="anchor" href="#exporting-gwasexperiment-objects-to-plink-files"></a>
</h2>
<p>PRISMA also provides facilities to export <code>GWASExperiment</code> objects to PLINK <code>bed/bim/fam</code> triplets (not PED files for the time being) through the usage of the <code>write.plink</code> function from the package snpStats along with other functionalities to attach phenotypes, metadata etc. and potentially split the output per chromosome. For example, the following will write one temporary PLINK binary triplet for the toy data included with the package, with the <code>"BMI"</code> included phenotype as the registered phenotype in the PLINK FAM file:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="../reference/writeFuns.html">writePlink</a></span><span class="op">(</span><span class="va">toy</span>,pheno<span class="op">=</span><span class="st">"BMI"</span>,outBase<span class="op">=</span><span class="va">out</span><span class="op">)</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpeiVKQg/file261325f0f8e2.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpeiVKQg/file261325f0f8e2.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpeiVKQg/file261325f0f8e2.bed (SNP-major mode)</span>
<span class="co">## NULL</span></code></pre></div>
<p>The following will write a binary PLINK files triplet, without registering a phenotype and splitting per chromosome:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="../reference/writeFuns.html">writePlink</a></span><span class="op">(</span><span class="va">toy</span>,outBase<span class="op">=</span><span class="va">out</span>,perChr<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## Preflight...</span>
<span class="co">## ===== Writing files for chromosome 18</span>
<span class="co">## Writing FAM file to /tmp/RtmpeiVKQg/file2613311e5611_chr18.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpeiVKQg/file2613311e5611_chr18.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpeiVKQg/file2613311e5611_chr18.bed (SNP-major mode)</span>
<span class="co">## ===== Writing files for chromosome 21</span>
<span class="co">## Writing FAM file to /tmp/RtmpeiVKQg/file2613311e5611_chr21.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpeiVKQg/file2613311e5611_chr21.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpeiVKQg/file2613311e5611_chr21.bed (SNP-major mode)</span>
<span class="co">## NULL</span></code></pre></div>
<p>The <code>writePlink</code> function also offers a SNP "salvation" mode (using the <code>salvage=TRUE</code> argument). Sometimes, PLINK files may be imported to PRISMA for further processing and calculations. It may occur that some SNPs in these files have missing location/position data, which will result in their exclusion from the output files. In this mode, <code>writePlink</code> connects to Ensembl through the R package biomaRt and tries to identify chromosomal coordinates for these SNPs.</p>
</div>
<div class="section level2">
<h2 id="other-io-and-coversion-tools">Other I/O and coversion tools<a class="anchor" aria-label="anchor" href="#other-io-and-coversion-tools"></a>
</h2>
<p>PRISMA implements additional file conversion and output functions. Specifically:</p>
<div class="section level3">
<h3 id="write-gds-files">Write GDS files<a class="anchor" aria-label="anchor" href="#write-gds-files"></a>
</h3>
<p>The <code>writeGdsFile</code> function can be used for writing GDS files from <code>GWASExperiment</code> objects:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

gdsf &lt;-<span class="st"> </span><span class="kw">tempfile</span>()
<span class="kw">writeGdsFile</span>(toy,<span class="dt">gdsfile=</span>gdsf)
## Writing FAM file to /tmp/RtmpeiVKQg/file2613785c658c.fam 
## Writing extended MAP file to /tmp/RtmpeiVKQg/file2613785c658c.bim 
## Writing BED file to /tmp/RtmpeiVKQg/file2613785c658c.bed (SNP-major mode)
## Start file conversion from PLINK BED to SNP GDS ...
##     BED file: '/tmp/RtmpeiVKQg/file2613785c658c.bed'
##         SNP-major mode (Sample X SNP), 24.4K
##     FAM file: '/tmp/RtmpeiVKQg/file2613785c658c.fam'
##     BIM file: '/tmp/RtmpeiVKQg/file2613785c658c.bim'
## Fri Jun  3 14:51:45 2022     (store sample id, snp id, position, and chromosome)
##     start writing: 100 samples, 1000 SNPs ...
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Fri Jun  3 14:51:45 2022     Done.
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file '/tmp/RtmpeiVKQg/file2613755857dc' (34.6K)
##     # of fragments: 50
##     save to '/tmp/RtmpeiVKQg/file2613755857dc.tmp'
##     rename '/tmp/RtmpeiVKQg/file2613755857dc.tmp' (34.3K, reduced: 312B)
##     # of fragments: 24
## [1] "/tmp/RtmpeiVKQg/file2613755857dc"</code></pre></div>
<p>The <code>GWASExperiment2GDS</code> function operates similarly but it uses the <code>gdsfile</code> accessor to get the object's GDS file and throws an error if it is not defined.</p>
</div>
<div class="section level3">
<h3 id="converters-and-utilities">Converters and utilities<a class="anchor" aria-label="anchor" href="#converters-and-utilities"></a>
</h3>
<p>The <code><a href="../reference/gweConvertFuns.html">GWASExperiment2gData()</a></code> function converts a <code>GWASExperiment</code> object to the <code>gData</code> class used by the functions in the package statgenGWAS, making sure that the phenotypes and covariates follow the rules of the <code>gData</code> class:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">gd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gweConvertFuns.html">GWASExperiment2gData</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span>
<span class="co"># Warnings may be produced if some covariates do not follow the rules of the</span>
<span class="co"># gData class.</span></code></pre></div>
<p>The <code>GWASExperimentLiftOver</code> converts the SNP positions between the human genome versions <code>hg19</code> and <code>hg38</code>. SNPs which fail to map are dropped with a notification:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">toy38</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gweConvertFuns.html">GWASExperimentLiftOver</a></span><span class="op">(</span><span class="va">toy</span>,from<span class="op">=</span><span class="st">"hg19"</span>,to<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span>
<span class="co">## Lifting over features</span>
<span class="co">##   lifting chr18</span>
<span class="co">##   lifting chr21</span></code></pre></div>
<p>If the human genome version is not known a priori, the <code><a href="../reference/gweConvertFuns.html">guessHumanGenomeVersion()</a></code> can be used to "guess" the human genome version by querying 5 random SNPs using the <code>ncbi_snp_query()</code> function from the package rsnps. Please note that if your overall data/experiment/study depends on SNP locations, only human hg19 and hg38 versions are currently supported by PRISMA.</p>
</div>
</div>
<div class="section level2">
<h2 id="filtering-gwasexperiment-objects">Filtering GWASExperiment objects<a class="anchor" aria-label="anchor" href="#filtering-gwasexperiment-objects"></a>
</h2>
<p>As PRISMA is intended to be a one-stop shop for PRS extraction, it also includes basic (or not-so-basic) functionalities for treating genotype data prior to GWA testing. These functionalities include genotype filtering based on sample and SNP properties as well as missing data imputation (the topic of another vignette). Therefore, PRISMA support several filters which follow the best practices according to the related literature. More specifically, the PRISMA filters follow losely <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.6605" class="external-link">this</a> publication which is also a tutorial on preprocessing genotype data with R.</p>
<p>The function that implements the filters in PRISMA is <code><a href="../reference/filterGWAS.html">filterGWAS()</a></code> and the filters to be performed are provided as a list of filter parameters through the <code>filters</code> argument. The following sections describe the supported filters.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="snp-filters">SNP filters<a class="anchor" aria-label="anchor" href="#snp-filters"></a>
</h3>
<p>The following SNP filters are supported (percentages are provided as fractions, i.e. a value between 0 and 1):</p>
<ul>
<li>SNP Call Rate: The call rate for a given SNP is defined as the proportion of samples for which the corresponding genotype is not missing. A typical value is 95%, meaning SNPs for which there is less than 5% missing data are retained. This could be smaller for datasets with fewer samples. The default value in PRISMA is 0.98 (<code>snpCallRate=0.98</code>).</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">snpCallRate</span>
<span class="co">## [1] 0.98</span></code></pre></div>
<ul>
<li>Minor Allele Frequency (MAF): Very often and especially in studies with small samples, genotypes for many markers exhibit a large degree of homogeneity (an extreme example of this could be an isolated population). This results in inadequate power to infer a statistically significant relationship between the SNP and the trait under study. This translates to very low MAFs so that the large majority of samples have two copies of the major allele. A typical strategy is to remove SNPs with MAF &lt; 1%. In smaller datasets, a loser cutoff may be applied (5%). The default value in PRISMA is 0.05 (<code>maf=0.05</code>).</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">maf</span>
<span class="co">## [1] 0.05</span></code></pre></div>
<ul>
<li>Hardy-Weinberg equilibrium (HWE): The Hardy-Weinberg equilibrium is a principle stating that the genetic variation in a population will remain constant from one generation to the next in the absence of disturbing factors. Violations of HWE can be an indication of the presence of population substructure or the occurrence of a genotyping error. A common strategy is to assume a genotyping error and remove SNPs for which HWE is violated. Violations from HWE are generally measured at a given SNP using a <span class="math inline">\(\chi^{2}\)</span> goodness-of-fit test between the observed and expected genotypes. SNPs presenting a p-value &lt; 10<sup>−6</sup> are typically removed. This value could be even smaller for smaller datasets. The default value in PRISMA is 1 x 10<sup>−6</sup> (<code>hwe=1e-6</code>).</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">hwe</span>
<span class="co">## [1] 1e-06</span></code></pre></div>
<ul>
<li>Linkage Disequilibrium (LD): Linkage disequilibrium is the nonrandom association of alleles of different loci. In other words (although not the single explanation) LD represents the strong correlation of close or distant loci genotypes, leading to strong collinear genptype effects and disturbing the downstream GWA testing. In PRISMA, LD is not applied directly (as suggested in the related litarature), but is a component of other filters and analyses such as PCA. The default cutoff value is 0.2 (<code>LD=0.2</code>). SNPs presenting pairwise relation with larger values are filtered.</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">LD</span>
<span class="co">## [1] 0.2</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="sample-filters">Sample filters<a class="anchor" aria-label="anchor" href="#sample-filters"></a>
</h3>
<p>The following sample filters are supported (percentages are provided as fractions, i.e. a value between 0 and 1):</p>
<ul>
<li>Sample Call Rate: Similar to the SNP Call Rate, samples with missing genotypes across more than a pre-defined percentage of the typed SNPs are excluded. This proportion of missingness across SNPs is referred to as the sample call rate. A typical threshold is 95%. That is, individuals who are missing genotype data for more than 5% of the typed SNPs are removed. The default value in PRISMA is 0.95 (<code>sampleCallRate=0.95</code>).</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">sampleCallRate</span>
<span class="co">## [1] 0.95</span></code></pre></div>
<ul>
<li>Heterozygosity: Heterozygosity refers to the presence of each of the two alleles at a given SNP within an individual. This is expected under HWE to occur with probability 2 x p x (1 − p), where p is the dominant allele frequency at that SNP (assuming a bi-allelic SNP). Excess heterozygosity across typed SNPs within an individual may be an indication of poor sample quality. Heterozygosity in PRISMA can be filtered in two ways:
<ul>
<li>Based on hard cutoff filter (<code>heteroHard</code>). Samples with heterozygosity above this value are excluded. The default value is <code>NA</code> (not used).</li>
<li>Based on the distribution of heterozygosity. The heterozygosity values of samples are used to create a distibution. Then, <code>heteroStat</code> (mean or median) of this distribution is calculated along with the respective measure of scale (standard deviation or IQR) which is multiplied by a factor <code>heteroFac</code>. Then samples with heterozygosity larger than <code>heteroStat + heteroFac*scale</code> are excluded. The default <code>heteroStat</code> in PRISMA is <code>"median"</code> and the default <code>heteroFac</code> is <code>3</code>.</li>
</ul>
</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">heteroHard</span>
<span class="co">## [1] NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">heteroStat</span>
<span class="co">## [1] "median"</span>
<span class="va">filts</span><span class="op">$</span><span class="va">heteroFac</span>
<span class="co">## [1] 3</span></code></pre></div>
<ul>
<li>Inbreed coefficient: as with heterozygosity, the Inbreed Coefficient can expose problems with deficient heterozygosity across samples. Inbreeding coefficients are defined in terms of the probability of identity in state of different pairs of genes. The probability of identity in state is simply the probability that two genes are of identical allelic type. In other words, the inbreed coefficient <em>F</em> is:</li>
</ul>
<p><span class="math display">\[|F| = 1 - \frac{O}{E} \]</span></p>
<p>where <em>O</em> is the observed and <em>E</em> the expected counts of heterozygous SNPs within an individual. The expected counts for each sample are calculated based on the observed SNPs for that sample. The default cutoff in PRISMA is 0.1 (<code>inbreed</code>=0.1). Samples with values above that are filtered out.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span>
<span class="co">## [1] 0.1</span></code></pre></div>
<ul>
<li>Identity By Descent (IBD): IBD refers to the case where two or more individuals/samples have inherited a common DNA segment (icluding also several common SNPs) from a common ancestor without recombination, that is, the segment has the same ancestral origin in these individuals. An IBD kinship coefficient of greater than 0.1 may suggest relatedness, duplicates, or sample mixture. The default value in PRISMA is 0.1 (<code>IBD=0.1</code>). Samples with values greater than this are excluded.</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">IBD</span>
<span class="co">## [1] 0.1</span></code></pre></div>
<ul>
<li>Principal Component Analysis (PCA): PCA can be used to visualize and classify samples into ancestry groups based on their observed genetic makeup. Such a genetic or ancestry group could be a simple case/control design. PCA can be used to identify outliers within such groups (either sample mixe-up or wrong self-reporting or even problematic genotyped samples). PCA in PRISMA is performed on LD-contolled SNPs, as suggested in the related literature and is used both for identifying uderlying genetic statification to be used later in downstream GWA and PRS extraction, as well as identifying samples that should be exluded from this analysis. Furthermore, PRISMA can automatically identify such samples (<em>experimental!</em>) through the application of robust PCA techniques (R package rrcov) which automatically identify outliers. However, such approaches should probably be applied to large enough datasets. PRISMA filters outliers based on the <code>pcaOut</code> filters parameter (<code>pcaOut=TRUE</code>) and this is done by performing robust PCA (<code>pcaRobust</code> parameter). <code>pcaRobust</code> can be one of <code>"hubert"</code> for the Hubert method or "grid" for the Grid-Seqrch algorith. Please see the documentation of the <a href="https://cran.r-project.org/web/packages/rrcov/index.html" class="external-link">rrcov</a> package. Please also <strong>note</strong> that robust PCA, especially with the <code>"grid"</code> method can take a <strong>lot</strong> of time, so it's probably best to apply it once on a <code>GWASExperiment</code> object and then continue any analysis with a new object.</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span>
<span class="co">## [1] FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaRobust</span>
<span class="co">## [1] "hubert"</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h3>
<p>Perorm some basic and lose filtering on the small toy dataset:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,
    <span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span>

<span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span>

<span class="va">toyF</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGWAS.html">filterGWAS</a></span><span class="op">(</span><span class="va">toy</span>,filters<span class="op">=</span><span class="va">filts</span>,imputeMissing<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 119</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span></code></pre></div>
<p>Check the filtering status using the <code><a href="../reference/GWASExperiment-class.html">filterRecord()</a></code> accessor:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GWASExperiment-class.html">filterRecord</a></span><span class="op">(</span><span class="va">toyF</span><span class="op">)</span>
<span class="co">##                     parameter                       name value   type filtered</span>
<span class="co">## snpCallRate       snpCallRate              SNP call rate 0.980    SNP        0</span>
<span class="co">## maf                       maf     Minor Allele Frequency 0.050    SNP      119</span>
<span class="co">## hwe                       hwe Hardy-Weinberg equilibrium 0.001    SNP        2</span>
<span class="co">## sampleCallRate sampleCallRate           Sample call rate 0.980 Sample        0</span>
<span class="co">## heteroHard         heteroHard             Heterozygosity    NA Sample        0</span>
<span class="co">## inbreed               inbreed                    Inbreed    NA Sample        0</span></code></pre></div>
<p>Add robust PCA to the filters and enable inbreed coefficient:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaRobust</span> <span class="op">&lt;-</span> <span class="st">"hubert"</span>

<span class="va">toyF</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGWAS.html">filterGWAS</a></span><span class="op">(</span><span class="va">toy</span>,filters<span class="op">=</span><span class="va">filts</span>,imputeMissing<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##     Inbreeding coefficient</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## Performing robust PCA for automatic outlier detection</span>
<span class="co">##   with Hubert method</span>
<span class="co">## </span>
<span class="co">## Reperforming LD and PCA analysis after outlier sample removal</span>
<span class="co">## Reading GDS file /tmp/RtmplLIKfg/temp_libpath4f71185f9e1c/prisma/extdata/toy.gds...</span>
<span class="co">## Performing LD pruning...</span>
<span class="co">## SNP pruning based on LD:</span>
<span class="co">## Excluding 120 SNPs (non-autosomes or non-selection)</span>
<span class="co">## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="co">##     # of samples: 86</span>
<span class="co">##     # of SNPs: 880</span>
<span class="co">##     using 1 thread</span>
<span class="co">##     sliding window: 500,000 basepairs, Inf SNPs</span>
<span class="co">##     |LD| threshold: 0.2</span>
<span class="co">##     method: composite</span>
<span class="co">## Chromosome 18: 20.40%, 134/657</span>
<span class="co">## Chromosome 21: 18.66%, 64/343</span>
<span class="co">## 198 markers are selected in total.</span>
<span class="co">## LD pruning finished! 198 SNPs will be used for PCA.</span>
<span class="co">## </span>
<span class="co">## Performing PCA...</span>
<span class="co">##   with Hubert method</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 119</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 10</span>
<span class="co">##   Robust PCA                : 4</span>
<span class="co">## </span></code></pre></div>
<p>Inspect the outcome:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GWASExperiment-class.html">filterRecord</a></span><span class="op">(</span><span class="va">toyF</span><span class="op">)</span>
<span class="co">##                     parameter                       name  value   type filtered</span>
<span class="co">## snpCallRate       snpCallRate              SNP call rate   0.98    SNP        0</span>
<span class="co">## maf                       maf     Minor Allele Frequency   0.05    SNP      119</span>
<span class="co">## hwe                       hwe Hardy-Weinberg equilibrium  0.001    SNP        2</span>
<span class="co">## sampleCallRate sampleCallRate           Sample call rate   0.98 Sample        0</span>
<span class="co">## heteroHard         heteroHard             Heterozygosity   &lt;NA&gt; Sample        0</span>
<span class="co">## inbreed               inbreed                    Inbreed    0.1 Sample       10</span>
<span class="co">## pcaOut                 pcaOut                 Robust PCA hubert Sample        4</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] prisma_0.0.1                SNPRelate_1.28.0           </span>
<span class="co">##  [3] gdsfmt_1.30.0               snpStats_1.44.0            </span>
<span class="co">##  [5] Matrix_1.4-1                survival_3.3-1             </span>
<span class="co">##  [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">## [11] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [15] matrixStats_0.62.0          lassosum_0.4.5             </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] knitr_1.39                              </span>
<span class="co">##  [19] rstudioapi_0.13                         </span>
<span class="co">##  [20] robustbase_0.95-0                       </span>
<span class="co">##  [21] TTR_0.24.3                              </span>
<span class="co">##  [22] listenv_0.8.0                           </span>
<span class="co">##  [23] optparse_1.7.1                          </span>
<span class="co">##  [24] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [25] bit64_4.0.5                             </span>
<span class="co">##  [26] rprojroot_2.0.3                         </span>
<span class="co">##  [27] parallelly_1.31.1                       </span>
<span class="co">##  [28] vctrs_0.4.1                             </span>
<span class="co">##  [29] generics_0.1.2                          </span>
<span class="co">##  [30] xfun_0.31                               </span>
<span class="co">##  [31] BiocFileCache_2.2.1                     </span>
<span class="co">##  [32] R6_2.5.1                                </span>
<span class="co">##  [33] rmdformats_1.0.3                        </span>
<span class="co">##  [34] bitops_1.0-7                            </span>
<span class="co">##  [35] cachem_1.0.6                            </span>
<span class="co">##  [36] DelayedArray_0.20.0                     </span>
<span class="co">##  [37] assertthat_0.2.1                        </span>
<span class="co">##  [38] gwasrapidd_0.99.12                      </span>
<span class="co">##  [39] BiocIO_1.4.0                            </span>
<span class="co">##  [40] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [41] scales_1.2.0                            </span>
<span class="co">##  [42] gtable_0.3.0                            </span>
<span class="co">##  [43] globals_0.15.0                          </span>
<span class="co">##  [44] rlang_1.0.2                             </span>
<span class="co">##  [45] systemfonts_1.0.4                       </span>
<span class="co">##  [46] splines_4.1.3                           </span>
<span class="co">##  [47] rtracklayer_1.54.0                      </span>
<span class="co">##  [48] rsnps_0.5.0.0                           </span>
<span class="co">##  [49] sitadela_1.3.1                          </span>
<span class="co">##  [50] BiocManager_1.30.16                     </span>
<span class="co">##  [51] yaml_2.3.5                              </span>
<span class="co">##  [52] reshape2_1.4.4                          </span>
<span class="co">##  [53] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [54] quantmod_0.4.20                         </span>
<span class="co">##  [55] RBGL_1.70.0                             </span>
<span class="co">##  [56] tools_4.1.3                             </span>
<span class="co">##  [57] lava_1.6.10                             </span>
<span class="co">##  [58] bookdown_0.26                           </span>
<span class="co">##  [59] liftOver_1.18.0                         </span>
<span class="co">##  [60] ggplot2_3.3.5                           </span>
<span class="co">##  [61] ellipsis_0.3.2                          </span>
<span class="co">##  [62] kableExtra_1.3.4                        </span>
<span class="co">##  [63] jquerylib_0.1.4                         </span>
<span class="co">##  [64] Rcpp_1.0.8.3                            </span>
<span class="co">##  [65] rrBLUP_4.6.1                            </span>
<span class="co">##  [66] plyr_1.8.7                              </span>
<span class="co">##  [67] progress_1.2.2                          </span>
<span class="co">##  [68] zlibbioc_1.40.0                         </span>
<span class="co">##  [69] purrr_0.3.4                             </span>
<span class="co">##  [70] RCurl_1.98-1.6                          </span>
<span class="co">##  [71] prettyunits_1.1.1                       </span>
<span class="co">##  [72] cowplot_1.1.1                           </span>
<span class="co">##  [73] zoo_1.8-10                              </span>
<span class="co">##  [74] fs_1.5.2                                </span>
<span class="co">##  [75] crul_1.2.0                              </span>
<span class="co">##  [76] magrittr_2.0.3                          </span>
<span class="co">##  [77] data.table_1.14.2                       </span>
<span class="co">##  [78] openxlsx_4.2.5                          </span>
<span class="co">##  [79] mvtnorm_1.1-3                           </span>
<span class="co">##  [80] survcomp_1.44.1                         </span>
<span class="co">##  [81] hms_1.1.1                               </span>
<span class="co">##  [82] evaluate_0.15                           </span>
<span class="co">##  [83] XML_3.99-0.9                            </span>
<span class="co">##  [84] RMTstat_0.3.1                           </span>
<span class="co">##  [85] shape_1.4.6                             </span>
<span class="co">##  [86] compiler_4.1.3                          </span>
<span class="co">##  [87] biomaRt_2.50.3                          </span>
<span class="co">##  [88] tibble_3.1.7                            </span>
<span class="co">##  [89] KernSmooth_2.23-20                      </span>
<span class="co">##  [90] crayon_1.5.1                            </span>
<span class="co">##  [91] R.oo_1.24.0                             </span>
<span class="co">##  [92] htmltools_0.5.2                         </span>
<span class="co">##  [93] pcaPP_2.0-1                             </span>
<span class="co">##  [94] tzdb_0.3.0                              </span>
<span class="co">##  [95] rrcov_1.7-0                             </span>
<span class="co">##  [96] DBI_1.1.2                               </span>
<span class="co">##  [97] SuppDists_1.1-9.7                       </span>
<span class="co">##  [98] dbplyr_2.1.1                            </span>
<span class="co">##  [99] MASS_7.3-56                             </span>
<span class="co">## [100] rappdirs_0.3.3                          </span>
<span class="co">## [101] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [102] getopt_1.20.3                           </span>
<span class="co">## [103] readr_2.1.2                             </span>
<span class="co">## [104] rmeta_3.0                               </span>
<span class="co">## [105] cli_3.3.0                               </span>
<span class="co">## [106] quadprog_1.5-8                          </span>
<span class="co">## [107] R.methodsS3_1.8.1                       </span>
<span class="co">## [108] parallel_4.1.3                          </span>
<span class="co">## [109] pkgconfig_2.0.3                         </span>
<span class="co">## [110] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [111] pkgdown_2.0.3.9000                      </span>
<span class="co">## [112] GenomicAlignments_1.30.0                </span>
<span class="co">## [113] signal_0.7-7                            </span>
<span class="co">## [114] xml2_1.3.3                              </span>
<span class="co">## [115] foreach_1.5.2                           </span>
<span class="co">## [116] svglite_2.1.0                           </span>
<span class="co">## [117] bslib_0.3.1                             </span>
<span class="co">## [118] webshot_0.5.3                           </span>
<span class="co">## [119] XVector_0.34.0                          </span>
<span class="co">## [120] prodlim_2019.11.13                      </span>
<span class="co">## [121] rvest_1.0.2                             </span>
<span class="co">## [122] stringr_1.4.0                           </span>
<span class="co">## [123] VariantAnnotation_1.40.0                </span>
<span class="co">## [124] digest_0.6.29                           </span>
<span class="co">## [125] graph_1.72.0                            </span>
<span class="co">## [126] httpcode_0.3.0                          </span>
<span class="co">## [127] Biostrings_2.62.0                       </span>
<span class="co">## [128] rmarkdown_2.14                          </span>
<span class="co">## [129] harmonicmeanp_3.0                       </span>
<span class="co">## [130] restfulr_0.0.13                         </span>
<span class="co">## [131] curl_4.3.2                              </span>
<span class="co">## [132] Rsamtools_2.10.0                        </span>
<span class="co">## [133] rjson_0.2.21                            </span>
<span class="co">## [134] tseries_0.10-51                         </span>
<span class="co">## [135] lifecycle_1.0.1                         </span>
<span class="co">## [136] jsonlite_1.8.0                          </span>
<span class="co">## [137] survivalROC_1.0.3                       </span>
<span class="co">## [138] desc_1.4.1                              </span>
<span class="co">## [139] viridisLite_0.4.0                       </span>
<span class="co">## [140] BSgenome_1.62.0                         </span>
<span class="co">## [141] fansi_1.0.3                             </span>
<span class="co">## [142] pillar_1.7.0                            </span>
<span class="co">## [143] lattice_0.20-45                         </span>
<span class="co">## [144] DEoptimR_1.0-11                         </span>
<span class="co">## [145] KEGGREST_1.34.0                         </span>
<span class="co">## [146] fastmap_1.1.0                           </span>
<span class="co">## [147] httr_1.4.3                              </span>
<span class="co">## [148] GO.db_3.14.0                            </span>
<span class="co">## [149] xts_0.12.1                              </span>
<span class="co">## [150] glue_1.6.2                              </span>
<span class="co">## [151] zip_2.2.0                               </span>
<span class="co">## [152] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [153] png_0.1-7                               </span>
<span class="co">## [154] iterators_1.0.14                        </span>
<span class="co">## [155] pander_0.6.5                            </span>
<span class="co">## [156] glmnet_4.1-4                            </span>
<span class="co">## [157] bit_4.0.4                               </span>
<span class="co">## [158] stringi_1.7.6                           </span>
<span class="co">## [159] sass_0.4.1                              </span>
<span class="co">## [160] bootstrap_2019.6                        </span>
<span class="co">## [161] blob_1.2.3                              </span>
<span class="co">## [162] statgenGWAS_1.0.8                       </span>
<span class="co">## [163] textshaping_0.3.6                       </span>
<span class="co">## [164] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [165] memoise_2.0.1                           </span>
<span class="co">## [166] FMStable_0.1-2                          </span>
<span class="co">## [167] dplyr_1.0.8                             </span>
<span class="co">## [168] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
