<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Imputation of missing genotypes, genotype extension and dataset merging â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Imputation of missing genotypes, genotype extension and dataset merging">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Imputation of missing genotypes, genotype extension and dataset merging</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/05-imputemerge.Rmd" class="external-link"><code>vignettes/05-imputemerge.Rmd</code></a></small>
      <div class="d-none name"><code>05-imputemerge.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="snp-imputation-in-prisma">SNP imputation in PRISMA<a class="anchor" aria-label="anchor" href="#snp-imputation-in-prisma"></a>
</h2>
<p>PRISMA offers functions to perform SNP genotype imputation at two levels. The first level is imputation of missing genotypes, for completeness of a dataset as many GWA tests do not operate with missing values. This is different from SNP imputation based on a large reference panel such as a population study (e.g. the 1000 genomes project). We refer to this imputation level as <em>internal imputation</em>. The second level is the imputation based on an external reference panel (as mentioned before). In this case, the genotypes are both imputed (in the case of missing data) and extended to infer SNPs and other variants not present in the initial dataset, based on the content of the initial dataset and sophisticated regression algorothims which use the reference panel and the actual dataset content to enrich the initial dataset with inferred SNPs with usually good accuracy. We refer to this level of imputation as <em>external imputation</em> or <em>extension</em>.</p>
<p>To achieve the two types of imputation PRISMA utilizes:</p>
<ul>
<li>For internal imputation, it implements a sequential imputation procedure based on the <code>snp.impute</code> function of the package snpStats, followed by genotype imputation with the package scrime for SNPs that fail to be imputed.</li>
<li>For external imputation, it wraps the <a href="https://mathgen.stats.ox.ac.uk/impute/impute_v2.html" class="external-link">IMPUTE2</a> algorithm and extends the <code>gfeatures</code> slot of the initial <code>GWASExperiment</code> object with the INFO score returned by IMPUTE2 and can be used to assess the quality of the extended genotypes and filter accordingly.</li>
</ul>
<div class="section level3">
<h3 id="internal-imputation">Internal imputation<a class="anchor" aria-label="anchor" href="#internal-imputation"></a>
</h3>
<p>As mentioned, internal imputation in PRISMA is performed by implementing a sequential imputation process using functions from the snpStats package. Therefore, the imputation can be performed in two possible ways:</p>
<ol style="list-style-type: lower-roman">
<li>Imputing with imputation rules based on the total dataset with or without missing values (<code>mode="single"</code>). Rules are created based on all SNPs in the dataset even if there are some missing values (snpStats takes care of that).</li>
<li>Imputing with spliting the dataset to a part with non-missing genotypes which is used to create the imputation rules (mode="split"), which are then applied on the dataset part with the missing values.</li>
</ol>
<p>Even though case (ii) may be more accurate, it is often non-applicable to smaller datasets where it is unlikely to find samples with no missing values at all. Case (ii) could also be applied with an external reference panel to add SNPs not genotyped in the initial dataset (another <em>external</em> imputation process) which is not implemented within PRISMA at this point.</p>
<p>All the functionalities are wrapped with the <code><a href="../reference/imputeGWAS.html">imputeGWAS()</a></code> function which operates on a <code>GWASExperiment</code> object. Below, we are using again the toy dataset included with PRISMA and we will artificially add a few missing values to demostrate case (i) (<code>mode="single"</code>):</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="co"># Make some genotypes missing</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>
<span class="va">toMiss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span><span class="op">)</span>,<span class="kw">function</span><span class="op">(</span><span class="va">j</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">20</span>,<span class="fl">1</span><span class="op">)</span>
    <span class="va">miss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/logical.html" class="external-link">logical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span><span class="op">)</span>
    <span class="va">miss</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span>,<span class="va">n</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">miss</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="va">G</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu"><a href="../reference/GWASExperiment-class.html">genotypes</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span>,<span class="st">"numeric"</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>
<span class="fu"><a href="https://rdrr.io/r/base/mode.html" class="external-link">mode</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"integer"</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
    <span class="va">G</span><span class="op">[</span><span class="va">toMiss</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span>,<span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0L</span>
<span class="va">G</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SnpMatrix.html">SnpMatrix</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span>
<span class="co">## coercing object of mode  numeric  to SnpMatrix</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">genotypes</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">G</span>

<span class="va">toyI</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imputeGWAS.html">imputeGWAS</a></span><span class="op">(</span><span class="va">toy</span>,fail<span class="op">=</span><span class="st">"none"</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Imputing missing values with snpStats rules</span>
<span class="co">## </span>
<span class="co">## Starting imputation analysis in 2 chunks</span>
<span class="co">## </span>
<span class="co">## ========== Imputing chromosome/part 18</span>
<span class="co">## -----&gt; Imputation iteration 1</span>
<span class="co">## Creating imputation rules</span>
<span class="co">## SNPs tagged by a single SNP: 45</span>
<span class="co">## SNPs tagged by multiple tag haplotypes (saturated model): 537</span>
<span class="co">## 97 samples have missing genotypes in 433 SNPs in total.</span>
<span class="co">## Imputing...</span>
<span class="co">## Applying imputations...</span>
<span class="co">## -----&gt; Imputation iteration 2</span>
<span class="co">## Creating imputation rules</span>
<span class="co">## SNPs tagged by a single SNP: 44</span>
<span class="co">## SNPs tagged by multiple tag haplotypes (saturated model): 538</span>
<span class="co">## 52 samples have missing genotypes in 47 SNPs in total.</span>
<span class="co">## Imputing...</span>
<span class="co">## Applying imputations...</span>
<span class="co">## </span>
<span class="co">## ========== Imputing chromosome/part 21</span>
<span class="co">## -----&gt; Imputation iteration 1</span>
<span class="co">## Creating imputation rules</span>
<span class="co">## SNPs tagged by a single SNP: 20</span>
<span class="co">## SNPs tagged by multiple tag haplotypes (saturated model): 301</span>
<span class="co">## 92 samples have missing genotypes in 221 SNPs in total.</span>
<span class="co">## Imputing...</span>
<span class="co">## Applying imputations...</span>
<span class="co">## -----&gt; Imputation iteration 2</span>
<span class="co">## Creating imputation rules</span>
<span class="co">## SNPs tagged by a single SNP: 20</span>
<span class="co">## SNPs tagged by multiple tag haplotypes (saturated model): 301</span>
<span class="co">## 27 samples have missing genotypes in 17 SNPs in total.</span>
<span class="co">## Imputing...</span>
<span class="co">## Applying imputations...</span>
<span class="co">## </span>
<span class="co">## Imputation finished, re-merging the output</span>
<span class="va">before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="../reference/GWASExperiment-class.html">genotypes</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">after</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="../reference/GWASExperiment-class.html">genotypes</a></span><span class="op">(</span><span class="va">toyI</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">before</span> <span class="op">&gt;</span> <span class="va">after</span> <span class="co"># TRUE</span>
<span class="co">## [1] TRUE</span></code></pre></div>
<p>The imputation process can be parallelized with the <code>rc</code> argument passed to the <code><a href="../reference/imputeGWAS.html">imputeGWAS()</a></code> function. The <code>fail</code> argument controls what to do with genotypes that failed to be imputed with the snpStats imputation rules. They either remain missing (<code>fail="none"</code>) or they are further imputed with the k-Nearest Neighbor algorithm implemented in the package scrime (<code>fail="scrime"</code>). It should be noted that some form of imputation <em>must</em> be performed prior to GWA tests and PRS analysis as many algorithms do not handle missing genotypes.</p>
<p>Also, internal imputation is part of the filtering process so as to procude objects ready for GWA testing. It can be turned off by setting <code>imputeMissing=FALSE</code> in the <code><a href="../reference/filterGWAS.html">filterGWAS()</a></code> function. This should be the case when playing with filters to bring data to acceptable form for later analysis.</p>
</div>
<div class="section level3">
<h3 id="external-imputation">External imputation<a class="anchor" aria-label="anchor" href="#external-imputation"></a>
</h3>
<p>As mentioned, <em>external imputation</em> in PRISMA refers to the extension of the dataset (e.g. SNP microarray) content with SNPs and other variants that do not exist in it, based on an external larger reference population panel such as the variants listed within the <a href="https://www.internationalgenome.org/" class="external-link">1000 Genomes Project</a>. This process is called <em>imputation</em> in the related literature but within PRISMA and in order to distinguish it from missing genotype imputation, we refer to it as <em>extension</em> or <em>external imputation</em>.</p>
<p>To perform external imputation PRISMA needs:</p>
<ul>
<li>The <a href="https://mathgen.stats.ox.ac.uk/impute/impute_v2.html" class="external-link">IMPUTE2</a> software</li>
<li>The <a href="https://mathgen.stats.ox.ac.uk/impute/1000GP_Phase3.tgz" class="external-link">IMPUTE2 reference panel files</a>
</li>
<li>The <a href="https://www.well.ox.ac.uk/~cfreeman/software/gwas/gtool.html" class="external-link">GTOOL</a> or <a href="https://www.well.ox.ac.uk/~gav/qctool_v2/" class="external-link">QCTOOL</a> tools for file conversion operations.</li>
</ul>
<p>IMPUTE2 by default imputes SNPs over shorter intervals and not over chromosomes. Briefly, the <code><a href="../reference/extendGWAS.html">extendGWAS()</a></code> function:</p>
<ul>
<li>Converts and exports the input <code>GWASExperiment</code> object to files that IMPUTE2 can understand</li>
<li>Creates the imputation intervals over the chromosomes present in the initial object</li>
<li>Paralelly (if possible) executes IMPUTE2 over the imputation intervals</li>
<li>Merges the produced files per chromosome and converts to PLINK format so that they can be read back to a <code>GWASExperiment</code>
</li>
<li>Creates a merged and extended <code>GWASExperiment</code> object with the imputed SNPs and adds the imputation <code>INFO</code> score (imputation accuracy) to the <code>gfeatures</code> of the object</li>
<li>Cleans up</li>
</ul>
<p>The following will perform imputation with writing intermediate files to a temporaray folder without removing intermediate files:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">refSpace</span> <span class="op">&lt;-</span> <span class="st">"PATH/TO/IMPUTE2/REFERENCE/PANEL"</span>
<span class="va">toyEx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extendGWAS.html">extendGWAS</a></span><span class="op">(</span><span class="va">toy</span>,refSpace<span class="op">=</span><span class="va">refSpace</span><span class="op">)</span></code></pre></div>
<p>The following will perform imputation with writing intermediate files to a. user specified folder, using QCTOOL as conversion tool and leaving some intermediate files back:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">refSpace</span> <span class="op">&lt;-</span> <span class="st">"PATH/TO/IMPUTE2/REFERENCE/PANEL"</span>
<span class="va">wspace</span> <span class="op">&lt;-</span> <span class="st">"PATH/TO/WORKSPACE"</span>

<span class="va">toyEx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extendGWAS.html">extendGWAS</a></span><span class="op">(</span><span class="va">toy</span>,wspace<span class="op">=</span><span class="va">wspace</span>,refSpace<span class="op">=</span><span class="va">refSpace</span>,convTool<span class="op">=</span><span class="st">"qctool"</span>,
    cleanup<span class="op">=</span><span class="st">"intemediate"</span><span class="op">)</span></code></pre></div>
<p>The following will perform imputation over intervals of 10Mbp (default is 1Mbp):</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>

<span class="va">refSpace</span> <span class="op">&lt;-</span> <span class="st">"PATH/TO/IMPUTE2/REFERENCE/PANEL"</span>
<span class="va">toyEx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extendGWAS.html">extendGWAS</a></span><span class="op">(</span><span class="va">toy</span>,intSize<span class="op">=</span><span class="fl">1e+7</span>,refSpace<span class="op">=</span><span class="va">refSpace</span><span class="op">)</span></code></pre></div>
<p>The two conversion tools, GTOOL and QCTOOL are both supported because GTOOL may not be compatible with some later Linux versions (e.g. &gt; Ubuntu 18.04). Generally, the process may require many hours to complete (especially for complete datasets) so it would be better to put the command in a script and run the command in the background with <code>Rscript</code>.</p>
<p>The following figure summarizes the imputation modes in PRISMA:</p>
<p><img src="imputation.png" width="800px"></p>
</div>
</div>
<div class="section level2">
<h2 id="dataset-merging">Dataset merging<a class="anchor" aria-label="anchor" href="#dataset-merging"></a>
</h2>
<p>With PRISMA, it is possible to merge two genotype datasets, for example datasets derived from different array platforms or different versions of the same platform or different arrays from the same provider. Merging is not a straigh-forward process as there are several factors - apart from the obvious which is non-common SNPs - such as reversed alleles, multiallelic SNPs with no sufficient representation of mixes in major and minor alleles between different datasets. Although there is software dedicated to the merging and harmonization of different datasets (for example <a href="https://github.com/molgenis/systemsgenetics/wiki/Genotype-Harmonizer" class="external-link">Genotype Harmonizer</a>), PRISMA facilitates also this process by offering a simple <code>GWASExperiment</code> merging function so that users do not have to use multiple packages and complex I/O operations. The merging process is explained below.</p>
<div class="section level3">
<h3 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<p>Generally, some prerequisites are assumed prior to merging:</p>
<ul>
<li>The human genome versions of the two object to be merged must be the same. Use the <code><a href="../reference/gweConvertFuns.html">guessHumanGenomeVersion()</a></code> and <code><a href="../reference/gweConvertFuns.html">GWASExperimentLiftOver()</a></code> functions to check and fix such issues where possible.</li>
<li>The SNP ids from each dataset must be in a common reference format. Ideally, the ids should be converted to dbSNP ids prior to any processing, as in this way the merging process can use online resources to resolve issues. Other ids are also possible like Illumina GSA ids, but online resolving resources cannot be used.</li>
<li>SNP locations must be available.</li>
</ul>
</div>
<div class="section level3">
<h3 id="merging-process">Merging process<a class="anchor" aria-label="anchor" href="#merging-process"></a>
</h3>
<p>This process is also described in the man page of the <code><a href="../reference/mergeGWAS.html">mergeGWAS()</a></code> function.</p>
<p>The following figure summarizes the imputation modes in PRISMA:</p>
<p><img src="merging.png" width="800px"></p>
<div class="section level4">
<h4 id="common-snps">Common SNPs<a class="anchor" aria-label="anchor" href="#common-snps"></a>
</h4>
<p>If the two datasets are compatible in terms of genome version and coordinates, the common SNPs are then identified between the two datasets. As these must point to the same information (location, minor and major alleles, same strand), the process then checks these attributes and tries to resolve any discrepancies.</p>
<p>The first check is whether non identical SNPs in terms of alleles can be solved with strand-fliping. Those that cannot be resolved with strand-fliping may not point to the same risk allele. This is resolved either by using one dataset as a reference (<code>mafResolve=1</code> or <code>mafResolve=2</code>) or even better, by querying online resources (e.g. Ensembl and dbSNP) to identify the true minor and major alleles based on summarized frequencies from population studies (there should be an internet connection for this). SNPs whose alleles cannot be resolved by any means (strand-flip, online MAF resolve) are completely dropped.</p>
<p>If the common SNPs between the two objects continue not being identical, it probably means that there are location/position mismatches potentially derived from older annotations. These can only be resolved online. If after this there are any remaining non-resolvable SNPs, they are dropped. Then, the genotypes of common SNPs are merged. The genotypes of the alleles that were flipped are also switched.</p>
<p>Finally, the common annotation elements and genotypes are combined. If <code>output="common"</code> the process stops. If <code>output="all"</code>, the unique elements of each dataset are also combined with the common. The genotypes of SNPs unique to each dataset are set to missing.</p>
<p>With respect to the online allele resolving of the common SNPs that cannot be resolved with strand-flipping:</p>
<ol style="list-style-type: decimal">
<li>SNPs are queried to <a href="https://www.ensembl.org/index.html" class="external-link">Ensembl</a> with <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html" class="external-link">biomaRt</a> to resolve allele frequencies. Genome version is not importantat this point as only alleles are used together with SNP IDs.</li>
<li>If minor allele is available from Ensembl, then the flip decision is based on it. If not (e.g. outdated/merged SNPs, dbSNP is queried with package <a href="https://cran.r-project.org/web/packages/rsnps/index.html" class="external-link">rsnps</a> and completes the Ensembl result.</li>
<li>Some SNPs may have been removed from dbSNP. Nothing can be done for these. Flipping for these will be based on the majority of the rest.</li>
<li>Some found SNPs may not have information on MAF and minor alleles. dbSNP is requeried with rsnps and filling missing info is retried. Again, some will not be resolved and will be flipped according to majority.</li>
</ol>
<p>Common SNPs that fail to be resolved in any of the aforementioned ways are dropped from the merged dataset.</p>
</div>
<div class="section level4">
<h4 id="non-common-snps">Non-common SNPs<a class="anchor" aria-label="anchor" href="#non-common-snps"></a>
</h4>
<p>With non common SNPs the process is easier as there is nothing to resolve. Non-common samples, SNPs and genotypes are appended to the common ones to form a final dataset. The genotypes of SNP that are unique to the first dataset are set to missing for the samples in the second dataset and vice-versa for SNPs unique to the second dataset. It is normal that this process produces many missing values. Internal imputation can be used to salvage some samples and SNPs from filtering. There is also the option to output only common SNPs to avoid having extreme numbers of missing values in the final dataset. This is to be decided by the user based on the nature of the datasets to be merged.</p>
</div>
<div class="section level4">
<h4 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h4>
<p>A simple example is presented here with the toy dataset, split in two. However, this is not very realistic and in practice, two datasets are merged in the end but with some potential loses.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
    
<span class="va">toy1</span> <span class="op">&lt;-</span> <span class="va">toy</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">800</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>
<span class="va">toy2</span> <span class="op">&lt;-</span> <span class="va">toy</span><span class="op">[</span><span class="fl">201</span><span class="op">:</span><span class="fl">1000</span>,<span class="fl">51</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span>

<span class="va">toym</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mergeGWAS.html">mergeGWAS</a></span><span class="op">(</span><span class="va">toy1</span>,<span class="va">toy2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">toym</span><span class="op">)</span> <span class="co"># 600 100</span>

<span class="va">toym</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mergeGWAS.html">mergeGWAS</a></span><span class="op">(</span><span class="va">toy1</span>,<span class="va">toy2</span>,output<span class="op">=</span><span class="st">"all"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">toym</span><span class="op">)</span> <span class="co"># 1000 100</span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] prisma_0.0.1                SNPRelate_1.28.0           </span>
<span class="co">##  [3] gdsfmt_1.30.0               snpStats_1.44.0            </span>
<span class="co">##  [5] Matrix_1.4-1                survival_3.3-1             </span>
<span class="co">##  [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">## [11] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [15] matrixStats_0.62.0          lassosum_0.4.5             </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] highr_0.9                               </span>
<span class="co">##  [19] knitr_1.39                              </span>
<span class="co">##  [20] rstudioapi_0.13                         </span>
<span class="co">##  [21] robustbase_0.95-0                       </span>
<span class="co">##  [22] TTR_0.24.3                              </span>
<span class="co">##  [23] listenv_0.8.0                           </span>
<span class="co">##  [24] optparse_1.7.1                          </span>
<span class="co">##  [25] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [26] bit64_4.0.5                             </span>
<span class="co">##  [27] rprojroot_2.0.3                         </span>
<span class="co">##  [28] parallelly_1.31.1                       </span>
<span class="co">##  [29] vctrs_0.4.1                             </span>
<span class="co">##  [30] generics_0.1.2                          </span>
<span class="co">##  [31] xfun_0.31                               </span>
<span class="co">##  [32] BiocFileCache_2.2.1                     </span>
<span class="co">##  [33] R6_2.5.1                                </span>
<span class="co">##  [34] rmdformats_1.0.3                        </span>
<span class="co">##  [35] bitops_1.0-7                            </span>
<span class="co">##  [36] cachem_1.0.6                            </span>
<span class="co">##  [37] DelayedArray_0.20.0                     </span>
<span class="co">##  [38] assertthat_0.2.1                        </span>
<span class="co">##  [39] gwasrapidd_0.99.12                      </span>
<span class="co">##  [40] BiocIO_1.4.0                            </span>
<span class="co">##  [41] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [42] scales_1.2.0                            </span>
<span class="co">##  [43] gtable_0.3.0                            </span>
<span class="co">##  [44] globals_0.15.0                          </span>
<span class="co">##  [45] rlang_1.0.2                             </span>
<span class="co">##  [46] systemfonts_1.0.4                       </span>
<span class="co">##  [47] splines_4.1.3                           </span>
<span class="co">##  [48] rtracklayer_1.54.0                      </span>
<span class="co">##  [49] rsnps_0.5.0.0                           </span>
<span class="co">##  [50] sitadela_1.3.1                          </span>
<span class="co">##  [51] BiocManager_1.30.16                     </span>
<span class="co">##  [52] yaml_2.3.5                              </span>
<span class="co">##  [53] reshape2_1.4.4                          </span>
<span class="co">##  [54] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [55] quantmod_0.4.20                         </span>
<span class="co">##  [56] RBGL_1.70.0                             </span>
<span class="co">##  [57] tools_4.1.3                             </span>
<span class="co">##  [58] lava_1.6.10                             </span>
<span class="co">##  [59] bookdown_0.26                           </span>
<span class="co">##  [60] liftOver_1.18.0                         </span>
<span class="co">##  [61] ggplot2_3.3.5                           </span>
<span class="co">##  [62] ellipsis_0.3.2                          </span>
<span class="co">##  [63] kableExtra_1.3.4                        </span>
<span class="co">##  [64] jquerylib_0.1.4                         </span>
<span class="co">##  [65] Rcpp_1.0.8.3                            </span>
<span class="co">##  [66] rrBLUP_4.6.1                            </span>
<span class="co">##  [67] plyr_1.8.7                              </span>
<span class="co">##  [68] progress_1.2.2                          </span>
<span class="co">##  [69] zlibbioc_1.40.0                         </span>
<span class="co">##  [70] purrr_0.3.4                             </span>
<span class="co">##  [71] RCurl_1.98-1.6                          </span>
<span class="co">##  [72] prettyunits_1.1.1                       </span>
<span class="co">##  [73] cowplot_1.1.1                           </span>
<span class="co">##  [74] zoo_1.8-10                              </span>
<span class="co">##  [75] fs_1.5.2                                </span>
<span class="co">##  [76] crul_1.2.0                              </span>
<span class="co">##  [77] magrittr_2.0.3                          </span>
<span class="co">##  [78] data.table_1.14.2                       </span>
<span class="co">##  [79] openxlsx_4.2.5                          </span>
<span class="co">##  [80] mvtnorm_1.1-3                           </span>
<span class="co">##  [81] survcomp_1.44.1                         </span>
<span class="co">##  [82] hms_1.1.1                               </span>
<span class="co">##  [83] evaluate_0.15                           </span>
<span class="co">##  [84] XML_3.99-0.9                            </span>
<span class="co">##  [85] RMTstat_0.3.1                           </span>
<span class="co">##  [86] shape_1.4.6                             </span>
<span class="co">##  [87] compiler_4.1.3                          </span>
<span class="co">##  [88] biomaRt_2.50.3                          </span>
<span class="co">##  [89] tibble_3.1.7                            </span>
<span class="co">##  [90] KernSmooth_2.23-20                      </span>
<span class="co">##  [91] crayon_1.5.1                            </span>
<span class="co">##  [92] R.oo_1.24.0                             </span>
<span class="co">##  [93] htmltools_0.5.2                         </span>
<span class="co">##  [94] pcaPP_2.0-1                             </span>
<span class="co">##  [95] tzdb_0.3.0                              </span>
<span class="co">##  [96] rrcov_1.7-0                             </span>
<span class="co">##  [97] DBI_1.1.2                               </span>
<span class="co">##  [98] SuppDists_1.1-9.7                       </span>
<span class="co">##  [99] dbplyr_2.1.1                            </span>
<span class="co">## [100] MASS_7.3-56                             </span>
<span class="co">## [101] rappdirs_0.3.3                          </span>
<span class="co">## [102] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [103] getopt_1.20.3                           </span>
<span class="co">## [104] readr_2.1.2                             </span>
<span class="co">## [105] rmeta_3.0                               </span>
<span class="co">## [106] cli_3.3.0                               </span>
<span class="co">## [107] quadprog_1.5-8                          </span>
<span class="co">## [108] R.methodsS3_1.8.1                       </span>
<span class="co">## [109] parallel_4.1.3                          </span>
<span class="co">## [110] pkgconfig_2.0.3                         </span>
<span class="co">## [111] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [112] pkgdown_2.0.3.9000                      </span>
<span class="co">## [113] GenomicAlignments_1.30.0                </span>
<span class="co">## [114] signal_0.7-7                            </span>
<span class="co">## [115] xml2_1.3.3                              </span>
<span class="co">## [116] foreach_1.5.2                           </span>
<span class="co">## [117] svglite_2.1.0                           </span>
<span class="co">## [118] bslib_0.3.1                             </span>
<span class="co">## [119] webshot_0.5.3                           </span>
<span class="co">## [120] XVector_0.34.0                          </span>
<span class="co">## [121] prodlim_2019.11.13                      </span>
<span class="co">## [122] rvest_1.0.2                             </span>
<span class="co">## [123] stringr_1.4.0                           </span>
<span class="co">## [124] VariantAnnotation_1.40.0                </span>
<span class="co">## [125] digest_0.6.29                           </span>
<span class="co">## [126] graph_1.72.0                            </span>
<span class="co">## [127] httpcode_0.3.0                          </span>
<span class="co">## [128] Biostrings_2.62.0                       </span>
<span class="co">## [129] rmarkdown_2.14                          </span>
<span class="co">## [130] harmonicmeanp_3.0                       </span>
<span class="co">## [131] restfulr_0.0.13                         </span>
<span class="co">## [132] curl_4.3.2                              </span>
<span class="co">## [133] Rsamtools_2.10.0                        </span>
<span class="co">## [134] rjson_0.2.21                            </span>
<span class="co">## [135] tseries_0.10-51                         </span>
<span class="co">## [136] lifecycle_1.0.1                         </span>
<span class="co">## [137] jsonlite_1.8.0                          </span>
<span class="co">## [138] survivalROC_1.0.3                       </span>
<span class="co">## [139] desc_1.4.1                              </span>
<span class="co">## [140] viridisLite_0.4.0                       </span>
<span class="co">## [141] BSgenome_1.62.0                         </span>
<span class="co">## [142] fansi_1.0.3                             </span>
<span class="co">## [143] pillar_1.7.0                            </span>
<span class="co">## [144] lattice_0.20-45                         </span>
<span class="co">## [145] DEoptimR_1.0-11                         </span>
<span class="co">## [146] KEGGREST_1.34.0                         </span>
<span class="co">## [147] fastmap_1.1.0                           </span>
<span class="co">## [148] httr_1.4.3                              </span>
<span class="co">## [149] GO.db_3.14.0                            </span>
<span class="co">## [150] xts_0.12.1                              </span>
<span class="co">## [151] glue_1.6.2                              </span>
<span class="co">## [152] zip_2.2.0                               </span>
<span class="co">## [153] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [154] png_0.1-7                               </span>
<span class="co">## [155] iterators_1.0.14                        </span>
<span class="co">## [156] pander_0.6.5                            </span>
<span class="co">## [157] glmnet_4.1-4                            </span>
<span class="co">## [158] bit_4.0.4                               </span>
<span class="co">## [159] stringi_1.7.6                           </span>
<span class="co">## [160] sass_0.4.1                              </span>
<span class="co">## [161] bootstrap_2019.6                        </span>
<span class="co">## [162] blob_1.2.3                              </span>
<span class="co">## [163] statgenGWAS_1.0.8                       </span>
<span class="co">## [164] textshaping_0.3.6                       </span>
<span class="co">## [165] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [166] memoise_2.0.1                           </span>
<span class="co">## [167] FMStable_0.1-2                          </span>
<span class="co">## [168] dplyr_1.0.8                             </span>
<span class="co">## [169] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
