<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Single trait association tests and summary statistics â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Single trait association tests and summary statistics">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Single trait association tests and summary statistics</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/06-gwasfuns.Rmd" class="external-link"><code>vignettes/06-gwasfuns.Rmd</code></a></small>
      <div class="d-none name"><code>06-gwasfuns.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>As a one-stop shop solution for the derivation of stable PRSs, PRISMA offers several methods to derive GWAS summary statistics. These summary statistics are then used to feed the PRS extraction process which is evaluated and reported in the end. The user is then able to select the best PRS or the PRS with combines relevant SNP content, reasonable number of SNPs and best evaluation metrics. The p-values of multiple tests can also be combined using one of the available p-value combination methods. See also the <code><a href="../reference/gwaFuns.html">gwa()</a></code> man page for details on the p-value combination.</p>
<p>PRISMA supports five methods for estimating SNP effects for single-trait associations. More specifically:</p>
<ol style="list-style-type: decimal">
<li>General Linear Models (GLM): Each SNP is regressed against the phenotype and the selected covariates. An additive SNP effect model is assumed.</li>
<li>Ridge regression through the R package <a href="https://cran.r-project.org/web/packages/rrBLUP/index.html" class="external-link">rrBLUP</a>. This package deploys penalized regression for shrinking the effects of less informative SNPs in relation with the phenotype.</li>
<li>Single-trait association tests with the <a href="https://biometris.github.io/statgenGWAS/" class="external-link">statgenGWAS</a> R package. The association is based on linear regression and the incorporation of an internally calculated kinship matrix.</li>
<li>Frequentist association tests with <a href="https://www.well.ox.ac.uk/~gav/snptest/" class="external-link">SNPTEST</a> (Oxford). Each SNP is regressed against the phenotype and the selected covariates using an additive model (all other SNPTEST options are also supported).</li>
<li>Association tests with <a href="https://www.cog-genomics.org/plink/" class="external-link">PLINK</a> (linear regression with covariates or simple X<sup>2</sup> association tests if no covariates are present).</li>
</ol>
<p>The SNP effects (regression coefficients and/or Odd Ratios) and p-values are reported in the end and registered to the related slots of the input <code>GWASExperiment</code> object. A <code>GWASExperiment</code> object can hold multiple results for the same genotypic/phenotypic data living in the object, as long as they are not the same models, i.e. a model with 2 PCs as covariates is different from a model with no PC covariates.</p>
<p>In the examples below we use the toy dataset included with PRISMA. For this reason we do not perform additional prior filtering although in a typical situation, genotypes should be filtered prior to any GWA testing. The main function for GWA testing is the <code><a href="../reference/gwaFuns.html">gwa()</a></code> function which attaches the results to the <code>GWASExperiment</code> object. Each GWA function (e.g. <code>gwaGlm</code>) can be used independently but it returns only a matrix of summary statistics. The following sections illustrate their usage.</p>
<div class="section level3">
<h3 id="summary-statistics-with-glm">Summary statistics with GLM<a class="anchor" aria-label="anchor" href="#summary-statistics-with-glm"></a>
</h3>
<p>We test each SNP in the toy dataset for association with the main phenotype (BMI) given age, sex and case/control status, along with 1 PC. Below, we run GLM independently just to get summary statistics without attaching to the <code>toy</code> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

<span class="co"># Only 1 PC will be returned because of dataset size</span>
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Mon Jun  6 20:32:06 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:32:06 2022    Begin (eigenvalues and eigenvectors)
## Mon Jun  6 20:32:06 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

glmSumstat &lt;-<span class="st"> </span><span class="kw">gwaGlm</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),<span class="dt">pcs=</span><span class="ot">TRUE</span>)
## 
## Performing GWAS with GLM over 1 chunks
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Regression  : gaussian (predicted)
## Use PCs     : Yes
##   testing SNPs from rs1009238 to rs2828316
## Found 42 associations (uncorrected p-values).</code></pre></div>
<p>The contents of the <code>glmSumstat</code> matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">glmSumstat</span><span class="op">)</span>
<span class="co">##             Estimate Std. Error    t value   Pr(&gt;|t|)</span>
<span class="co">## rs1009238 -0.6695782  0.6649037 -1.0070303 0.31647832</span>
<span class="co">## rs1940646  0.4810455  0.6722091  0.7156188 0.47598195</span>
<span class="co">## rs1940651 -1.4435680  0.6911187 -2.0887412 0.03940511</span>
<span class="co">## rs1940658 -0.9054984  0.6533306 -1.3859727 0.16900001</span>
<span class="co">## rs1940670  0.8641098  0.6855611  1.2604417 0.21059720</span>
<span class="co">## rs1954825  0.1746603  0.6480141  0.2695316 0.78810519</span></code></pre></div>
<p>Now attach such results to the <code>toy</code> object with and without PCs</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># With 1 PC</span>
<span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 24 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 18 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 18 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 24 associations found out of 657 markers</span>
<span class="co">##   Part 21: 18 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 42 associations with (uncorrected) combined p-values).</span>
    
<span class="co"># With no PCs</span>
<span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 25 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 25 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 16 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 25 associations found out of 657 markers</span>
<span class="co">##   Part 21: 16 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 41 associations with (uncorrected) combined p-values).</span>
    
<span class="co"># With 1 PCs and one more covariate</span>
<span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 54 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 54 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 20 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 20 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 54 associations found out of 657 markers</span>
<span class="co">##   Part 21: 20 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 74 associations with (uncorrected) combined p-values).</span></code></pre></div>
<p>Change the phenotype to <code>Case_Control</code>. Now, GLM will run in binomial regression mode as the <code>Case_Control</code> phenotype is binary.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># With 1 PC</span>
<span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 35 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 19 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 35 associations found out of 657 markers</span>
<span class="co">##   Part 21: 19 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 54 associations with (uncorrected) combined p-values).</span>
    
<span class="co"># With no PCs</span>
<span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 60 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 60 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 28 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 28 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 60 associations found out of 657 markers</span>
<span class="co">##   Part 21: 28 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 88 associations with (uncorrected) combined p-values).</span></code></pre></div>
<p>Now, use the <code><a href="../reference/listAttrFuns.html">listGwa()</a></code> function to see what test results are attached to the <code>toy</code> object:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listAttrFuns.html">listGwa</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span>
<span class="co">## 1 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + PC1 + [SNP]</span>
<span class="co">## 2 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: -</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + [SNP]</span>
<span class="co">## 3 -&gt; Phenotype: BMI | Covariates: Age, Case_Control, Gender | PCs: -</span>
<span class="co">##      Putative formula: BMI ~ Age + Case_Control + Gender + [SNP]</span>
<span class="co">## 4 -&gt; Phenotype: Case_Control | Covariates: Age, BMI, Gender | PCs: 1</span>
<span class="co">##      Putative formula: Case_Control ~ Age + BMI + Gender + PC1 + [SNP]</span>
<span class="co">## 5 -&gt; Phenotype: Case_Control | Covariates: Age, BMI, Gender | PCs: -</span>
<span class="co">##      Putative formula: Case_Control ~ Age + BMI + Gender + [SNP]</span>
<span class="co">## </span>
<span class="co">## Note: The PC variables have putative names based on the number of PCs included in the model.</span>
<span class="co">## For example, instead of PC1, PC2, PC3 the actual PCs may be PC1, PC3, PC4</span></code></pre></div>
<p>In the following examples that illustrate other GWA methods, we will run a smaller number of command except the section where running more than one algorithm is illustrated.</p>
</div>
<div class="section level3">
<h3 id="summary-statistics-with-rrblup">Summary statistics with rrBLUP<a class="anchor" aria-label="anchor" href="#summary-statistics-with-rrblup"></a>
</h3>
<p>We test each SNP in the toy dataset for association with the main phenotype (BMI) given age, sex and case/control status, along with 1 PC. Below, we run rrBLUP independently just to get summary statistics. Please note that rrBLUP is <strong>not</strong> suitable for binary or categorical variables. It works only with continuous variables.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

<span class="co"># Only 1 PC will be returned because of dataset size</span>
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Mon Jun  6 20:33:00 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:33:00 2022    Begin (eigenvalues and eigenvectors)
## Mon Jun  6 20:33:00 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

rrSumstat &lt;-<span class="st"> </span><span class="kw">gwaBlup</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,
    <span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),<span class="dt">usepc=</span><span class="st">"rrint"</span>,<span class="dt">npcs=</span><span class="dv">1</span>)
## 
## Preparing genotypes for rrBLUP...
## Performing GWAS with rrBLUP model
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## Found 55 associations (uncorrected p-values).</code></pre></div>
<p>The contents of the <code>rrSumstat</code> matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rrSumstat</span><span class="op">)</span>
<span class="co">##                 snp chrom bp        BMI        effect      pvalue</span>
<span class="co">## rs1009238 rs1009238    NA NA 0.77883380 -3.734729e-08 0.166404936</span>
<span class="co">## rs1940646 rs1940646    NA NA 0.53061081  2.095420e-08 0.294706141</span>
<span class="co">## rs1940651 rs1940651    NA NA 2.57551170 -6.830224e-08 0.002657592</span>
<span class="co">## rs1940658 rs1940658    NA NA 1.31138453 -4.537714e-08 0.048821989</span>
<span class="co">## rs1940670 rs1940670    NA NA 1.33929350  5.995759e-08 0.045783237</span>
<span class="co">## rs1954825 rs1954825    NA NA 0.01160779 -1.675637e-08 0.973626111</span></code></pre></div>
<p>Also, rrBLUP without PCs:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rrSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwaBlup</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,
    usepc<span class="op">=</span><span class="st">"none"</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Preparing genotypes for rrBLUP...</span>
<span class="co">## Performing GWAS with rrBLUP model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 48 associations (uncorrected p-values).</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summary-statistics-with-statgengwas">Summary statistics with statgenGWAS<a class="anchor" aria-label="anchor" href="#summary-statistics-with-statgengwas"></a>
</h3>
<p>We test each SNP in the toy dataset for association with similar settings as the previous examples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

<span class="co"># Only 1 PC will be returned because of dataset size</span>
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Mon Jun  6 20:33:05 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:33:05 2022    Begin (eigenvalues and eigenvectors)
## Mon Jun  6 20:33:05 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

sgSumstat &lt;-<span class="st"> </span><span class="kw">gwaStatgen</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),
    <span class="dt">pcs=</span><span class="ot">TRUE</span>)
## 
## Preparing data for statgenGWAS...
## Performing GWAS with statgenGWAS model
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## Found 42 associations (uncorrected p-values).</code></pre></div>
<p>The contents of the <code>sgSumstat</code> matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sgSumstat</span><span class="op">)</span>
<span class="co">##               effect  effectSe     pValue</span>
<span class="co">## rs1009238 -0.6695949 0.6649569 0.31646937</span>
<span class="co">## rs1940646  0.4810610 0.6705028 0.47597033</span>
<span class="co">## rs1940651 -1.4435833 0.7031226 0.03940392</span>
<span class="co">## rs1940658 -0.9055187 0.6564590 0.16899183</span>
<span class="co">## rs1940670  0.8641115 0.6876647 0.21059949</span>
<span class="co">## rs1954825  0.1746483 0.6448801 0.78812052</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summary-statistics-with-snptest">Summary statistics with SNPTEST<a class="anchor" aria-label="anchor" href="#summary-statistics-with-snptest"></a>
</h3>
<p>We test each SNP in the toy dataset for association with similar settings as the previous examples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

<span class="co"># Only 1 PC will be returned because of dataset size</span>
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Mon Jun  6 20:33:06 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:33:06 2022    Begin (eigenvalues and eigenvectors)
## Mon Jun  6 20:33:06 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

<span class="cf">if</span> (<span class="kw">Sys.which</span>(<span class="st">"snptest"</span>) <span class="op">!=</span><span class="st"> ""</span>) {
    snpSumstat &lt;-<span class="st"> </span><span class="kw">gwaSnptest</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),
        <span class="dt">pcs=</span><span class="ot">TRUE</span>)
} <span class="cf">else</span> {
    snpSumstat &lt;-<span class="st"> </span><span class="kw">matrix</span>()
}
## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_SLDQQ5898S
## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_SLDQQ5898S
##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_SLDQQ5898S
##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_SLDQQ5898S
## Preflight...
## Writing FAM file to /tmp/RtmpPHwuf6/snptest_SLDQQ5898S/plink_snptest_RZAMU4649H.fam 
## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_SLDQQ5898S/plink_snptest_RZAMU4649H.bim 
## Writing BED file to /tmp/RtmpPHwuf6/snptest_SLDQQ5898S/plink_snptest_RZAMU4649H.bed (SNP-major mode)
## Preparation done!
## 
## Performing GWAS with SNPTEST
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## 
## Executing:
## /home/panos/.local/share/R/snptest-2.5.4/snptest \
##   -data /tmp/RtmpPHwuf6/snptest_SLDQQ5898S/plink_snptest_RZAMU4649H.bed \
##     /tmp/RtmpPHwuf6/snptest_SLDQQ5898S/pheno.sample \
##   -frequentist 1 \
##   -method score \
##   -pheno BMI \
##   -cov_all \
##   -o /tmp/RtmpPHwuf6/snptest_SLDQQ5898S/snptest.out
## Found 38 associations (uncorrected p-values).</code></pre></div>
<p>The contents of the <code>snpSumstat</code> matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">snpSumstat</span><span class="op">)</span>
<span class="co">##               effect       se    pvalue</span>
<span class="co">## rs1009238 -0.1357960 0.134422 0.3123870</span>
<span class="co">## rs1940646  0.1001220 0.137311 0.4659020</span>
<span class="co">## rs1940651 -0.2962890 0.142989 0.0382558</span>
<span class="co">## rs1940658 -0.1868070 0.133843 0.1627990</span>
<span class="co">## rs1940670  0.1679460 0.136085 0.2171560</span>
<span class="co">## rs1954825  0.0347602 0.129139 0.7877990</span></code></pre></div>
<p>SNPTEST with a dominant effect model and adding <code>Case_Control</code> as covariate:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">snpSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwaSnptest</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">snpSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_BYLLO1875F</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_BYLLO1875F</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_BYLLO1875F</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_BYLLO1875F</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_BYLLO1875F/plink_snptest_ZPSCO2758C.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_BYLLO1875F/plink_snptest_ZPSCO2758C.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_BYLLO1875F/plink_snptest_ZPSCO2758C.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_BYLLO1875F/plink_snptest_ZPSCO2758C.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_BYLLO1875F/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_BYLLO1875F/snptest.out</span>
<span class="co">## Found 36 associations (uncorrected p-values).</span></code></pre></div>
<p>SNPTEST with <code>Case_Control</code> as the phenotype. In this case we must convert the <code>Case_Control</code> values from <code>1</code> and <code>2</code> to <code>0</code> and <code>1</code> as SNPTEST complains...</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">phenotypes</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span>
<span class="va">p</span><span class="op">$</span><span class="va">Case_Control2</span> <span class="op">&lt;-</span> <span class="va">p</span><span class="op">$</span><span class="va">Case_Control</span>
<span class="va">p</span><span class="op">$</span><span class="va">Case_Control2</span><span class="op">[</span><span class="va">p</span><span class="op">$</span><span class="va">Case_Control</span><span class="op">==</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>
<span class="va">p</span><span class="op">$</span><span class="va">Case_Control2</span><span class="op">[</span><span class="va">p</span><span class="op">$</span><span class="va">Case_Control</span><span class="op">==</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">phenotypes</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">p</span>

<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">snpSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwaSnptest</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control2"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">snpSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_OGHNM0655C</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_OGHNM0655C</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_OGHNM0655C</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_OGHNM0655C</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_OGHNM0655C/plink_snptest_PTVGW6658Z.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_OGHNM0655C/plink_snptest_PTVGW6658Z.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_OGHNM0655C/plink_snptest_PTVGW6658Z.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : Case_Control2</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_OGHNM0655C/plink_snptest_PTVGW6658Z.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_OGHNM0655C/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno Case_Control2 \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_OGHNM0655C/snptest.out</span>
<span class="co">## Found 63 associations (uncorrected p-values).</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summary-statistics-with-plink">Summary statistics with PLINK<a class="anchor" aria-label="anchor" href="#summary-statistics-with-plink"></a>
</h3>
<p>We test each SNP in the toy dataset for association with similar settings as the previous examples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

<span class="co"># Only 1 PC will be returned because of dataset size</span>
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Mon Jun  6 20:33:18 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:33:18 2022    Begin (eigenvalues and eigenvectors)
## Mon Jun  6 20:33:18 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

<span class="cf">if</span> (<span class="kw">Sys.which</span>(<span class="st">"plink"</span>) <span class="op">!=</span><span class="st"> ""</span>) {
    plSumstat &lt;-<span class="st"> </span><span class="kw">gwaPlink</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),
        <span class="dt">pcs=</span><span class="ot">TRUE</span>)
} <span class="cf">else</span> {
    plSumstat &lt;-<span class="st"> </span><span class="kw">matrix</span>()
}
## plink operations will be executed at /tmp/RtmpPHwuf6/plink_VFMGY3290Q
## Preparing PLINK association run at workspace directory: /tmp/RtmpPHwuf6/plink_VFMGY3290Q
##   writing the PLINK phenotype file...
##   writing the covariates file...
##   writing PLINK genotype files in /tmp/RtmpPHwuf6/plink_VFMGY3290Q
## Preflight...
## Writing FAM file to /tmp/RtmpPHwuf6/plink_VFMGY3290Q/plink_plink_BIEPK7257V.fam 
## Writing extended MAP file to /tmp/RtmpPHwuf6/plink_VFMGY3290Q/plink_plink_BIEPK7257V.bim 
## Writing BED file to /tmp/RtmpPHwuf6/plink_VFMGY3290Q/plink_plink_BIEPK7257V.bed (SNP-major mode)
## Preparation done!
## 
## Performing GWAS with PLINK
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## 
## Executing:
## /home/panos/.local/share/R/plink-1.90/plink \
##   --bfile /tmp/RtmpPHwuf6/plink_VFMGY3290Q/plink_plink_BIEPK7257V \
##   --pheno /tmp/RtmpPHwuf6/plink_VFMGY3290Q/pheno_BIEPK7257V \
##   --allow-no-sex \
##   --ci 0.95 \
##   --seed 42 \
##   --out /tmp/RtmpPHwuf6/plink_VFMGY3290Q/out_BIEPK7257V \
##   --covar /tmp/RtmpPHwuf6/plink_VFMGY3290Q/covar_BIEPK7257V \
##   --linear perm
## Found 43 associations (uncorrected p-values).</code></pre></div>
<p>The contents of the <code>plSumstat</code> matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">plSumstat</span><span class="op">)</span>
<span class="co">##            effect     se    stat pvalue</span>
<span class="co">## rs1009238  0.6696 0.6649  1.0070 0.3390</span>
<span class="co">## rs1940646 -0.4810 0.6722 -0.7156 0.4474</span>
<span class="co">## rs1940651  1.4440 0.6911  2.0890 0.0559</span>
<span class="co">## rs1940658  0.9055 0.6533  1.3860 0.1488</span>
<span class="co">## rs1940670 -0.8641 0.6856 -1.2600 0.2212</span>
<span class="co">## rs1954825 -0.1747 0.6480 -0.2695 1.0000</span></code></pre></div>
<p>PLINK with <code>Case_Control</code> as covariate:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">plSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwaPlink</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">plSumstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## plink operations will be executed at /tmp/RtmpPHwuf6/plink_GNDFN5486O</span>
<span class="co">## Preparing PLINK association run at workspace directory: /tmp/RtmpPHwuf6/plink_GNDFN5486O</span>
<span class="co">##   writing the PLINK phenotype file...</span>
<span class="co">##   writing the covariates file...</span>
<span class="co">##   writing PLINK genotype files in /tmp/RtmpPHwuf6/plink_GNDFN5486O</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/plink_GNDFN5486O/plink_plink_KKRFY4296S.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/plink_GNDFN5486O/plink_plink_KKRFY4296S.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/plink_GNDFN5486O/plink_plink_KKRFY4296S.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with PLINK</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="co">##   --bfile /tmp/RtmpPHwuf6/plink_GNDFN5486O/plink_plink_KKRFY4296S \</span>
<span class="co">##   --pheno /tmp/RtmpPHwuf6/plink_GNDFN5486O/pheno_KKRFY4296S \</span>
<span class="co">##   --allow-no-sex \</span>
<span class="co">##   --ci 0.95 \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out /tmp/RtmpPHwuf6/plink_GNDFN5486O/out_KKRFY4296S \</span>
<span class="co">##   --covar /tmp/RtmpPHwuf6/plink_GNDFN5486O/covar_KKRFY4296S \</span>
<span class="co">##   --logistic beta perm</span>
<span class="co">## Found 54 associations (uncorrected p-values).</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summary-statistics-with-more-than-one-test">Summary statistics with more than one test<a class="anchor" aria-label="anchor" href="#summary-statistics-with-more-than-one-test"></a>
</h3>
<p>Here we demonstrate the usage of the <code><a href="../reference/gwaFuns.html">gwa()</a></code> function which wraps over many tests for summary statistic acquisition and regsiters the results in the respective <code>GWASExperiment</code> slots. We test each SNP in the toy dataset for association with the main phenotype (BMI) given age, sex and case/control status, along with 1 PC. We also try additional associations with several tests and register them to the toy dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

<span class="co"># Only 1 PC will be returned because of dataset size</span>
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Mon Jun  6 20:33:54 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Mon Jun  6 20:33:54 2022    Begin (eigenvalues and eigenvectors)
## Mon Jun  6 20:33:54 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

<span class="cf">if</span> (<span class="kw">Sys.which</span>(<span class="st">"snptest"</span>) <span class="op">!=</span><span class="st"> ""</span>) {
    toy &lt;-<span class="st"> </span><span class="kw">gwa</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),<span class="dt">pcs=</span><span class="ot">TRUE</span>,
        <span class="dt">psig=</span><span class="fl">0.05</span>,<span class="dt">methods=</span><span class="kw">c</span>(<span class="st">"glm"</span>,<span class="st">"statgen"</span>,<span class="st">"snptest"</span>))
} <span class="cf">else</span> {
    toy &lt;-<span class="st"> </span><span class="kw">gwa</span>(toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>),<span class="dt">pcs=</span><span class="ot">TRUE</span>,
        <span class="dt">psig=</span><span class="fl">0.05</span>,<span class="dt">methods=</span><span class="kw">c</span>(<span class="st">"glm"</span>,<span class="st">"statgen"</span>))
}
## 
## Starting GWA analysis in 2 chunks with 3 methods: glm, statgen, snptest
## 
## ========== Analyzing chromosome/part 18
## 
## Performing GWAS with GLM over 1 chunks
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Regression  : gaussian (predicted)
## Use PCs     : Yes
##   testing SNPs from rs1009238 to rs1568327
## Found 24 associations (uncorrected p-values).
## 
## Preparing data for statgenGWAS...
## Performing GWAS with statgenGWAS model
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## Found 24 associations (uncorrected p-values).
## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_COQKD2277R
## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_COQKD2277R
##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_COQKD2277R
##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_COQKD2277R
## Preflight...
## Writing FAM file to /tmp/RtmpPHwuf6/snptest_COQKD2277R/plink_snptest_UFSHT8740V.fam 
## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_COQKD2277R/plink_snptest_UFSHT8740V.bim 
## Writing BED file to /tmp/RtmpPHwuf6/snptest_COQKD2277R/plink_snptest_UFSHT8740V.bed (SNP-major mode)
## Preparation done!
## 
## Performing GWAS with SNPTEST
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## 
## Executing:
## /home/panos/.local/share/R/snptest-2.5.4/snptest \
##   -data /tmp/RtmpPHwuf6/snptest_COQKD2277R/plink_snptest_UFSHT8740V.bed \
##     /tmp/RtmpPHwuf6/snptest_COQKD2277R/pheno.sample \
##   -frequentist 1 \
##   -method score \
##   -pheno BMI \
##   -cov_all \
##   -o /tmp/RtmpPHwuf6/snptest_COQKD2277R/snptest.out
## Found 22 associations (uncorrected p-values).
## 
## Combining p-values from multi-GWAS using fisher method
## 
## Overall, found 72 associations with (uncorrected) combined p-values).
## 
## ========== Analyzing chromosome/part 21
## 
## Performing GWAS with GLM over 1 chunks
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Regression  : gaussian (predicted)
## Use PCs     : Yes
##   testing SNPs from rs2824974 to rs2828316
## Found 18 associations (uncorrected p-values).
## 
## Preparing data for statgenGWAS...
## Performing GWAS with statgenGWAS model
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## Found 18 associations (uncorrected p-values).
## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_QPVCQ0883T
## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_QPVCQ0883T
##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_QPVCQ0883T
##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_QPVCQ0883T
## Preflight...
## Writing FAM file to /tmp/RtmpPHwuf6/snptest_QPVCQ0883T/plink_snptest_EQHBW5154V.fam 
## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_QPVCQ0883T/plink_snptest_EQHBW5154V.bim 
## Writing BED file to /tmp/RtmpPHwuf6/snptest_QPVCQ0883T/plink_snptest_EQHBW5154V.bed (SNP-major mode)
## Preparation done!
## 
## Performing GWAS with SNPTEST
## Trait(s)    : BMI
## Covariate(s): Age, Gender
## Use PCs     : Yes
## 
## Executing:
## /home/panos/.local/share/R/snptest-2.5.4/snptest \
##   -data /tmp/RtmpPHwuf6/snptest_QPVCQ0883T/plink_snptest_EQHBW5154V.bed \
##     /tmp/RtmpPHwuf6/snptest_QPVCQ0883T/pheno.sample \
##   -frequentist 1 \
##   -method score \
##   -pheno BMI \
##   -cov_all \
##   -o /tmp/RtmpPHwuf6/snptest_QPVCQ0883T/snptest.out
## Found 16 associations (uncorrected p-values).
## 
## Combining p-values from multi-GWAS using fisher method
## 
## Overall, found 42 associations with (uncorrected) combined p-values).
## 
## Finished, the following putative associations were found per part:
##   Part 18: 72 associations found out of 657 markers
##   Part 21: 42 associations found out of 343 markers
## 
## Overall, found 114 associations with (uncorrected) combined p-values).</code></pre></div>
<p>Additional models without PCs:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">FALSE</span>,
        psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span>,<span class="st">"snptest"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">FALSE</span>,
        psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 3 methods: glm, statgen, snptest</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 25 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 25 associations (uncorrected p-values).</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_FGJRT6809P</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_FGJRT6809P</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_FGJRT6809P</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_FGJRT6809P</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_FGJRT6809P/plink_snptest_XAHDO5210E.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_FGJRT6809P/plink_snptest_XAHDO5210E.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_FGJRT6809P/plink_snptest_XAHDO5210E.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_FGJRT6809P/plink_snptest_XAHDO5210E.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_FGJRT6809P/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_FGJRT6809P/snptest.out</span>
<span class="co">## Found 28 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 71 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_ODLGM4531L</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_ODLGM4531L</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_ODLGM4531L</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_ODLGM4531L</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_ODLGM4531L/plink_snptest_QFZPA8763Z.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_ODLGM4531L/plink_snptest_QFZPA8763Z.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_ODLGM4531L/plink_snptest_QFZPA8763Z.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_ODLGM4531L/plink_snptest_QFZPA8763Z.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_ODLGM4531L/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_ODLGM4531L/snptest.out</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 40 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 71 associations found out of 657 markers</span>
<span class="co">##   Part 21: 40 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 111 associations with (uncorrected) combined p-values).</span></code></pre></div>
<p>Let's add <code>Case_Control</code> in the covariates:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># With 1 PC</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span>,<span class="st">"snptest"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 3 methods: glm, statgen, snptest</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 22 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 53 associations (uncorrected p-values).</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_EPPXW5838M</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_EPPXW5838M</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_EPPXW5838M</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_EPPXW5838M</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_EPPXW5838M/plink_snptest_FVSNY4040A.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_EPPXW5838M/plink_snptest_FVSNY4040A.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_EPPXW5838M/plink_snptest_FVSNY4040A.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_EPPXW5838M/plink_snptest_FVSNY4040A.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_EPPXW5838M/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_EPPXW5838M/snptest.out</span>
<span class="co">## Found 22 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 71 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_WRGTO6972S</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_WRGTO6972S</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_WRGTO6972S</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_WRGTO6972S</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_WRGTO6972S/plink_snptest_LTIDH4684E.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_WRGTO6972S/plink_snptest_LTIDH4684E.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_WRGTO6972S/plink_snptest_LTIDH4684E.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_WRGTO6972S/plink_snptest_LTIDH4684E.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_WRGTO6972S/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_WRGTO6972S/snptest.out</span>
<span class="co">## Found 14 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 36 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 71 associations found out of 657 markers</span>
<span class="co">##   Part 21: 36 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 107 associations with (uncorrected) combined p-values).</span>

<span class="co"># With no PCs</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span>,<span class="st">"snptest"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 3 methods: glm, statgen, snptest</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 54 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 23 associations (uncorrected p-values).</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_PRMMF1810Y</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_PRMMF1810Y</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_PRMMF1810Y</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_PRMMF1810Y</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_PRMMF1810Y/plink_snptest_DKVSR3365R.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_PRMMF1810Y/plink_snptest_DKVSR3365R.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_PRMMF1810Y/plink_snptest_DKVSR3365R.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_PRMMF1810Y/plink_snptest_DKVSR3365R.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_PRMMF1810Y/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_PRMMF1810Y/snptest.out</span>
<span class="co">## Found 22 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 103 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 20 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## snptest operations will be executed at /tmp/RtmpPHwuf6/snptest_IFMWK3677M</span>
<span class="co">## Preparing SNPTEST association run at workspace directory: /tmp/RtmpPHwuf6/snptest_IFMWK3677M</span>
<span class="co">##  writing SNPTEST sample file in /tmp/RtmpPHwuf6/snptest_IFMWK3677M</span>
<span class="co">##   writing PLINK files in /tmp/RtmpPHwuf6/snptest_IFMWK3677M</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/snptest_IFMWK3677M/plink_snptest_JNNHN6437Z.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/snptest_IFMWK3677M/plink_snptest_JNNHN6437Z.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/snptest_IFMWK3677M/plink_snptest_JNNHN6437Z.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with SNPTEST</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="co">##   -data /tmp/RtmpPHwuf6/snptest_IFMWK3677M/plink_snptest_JNNHN6437Z.bed \</span>
<span class="co">##     /tmp/RtmpPHwuf6/snptest_IFMWK3677M/pheno.sample \</span>
<span class="co">##   -frequentist 1 \</span>
<span class="co">##   -method score \</span>
<span class="co">##   -pheno BMI \</span>
<span class="co">##   -cov_all \</span>
<span class="co">##   -o /tmp/RtmpPHwuf6/snptest_IFMWK3677M/snptest.out</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 44 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 103 associations found out of 657 markers</span>
<span class="co">##   Part 21: 44 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 147 associations with (uncorrected) combined p-values).</span></code></pre></div>
<p>Change the phenotype to <code>Case_Control</code> and change SNPTEST to PLINK.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># With 1 PC</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span>,<span class="st">"plink"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 3 methods: glm, statgen, plink</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 35 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 33 associations (uncorrected p-values).</span>
<span class="co">## plink operations will be executed at /tmp/RtmpPHwuf6/plink_CRUVJ8812F</span>
<span class="co">## Preparing PLINK association run at workspace directory: /tmp/RtmpPHwuf6/plink_CRUVJ8812F</span>
<span class="co">##   writing the PLINK phenotype file...</span>
<span class="co">##   writing the covariates file...</span>
<span class="co">##   writing PLINK genotype files in /tmp/RtmpPHwuf6/plink_CRUVJ8812F</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/plink_CRUVJ8812F/plink_plink_QBLSL8132T.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/plink_CRUVJ8812F/plink_plink_QBLSL8132T.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/plink_CRUVJ8812F/plink_plink_QBLSL8132T.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with PLINK</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="co">##   --bfile /tmp/RtmpPHwuf6/plink_CRUVJ8812F/plink_plink_QBLSL8132T \</span>
<span class="co">##   --pheno /tmp/RtmpPHwuf6/plink_CRUVJ8812F/pheno_QBLSL8132T \</span>
<span class="co">##   --allow-no-sex \</span>
<span class="co">##   --ci 0.95 \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out /tmp/RtmpPHwuf6/plink_CRUVJ8812F/out_QBLSL8132T \</span>
<span class="co">##   --covar /tmp/RtmpPHwuf6/plink_CRUVJ8812F/covar_QBLSL8132T \</span>
<span class="co">##   --logistic beta perm</span>
<span class="co">## Found 36 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 79 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 22 associations (uncorrected p-values).</span>
<span class="co">## plink operations will be executed at /tmp/RtmpPHwuf6/plink_YFGBR5935O</span>
<span class="co">## Preparing PLINK association run at workspace directory: /tmp/RtmpPHwuf6/plink_YFGBR5935O</span>
<span class="co">##   writing the PLINK phenotype file...</span>
<span class="co">##   writing the covariates file...</span>
<span class="co">##   writing PLINK genotype files in /tmp/RtmpPHwuf6/plink_YFGBR5935O</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/plink_YFGBR5935O/plink_plink_ZWXDN2389E.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/plink_YFGBR5935O/plink_plink_ZWXDN2389E.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/plink_YFGBR5935O/plink_plink_ZWXDN2389E.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with PLINK</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="co">##   --bfile /tmp/RtmpPHwuf6/plink_YFGBR5935O/plink_plink_ZWXDN2389E \</span>
<span class="co">##   --pheno /tmp/RtmpPHwuf6/plink_YFGBR5935O/pheno_ZWXDN2389E \</span>
<span class="co">##   --allow-no-sex \</span>
<span class="co">##   --ci 0.95 \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out /tmp/RtmpPHwuf6/plink_YFGBR5935O/out_ZWXDN2389E \</span>
<span class="co">##   --covar /tmp/RtmpPHwuf6/plink_YFGBR5935O/covar_ZWXDN2389E \</span>
<span class="co">##   --logistic beta perm</span>
<span class="co">## Found 18 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 41 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 79 associations found out of 657 markers</span>
<span class="co">##   Part 21: 41 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 120 associations with (uncorrected) combined p-values).</span>

<span class="co"># With no PCs</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span>,<span class="st">"plink"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 3 methods: glm, statgen, plink</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 60 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 31 associations (uncorrected p-values).</span>
<span class="co">## plink operations will be executed at /tmp/RtmpPHwuf6/plink_SNTJN9609W</span>
<span class="co">## Preparing PLINK association run at workspace directory: /tmp/RtmpPHwuf6/plink_SNTJN9609W</span>
<span class="co">##   writing the PLINK phenotype file...</span>
<span class="co">##   writing the covariates file...</span>
<span class="co">##   writing PLINK genotype files in /tmp/RtmpPHwuf6/plink_SNTJN9609W</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/plink_SNTJN9609W/plink_plink_ILQRX2634W.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/plink_SNTJN9609W/plink_plink_ILQRX2634W.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/plink_SNTJN9609W/plink_plink_ILQRX2634W.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with PLINK</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="co">##   --bfile /tmp/RtmpPHwuf6/plink_SNTJN9609W/plink_plink_ILQRX2634W \</span>
<span class="co">##   --pheno /tmp/RtmpPHwuf6/plink_SNTJN9609W/pheno_ILQRX2634W \</span>
<span class="co">##   --allow-no-sex \</span>
<span class="co">##   --ci 0.95 \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out /tmp/RtmpPHwuf6/plink_SNTJN9609W/out_ILQRX2634W \</span>
<span class="co">##   --covar /tmp/RtmpPHwuf6/plink_SNTJN9609W/covar_ILQRX2634W \</span>
<span class="co">##   --logistic beta perm</span>
<span class="co">## Found 32 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 103 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Regression  : binomial (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 28 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 20 associations (uncorrected p-values).</span>
<span class="co">## plink operations will be executed at /tmp/RtmpPHwuf6/plink_CZNMI1328N</span>
<span class="co">## Preparing PLINK association run at workspace directory: /tmp/RtmpPHwuf6/plink_CZNMI1328N</span>
<span class="co">##   writing the PLINK phenotype file...</span>
<span class="co">##   writing the covariates file...</span>
<span class="co">##   writing PLINK genotype files in /tmp/RtmpPHwuf6/plink_CZNMI1328N</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpPHwuf6/plink_CZNMI1328N/plink_plink_PZZQS2762N.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpPHwuf6/plink_CZNMI1328N/plink_plink_PZZQS2762N.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpPHwuf6/plink_CZNMI1328N/plink_plink_PZZQS2762N.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with PLINK</span>
<span class="co">## Trait(s)    : Case_Control</span>
<span class="co">## Covariate(s): Age, Gender, BMI</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="co">##   --bfile /tmp/RtmpPHwuf6/plink_CZNMI1328N/plink_plink_PZZQS2762N \</span>
<span class="co">##   --pheno /tmp/RtmpPHwuf6/plink_CZNMI1328N/pheno_PZZQS2762N \</span>
<span class="co">##   --allow-no-sex \</span>
<span class="co">##   --ci 0.95 \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out /tmp/RtmpPHwuf6/plink_CZNMI1328N/out_PZZQS2762N \</span>
<span class="co">##   --covar /tmp/RtmpPHwuf6/plink_CZNMI1328N/covar_PZZQS2762N \</span>
<span class="co">##   --logistic beta perm</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 49 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 103 associations found out of 657 markers</span>
<span class="co">##   Part 21: 49 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 152 associations with (uncorrected) combined p-values).</span></code></pre></div>
<p>Now, use the <code><a href="../reference/listAttrFuns.html">listGwa()</a></code> function to see what test results are attached to the <code>toy</code> object:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listAttrFuns.html">listGwa</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span>
<span class="co">## 1 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + PC1 + [SNP]</span>
<span class="co">## 2 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: -</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + [SNP]</span>
<span class="co">## 3 -&gt; Phenotype: BMI | Covariates: Age, Case_Control, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Case_Control + Gender + PC1 + [SNP]</span>
<span class="co">## 4 -&gt; Phenotype: BMI | Covariates: Age, Case_Control, Gender | PCs: -</span>
<span class="co">##      Putative formula: BMI ~ Age + Case_Control + Gender + [SNP]</span>
<span class="co">## 5 -&gt; Phenotype: Case_Control | Covariates: Age, BMI, Gender | PCs: 1</span>
<span class="co">##      Putative formula: Case_Control ~ Age + BMI + Gender + PC1 + [SNP]</span>
<span class="co">## 6 -&gt; Phenotype: Case_Control | Covariates: Age, BMI, Gender | PCs: -</span>
<span class="co">##      Putative formula: Case_Control ~ Age + BMI + Gender + [SNP]</span>
<span class="co">## </span>
<span class="co">## Note: The PC variables have putative names based on the number of PCs included in the model.</span>
<span class="co">## For example, instead of PC1, PC2, PC3 the actual PCs may be PC1, PC3, PC4</span></code></pre></div>
<p>Let's see the effects and p-values for some of the attached models. Let's start with the first one where the phenotype was <code>BMI</code> and the covariates were <code>Age</code> and <code>Gender</code> along with 1 PC:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">pvalues</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">##                  glm    statgen   snptest      fisher</span>
<span class="co">## rs1009238 0.31647832 0.31646353 0.3123870 0.327451895</span>
<span class="co">## rs1940646 0.47598195 0.47596004 0.4659020 0.609719029</span>
<span class="co">## rs1940651 0.03940511 0.03940241 0.0382558 0.003449833</span>
<span class="co">## rs1940658 0.16900001 0.16898846 0.1627990 0.096683730</span>
<span class="co">## rs1940670 0.21059720 0.21060807 0.2171560 0.158151740</span>
<span class="co">## rs1954825 0.78810519 0.78809447 0.7877990 0.964036429</span>

<span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">effects</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span>
<span class="co">##                  glm    statgen    snptest</span>
<span class="co">## rs1009238 -0.6695782 -0.6696046 -0.1357960</span>
<span class="co">## rs1940646  0.4810455  0.4810737  0.1001220</span>
<span class="co">## rs1940651 -1.4435680 -1.4435983 -0.2962890</span>
<span class="co">## rs1940658 -0.9054984 -0.9055274 -0.1868070</span>
<span class="co">## rs1940670  0.8641098  0.8640962  0.1679460</span>
<span class="co">## rs1954825  0.1746603  0.1746709  0.0347602</span></code></pre></div>
<p>You will notice that the is an additional column named <code>fisher</code> to the matrix of p-values. This represents the combined p-value out of three tests using the <a href="https://en.wikipedia.org/wiki/Fisher%27s_method" class="external-link">Fisher's p-value combination method</a>. The <code>e</code> matrix provides the regression coefficients for each method and SNP. These are used later for the calculation of weighted PRSs.</p>
<p>Furthermore, let's access the case where the phenotype was <code>BMI</code> and the covariates were <code>Age</code>, <code>Gender</code> and <code>Case_Control</code> with no PCs:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">pvalues</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">##                  glm    statgen   snptest       fisher</span>
<span class="co">## rs1009238 0.14212106 0.14211747 0.1395850 0.0679656741</span>
<span class="co">## rs1940646 0.08739586 0.08738919 0.0844680 0.0227911189</span>
<span class="co">## rs1940651 0.01375438 0.01375287 0.0136554 0.0002496301</span>
<span class="co">## rs1940658 0.18657047 0.18656332 0.1805250 0.1188935709</span>
<span class="co">## rs1940670 0.40166435 0.40168686 0.4052340 0.4869093573</span>
<span class="co">## rs1954825 0.77932398 0.77934969 0.7769420 0.9593598078</span>

<span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">effects</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span>
<span class="co">##                  glm    statgen    snptest</span>
<span class="co">## rs1009238 -0.7981357 -0.7981498 -0.1621830</span>
<span class="co">## rs1940646  0.9427230  0.9427508  0.1934790</span>
<span class="co">## rs1940651 -1.4033121 -1.4033440 -0.2879970</span>
<span class="co">## rs1940658 -0.7112943 -0.7113093 -0.1463780</span>
<span class="co">## rs1940670  0.4717259  0.4717071  0.0925017</span>
<span class="co">## rs1954825 -0.1472900 -0.1472737 -0.0298792</span></code></pre></div>
<p>Let's also access the case where the phenotype was <code>BMI</code> and the covariate was <code>Age</code> with no PCs:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">pvalues</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="st">"Age"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">## NULL</span>
<span class="co"># NULL</span></code></pre></div>
<p>Whops! We never did fit that model...</p>
<p>Finally, let's also see the case where the phenotype was <code>Case_Control</code> and the covariates were <code>Age</code>, <code>Gender</code>, <code>BMI</code> with 1 PC:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">pvalues</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
    npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">##                 glm    statgen  plink     fisher</span>
<span class="co">## rs1009238 0.5049708 0.26931142 0.5385 0.51489328</span>
<span class="co">## rs1940646 0.1200773 0.05172862 0.1295 0.02695589</span>
<span class="co">## rs1940651 0.2967032 0.16564973 0.3519 0.22982557</span>
<span class="co">## rs1940658 0.9764899 0.75390414 0.7778 0.98086897</span>
<span class="co">## rs1940670 0.8260349 0.70498567 0.8571 0.96647635</span>
<span class="co">## rs1954825 0.3453345 0.43659563 0.3393 0.42929793</span>

<span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">effects</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"Case_Control"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"BMI"</span><span class="op">)</span>,
    npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span>
<span class="co">##                   glm     statgen    plink</span>
<span class="co">## rs1009238  0.33192525  0.05973733 -0.33190</span>
<span class="co">## rs1940646 -0.86898631 -0.10506731  0.86900</span>
<span class="co">## rs1940651  0.59339897  0.08035293 -0.59340</span>
<span class="co">## rs1940658 -0.01394122  0.01677897  0.01394</span>
<span class="co">## rs1940670  0.11419298  0.02124142 -0.11420</span>
<span class="co">## rs1954825  0.46707205  0.04064121 -0.46710</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] prisma_0.0.2                SNPRelate_1.28.0           </span>
<span class="co">##  [3] gdsfmt_1.30.0               snpStats_1.44.0            </span>
<span class="co">##  [5] Matrix_1.4-1                survival_3.3-1             </span>
<span class="co">##  [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">## [11] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [15] matrixStats_0.62.0          lassosum_0.4.5             </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] knitr_1.39                              </span>
<span class="co">##  [19] rstudioapi_0.13                         </span>
<span class="co">##  [20] robustbase_0.95-0                       </span>
<span class="co">##  [21] TTR_0.24.3                              </span>
<span class="co">##  [22] listenv_0.8.0                           </span>
<span class="co">##  [23] optparse_1.7.1                          </span>
<span class="co">##  [24] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [25] bit64_4.0.5                             </span>
<span class="co">##  [26] rprojroot_2.0.3                         </span>
<span class="co">##  [27] parallelly_1.31.1                       </span>
<span class="co">##  [28] vctrs_0.4.1                             </span>
<span class="co">##  [29] generics_0.1.2                          </span>
<span class="co">##  [30] xfun_0.31                               </span>
<span class="co">##  [31] BiocFileCache_2.2.1                     </span>
<span class="co">##  [32] R6_2.5.1                                </span>
<span class="co">##  [33] rmdformats_1.0.3                        </span>
<span class="co">##  [34] bitops_1.0-7                            </span>
<span class="co">##  [35] cachem_1.0.6                            </span>
<span class="co">##  [36] DelayedArray_0.20.0                     </span>
<span class="co">##  [37] assertthat_0.2.1                        </span>
<span class="co">##  [38] gwasrapidd_0.99.12                      </span>
<span class="co">##  [39] BiocIO_1.4.0                            </span>
<span class="co">##  [40] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [41] scales_1.2.0                            </span>
<span class="co">##  [42] gtable_0.3.0                            </span>
<span class="co">##  [43] globals_0.15.0                          </span>
<span class="co">##  [44] rlang_1.0.2                             </span>
<span class="co">##  [45] systemfonts_1.0.4                       </span>
<span class="co">##  [46] splines_4.1.3                           </span>
<span class="co">##  [47] rtracklayer_1.54.0                      </span>
<span class="co">##  [48] rsnps_0.5.0.0                           </span>
<span class="co">##  [49] sitadela_1.3.1                          </span>
<span class="co">##  [50] BiocManager_1.30.16                     </span>
<span class="co">##  [51] yaml_2.3.5                              </span>
<span class="co">##  [52] reshape2_1.4.4                          </span>
<span class="co">##  [53] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [54] quantmod_0.4.20                         </span>
<span class="co">##  [55] RBGL_1.70.0                             </span>
<span class="co">##  [56] tools_4.1.3                             </span>
<span class="co">##  [57] lava_1.6.10                             </span>
<span class="co">##  [58] bookdown_0.26                           </span>
<span class="co">##  [59] liftOver_1.18.0                         </span>
<span class="co">##  [60] ggplot2_3.3.5                           </span>
<span class="co">##  [61] ellipsis_0.3.2                          </span>
<span class="co">##  [62] kableExtra_1.3.4                        </span>
<span class="co">##  [63] jquerylib_0.1.4                         </span>
<span class="co">##  [64] Rcpp_1.0.8.3                            </span>
<span class="co">##  [65] rrBLUP_4.6.1                            </span>
<span class="co">##  [66] plyr_1.8.7                              </span>
<span class="co">##  [67] progress_1.2.2                          </span>
<span class="co">##  [68] zlibbioc_1.40.0                         </span>
<span class="co">##  [69] purrr_0.3.4                             </span>
<span class="co">##  [70] RCurl_1.98-1.6                          </span>
<span class="co">##  [71] prettyunits_1.1.1                       </span>
<span class="co">##  [72] cowplot_1.1.1                           </span>
<span class="co">##  [73] zoo_1.8-10                              </span>
<span class="co">##  [74] fs_1.5.2                                </span>
<span class="co">##  [75] crul_1.2.0                              </span>
<span class="co">##  [76] magrittr_2.0.3                          </span>
<span class="co">##  [77] data.table_1.14.2                       </span>
<span class="co">##  [78] openxlsx_4.2.5                          </span>
<span class="co">##  [79] mvtnorm_1.1-3                           </span>
<span class="co">##  [80] survcomp_1.44.1                         </span>
<span class="co">##  [81] hms_1.1.1                               </span>
<span class="co">##  [82] evaluate_0.15                           </span>
<span class="co">##  [83] XML_3.99-0.9                            </span>
<span class="co">##  [84] RMTstat_0.3.1                           </span>
<span class="co">##  [85] shape_1.4.6                             </span>
<span class="co">##  [86] compiler_4.1.3                          </span>
<span class="co">##  [87] biomaRt_2.50.3                          </span>
<span class="co">##  [88] tibble_3.1.7                            </span>
<span class="co">##  [89] KernSmooth_2.23-20                      </span>
<span class="co">##  [90] crayon_1.5.1                            </span>
<span class="co">##  [91] R.oo_1.24.0                             </span>
<span class="co">##  [92] htmltools_0.5.2                         </span>
<span class="co">##  [93] pcaPP_2.0-1                             </span>
<span class="co">##  [94] tzdb_0.3.0                              </span>
<span class="co">##  [95] rrcov_1.7-0                             </span>
<span class="co">##  [96] DBI_1.1.2                               </span>
<span class="co">##  [97] SuppDists_1.1-9.7                       </span>
<span class="co">##  [98] dbplyr_2.1.1                            </span>
<span class="co">##  [99] MASS_7.3-56                             </span>
<span class="co">## [100] rappdirs_0.3.3                          </span>
<span class="co">## [101] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [102] getopt_1.20.3                           </span>
<span class="co">## [103] readr_2.1.2                             </span>
<span class="co">## [104] rmeta_3.0                               </span>
<span class="co">## [105] cli_3.3.0                               </span>
<span class="co">## [106] quadprog_1.5-8                          </span>
<span class="co">## [107] R.methodsS3_1.8.1                       </span>
<span class="co">## [108] parallel_4.1.3                          </span>
<span class="co">## [109] pkgconfig_2.0.3                         </span>
<span class="co">## [110] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [111] pkgdown_2.0.3.9000                      </span>
<span class="co">## [112] GenomicAlignments_1.30.0                </span>
<span class="co">## [113] signal_0.7-7                            </span>
<span class="co">## [114] xml2_1.3.3                              </span>
<span class="co">## [115] foreach_1.5.2                           </span>
<span class="co">## [116] svglite_2.1.0                           </span>
<span class="co">## [117] bslib_0.3.1                             </span>
<span class="co">## [118] webshot_0.5.3                           </span>
<span class="co">## [119] XVector_0.34.0                          </span>
<span class="co">## [120] prodlim_2019.11.13                      </span>
<span class="co">## [121] rvest_1.0.2                             </span>
<span class="co">## [122] stringr_1.4.0                           </span>
<span class="co">## [123] VariantAnnotation_1.40.0                </span>
<span class="co">## [124] digest_0.6.29                           </span>
<span class="co">## [125] graph_1.72.0                            </span>
<span class="co">## [126] httpcode_0.3.0                          </span>
<span class="co">## [127] Biostrings_2.62.0                       </span>
<span class="co">## [128] rmarkdown_2.14                          </span>
<span class="co">## [129] harmonicmeanp_3.0                       </span>
<span class="co">## [130] restfulr_0.0.13                         </span>
<span class="co">## [131] curl_4.3.2                              </span>
<span class="co">## [132] Rsamtools_2.10.0                        </span>
<span class="co">## [133] rjson_0.2.21                            </span>
<span class="co">## [134] tseries_0.10-51                         </span>
<span class="co">## [135] lifecycle_1.0.1                         </span>
<span class="co">## [136] jsonlite_1.8.0                          </span>
<span class="co">## [137] survivalROC_1.0.3                       </span>
<span class="co">## [138] desc_1.4.1                              </span>
<span class="co">## [139] viridisLite_0.4.0                       </span>
<span class="co">## [140] BSgenome_1.62.0                         </span>
<span class="co">## [141] fansi_1.0.3                             </span>
<span class="co">## [142] pillar_1.7.0                            </span>
<span class="co">## [143] lattice_0.20-45                         </span>
<span class="co">## [144] DEoptimR_1.0-11                         </span>
<span class="co">## [145] KEGGREST_1.34.0                         </span>
<span class="co">## [146] fastmap_1.1.0                           </span>
<span class="co">## [147] httr_1.4.3                              </span>
<span class="co">## [148] GO.db_3.14.0                            </span>
<span class="co">## [149] xts_0.12.1                              </span>
<span class="co">## [150] glue_1.6.2                              </span>
<span class="co">## [151] zip_2.2.0                               </span>
<span class="co">## [152] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [153] png_0.1-7                               </span>
<span class="co">## [154] iterators_1.0.14                        </span>
<span class="co">## [155] pander_0.6.5                            </span>
<span class="co">## [156] glmnet_4.1-4                            </span>
<span class="co">## [157] bit_4.0.4                               </span>
<span class="co">## [158] stringi_1.7.6                           </span>
<span class="co">## [159] sass_0.4.1                              </span>
<span class="co">## [160] bootstrap_2019.6                        </span>
<span class="co">## [161] blob_1.2.3                              </span>
<span class="co">## [162] statgenGWAS_1.0.8                       </span>
<span class="co">## [163] textshaping_0.3.6                       </span>
<span class="co">## [164] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [165] memoise_2.0.1                           </span>
<span class="co">## [166] FMStable_0.1-2                          </span>
<span class="co">## [167] dplyr_1.0.8                             </span>
<span class="co">## [168] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
