<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Derivation of PRS candidates and evaluation â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Derivation of PRS candidates and evaluation">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Derivation of PRS candidates and evaluation</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/07-prsfuns.Rmd" class="external-link"><code>vignettes/07-prsfuns.Rmd</code></a></small>
      <div class="d-none name"><code>07-prsfuns.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The main purpose of PRISMA development was to provide an efficient wrapper that would integrate the several steps required for the derivation of PRSs, from the derivation of summary statistics in an iterative fashion for smaller datsets, up to the actual extraction and evaluation of PRS, all in an iterative and stabilizing way. After the derivation of the summary statistics, this step concerns the actual PRS extraction from the genotypes and summary statistics, for a queried phenotype and covariates. The user can perform PRS analysis with the available algorithms supported by PRISMA and gather the output data in easily interpretable formats, which can be used for PRS evaluation.</p>
<p>PRISMA currently supports two methods for extracting PRSs. More specifically:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://www.prsice.info/" class="external-link">PRSice2</a>, which operates based on the PLINK framework and automates a PRS extraction process based on clumping and thresholding followed by weighted PRS construction and evaluation with linear regression.</li>
<li>
<a href="https://github.com/tshmak/lassosum" class="external-link">lassosum</a>, which utilizes penalized (LASSO) regression to shrink the effects of less informative variants without completely removing them from the PRS.</li>
</ol>
<p>In the examples below we use the toy dataset included with PRISMA. For this reason we do not perform additional prior filtering although in a typical situation, genotypes should be filtered prior to any GWA testing and subsequent PRS analysis. The main function for PRS extraction is testing is the <code><a href="../reference/prsFuns.html">runPRS()</a></code> function which attaches the results to the <code>GWASExperiment</code> object. Each PRS extraction function (e.g. <code><a href="../reference/prsFuns.html">prsicePRS()</a></code>) can be used independently but it returns only a data frame with the SNPs in the PRS of a single run. The following sections illustrate their usage.</p>
<div class="section level3">
<h3 id="prs-analysis-with-lassosum">PRS analysis with lassosum<a class="anchor" aria-label="anchor" href="#prs-analysis-with-lassosum"></a>
</h3>
<p>The toy dataset is split in two parts, one for <em>training</em>, that is derivation of summary statistics, and one for <em>testing</em>, that is extraction of PRS. We are using 80% of samples for training (<code>base</code>) and 20% for testing (<code>target</code>). Theoretically, one can use the whole dataset for training and testing but this could lead to extreme overfitting and PRSs of poor predictive value. We test each SNP in the toy dataset for association with the main phenotype (BMI) given age, sex and case/control status, along with 1 PC using GLM and statgenGWAS. Then PRS extraction analysis follows. The output of summary statistics and the outcome of PRS analysis is registered to the <code>GWASExperiment</code> object.</p>
<p>Let's load the data and partition the dataset and perform PCA. The training and test sets should have the same number of PCs indicating any underlying population stratification in both datasets:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)

tt &lt;-<span class="st"> </span><span class="kw">partitionGWAS</span>(toy,<span class="dt">by=</span><span class="st">"BMI"</span>,<span class="dt">out=</span><span class="st">"ttboth"</span>)
## Partitioning dataset into 1 partitions by BMI
base &lt;-<span class="st"> </span>tt<span class="op">$</span>train
target &lt;-<span class="st"> </span>tt<span class="op">$</span>test
base &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(base,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmplLIKfg/temp_libpath4f71185f9e1c/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 54 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 50
##     # of SNPs: 946
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 16.74%, 110/657
## Chromosome 21: 15.74%, 54/343
## 164 markers are selected in total.
## LD pruning finished! 164 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 836 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 50
##     # of SNPs: 164
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 3900
## CPU capabilities: Double-Precision SSE2
## Fri Jun  3 14:57:48 2022    (internal increment: 26048)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Fri Jun  3 14:57:48 2022    Begin (eigenvalues and eigenvectors)
## Fri Jun  3 14:57:48 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs
target &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(target,<span class="dt">method=</span><span class="st">"snprel"</span>,<span class="dt">npc=</span><span class="dv">1</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmplLIKfg/temp_libpath4f71185f9e1c/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 62 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 50
##     # of SNPs: 938
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 16.13%, 106/657
## Chromosome 21: 17.20%, 59/343
## 165 markers are selected in total.
## LD pruning finished! 165 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 835 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 50
##     # of SNPs: 165
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 3987
## CPU capabilities: Double-Precision SSE2
## Fri Jun  3 14:57:49 2022    (internal increment: 26048)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 0s
## Fri Jun  3 14:57:49 2022    Begin (eigenvalues and eigenvectors)
## Fri Jun  3 14:57:49 2022    Done.
## Selecting number of PCs to include
##   with hard-cutoff at 1 PCs</code></pre></div>
<p>Now, summary statistics</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># With 1 PC</span>
<span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">base</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 31 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 22 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 24 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 38 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 31 associations found out of 657 markers</span>
<span class="co">##   Part 21: 38 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 69 associations with (uncorrected) combined p-values).</span>

<span class="co"># With no PCs</span>
<span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">base</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 46 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 25 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 21 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 34 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 46 associations found out of 657 markers</span>
<span class="co">##   Part 21: 34 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 80 associations with (uncorrected) combined p-values).</span>
    
<span class="co"># With 1 PCs and one more covariate</span>
<span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">base</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,
    pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 69 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 25 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 61 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 33 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Age, Gender, Case_Control</span>
<span class="co">## Use PCs     : Yes</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Combining p-values from multi-GWAS using fisher method</span>
<span class="co">## </span>
<span class="co">## Overall, found 31 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 61 associations found out of 657 markers</span>
<span class="co">##   Part 21: 31 associations found out of 343 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 92 associations with (uncorrected) combined p-values).</span></code></pre></div>
<p>Verify that we have three different models:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listAttrFuns.html">listGwa</a></span><span class="op">(</span><span class="va">base</span><span class="op">)</span>
<span class="co">## 1 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + PC1 + [SNP]</span>
<span class="co">## 2 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: -</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + [SNP]</span>
<span class="co">## 3 -&gt; Phenotype: BMI | Covariates: Age, Case_Control, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Case_Control + Gender + PC1 + [SNP]</span>
<span class="co">## </span>
<span class="co">## Note: The PC variables have putative names based on the number of PCs included in the model.</span>
<span class="co">## For example, instead of PC1, PC2, PC3 the actual PCs may be PC1, PC3, PC4</span></code></pre></div>
<p>Derivation of lassosum PRS for the three models:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lassoPrs1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">lassosumPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_ETXUS0626J</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_ETXUS0626J</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_ETXUS0626J/plink_lassosum_target_MKJKH0960I.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_ETXUS0626J/plink_lassosum_target_MKJKH0960I.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/lassosum_ETXUS0626J/plink_lassosum_target_MKJKH0960I.bed (SNP-major mode)</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Coordinating summary stats with reference panel...</span>
<span class="co">## Coordinating summary stats with test data...</span>
<span class="co">## Coordinating summary stats, reference panel, and test data...</span>
<span class="co">## Splitting genome by LD blocks ...</span>
<span class="co">## Running lassosum ...</span>
<span class="co">## s =  0.2 </span>
<span class="co">## s =  0.5 </span>
<span class="co">## s =  0.9 </span>
<span class="co">## Running lassosum with s=1...</span>
<span class="co">## Calculating polygenic scores ...</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 50 out of 50 samples kept in pheno.</span>
<span class="co">## 50 out of 50 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0175370886238609</span>

<span class="va">lassoPrs2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">lassosumPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
    covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_CQWHL7766Y</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_CQWHL7766Y</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_CQWHL7766Y/plink_lassosum_target_BUQPA4749F.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_CQWHL7766Y/plink_lassosum_target_BUQPA4749F.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/lassosum_CQWHL7766Y/plink_lassosum_target_BUQPA4749F.bed (SNP-major mode)</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Coordinating summary stats with reference panel...</span>
<span class="co">## Coordinating summary stats with test data...</span>
<span class="co">## Coordinating summary stats, reference panel, and test data...</span>
<span class="co">## Splitting genome by LD blocks ...</span>
<span class="co">## Running lassosum ...</span>
<span class="co">## s =  0.2 </span>
<span class="co">## s =  0.5 </span>
<span class="co">## s =  0.9 </span>
<span class="co">## Running lassosum with s=1...</span>
<span class="co">## Calculating polygenic scores ...</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 50 out of 50 samples kept in pheno.</span>
<span class="co">## 50 out of 50 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0528354299230865</span>
        
<span class="va">lassoPrs3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">lassosumPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
    covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_PRDBL7622F</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_PRDBL7622F</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_PRDBL7622F/plink_lassosum_target_GKDRD4751K.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_PRDBL7622F/plink_lassosum_target_GKDRD4751K.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/lassosum_PRDBL7622F/plink_lassosum_target_GKDRD4751K.bed (SNP-major mode)</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Coordinating summary stats with reference panel...</span>
<span class="co">## Coordinating summary stats with test data...</span>
<span class="co">## Coordinating summary stats, reference panel, and test data...</span>
<span class="co">## Splitting genome by LD blocks ...</span>
<span class="co">## Running lassosum ...</span>
<span class="co">## s =  0.2 </span>
<span class="co">## s =  0.5 </span>
<span class="co">## s =  0.9 </span>
<span class="co">## Running lassosum with s=1...</span>
<span class="co">## Calculating polygenic scores ...</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 50 out of 50 samples kept in pheno.</span>
<span class="co">## 50 out of 50 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0188222631225712</span></code></pre></div>
<p>Peak at the contents of the lassosum PRS:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lassoPrs1</span><span class="op">)</span>
<span class="co">##           chromosome position variant_id risk_allele reference_allele</span>
<span class="co">## rs1009238         18  2000079  rs1009238           G                A</span>
<span class="co">## rs1940646         18  2002665  rs1940646           A                G</span>
<span class="co">## rs1940651         18  2006123  rs1940651           G                A</span>
<span class="co">## rs1940658         18  2029280  rs1940658           A                G</span>
<span class="co">## rs1940670         18  2042445  rs1940670           A                G</span>
<span class="co">## rs508857          18  2101958   rs508857           A                G</span>
<span class="co">##           locus_name effect_weight        OR  asm</span>
<span class="co">## rs1009238         NA   -0.03356763 0.9669895 hg19</span>
<span class="co">## rs1940646         NA    0.02301112 1.0232779 hg19</span>
<span class="co">## rs1940651         NA   -0.14073585 0.8687188 hg19</span>
<span class="co">## rs1940658         NA   -0.01327724 0.9868105 hg19</span>
<span class="co">## rs1940670         NA    0.40244651 1.4954789 hg19</span>
<span class="co">## rs508857          NA   -0.07546377 0.9273133 hg19</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lassoPrs3</span><span class="op">)</span>
<span class="co">##           chromosome position variant_id risk_allele reference_allele</span>
<span class="co">## rs1009238         18  2000079  rs1009238           G                A</span>
<span class="co">## rs1940646         18  2002665  rs1940646           A                G</span>
<span class="co">## rs1940651         18  2006123  rs1940651           G                A</span>
<span class="co">## rs1940658         18  2029280  rs1940658           A                G</span>
<span class="co">## rs1940670         18  2042445  rs1940670           A                G</span>
<span class="co">## rs579292          18  2075770   rs579292           A                G</span>
<span class="co">##           locus_name effect_weight        OR  asm</span>
<span class="co">## rs1009238         NA   -0.10536140 0.8999992 hg19</span>
<span class="co">## rs1940646         NA    0.21153851 1.2355775 hg19</span>
<span class="co">## rs1940651         NA   -0.17171852 0.8422162 hg19</span>
<span class="co">## rs1940658         NA   -0.07363955 0.9290065 hg19</span>
<span class="co">## rs1940670         NA    0.36186276 1.4360018 hg19</span>
<span class="co">## rs579292          NA   -0.04268063 0.9582174 hg19</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prs-analysis-with-prsice2">PRS analysis with PRSice2<a class="anchor" aria-label="anchor" href="#prs-analysis-with-prsice2"></a>
</h3>
<p>We assume the same dataset split and the same summary statistics, calculated once:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="va">.Platform</span><span class="op">$</span><span class="va">OS.type</span> <span class="op">==</span> <span class="st">"unix"</span> <span class="op">&amp;&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^darwin"</span>,<span class="va">R.version</span><span class="op">$</span><span class="va">os</span><span class="op">)</span> 
    <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">prsicePrs1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">prsicePRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>

    <span class="va">prsicePrs2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">prsicePRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">FALSE</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>
            
    <span class="va">prsicePrs3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">prsicePRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_OIPDH5538B</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_OIPDH5538B</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/prsice_OIPDH5538B/plink_prsice_PBBJC2503E.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_OIPDH5538B/plink_prsice_PBBJC2503E.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/prsice_OIPDH5538B/plink_prsice_PBBJC2503E.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpvgB5i7/prsice_OIPDH5538B/base_PBBJC2503E \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpvgB5i7/prsice_OIPDH5538B/pheno_PBBJC2503E \</span>
<span class="co">##   --target /tmp/RtmpvgB5i7/prsice_OIPDH5538B/plink_prsice_PBBJC2503E \</span>
<span class="co">##   --cov /tmp/RtmpvgB5i7/prsice_OIPDH5538B/covar_PBBJC2503E \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 1 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_PBBJC2503E \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0369313</span>
<span class="co">## Full model R^2 is: 0.146879</span>
<span class="co">## Null model R^2 is: 0.109948</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_QLFZL1758J</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_QLFZL1758J</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/prsice_QLFZL1758J/plink_prsice_ZODXA2706J.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_QLFZL1758J/plink_prsice_ZODXA2706J.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/prsice_QLFZL1758J/plink_prsice_ZODXA2706J.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpvgB5i7/prsice_QLFZL1758J/base_ZODXA2706J \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpvgB5i7/prsice_QLFZL1758J/pheno_ZODXA2706J \</span>
<span class="co">##   --target /tmp/RtmpvgB5i7/prsice_QLFZL1758J/plink_prsice_ZODXA2706J \</span>
<span class="co">##   --cov /tmp/RtmpvgB5i7/prsice_QLFZL1758J/covar_ZODXA2706J \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 1 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_ZODXA2706J \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0367948</span>
<span class="co">## Full model R^2 is: 0.142178</span>
<span class="co">## Null model R^2 is: 0.105384</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_GWKES1830D</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_GWKES1830D</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/prsice_GWKES1830D/plink_prsice_DUIIJ9952P.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_GWKES1830D/plink_prsice_DUIIJ9952P.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/prsice_GWKES1830D/plink_prsice_DUIIJ9952P.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpvgB5i7/prsice_GWKES1830D/base_DUIIJ9952P \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpvgB5i7/prsice_GWKES1830D/pheno_DUIIJ9952P \</span>
<span class="co">##   --target /tmp/RtmpvgB5i7/prsice_GWKES1830D/plink_prsice_DUIIJ9952P \</span>
<span class="co">##   --cov /tmp/RtmpvgB5i7/prsice_GWKES1830D/covar_DUIIJ9952P \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 1 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_DUIIJ9952P \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0218937</span>
<span class="co">## Full model R^2 is: 0.375997</span>
<span class="co">## Null model R^2 is: 0.354104</span></code></pre></div>
<p>Peak at the contents of the lassosum PRS:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prsicePrs1</span><span class="op">)</span>
<span class="co">##            chromosome position variant_id risk_allele reference_allele</span>
<span class="co">## rs11876399         18  3615943 rs11876399           G                A</span>
<span class="co">##            locus_name effect_weight         OR  asm</span>
<span class="co">## rs11876399         NA     -2.780566 0.06200341 hg19</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prsicePrs3</span><span class="op">)</span>
<span class="co">##           chromosome position variant_id risk_allele reference_allele</span>
<span class="co">## rs979916          21 23607112   rs979916           A                G</span>
<span class="co">## rs2826023         21 21498290  rs2826023           A                G</span>
<span class="co">## rs9977668         21 22044940  rs9977668           G                A</span>
<span class="co">## rs7280065         21 24827764  rs7280065           G                A</span>
<span class="co">## rs2826810         21 22727138  rs2826810           C                A</span>
<span class="co">## rs1940670         18  2042445  rs1940670           A                G</span>
<span class="co">##           locus_name effect_weight         OR  asm</span>
<span class="co">## rs979916          NA     -3.733789 0.02390211 hg19</span>
<span class="co">## rs2826023         NA     -2.556728 0.07755806 hg19</span>
<span class="co">## rs9977668         NA     -2.999237 0.04982508 hg19</span>
<span class="co">## rs7280065         NA     -2.743046 0.06437396 hg19</span>
<span class="co">## rs2826810         NA     -2.819718 0.05962272 hg19</span>
<span class="co">## rs1940670         NA      1.725512 5.61539632 hg19</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prs-analysis-with-more-than-one-algorithm">PRS analysis with more than one algorithm<a class="anchor" aria-label="anchor" href="#prs-analysis-with-more-than-one-algorithm"></a>
</h3>
<p>Here we demonstrate the usage of the <code><a href="../reference/prsFuns.html">runPRS()</a></code> function which wraps over more than one PRS extraction algorithm and regsiters the results (which SNPs participate and their effects) in the respective <code>GWASExperiment</code> slot (<code>prsbetas</code>).</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="va">.Platform</span><span class="op">$</span><span class="va">OS.type</span> <span class="op">==</span> <span class="st">"unix"</span> <span class="op">&amp;&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^darwin"</span>,<span class="va">R.version</span><span class="op">$</span><span class="va">os</span><span class="op">)</span> 
    <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">prsiceOpts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span>
    <span class="va">prsiceOpts</span><span class="op">$</span><span class="va">perm</span> <span class="op">&lt;-</span> <span class="fl">100</span>
    
    <span class="va">target</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">runPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span>,prsiceOpts<span class="op">=</span><span class="va">prsiceOpts</span><span class="op">)</span>
    
    <span class="va">target</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">runPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">FALSE</span>,prsiceOpts<span class="op">=</span><span class="va">prsiceOpts</span><span class="op">)</span>
        
    <span class="va">target</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prsFuns.html">runPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">base</span>,target<span class="op">=</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,
        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span>,
        prsiceOpts<span class="op">=</span><span class="va">prsiceOpts</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_LCAQR0699L</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_LCAQR0699L</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_LCAQR0699L/plink_lassosum_target_STYWF0523P.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_LCAQR0699L/plink_lassosum_target_STYWF0523P.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/lassosum_LCAQR0699L/plink_lassosum_target_STYWF0523P.bed (SNP-major mode)</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Coordinating summary stats with reference panel...</span>
<span class="co">## Coordinating summary stats with test data...</span>
<span class="co">## Coordinating summary stats, reference panel, and test data...</span>
<span class="co">## Splitting genome by LD blocks ...</span>
<span class="co">## Running lassosum ...</span>
<span class="co">## s =  0.2 </span>
<span class="co">## s =  0.5 </span>
<span class="co">## s =  0.9 </span>
<span class="co">## Running lassosum with s=1...</span>
<span class="co">## Calculating polygenic scores ...</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 50 out of 50 samples kept in pheno.</span>
<span class="co">## 50 out of 50 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0175370886238609</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_CORHL2531V</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_CORHL2531V</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/prsice_CORHL2531V/plink_prsice_OFVZE4193M.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_CORHL2531V/plink_prsice_OFVZE4193M.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/prsice_CORHL2531V/plink_prsice_OFVZE4193M.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpvgB5i7/prsice_CORHL2531V/base_OFVZE4193M \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpvgB5i7/prsice_CORHL2531V/pheno_OFVZE4193M \</span>
<span class="co">##   --target /tmp/RtmpvgB5i7/prsice_CORHL2531V/plink_prsice_OFVZE4193M \</span>
<span class="co">##   --cov /tmp/RtmpvgB5i7/prsice_CORHL2531V/covar_OFVZE4193M \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 1 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_OFVZE4193M \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0369313</span>
<span class="co">## Full model R^2 is: 0.146879</span>
<span class="co">## Null model R^2 is: 0.109948</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_HJPIH0663S</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_HJPIH0663S</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_HJPIH0663S/plink_lassosum_target_KUAYB3231B.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_HJPIH0663S/plink_lassosum_target_KUAYB3231B.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/lassosum_HJPIH0663S/plink_lassosum_target_KUAYB3231B.bed (SNP-major mode)</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Coordinating summary stats with reference panel...</span>
<span class="co">## Coordinating summary stats with test data...</span>
<span class="co">## Coordinating summary stats, reference panel, and test data...</span>
<span class="co">## Splitting genome by LD blocks ...</span>
<span class="co">## Running lassosum ...</span>
<span class="co">## s =  0.2 </span>
<span class="co">## s =  0.5 </span>
<span class="co">## s =  0.9 </span>
<span class="co">## Running lassosum with s=1...</span>
<span class="co">## Calculating polygenic scores ...</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 50 out of 50 samples kept in pheno.</span>
<span class="co">## 50 out of 50 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0528354299230865</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_VBYHS2459E</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_VBYHS2459E</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/prsice_VBYHS2459E/plink_prsice_HNJXC6565F.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_VBYHS2459E/plink_prsice_HNJXC6565F.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/prsice_VBYHS2459E/plink_prsice_HNJXC6565F.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpvgB5i7/prsice_VBYHS2459E/base_HNJXC6565F \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpvgB5i7/prsice_VBYHS2459E/pheno_HNJXC6565F \</span>
<span class="co">##   --target /tmp/RtmpvgB5i7/prsice_VBYHS2459E/plink_prsice_HNJXC6565F \</span>
<span class="co">##   --cov /tmp/RtmpvgB5i7/prsice_VBYHS2459E/covar_HNJXC6565F \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 1 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_HNJXC6565F \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0367948</span>
<span class="co">## Full model R^2 is: 0.142178</span>
<span class="co">## Null model R^2 is: 0.105384</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_JNGBP4362E</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_JNGBP4362E</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_JNGBP4362E/plink_lassosum_target_KZUAH0700H.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_JNGBP4362E/plink_lassosum_target_KZUAH0700H.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/lassosum_JNGBP4362E/plink_lassosum_target_KZUAH0700H.bed (SNP-major mode)</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Coordinating summary stats with reference panel...</span>
<span class="co">## Coordinating summary stats with test data...</span>
<span class="co">## Coordinating summary stats, reference panel, and test data...</span>
<span class="co">## Splitting genome by LD blocks ...</span>
<span class="co">## Running lassosum ...</span>
<span class="co">## s =  0.2 </span>
<span class="co">## s =  0.5 </span>
<span class="co">## s =  0.9 </span>
<span class="co">## Running lassosum with s=1...</span>
<span class="co">## Calculating polygenic scores ...</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 50 out of 50 samples kept in pheno.</span>
<span class="co">## 50 out of 50 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0188222631225712</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_LCNEC7673G</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_LCNEC7673G</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Writing FAM file to /tmp/RtmpvgB5i7/prsice_LCNEC7673G/plink_prsice_GSEYM2498C.fam </span>
<span class="co">## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_LCNEC7673G/plink_prsice_GSEYM2498C.bim </span>
<span class="co">## Writing BED file to /tmp/RtmpvgB5i7/prsice_LCNEC7673G/plink_prsice_GSEYM2498C.bed (SNP-major mode)</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpvgB5i7/prsice_LCNEC7673G/base_GSEYM2498C \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpvgB5i7/prsice_LCNEC7673G/pheno_GSEYM2498C \</span>
<span class="co">##   --target /tmp/RtmpvgB5i7/prsice_LCNEC7673G/plink_prsice_GSEYM2498C \</span>
<span class="co">##   --cov /tmp/RtmpvgB5i7/prsice_LCNEC7673G/covar_GSEYM2498C \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 1 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_GSEYM2498C \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0218937</span>
<span class="co">## Full model R^2 is: 0.375997</span>
<span class="co">## Null model R^2 is: 0.354104</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span></code></pre></div>
<p>Now, use the <code><a href="../reference/listAttrFuns.html">listPrs()</a></code> function to see what PRS extraction results are attached to the <code>target</code> object:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listAttrFuns.html">listPrs</a></span><span class="op">(</span><span class="va">target</span><span class="op">)</span>
<span class="co">## 1 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + PC1 + [PRS]</span>
<span class="co">## 2 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: -</span>
<span class="co">##      Putative formula: BMI ~ Age + Gender + [PRS]</span>
<span class="co">## 3 -&gt; Phenotype: BMI | Covariates: Age, Case_Control, Gender | PCs: 1</span>
<span class="co">##      Putative formula: BMI ~ Age + Case_Control + Gender + PC1 + [PRS]</span>
<span class="co">## </span>
<span class="co">## Note: The PC variables have putative names based on the number of PCs included in the model.</span>
<span class="co">## For example, instead of PC1, PC2, PC3 the actual PCs may be PC1, PC3, PC4</span></code></pre></div>
<p>Let's see the prsbetas and p-values for some of the extracted PRSs. Let's start with the first one where the phenotype was <code>BMI</code> and the covariates were <code>Age</code> and <code>Gender</code> along with 1 PC:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">prsbetas</a></span><span class="op">(</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">##              lassosum prsice</span>
<span class="co">## rs1009238 -0.03356763      0</span>
<span class="co">## rs1940646  0.02301112      0</span>
<span class="co">## rs1940651 -0.14073585      0</span>
<span class="co">## rs1940658 -0.01327724      0</span>
<span class="co">## rs1940670  0.40244651      0</span>
<span class="co">## rs1954825  0.00000000      0</span></code></pre></div>
<p>Or another model:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWASExperiment-class.html">prsbetas</a></span><span class="op">(</span><span class="va">target</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,
    <span class="st">"Case_Control"</span><span class="op">)</span>,npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">##              lassosum   prsice</span>
<span class="co">## rs1009238 -0.10536140 0.000000</span>
<span class="co">## rs1940646  0.21153851 0.000000</span>
<span class="co">## rs1940651 -0.17171852 0.000000</span>
<span class="co">## rs1940658 -0.07363955 0.000000</span>
<span class="co">## rs1940670  0.36186276 1.725512</span>
<span class="co">## rs1954825  0.00000000 0.000000</span></code></pre></div>
<p>It is possible of course to use the total dataset for PRS analysis with a lot of overfit:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(toy,<span class="dt">package=</span><span class="st">"prisma"</span>)

<span class="kw">gdsfile</span>(toy) &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span>)
toy &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">calcPcaCovar</span>(toy,<span class="dt">method=</span><span class="st">"snprel"</span>))
## Performing PCA to select covariates to include in associations
## Reading GDS file /tmp/RtmplLIKfg/temp_libpath4f71185f9e1c/prisma/extdata/toy.gds...
## Performing LD pruning...
## SNP pruning based on LD:
## Excluding 0 SNP (non-autosomes or non-selection)
## Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 962
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chromosome 18: 23.44%, 154/657
## Chromosome 21: 21.28%, 73/343
## 227 markers are selected in total.
## LD pruning finished! 227 SNPs will be used for PCA.
## 
## Performing PCA...
## Principal Component Analysis (PCA) on genotypes:
## Excluding 773 SNPs (non-autosomes or non-selection)
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 100
##     # of SNPs: 227
##     using 1 thread
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 10268
## CPU capabilities: Double-Precision SSE2
## Fri Jun  3 14:59:20 2022    (internal increment: 13024)
## 
[..................................................]  <span class="dv">0</span>%, ETC<span class="op">:</span><span class="st"> </span><span class="op">---</span><span class="st">        </span>
[<span class="op">==</span><span class="er">================================================</span>] <span class="dv">100</span>%, completed, 1s
## Fri Jun  3 14:59:21 2022    Begin (eigenvalues and eigenvectors)
## Fri Jun  3 14:59:21 2022    Done.
## Selecting number of PCs to include
##   with Tracy-Widom test for significant eigenvalues
##     selected 1 PCs

toy &lt;-<span class="st"> </span><span class="kw">gwa</span>(base,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span>),
    <span class="dt">pcs=</span><span class="ot">TRUE</span>,<span class="dt">psig=</span><span class="fl">0.05</span>,<span class="dt">methods=</span><span class="st">"statgen"</span>)
## 
## Starting GWA analysis in 2 chunks with 1 methods: statgen
## 
## ========== Analyzing chromosome/part 18
## 
## Preparing data for statgenGWAS...
## Performing GWAS with statgenGWAS model
## Trait(s)    : BMI
## Covariate(s): Age, Gender, Case_Control
## Use PCs     : Yes
## Found 25 associations (uncorrected p-values).
## 
## Overall, found 25 associations with (uncorrected) combined p-values).
## 
## ========== Analyzing chromosome/part 21
## 
## Preparing data for statgenGWAS...
## Performing GWAS with statgenGWAS model
## Trait(s)    : BMI
## Covariate(s): Age, Gender, Case_Control
## Use PCs     : Yes
## Found 16 associations (uncorrected p-values).
## 
## Overall, found 16 associations with (uncorrected) combined p-values).
## 
## Finished, the following putative associations were found per part:
##   Part 18: 25 associations found out of 657 markers
##   Part 21: 16 associations found out of 343 markers
## 
## Overall, found 41 associations with (uncorrected) combined p-values).

<span class="cf">if</span> (.Platform<span class="op">$</span>OS.type <span class="op">==</span><span class="st"> "unix"</span> <span class="op">&amp;&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">grepl</span>(<span class="st">"^darwin"</span>,R.version<span class="op">$</span>os) 
    <span class="op">&amp;&amp;</span><span class="st"> </span><span class="kw">Sys.which</span>(<span class="st">"PRSice_linux"</span>) <span class="op">!=</span><span class="st"> ""</span>) {
    prsiceOpts &lt;-<span class="st"> </span><span class="kw">getDefaults</span>(<span class="st">"prsice"</span>)
    prsiceOpts<span class="op">$</span>perm &lt;-<span class="st"> </span><span class="dv">100</span>
    
    toy &lt;-<span class="st"> </span><span class="kw">runPRS</span>(<span class="dt">base=</span>toy,<span class="dt">response=</span><span class="st">"BMI"</span>,<span class="dt">covariates=</span><span class="kw">c</span>(<span class="st">"Age"</span>,<span class="st">"Gender"</span>,
        <span class="st">"Case_Control"</span>),<span class="dt">pcs=</span><span class="ot">TRUE</span>,<span class="dt">prsiceOpts=</span>prsiceOpts) 
}
## 
## Starting PRS analysis with lassosum, prsice
## 
## ========== Running PRS analysis with lassosum
## lassosum operations will be executed at /tmp/RtmpvgB5i7/lassosum_LAXEO6496D
## Preparing lassosum run at workspace directory: /tmp/RtmpvgB5i7/lassosum_LAXEO6496D
##   preparing summary statistics...
##   preparing PLINK temporary files for target...
## Preflight...
## Writing FAM file to /tmp/RtmpvgB5i7/lassosum_LAXEO6496D/plink_lassosum_target_JJJFY5019G.fam 
## Writing extended MAP file to /tmp/RtmpvgB5i7/lassosum_LAXEO6496D/plink_lassosum_target_JJJFY5019G.bim 
## Writing BED file to /tmp/RtmpvgB5i7/lassosum_LAXEO6496D/plink_lassosum_target_JJJFY5019G.bed (SNP-major mode)
##   copying linkage disequilibrium block files...
## Preparation done!
## Running lassosum PRS algorithm
## Coordinating summary stats with reference panel...
## Coordinating summary stats with test data...
## Coordinating summary stats, reference panel, and test data...
## Splitting genome by LD blocks ...
## Running lassosum ...
## s =  0.2 
## s =  0.5 
## s =  0.9 
## Running lassosum with s=1...
## Calculating polygenic scores ...
## Running lassosum validation and PRS calculation
## 50 out of 50 samples kept in pheno.
## 50 out of 50 samples kept in covar.
## Done! R^2 is: 0.930939526831895
## 
## ========== Running PRS analysis with prsice
## prsice operations will be executed at /tmp/RtmpvgB5i7/prsice_AEPWO5989T
## Preparing PRSice run at workspace directory: /tmp/RtmpvgB5i7/prsice_AEPWO5989T
##   preparing summary statistics for the base file...
##   preparing the covariates file...
##   preparing the phenotype file...
##   preparing PLINK temporary files - target...
## Preflight...
## Writing FAM file to /tmp/RtmpvgB5i7/prsice_AEPWO5989T/plink_prsice_TMODH7415L.fam 
## Writing extended MAP file to /tmp/RtmpvgB5i7/prsice_AEPWO5989T/plink_prsice_TMODH7415L.bim 
## Writing BED file to /tmp/RtmpvgB5i7/prsice_AEPWO5989T/plink_prsice_TMODH7415L.bed (SNP-major mode)
## Preparation done!
## Running PRSice PRS algorithm in calculate mode
## 
## Executing:
## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \
##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \
##   --base /tmp/RtmpvgB5i7/prsice_AEPWO5989T/base_TMODH7415L \
##   --beta \
##   --binary-target F \
##   --pheno /tmp/RtmpvgB5i7/prsice_AEPWO5989T/pheno_TMODH7415L \
##   --target /tmp/RtmpvgB5i7/prsice_AEPWO5989T/plink_prsice_TMODH7415L \
##   --cov /tmp/RtmpvgB5i7/prsice_AEPWO5989T/covar_TMODH7415L \
##   --clump-kb 250 \
##   --clump-r2 0.1 \
##   --clump-p 1 \
##   --score avg \
##   --seed 42 \
##   --out prsice_out_TMODH7415L \
##   --print-snp \
##   --perm 100
## Done!
## Adjusted PRSice R^2 is: 0.395084
## Full model R^2 is: 0.971816
## Null model R^2 is: 0.576731
## 
## Running PRS algorithms finished! Processing the results...
## Done!</code></pre></div>
<p>You will notice the larger R<sup>2</sup> values in the outputs.</p>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] prisma_0.0.1                SNPRelate_1.28.0           </span>
<span class="co">##  [3] gdsfmt_1.30.0               snpStats_1.44.0            </span>
<span class="co">##  [5] Matrix_1.4-1                survival_3.3-1             </span>
<span class="co">##  [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">## [11] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [15] matrixStats_0.62.0          lassosum_0.4.5             </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] knitr_1.39                              </span>
<span class="co">##  [19] rstudioapi_0.13                         </span>
<span class="co">##  [20] robustbase_0.95-0                       </span>
<span class="co">##  [21] TTR_0.24.3                              </span>
<span class="co">##  [22] listenv_0.8.0                           </span>
<span class="co">##  [23] optparse_1.7.1                          </span>
<span class="co">##  [24] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [25] bit64_4.0.5                             </span>
<span class="co">##  [26] rprojroot_2.0.3                         </span>
<span class="co">##  [27] parallelly_1.31.1                       </span>
<span class="co">##  [28] vctrs_0.4.1                             </span>
<span class="co">##  [29] generics_0.1.2                          </span>
<span class="co">##  [30] xfun_0.31                               </span>
<span class="co">##  [31] BiocFileCache_2.2.1                     </span>
<span class="co">##  [32] R6_2.5.1                                </span>
<span class="co">##  [33] rmdformats_1.0.3                        </span>
<span class="co">##  [34] bitops_1.0-7                            </span>
<span class="co">##  [35] cachem_1.0.6                            </span>
<span class="co">##  [36] DelayedArray_0.20.0                     </span>
<span class="co">##  [37] assertthat_0.2.1                        </span>
<span class="co">##  [38] gwasrapidd_0.99.12                      </span>
<span class="co">##  [39] BiocIO_1.4.0                            </span>
<span class="co">##  [40] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [41] scales_1.2.0                            </span>
<span class="co">##  [42] gtable_0.3.0                            </span>
<span class="co">##  [43] globals_0.15.0                          </span>
<span class="co">##  [44] rlang_1.0.2                             </span>
<span class="co">##  [45] systemfonts_1.0.4                       </span>
<span class="co">##  [46] splines_4.1.3                           </span>
<span class="co">##  [47] rtracklayer_1.54.0                      </span>
<span class="co">##  [48] rsnps_0.5.0.0                           </span>
<span class="co">##  [49] sitadela_1.3.1                          </span>
<span class="co">##  [50] BiocManager_1.30.16                     </span>
<span class="co">##  [51] yaml_2.3.5                              </span>
<span class="co">##  [52] reshape2_1.4.4                          </span>
<span class="co">##  [53] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [54] quantmod_0.4.20                         </span>
<span class="co">##  [55] RBGL_1.70.0                             </span>
<span class="co">##  [56] tools_4.1.3                             </span>
<span class="co">##  [57] lava_1.6.10                             </span>
<span class="co">##  [58] bookdown_0.26                           </span>
<span class="co">##  [59] liftOver_1.18.0                         </span>
<span class="co">##  [60] ggplot2_3.3.5                           </span>
<span class="co">##  [61] ellipsis_0.3.2                          </span>
<span class="co">##  [62] kableExtra_1.3.4                        </span>
<span class="co">##  [63] jquerylib_0.1.4                         </span>
<span class="co">##  [64] Rcpp_1.0.8.3                            </span>
<span class="co">##  [65] rrBLUP_4.6.1                            </span>
<span class="co">##  [66] plyr_1.8.7                              </span>
<span class="co">##  [67] progress_1.2.2                          </span>
<span class="co">##  [68] zlibbioc_1.40.0                         </span>
<span class="co">##  [69] purrr_0.3.4                             </span>
<span class="co">##  [70] RCurl_1.98-1.6                          </span>
<span class="co">##  [71] prettyunits_1.1.1                       </span>
<span class="co">##  [72] cowplot_1.1.1                           </span>
<span class="co">##  [73] zoo_1.8-10                              </span>
<span class="co">##  [74] fs_1.5.2                                </span>
<span class="co">##  [75] crul_1.2.0                              </span>
<span class="co">##  [76] magrittr_2.0.3                          </span>
<span class="co">##  [77] data.table_1.14.2                       </span>
<span class="co">##  [78] openxlsx_4.2.5                          </span>
<span class="co">##  [79] mvtnorm_1.1-3                           </span>
<span class="co">##  [80] survcomp_1.44.1                         </span>
<span class="co">##  [81] hms_1.1.1                               </span>
<span class="co">##  [82] evaluate_0.15                           </span>
<span class="co">##  [83] XML_3.99-0.9                            </span>
<span class="co">##  [84] RMTstat_0.3.1                           </span>
<span class="co">##  [85] shape_1.4.6                             </span>
<span class="co">##  [86] compiler_4.1.3                          </span>
<span class="co">##  [87] biomaRt_2.50.3                          </span>
<span class="co">##  [88] tibble_3.1.7                            </span>
<span class="co">##  [89] KernSmooth_2.23-20                      </span>
<span class="co">##  [90] crayon_1.5.1                            </span>
<span class="co">##  [91] R.oo_1.24.0                             </span>
<span class="co">##  [92] htmltools_0.5.2                         </span>
<span class="co">##  [93] pcaPP_2.0-1                             </span>
<span class="co">##  [94] tzdb_0.3.0                              </span>
<span class="co">##  [95] rrcov_1.7-0                             </span>
<span class="co">##  [96] DBI_1.1.2                               </span>
<span class="co">##  [97] SuppDists_1.1-9.7                       </span>
<span class="co">##  [98] dbplyr_2.1.1                            </span>
<span class="co">##  [99] MASS_7.3-56                             </span>
<span class="co">## [100] rappdirs_0.3.3                          </span>
<span class="co">## [101] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [102] getopt_1.20.3                           </span>
<span class="co">## [103] readr_2.1.2                             </span>
<span class="co">## [104] rmeta_3.0                               </span>
<span class="co">## [105] cli_3.3.0                               </span>
<span class="co">## [106] quadprog_1.5-8                          </span>
<span class="co">## [107] R.methodsS3_1.8.1                       </span>
<span class="co">## [108] parallel_4.1.3                          </span>
<span class="co">## [109] pkgconfig_2.0.3                         </span>
<span class="co">## [110] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [111] pkgdown_2.0.3.9000                      </span>
<span class="co">## [112] GenomicAlignments_1.30.0                </span>
<span class="co">## [113] signal_0.7-7                            </span>
<span class="co">## [114] xml2_1.3.3                              </span>
<span class="co">## [115] foreach_1.5.2                           </span>
<span class="co">## [116] svglite_2.1.0                           </span>
<span class="co">## [117] bslib_0.3.1                             </span>
<span class="co">## [118] webshot_0.5.3                           </span>
<span class="co">## [119] XVector_0.34.0                          </span>
<span class="co">## [120] prodlim_2019.11.13                      </span>
<span class="co">## [121] rvest_1.0.2                             </span>
<span class="co">## [122] stringr_1.4.0                           </span>
<span class="co">## [123] VariantAnnotation_1.40.0                </span>
<span class="co">## [124] digest_0.6.29                           </span>
<span class="co">## [125] graph_1.72.0                            </span>
<span class="co">## [126] httpcode_0.3.0                          </span>
<span class="co">## [127] Biostrings_2.62.0                       </span>
<span class="co">## [128] rmarkdown_2.14                          </span>
<span class="co">## [129] harmonicmeanp_3.0                       </span>
<span class="co">## [130] restfulr_0.0.13                         </span>
<span class="co">## [131] curl_4.3.2                              </span>
<span class="co">## [132] Rsamtools_2.10.0                        </span>
<span class="co">## [133] rjson_0.2.21                            </span>
<span class="co">## [134] tseries_0.10-51                         </span>
<span class="co">## [135] lifecycle_1.0.1                         </span>
<span class="co">## [136] jsonlite_1.8.0                          </span>
<span class="co">## [137] survivalROC_1.0.3                       </span>
<span class="co">## [138] desc_1.4.1                              </span>
<span class="co">## [139] viridisLite_0.4.0                       </span>
<span class="co">## [140] BSgenome_1.62.0                         </span>
<span class="co">## [141] fansi_1.0.3                             </span>
<span class="co">## [142] pillar_1.7.0                            </span>
<span class="co">## [143] lattice_0.20-45                         </span>
<span class="co">## [144] DEoptimR_1.0-11                         </span>
<span class="co">## [145] KEGGREST_1.34.0                         </span>
<span class="co">## [146] fastmap_1.1.0                           </span>
<span class="co">## [147] httr_1.4.3                              </span>
<span class="co">## [148] GO.db_3.14.0                            </span>
<span class="co">## [149] xts_0.12.1                              </span>
<span class="co">## [150] glue_1.6.2                              </span>
<span class="co">## [151] zip_2.2.0                               </span>
<span class="co">## [152] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [153] png_0.1-7                               </span>
<span class="co">## [154] iterators_1.0.14                        </span>
<span class="co">## [155] pander_0.6.5                            </span>
<span class="co">## [156] glmnet_4.1-4                            </span>
<span class="co">## [157] bit_4.0.4                               </span>
<span class="co">## [158] stringi_1.7.6                           </span>
<span class="co">## [159] sass_0.4.1                              </span>
<span class="co">## [160] bootstrap_2019.6                        </span>
<span class="co">## [161] blob_1.2.3                              </span>
<span class="co">## [162] statgenGWAS_1.0.8                       </span>
<span class="co">## [163] textshaping_0.3.6                       </span>
<span class="co">## [164] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [165] memoise_2.0.1                           </span>
<span class="co">## [166] FMStable_0.1-2                          </span>
<span class="co">## [167] dplyr_1.0.8                             </span>
<span class="co">## [168] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
