<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Complete PRISMA pipelines â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Complete PRISMA pipelines">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Complete PRISMA pipelines</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/08-pipelines.Rmd" class="external-link"><code>vignettes/08-pipelines.Rmd</code></a></small>
      <div class="d-none name"><code>08-pipelines.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Although the individual steps of a PRS extraction process can be performed in a stepwise manner, PRISMA was designed to offer complete RPS extraction pipelines with evaulation and related visualizations. The results of the iterative PRS extraction as well as the evaluation metrics and the reports are registered to an object alltogether which facilitates reporting. This is not as easily possible when using individual steps. In addition, as the whole process involves many complex steps and inclusion of external software, it may be prone to crashes. The pipelines implement several checkpoints that can be used to restart the pipeline from the crash point (if any). Furthermore, there is a pipeline wrapper which iteratively executes the selected pipeline until finished.</p>
<p>The main steps implemented in the PRISMA pipelines which offer an end-to-end PRS extraction pipeline are:</p>
<ol style="list-style-type: decimal">
<li>Preparation of the main total dataset consisting of basic quaility control filtering and missing genotype imputation. Optionally, holding-out a completely independent object to be used for evaluation.</li>
<li>Iterative PRS extraction consising of
<ol style="list-style-type: lower-alpha">
<li>Splitting of the main <code>GWASExperiment</code> in a base (training) object which will be used for summary statistics and a target (test) object which will be used for PRS extraction and evaluation.</li>
<li>Quality control of the base object, essentially for filtering out SNPs with small MAF and possibly violating HWE after the dataset splitting. Further sample filtering is also possible. PCA is optionally performed in both base and target objects.</li>
<li>Calculation of summary statistics with many available methods using the base object.</li>
<li>Derivation of PRS using the summary statistics of (2c) and the target object, with more than one available methods.</li>
</ol>
</li>
<li>Aggregation of the SNPs suggested in the iterative PRS extraction process and summarization of SNP effects, followed by semi-automatic selection of the best PRS candidates.</li>
<li>Evaluation of the PRS candidates using the total dataset and/or optionally the independent testing samples held-out. In this step, an externally retrieved PRS (e.g. from PGS Catalog) can be plugged in for evaluation.</li>
<li>Lookup of the SNPs included in the final PRS candidates to annotate from online resources for known traits (GWAS Catalog) followed by reporting and the selection of the final PRS.</li>
</ol>
<p>The main PRISMA pipeline is depicted in the following figure:</p>
<p><img src="pipeline.png" width="800px"></p>
</div>
<div class="section level2">
<h2 id="prisma-pipelines">PRISMA pipelines<a class="anchor" aria-label="anchor" href="#prisma-pipelines"></a>
</h2>
<p>At the heart of the PRISMA pipelines is the main <code><a href="../reference/prismaFuns.html">prisma()</a></code> wrapper, along with other functions that implement the steps such as <code><a href="../reference/prismaFuns.html">prsPipeline()</a></code>, <code><a href="../reference/prismaFuns.html">prsSelection()</a></code>, <code><a href="../reference/selectPrs.html">selectPrs()</a></code> and other validation functions such as <code><a href="../reference/prismaCvFuns.html">prismaCrossValidate()</a></code>. Parallelization is implemented wherever possible. The following subsections present the different PRISMA pipelines which implement the PRS extraction and evaluation steps outlined above.</p>
<div class="section level3">
<h3 id="de-novo-prs-derivation-pipeline">De novo PRS derivation pipeline<a class="anchor" aria-label="anchor" href="#de-novo-prs-derivation-pipeline"></a>
</h3>
<p>This is the main PRS candidate extraction pipeline and performs the following steps: 1. Split of the input data object to a usually larger training set and a smaller independent test set. 2. De novo quality control on the training and test sets as well as optional PCA in the training set to capture local population structure to be used as covariates in the subsequent GWA tests. 3. Execution of GWA tests against a phenotype using one or more of the supported methods to extract summary statistics (SNP effects and p-values). 4. Execution of one or more PRS extraction algorithms to derive PRS candidate SNPs.</p>
<p>Steps (1) â€“ (4) are executed N times with different training and test sets to capture as much variation in the selection process as possible. Subsequently:</p>
<ol start="5" style="list-style-type: decimal">
<li>The candidate SNPs from PRSs extracted at each iteration are aggregated and the frequency of their appearance is recorded. At the same time, new effects are calculated either by averaging the SNP effects from each iteration, a <a href="https://www.biorxiv.org/content/10.1101/133785v8.full.pdf" class="external-link">statistically valid process</a> which generally corrects for overfitting, or by weighted averaging of the effects based on the performance of PRSs constructed at each iteration (as measured by the adjusted PRS R<sup>2</sup> returned by PRSice2).</li>
</ol>
<p>Here, <code><a href="../reference/prismaFuns.html">prsPipeline()</a></code> produces a list either with <code>GWASExperiment</code> objects with the <code>prsbetas</code> and <code>effects</code> or summary lists, not holding the genotypic data for saving memory and space. This list practically hold the derived PRSs in each iteration, through the <code>prsbetas</code> slot.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmoulos/prisma" class="external-link">prisma</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span>

<span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d1</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">dnList</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsPipeline</a></span><span class="op">(</span>
        gwe<span class="op">=</span><span class="va">toy</span>,
        phenotype<span class="op">=</span><span class="va">response</span>,
        covariates<span class="op">=</span><span class="va">covariates</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,
        trainSize<span class="op">=</span><span class="fl">0.8</span>,
        niter<span class="op">=</span><span class="fl">4</span>,
        filters<span class="op">=</span><span class="va">filts</span>,
        pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
        imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
        gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
        prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        prsWorkspace<span class="op">=</span><span class="va">wspace1</span>,
        cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
        logging<span class="op">=</span><span class="st">"file"</span>,
        rc<span class="op">=</span><span class="fl">0.25</span>
    <span class="op">)</span>
<span class="op">}</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input pipeline list (dnList) : a list of PRS pipeline summaries</span>
<span class="co">## Input dataset object (gwe)   : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : single</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : fisher</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 0.75</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 100</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)              : 80%</span>
<span class="co">## Number of PRS selection iterations (niter) : 4</span>
<span class="co">## PRS evaluation framework (evalWith)        : vanilla</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## Output type (output)                     : gwaslist</span>
<span class="co">## Run ID (runId)                           : GPPIJYMHNUD1847P</span>
<span class="co">## Available cores fraction (rc)            : single core</span>
<span class="co">## </span>
<span class="co">## Executing denovo pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 0</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 23 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 23 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 12 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 12 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 23 associations found out of 564 markers</span>
<span class="co">##   Part 21: 12 associations found out of 312 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.00227010528612243</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1/base_CWGOX7416E \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1/pheno_CWGOX7416E \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1/plink_prsice_CWGOX7416E \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1/covar_CWGOX7416E \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_CWGOX7416E \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0912606</span>
<span class="co">## Full model R^2 is: 0.485626</span>
<span class="co">## Null model R^2 is: 0.394365</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_1/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 124</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 116</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 19 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 16 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 19 associations found out of 561 markers</span>
<span class="co">##   Part 21: 16 associations found out of 314 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.00341981840311068</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2/base_WXSHL6566G \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2/pheno_WXSHL6566G \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2/plink_prsice_WXSHL6566G \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2/covar_WXSHL6566G \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_WXSHL6566G \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.068263</span>
<span class="co">## Full model R^2 is: 0.558609</span>
<span class="co">## Null model R^2 is: 0.490346</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_2/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 123</span>
<span class="co">##   Hardy-Weinberg equilibrium: 4</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 23 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 23 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 8 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 8 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 23 associations found out of 566 markers</span>
<span class="co">##   Part 21: 8 associations found out of 309 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 31 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0345324572269246</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3/base_AVPTN6937J \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3/pheno_AVPTN6937J \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3/plink_prsice_AVPTN6937J \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3/covar_AVPTN6937J \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_AVPTN6937J \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.105466</span>
<span class="co">## Full model R^2 is: 0.711932</span>
<span class="co">## Null model R^2 is: 0.606466</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_3/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 124</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 119</span>
<span class="co">##   Hardy-Weinberg equilibrium: 0</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 23 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 23 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 23 associations found out of 564 markers</span>
<span class="co">##   Part 21: 13 associations found out of 312 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 36 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0036170577218724</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4/base_KYAIG6850M \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4/pheno_KYAIG6850M \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4/plink_prsice_KYAIG6850M \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4/covar_KYAIG6850M \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_KYAIG6850M \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0432417</span>
<span class="co">## Full model R^2 is: 0.61248</span>
<span class="co">## Null model R^2 is: 0.569238</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b/GPPIJYMHNUD1847P_denovo_4/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f76bcde5b</span></code></pre></div>
<p>Now, aggregate the outcome to a pool of SNPs to be evaluated for PRS candidates:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snpSummary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/aggregatePrsMarkers.html">aggregatePrsMarkers</a></span><span class="op">(</span><span class="va">dnList</span>,mode<span class="op">=</span><span class="st">"intersect"</span>,qcut<span class="op">=</span><span class="fl">0</span><span class="op">)</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 9 markers</span>
<span class="co">##   iteration 2 yields 66 markers</span>
<span class="co">##   iteration 3 yields 5 markers</span>
<span class="co">##   iteration 4 yields 19 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">snpSummary</span><span class="op">)</span>
<span class="co">##                   snp freq    effect</span>
<span class="co">## rs508857     rs508857    4  2.169076</span>
<span class="co">## rs11876399 rs11876399    3  1.456026</span>
<span class="co">## rs16943205 rs16943205    3 -1.534372</span>
<span class="co">## rs1740090   rs1740090    3 -1.845601</span>
<span class="co">## rs7280065   rs7280065    3  2.324420</span>
<span class="co">## rs1442343   rs1442343    2 -1.173113</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prs-evaluation-pipeline-based-on-the-original-traintest-split-with-prsice2">PRS evaluation pipeline based on the original train/test split with PRSice2<a class="anchor" aria-label="anchor" href="#prs-evaluation-pipeline-based-on-the-original-traintest-split-with-prsice2"></a>
</h3>
<p>This pipeline is similar to the de novo PRS evaluation pipeline but it uses the initial train/test split described above for PRS evaluation. The steps are similar to those described above and the evaluation is performed with the PRSice2 framework, limiting somehow the available model evaluation metrics. Also, averaged SNP effects are used. In the example below, we provide a PRS candidate (<code>snpSelection</code> argument) based on the previous example, with SNPs appearing more than 1 time. We also request the evaluation to be performed in the same split set (<code>useDenovoWorkspace</code> argument).</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snpSelection</span> <span class="op">&lt;-</span> <span class="va">snpSummary</span><span class="op">[</span><span class="va">snpSummary</span><span class="op">$</span><span class="va">freq</span><span class="op">&gt;</span><span class="fl">1</span>,<span class="op">]</span>

<span class="va">d2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d2</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsPipeline</a></span><span class="op">(</span>
        gwe<span class="op">=</span><span class="va">toy</span>,
        phenotype<span class="op">=</span><span class="va">response</span>,
        covariates<span class="op">=</span><span class="va">covariates</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,
        trainSize<span class="op">=</span><span class="fl">0.8</span>,
        niter<span class="op">=</span><span class="fl">4</span>,
        filters<span class="op">=</span><span class="va">filts</span>,
        pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
        imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
        snpSelection<span class="op">=</span><span class="va">snpSelection</span>,
        gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
        prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        prsWorkspace<span class="op">=</span><span class="va">wspace2</span>,
        cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
        logging<span class="op">=</span><span class="st">"file"</span>,
        evalWith<span class="op">=</span><span class="st">"prsice"</span>,
        useDenovoWorkspace<span class="op">=</span><span class="va">dnList</span>,
        rc<span class="op">=</span><span class="fl">0.25</span>
    <span class="op">)</span>
<span class="op">}</span></code></pre></div>
<pre><code><span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Input</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## Input pipeline list (dnList) : a list of PRS pipeline summaries</span></code></pre>
<pre><code><span class="co">## Input dataset object (gwe)   : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Regression parameters</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## Phenotypic response (phenotype)          : BMI</span></code></pre>
<pre><code><span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span></code></pre>
<pre><code><span class="co">## Principal Components in covariates (pcs) : No</span></code></pre>
<pre><code><span class="co">## Number of Principal Components (npcs)    : -</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Preprocessing options</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## Sample and genotype filtering options (filters)     :</span></code></pre>
<pre><code><span class="co">##   snpCallRate    : 0.98</span></code></pre>
<pre><code><span class="co">##   sampleCallRate : 0.95</span></code></pre>
<pre><code><span class="co">##   maf            : 0.05</span></code></pre>
<pre><code><span class="co">##   hwe            : 0.001</span></code></pre>
<pre><code><span class="co">##   heteroStat     : median</span></code></pre>
<pre><code><span class="co">##   heteroFac      : 3</span></code></pre>
<pre><code><span class="co">##   heteroHard     : NA</span></code></pre>
<pre><code><span class="co">##   pcaOut         : No</span></code></pre>
<pre><code><span class="co">##   pcaRobust      : hubert</span></code></pre>
<pre><code><span class="co">##   nPC            : NA</span></code></pre>
<pre><code><span class="co">##   LD             : 0.2</span></code></pre>
<pre><code><span class="co">##   IBD            : NA</span></code></pre>
<pre><code><span class="co">##   inbreed        : NA</span></code></pre>
<pre><code><span class="co">## Principal Components calculation method (pcaMethod) : snprel</span></code></pre>
<pre><code><span class="co">## Impute missing genotypes (imputeMissing)            : No</span></code></pre>
<pre><code><span class="co">## Missing genotype imputation method (imputeMethod)   : single</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## GWA analysis options</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## Genome-Wide Association tests (gwaMethods)        : statgen</span></code></pre>
<pre><code><span class="co">## Association tests combination method (gwaCombine) : fisher</span></code></pre>
<pre><code><span class="co">## Association tests options                         :</span></code></pre>
<pre><code><span class="co">##   statgen :</span></code></pre>
<pre><code><span class="co">##     kinship : astle</span></code></pre>
<pre><code><span class="co">##     reml    : EMMA</span></code></pre>
<pre><code><span class="co">##     gls     : single</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## PRS candidate extraction options</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span></code></pre>
<pre><code><span class="co">## PRS algorithms options (prsMethods)    :</span></code></pre>
<pre><code><span class="co">##   lassosum :</span></code></pre>
<pre><code><span class="co">##     anc   : eur</span></code></pre>
<pre><code><span class="co">##     valid : auto</span></code></pre>
<pre><code><span class="co">##   prsice :</span></code></pre>
<pre><code><span class="co">##     clump_kb : 250</span></code></pre>
<pre><code><span class="co">##     clump_r2 : 0.1</span></code></pre>
<pre><code><span class="co">##     clump_p  : 0.75</span></code></pre>
<pre><code><span class="co">##     score    : avg</span></code></pre>
<pre><code><span class="co">##     perm     : 100</span></code></pre>
<pre><code><span class="co">##     seed     : 42</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## Training set size (trainSize)              : 80%</span></code></pre>
<pre><code><span class="co">## Number of PRS selection iterations (niter) : 4</span></code></pre>
<pre><code><span class="co">## PRS evaluation framework (evalWith)        : prsice</span></code></pre>
<pre><code><span class="co">## Provided candidate SNPs for PRS            : 15 SNPs</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Miscellaneous options</span></code></pre>
<pre><code><span class="co">## --------------------------------------------------------------------</span></code></pre>
<pre><code><span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01</span></code></pre>
<pre><code><span class="co">## Cleanup level (cleanup)                  : intermediate</span></code></pre>
<pre><code><span class="co">## Output type (output)                     : summaries</span></code></pre>
<pre><code><span class="co">## Run ID (runId)                           : UHGYAAXXFJR5949F</span></code></pre>
<pre><code><span class="co">## Available cores fraction (rc)            : single core</span></code></pre>
<pre><code><span class="co">## </span></code></pre>
<pre><code><span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_iter_1.log</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## -----&gt; Evaluation pipeline iteration 1</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span></code></pre>
<pre><code><span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1</span></code></pre>
<pre><code><span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1</span></code></pre>
<pre><code><span class="co">##   preparing summary statistics for the base file...</span></code></pre>
<pre><code><span class="co">##   preparing the covariates file...</span></code></pre>
<pre><code><span class="co">##   preparing the phenotype file...</span></code></pre>
<pre><code><span class="co">##   preparing PLINK temporary files - target...</span></code></pre>
<pre><code><span class="co">## Preflight...</span></code></pre>
<pre><code><span class="co">## Preparation done!</span></code></pre>
<pre><code><span class="co">## Running PRSice PRS algorithm in apply mode</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1/base_ASWDG9857Y \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1/pheno_ASWDG9857Y \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1/plink_prsice_ASWDG9857Y \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1/covar_ASWDG9857Y \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_ASWDG9857Y \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span></code></pre>
<pre><code><span class="co">## Done!</span></code></pre>
<pre><code><span class="co">## Adjusted PRSice R^2 is: 0.0772949</span></code></pre>
<pre><code><span class="co">## Full model R^2 is: 0.47166</span></code></pre>
<pre><code><span class="co">## Null model R^2 is: 0.394365</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1/evaluate_result.RData-----</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_iter_2.log</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## -----&gt; Evaluation pipeline iteration 2</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span></code></pre>
<pre><code><span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2</span></code></pre>
<pre><code><span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2</span></code></pre>
<pre><code><span class="co">##   preparing summary statistics for the base file...</span></code></pre>
<pre><code><span class="co">##   preparing the covariates file...</span></code></pre>
<pre><code><span class="co">##   preparing the phenotype file...</span></code></pre>
<pre><code><span class="co">##   preparing PLINK temporary files - target...</span></code></pre>
<pre><code><span class="co">## Preflight...</span></code></pre>
<pre><code><span class="co">## Preparation done!</span></code></pre>
<pre><code><span class="co">## Running PRSice PRS algorithm in apply mode</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2/base_GBEMY7983Y \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2/pheno_GBEMY7983Y \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2/plink_prsice_GBEMY7983Y \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2/covar_GBEMY7983Y \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_GBEMY7983Y \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span></code></pre>
<pre><code><span class="co">## Done!</span></code></pre>
<pre><code><span class="co">## Adjusted PRSice R^2 is: 0.0340179</span></code></pre>
<pre><code><span class="co">## Full model R^2 is: 0.524364</span></code></pre>
<pre><code><span class="co">## Null model R^2 is: 0.490346</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_2/evaluate_result.RData-----</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_iter_3.log</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## -----&gt; Evaluation pipeline iteration 3</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span></code></pre>
<pre><code><span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3</span></code></pre>
<pre><code><span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3</span></code></pre>
<pre><code><span class="co">##   preparing summary statistics for the base file...</span></code></pre>
<pre><code><span class="co">##   preparing the covariates file...</span></code></pre>
<pre><code><span class="co">##   preparing the phenotype file...</span></code></pre>
<pre><code><span class="co">##   preparing PLINK temporary files - target...</span></code></pre>
<pre><code><span class="co">## Preflight...</span></code></pre>
<pre><code><span class="co">## Preparation done!</span></code></pre>
<pre><code><span class="co">## Running PRSice PRS algorithm in apply mode</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3/base_OIMFI4115V \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3/pheno_OIMFI4115V \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3/plink_prsice_OIMFI4115V \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3/covar_OIMFI4115V \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_OIMFI4115V \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span></code></pre>
<pre><code><span class="co">## Done!</span></code></pre>
<pre><code><span class="co">## Adjusted PRSice R^2 is: 0.0710682</span></code></pre>
<pre><code><span class="co">## Full model R^2 is: 0.677534</span></code></pre>
<pre><code><span class="co">## Null model R^2 is: 0.606466</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_3/evaluate_result.RData-----</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_iter_4.log</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## -----&gt; Evaluation pipeline iteration 4</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span></code></pre>
<pre><code><span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4</span></code></pre>
<pre><code><span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4</span></code></pre>
<pre><code><span class="co">##   preparing summary statistics for the base file...</span></code></pre>
<pre><code><span class="co">##   preparing the covariates file...</span></code></pre>
<pre><code><span class="co">##   preparing the phenotype file...</span></code></pre>
<pre><code><span class="co">##   preparing PLINK temporary files - target...</span></code></pre>
<pre><code><span class="co">## Preflight...</span></code></pre>
<pre><code><span class="co">## Preparation done!</span></code></pre>
<pre><code><span class="co">## Running PRSice PRS algorithm in apply mode</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4/base_OOIQL9806D \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4/pheno_OOIQL9806D \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4/plink_prsice_OOIQL9806D \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4/covar_OOIQL9806D \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_OOIQL9806D \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span></code></pre>
<pre><code><span class="co">## Done!</span></code></pre>
<pre><code><span class="co">## Adjusted PRSice R^2 is: 0.0188534</span></code></pre>
<pre><code><span class="co">## Full model R^2 is: 0.588091</span></code></pre>
<pre><code><span class="co">## Null model R^2 is: 0.569238</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_4/evaluate_result.RData-----</span></code></pre>
<pre><code><span class="co">## ================================================================</span></code></pre>
<pre><code><span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01</span></code></pre>
<p>The <code>out</code> argument is a list of data frames containing a more elegant format of the SNPs in <code>snpSelection</code>. Since we requested evaluation with PRSice2, each data frame has a list of attributes attached to it, summarizing this evaluation:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">out</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">## $names</span>
<span class="co">## [1] "chromosome"       "position"         "variant_id"       "risk_allele"     </span>
<span class="co">## [5] "reference_allele" "locus_name"       "effect_weight"    "OR"              </span>
<span class="co">## [9] "asm"             </span>
<span class="co">## </span>
<span class="co">## $row.names</span>
<span class="co">##  [1] "rs508857"   "rs11876399" "rs16943205" "rs1740090"  "rs7280065" </span>
<span class="co">##  [6] "rs1442343"  "rs1567041"  "rs1582761"  "rs1785706"  "rs1940651" </span>
<span class="co">## [11] "rs2827694"  "rs8093690"  "rs875356"   "rs9303974"  "rs9977668" </span>
<span class="co">## </span>
<span class="co">## $class</span>
<span class="co">## [1] "data.frame"</span>
<span class="co">## </span>
<span class="co">## $workspace</span>
<span class="co">## [1] "/tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f1e26fb01/UHGYAAXXFJR5949F_evaluate_1"</span>
<span class="co">## </span>
<span class="co">## $PR2</span>
<span class="co">##        r2m        r2n        r2p          p       nsnp </span>
<span class="co">##  0.4716600  0.3943650  0.0772949  0.1540910 15.0000000 </span>
<span class="co">## </span>
<span class="co">## $nsnp</span>
<span class="co">## [1] 15</span></code></pre></div>
<p>The <code>PR2</code> attribute contains the PRsice2 evalutation metrics which are not identical between two outputs as they refer to different dataset splits in training and testing.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">out</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,<span class="st">"PR2"</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">out</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,<span class="st">"PR2"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] FALSE</span></code></pre>
<p>These metrics are properly gathered and returned by the <code><a href="../reference/prismaFuns.html">prsSelection()</a></code> function.</p>
</div>
<div class="section level3">
<h3 id="prs-evaluation-pipeline-based-on-the-original-traintest-split-with-r">PRS evaluation pipeline based on the original train/test split with R<a class="anchor" aria-label="anchor" href="#prs-evaluation-pipeline-based-on-the-original-traintest-split-with-r"></a>
</h3>
<p>Again, this pipeline is similar to the one described above but this time evaluation is done with native R functions (manual PRS construction outside PRSice2 followed by GLM regressions). This is the fastest evaluation pipeline and allows for the collection of many additional metrics. We change the <code>evalWith</code> argument to <code>"vanilla"</code> which is also the default.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snpSelection</span> <span class="op">&lt;-</span> <span class="va">snpSummary</span><span class="op">[</span><span class="va">snpSummary</span><span class="op">$</span><span class="va">freq</span><span class="op">&gt;</span><span class="fl">1</span>,<span class="op">]</span>

<span class="va">d3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d3</span><span class="op">)</span>

<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsPipeline</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    snpSelection<span class="op">=</span><span class="va">snpSelection</span>,
    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace3</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    evalWith<span class="op">=</span><span class="st">"vanilla"</span>,
    useDenovoWorkspace<span class="op">=</span><span class="va">dnList</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>
<span class="op">)</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input pipeline list (dnList) : a list of PRS pipeline summaries</span>
<span class="co">## Input dataset object (gwe)   : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : single</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : fisher</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 1</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 10000</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)              : 80%</span>
<span class="co">## Number of PRS selection iterations (niter) : 4</span>
<span class="co">## PRS evaluation framework (evalWith)        : vanilla</span>
<span class="co">## Provided candidate SNPs for PRS            : 15 SNPs</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## Output type (output)                     : summaries</span>
<span class="co">## Run ID (runId)                           : XIIJHIFRCQR2863W</span>
<span class="co">## Available cores fraction (rc)            : single core</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/XIIJHIFRCQR2863W_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8</span></code></pre></div>
<p>This time out is a list of data frames with the collected evaluation metrics. These are described in detail in the respective man pages.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">"rbind"</span>,<span class="va">out</span><span class="op">)</span>

<span class="va">m</span>
<span class="co">##   n_snp freq   full_r2  full_pvalue   prs_pvalue  full_aic reduced_r2</span>
<span class="co">## 1    15    2 0.7536168 1.819886e-04 0.0002981207 103.57851  0.3943653</span>
<span class="co">## 2    15    2 0.6829144 1.110981e-03 0.0086460253  98.23304  0.4903465</span>
<span class="co">## 3    15    2 0.8301268 1.215715e-05 0.0004733129 101.36053  0.6064660</span>
<span class="co">## 4    15    2 0.6308640 3.252108e-03 0.1343960078  96.98730  0.5692381</span>
<span class="co">##       prs_r2 reduced_pvalue reduced_aic    prs_aic</span>
<span class="co">## 1 0.35925144    0.040976542   119.56629 -15.987775</span>
<span class="co">## 2 0.19256798    0.011227704   105.72423  -7.491190</span>
<span class="co">## 3 0.22366079    0.001547880   116.16283 -14.802303</span>
<span class="co">## 4 0.06162589    0.003107231    98.07511  -1.087806</span></code></pre></div>
<p>These metrics are properly gathered and returned by the <code><a href="../reference/prismaFuns.html">prsSelection()</a></code> function.</p>
</div>
<div class="section level3">
<h3 id="de-novo-prs-evaluation-pipeline">De novo PRS evaluation pipeline<a class="anchor" aria-label="anchor" href="#de-novo-prs-evaluation-pipeline"></a>
</h3>
<p>This evaluation pipeline differs from the other two in that it creates a new dataset split with related preprocessing and thus taking much more time than the other two. See also the Introduction vignette for more details on this process. It is controlled by the lack of the <code>useDenovoWorkspace</code> argument or <code>useDenovoWorkspace=NULL</code>. Evaluation is done with PRSice2 resulting in a list of data frames with attributes as above.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snpSelection</span> <span class="op">&lt;-</span> <span class="va">snpSummary</span><span class="op">[</span><span class="va">snpSummary</span><span class="op">$</span><span class="va">freq</span><span class="op">&gt;</span><span class="fl">1</span>,<span class="op">]</span>

<span class="va">d4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d4</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsPipeline</a></span><span class="op">(</span>
        gwe<span class="op">=</span><span class="va">toy</span>,
        phenotype<span class="op">=</span><span class="va">response</span>,
        covariates<span class="op">=</span><span class="va">covariates</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,
        trainSize<span class="op">=</span><span class="fl">0.8</span>,
        niter<span class="op">=</span><span class="fl">4</span>,
        filters<span class="op">=</span><span class="va">filts</span>,
        pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
        imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
        snpSelection<span class="op">=</span><span class="va">snpSelection</span>,
        gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
        prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        prsWorkspace<span class="op">=</span><span class="va">wspace3</span>,
        cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
        logging<span class="op">=</span><span class="st">"file"</span>,
        rc<span class="op">=</span><span class="fl">0.25</span>
    <span class="op">)</span>
<span class="op">}</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input pipeline list (dnList) : a list of PRS pipeline summaries</span>
<span class="co">## Input dataset object (gwe)   : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : single</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : fisher</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 0.75</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 100</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)              : 80%</span>
<span class="co">## Number of PRS selection iterations (niter) : 4</span>
<span class="co">## PRS evaluation framework (evalWith)        : vanilla</span>
<span class="co">## Provided candidate SNPs for PRS            : 15 SNPs</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## Output type (output)                     : summaries</span>
<span class="co">## Run ID (runId)                           : GKJMITDYNHO1263B</span>
<span class="co">## Available cores fraction (rc)            : single core</span>
<span class="co">## </span>
<span class="co">## Executing external pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; External pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Base GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 30 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 30 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 8 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 8 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 30 associations found out of 565 markers</span>
<span class="co">##   Part 21: 8 associations found out of 313 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 38 associations with (uncorrected) combined p-values).</span>
<span class="co">## 1 SNPs not found in base data! Excluding...</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in apply mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1/base_VNCPN4677K \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1/pheno_VNCPN4677K \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1/plink_prsice_VNCPN4677K \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1/covar_VNCPN4677K \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_VNCPN4677K \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.324046</span>
<span class="co">## Full model R^2 is: 0.60757</span>
<span class="co">## Null model R^2 is: 0.283524</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_1/external_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing external pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; External pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 129</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Base GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 16 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 17 associations found out of 562 markers</span>
<span class="co">##   Part 21: 16 associations found out of 315 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 33 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in apply mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2/base_JDLFX7890Y \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2/pheno_JDLFX7890Y \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2/plink_prsice_JDLFX7890Y \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2/covar_JDLFX7890Y \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_JDLFX7890Y \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.218137</span>
<span class="co">## Full model R^2 is: 0.859661</span>
<span class="co">## Null model R^2 is: 0.641524</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_2/external_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing external pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; External pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 116</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Base GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 16 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 15 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 15 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 16 associations found out of 564 markers</span>
<span class="co">##   Part 21: 15 associations found out of 312 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 31 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in apply mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3/base_LFXXL9085W \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3/pheno_LFXXL9085W \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3/plink_prsice_LFXXL9085W \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3/covar_LFXXL9085W \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_LFXXL9085W \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.108571</span>
<span class="co">## Full model R^2 is: 0.804952</span>
<span class="co">## Null model R^2 is: 0.696381</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_3/external_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing external pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; External pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 0</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 130</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Base GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 21 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 21 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 10 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 10 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 21 associations found out of 567 markers</span>
<span class="co">##   Part 21: 10 associations found out of 312 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 31 associations with (uncorrected) combined p-values).</span>
<span class="co">## 1 SNPs not found in base data! Excluding...</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in apply mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4/base_MGMQU6405C \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4/pheno_MGMQU6405C \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4/plink_prsice_MGMQU6405C \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4/covar_MGMQU6405C \</span>
<span class="co">##   --no-clump \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_MGMQU6405C \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --bar-levels 1 \</span>
<span class="co">##   --fastscore</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.374945</span>
<span class="co">## Full model R^2 is: 0.928113</span>
<span class="co">## Null model R^2 is: 0.553168</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8/GKJMITDYNHO1263B_external_4/external_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f7f5c3ae8</span></code></pre></div>
<p>The evaluation metrics are properly gathered and returned by the <code><a href="../reference/prismaFuns.html">prsSelection()</a></code> function.</p>
</div>
<div class="section level3">
<h3 id="wrapping-the-evaluation-process-with-prsselection">Wrapping the evaluation process with <code>prsSelection()</code><a class="anchor" aria-label="anchor" href="#wrapping-the-evaluation-process-with-prsselection"></a>
</h3>
<p>As mentioned, PRISMA tries to select the best PRS candidates by subsetting the aggregated SNPs and iteratively evaluating. More precisely the <code><a href="../reference/prismaFuns.html">prsSelection()</a></code> function is responsible for the following:</p>
<ol style="list-style-type: decimal">
<li>Create sets of PRS candidates from aggregated PRS extraction SNPs based on:
<ol style="list-style-type: lower-alpha">
<li>Fixed quantiles of the frequency distribution (e.g. 10%, 20%, 30%, 40%, 50%, 60%, 75%, 80%, 90%, 95%, 99%).</li>
<li>A stepwise set creation process based on frequency of appearance, i.e. each set contains SNPs that appear 5, 6, â€¦ up to the maximum number of appearances.</li>
</ol>
</li>
<li>For each set created at (1), run an evaluation pipeline and collect evaluation metrics in a format proper for downstream analysis.</li>
</ol>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d5</span><span class="op">)</span>

<span class="va">evalMetrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prsSelection</a></span><span class="op">(</span>
    dnList<span class="op">=</span><span class="va">dnList</span>,
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace5</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"summary"</span>,
    useDenovoWorkspace<span class="op">=</span><span class="va">dnList</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>
<span class="op">)</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 9 markers</span>
<span class="co">##   iteration 2 yields 66 markers</span>
<span class="co">##   iteration 3 yields 5 markers</span>
<span class="co">##   iteration 4 yields 19 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : single</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : simes</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 1</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 10000</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)                               : 80%</span>
<span class="co">## Number of PRS selection iterations (niter)                  : 4</span>
<span class="co">## PRS selection resolution method (resolution)                : frequency</span>
<span class="co">## PRS selection resolution step (step)                        : 1</span>
<span class="co">## Minimum SNP frequency in PRS (minFreq)                      : 2</span>
<span class="co">## Minimum number of SNPs in PRS (minSnps)                     : 2</span>
<span class="co">## Drop quantiles with same number of SNPs (dropSameQuantiles) : Yes</span>
<span class="co">## PRS aggregation method (aggregation)                        : intersection</span>
<span class="co">## SNP effect summarization method (effectWeight)              : mean</span>
<span class="co">## PRS evaluation framework (evalWith)                         : vanilla</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## Output type (output)                     : summary</span>
<span class="co">## Run ID (runId)                           : auto</span>
<span class="co">## Available cores fraction (rc)            : 25%</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 2 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs9977668</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/2</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 3 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs508857 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f10c862b8/3</span></code></pre></div>
<p><code>evalMetrics</code> is a list with contents used by the subsequent <code><a href="../reference/selectPrs.html">selectPrs()</a></code> function:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">candidates</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectPrs.html">selectPrs</a></span><span class="op">(</span>
    metrics<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">metrics</span>,
    snpSelection<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">pgs</span>,
    gwe<span class="op">=</span><span class="va">toy</span>,
    base<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">baseline</span>
<span class="op">)</span>
<span class="co">## The optimal number of markers for PRS based on prs_r2 is 5 at prs_r2=0.14494</span></code></pre></div>
<p>The <code>candidates</code> variable is a list of candidate PRSs.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">candidates</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">##            chromosome position variant_id risk_allele reference_allele</span>
<span class="co">## rs508857           18  2101958   rs508857           A                G</span>
<span class="co">## rs11876399         18  3615943 rs11876399           G                A</span>
<span class="co">## rs16943205         18  2412278 rs16943205           G                A</span>
<span class="co">## rs1740090          21 23258743  rs1740090           G                A</span>
<span class="co">## rs7280065          21 24827764  rs7280065           G                A</span>
<span class="co">##            locus_name effect_weight         OR  asm freq</span>
<span class="co">## rs508857           NA      2.169076  8.7501978 hg19    4</span>
<span class="co">## rs11876399         NA      1.456026  4.2888806 hg19    3</span>
<span class="co">## rs16943205         NA     -1.534372  0.2155911 hg19    3</span>
<span class="co">## rs1740090          NA     -1.845601  0.1579304 hg19    3</span>
<span class="co">## rs7280065          NA      2.324420 10.2207480 hg19    3</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="wrapping-the-whole-process-with-the-prisma-function">Wrapping the whole process with the <code>prisma()</code> function<a class="anchor" aria-label="anchor" href="#wrapping-the-whole-process-with-the-prisma-function"></a>
</h3>
<p>The aforementioned processes can be executed with the <code><a href="../reference/prismaFuns.html">prisma()</a></code> function. This will output a list object with the PRS candidates and evaluatiom metrics which is then fed to cross-validation, lookup and reporting procedures.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Workspace</span>
<span class="va">d6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d6</span><span class="op">)</span>

<span class="co"># Complete 4-iteration PRS pipeline - ~1 min to run with 4 cores - no PCA</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">prismaOut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prisma</a></span><span class="op">(</span>
        gwe<span class="op">=</span><span class="va">toy</span>,
        phenotype<span class="op">=</span><span class="va">response</span>,
        covariates<span class="op">=</span><span class="va">covariates</span>,
        pcs<span class="op">=</span><span class="cn">FALSE</span>,
        trainSize<span class="op">=</span><span class="fl">0.8</span>,
        niter<span class="op">=</span><span class="fl">4</span>,
        resolution<span class="op">=</span><span class="st">"frequency"</span>,
        minSnps<span class="op">=</span><span class="fl">2</span>,
        filters<span class="op">=</span><span class="va">filts</span>,
        pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
        imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
        gwaMethods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span>,
        prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        prsWorkspace<span class="op">=</span><span class="va">wspace6</span>,
        cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
        logging<span class="op">=</span><span class="st">"file"</span>,
        output<span class="op">=</span><span class="st">"normal"</span>,
        rc<span class="op">=</span><span class="fl">0.25</span>
    <span class="op">)</span>
<span class="op">}</span>
<span class="co">## </span>
<span class="co">## 2022-06-03 15:02:32: Data processing started...</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : singlesplit</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : glm, statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : fishersimesmaxminharmonicwhitlockpandora</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   glm :</span>
<span class="co">##     family : -</span>
<span class="co">##     size   : 1</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 0.75</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 100</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)                               : 80%</span>
<span class="co">## Number of PRS selection iterations (niter)                  : 4</span>
<span class="co">## PRS selection resolution method (resolution)                : frequency</span>
<span class="co">## PRS selection resolution step (step)                        : 1</span>
<span class="co">## Minimum SNP frequency in PRS (minFreq)                      : 2</span>
<span class="co">## Minimum number of SNPs in PRS (minSnps)                     : 2</span>
<span class="co">## Drop quantiles with same number of SNPs (dropSameQuantiles) : Yes</span>
<span class="co">## PRS aggregation method (aggregation)                        : intersectionunion</span>
<span class="co">## SNP effect summarization method (effectWeight)              : meanmedianweight</span>
<span class="co">## PRS evaluation framework (evalWith)                         : vanillaprscice</span>
<span class="co">## </span>
<span class="co">## PRS selection options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS selection method (prsSelectMethod)           : maxima</span>
<span class="co">## PRS selection criterion (prsSelectCrit)          : prs_r2</span>
<span class="co">## PRS criterion summarization (prsSelectStat)      : mean</span>
<span class="co">## R^2 type if criterion includes R^2 (prsSelectR2) : adjusted</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## PRS pipeline output type (dnOutput)      : normal</span>
<span class="co">## Output level (output)                    : normal</span>
<span class="co">## Run ID (runId)                           : auto</span>
<span class="co">## Available cores fraction (rc)            : 25%</span>
<span class="co">## </span>
<span class="co">## ****************************************************************</span>
<span class="co">## Executing PRISMA pipeline with glm</span>
<span class="co">## ****************************************************************</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 1. Discovery of de novo PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline</span>
<span class="co">## </span>
<span class="co">## Executing denovo pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 123</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 123</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 26 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 26 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 10 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 10 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 26 associations found out of 563 markers</span>
<span class="co">##   Part 21: 10 associations found out of 314 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 36 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.00136195902055067</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1/base_DXTGW1434R \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1/pheno_DXTGW1434R \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1/plink_prsice_DXTGW1434R \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1/covar_DXTGW1434R \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_DXTGW1434R \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.152345</span>
<span class="co">## Full model R^2 is: 0.723958</span>
<span class="co">## Null model R^2 is: 0.571613</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_1/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 118</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 128</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 19 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 19 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 18 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 18 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 19 associations found out of 568 markers</span>
<span class="co">##   Part 21: 18 associations found out of 314 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 37 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.011030722261469</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2/base_GWDHF8969Z \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2/pheno_GWDHF8969Z \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2/plink_prsice_GWDHF8969Z \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2/covar_GWDHF8969Z \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_GWDHF8969Z \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.168935</span>
<span class="co">## Full model R^2 is: 0.631064</span>
<span class="co">## Null model R^2 is: 0.462129</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_2/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 0</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 20 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 20 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 8 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 8 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 20 associations found out of 563 markers</span>
<span class="co">##   Part 21: 8 associations found out of 315 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 28 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.163301092128739</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3/base_MDJIV0680I \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3/pheno_MDJIV0680I \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3/plink_prsice_MDJIV0680I \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3/covar_MDJIV0680I \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_MDJIV0680I \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0553973</span>
<span class="co">## Full model R^2 is: 0.546024</span>
<span class="co">## Null model R^2 is: 0.490626</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_3/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 117</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 128</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs1009238 to rs1568327</span>
<span class="co">## Found 23 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 23 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Performing GWAS with GLM over 1 chunks</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">##   testing SNPs from rs2824974 to rs2828316</span>
<span class="co">## Found 18 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 18 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 23 associations found out of 567 markers</span>
<span class="co">##   Part 21: 18 associations found out of 316 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 41 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.290977538053044</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4/base_CTMGC4503V \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4/pheno_CTMGC4503V \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4/plink_prsice_CTMGC4503V \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4/covar_CTMGC4503V \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_CTMGC4503V \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0451768</span>
<span class="co">## Full model R^2 is: 0.636041</span>
<span class="co">## Null model R^2 is: 0.590865</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline/HCMHTBCBFOB1103Q_denovo_4/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/baseline</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 2. Calculation of PRS evaluation metrics</span>
<span class="co">## ################################################################</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 14 markers</span>
<span class="co">##   iteration 2 yields 18 markers</span>
<span class="co">##   iteration 3 yields 14 markers</span>
<span class="co">##   iteration 4 yields 1 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 2 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs8093690</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/2</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 3 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs9303974</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs9303974</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs9303974</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs9303974</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/glm/selection/3</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 3. Selection of best PRS candidate markers</span>
<span class="co">## ################################################################</span>
<span class="co">## The optimal number of markers for PRS based on prs_r2 is 11 at prs_r2=0.16861</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 4. Wrap-up and graphics</span>
<span class="co">## ################################################################</span>
<span class="co">## sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="co">## Getting latest annotation on the fly for hg19 from refseq</span>
<span class="co">## Retrieving genome information for hg19 from refseq</span>
<span class="co">## Retrieving latest gene annotation for hg19 from refseq</span>
<span class="co">## Loading required namespace: RMySQL</span>
<span class="co">## Converting annotation to GenomicRanges object...</span>
<span class="co">## </span>
<span class="co">## Attaching package: 'Biostrings'</span>
<span class="co">## The following object is masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     strsplit</span>
<span class="co">## Getting DNA sequences...</span>
<span class="co">## Getting GC content...</span>
<span class="co">## </span>
<span class="co">## ****************************************************************</span>
<span class="co">## Executing PRISMA pipeline with statgen</span>
<span class="co">## ****************************************************************</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 1. Discovery of de novo PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline</span>
<span class="co">## </span>
<span class="co">## Executing denovo pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 120</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 122</span>
<span class="co">##   Hardy-Weinberg equilibrium: 6</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 10 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 10 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 17 associations found out of 561 markers</span>
<span class="co">##   Part 21: 10 associations found out of 317 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 27 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.121778346166596</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1/base_UKIIH0402C \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1/pheno_UKIIH0402C \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1/plink_prsice_UKIIH0402C \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1/covar_UKIIH0402C \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_UKIIH0402C \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0652354</span>
<span class="co">## Full model R^2 is: 0.585825</span>
<span class="co">## Null model R^2 is: 0.52059</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_1/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 119</span>
<span class="co">##   Hardy-Weinberg equilibrium: 4</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 127</span>
<span class="co">##   Hardy-Weinberg equilibrium: 7</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 23 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 23 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 16 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 16 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 23 associations found out of 565 markers</span>
<span class="co">##   Part 21: 16 associations found out of 313 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 39 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.117569957659734</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2/base_ZRAKA2461W \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2/pheno_ZRAKA2461W \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2/plink_prsice_ZRAKA2461W \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2/covar_ZRAKA2461W \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_ZRAKA2461W \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.144921</span>
<span class="co">## Full model R^2 is: 0.632141</span>
<span class="co">## Null model R^2 is: 0.48722</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_2/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 123</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 118</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 12 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 12 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 17 associations found out of 561 markers</span>
<span class="co">##   Part 21: 12 associations found out of 315 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 29 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 7.59991983284998e-05</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3/base_KNVMD0699O \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3/pheno_KNVMD0699O \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3/plink_prsice_KNVMD0699O \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3/covar_KNVMD0699O \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_KNVMD0699O \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0842771</span>
<span class="co">## Full model R^2 is: 0.454247</span>
<span class="co">## Null model R^2 is: 0.36997</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_3/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 125</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 115</span>
<span class="co">##   Hardy-Weinberg equilibrium: 0</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 21 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 21 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 14 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 21 associations found out of 562 markers</span>
<span class="co">##   Part 21: 14 associations found out of 311 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0200701925748578</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4/base_AVSOU5582A \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4/pheno_AVSOU5582A \</span>
<span class="co">##   --target /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4/plink_prsice_AVSOU5582A \</span>
<span class="co">##   --cov /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4/covar_AVSOU5582A \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_AVSOU5582A \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0325911</span>
<span class="co">## Full model R^2 is: 0.65919</span>
<span class="co">## Null model R^2 is: 0.626598</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline/HCMHTBCBFOB1103Q_denovo_4/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/baseline</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 2. Calculation of PRS evaluation metrics</span>
<span class="co">## ################################################################</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 2 markers</span>
<span class="co">##   iteration 2 yields 20 markers</span>
<span class="co">##   iteration 3 yields 2 markers</span>
<span class="co">##   iteration 4 yields 10 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 2 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs508857</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs508857</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs508857</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs508857</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/2</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 3 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs11876399 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/RtmpiheQyu//tmp/RtmpiheQyu/file6f1f21661d05/statgen/selection/3</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 3. Selection of best PRS candidate markers</span>
<span class="co">## ################################################################</span>
<span class="co">## The optimal number of markers for PRS based on prs_r2 is 5 at prs_r2=0.13692</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 4. Wrap-up and graphics</span>
<span class="co">## ################################################################</span>
<span class="co">## sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="co">## Getting latest annotation on the fly for hg19 from refseq</span>
<span class="co">## Retrieving genome information for hg19 from refseq</span>
<span class="co">## Retrieving latest gene annotation for hg19 from refseq</span>
<span class="co">## Converting annotation to GenomicRanges object...</span>
<span class="co">## Getting DNA sequences...</span>
<span class="co">## Getting GC content...</span>
<span class="co">## </span>
<span class="co">## 2022-06-03 15:06:10: Data processing finished!</span>
<span class="co">## Total processing time: 03 minutes 38 seconds</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="prs-candidate-evaluation-lookup-and-reporting">PRS candidate evaluation, lookup and reporting<a class="anchor" aria-label="anchor" href="#prs-candidate-evaluation-lookup-and-reporting"></a>
</h2>
<p>After the completion of the PRISMA pipeline, we are using the total dataset to evaluate the candidates. We also lookup for the SNP content in GWAS Catalog and finally report the results in an HTML report.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cvMetrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaCvFuns.html">prismaCrossValidate</a></span><span class="op">(</span>
    prismaOut<span class="op">=</span><span class="va">prismaOut</span>,
    gwe<span class="op">=</span><span class="va">toy</span>,
    response<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    leaveOut<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>,<span class="fl">0.25</span>,<span class="fl">0.5</span><span class="op">)</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>
<span class="op">)</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## Running cross-validations for glm with 3 leave-out cases across 1 PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Cross-validating PRS candidate with 11 markers</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 10% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 25% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 50% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## Running cross-validations for statgen with 3 leave-out cases across 1 PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Cross-validating PRS candidate with 5 markers</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 10% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 25% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ----- Leave-out percentage 50% -----</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Running cross-validation iteration 1</span>
<span class="co">## Running cross-validation iteration 2</span>
<span class="co">## Running cross-validation iteration 3</span>
<span class="co">## Running cross-validation iteration 4</span>
<span class="co">## Running cross-validation iteration 5</span>
<span class="co">## Running cross-validation iteration 6</span>
<span class="co">## Running cross-validation iteration 7</span>
<span class="co">## Running cross-validation iteration 8</span>
<span class="co">## Running cross-validation iteration 9</span>
<span class="co">## Running cross-validation iteration 10</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## Done! Use summarizeCvMetrics() and plotCvMetrics() to summarize and inspect the results.</span>
<span class="co">## ################################################################</span></code></pre></div>
<p><code>cvMetrics</code> is a list of validation metrics used with the reporting function. You may inspect it but it has not great meaning outside the report.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lookup</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ebiQueryFuns.html">prismaLookup</a></span><span class="op">(</span><span class="va">prismaOut</span><span class="op">)</span>
<span class="co">## -----&gt; Querying for glm</span>
<span class="co">## Querying GWAS Catalog with the largest PRS</span>
<span class="co">## -----&gt; Querying for statgen</span>
<span class="co">## Querying GWAS Catalog with the largest PRS</span></code></pre></div>
<p>Similarly, <code>lookup</code> is a list with findings on GWAS Catalog, regarding the SNP content of the PRS candidates and potential associated traits.</p>
<p>Finally, create the report:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/prismaOutFuns.html">prismaReport</a></span><span class="op">(</span><span class="va">toy</span>,<span class="va">prismaOut</span>,<span class="va">cvMetrics</span>,<span class="va">lookup</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="wrapping-the-whole-process-with-the-prismapipeline-function">Wrapping the whole process with the <code>prismaPipeline()</code> function<a class="anchor" aria-label="anchor" href="#wrapping-the-whole-process-with-the-prismapipeline-function"></a>
</h3>
<p>The difference with the <code><a href="../reference/prismaFuns.html">prisma()</a></code> function is that the <code><a href="../reference/prismaFuns.html">prismaPipeline()</a></code> function wraps the <code><a href="../reference/prismaFuns.html">prisma()</a></code> call, the calculation of evaluation and cross-validation metrics, the lookup and report. The report will be written in <code>outPath</code>.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">d7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,<span class="va">d7</span><span class="op">)</span>

<span class="fu"><a href="../reference/prismaFuns.html">prismaPipeline</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace4</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"normal"</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>,
    outPath<span class="op">=</span><span class="va">outPath</span>
<span class="op">)</span></code></pre></div>
<p>The output report for this case is shown embedded below. Plese <strong>note</strong> that this is <strong>not</strong> a realistic report. The normal one is richer and with more PRS candidates to evaluate.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"file://"</span>,<span class="va">outPath</span>,<span class="st">"index.html"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.62.0                  </span>
<span class="co">##  [3] rtracklayer_1.54.0                Biostrings_2.62.0                </span>
<span class="co">##  [5] XVector_0.34.0                    prisma_0.0.1                     </span>
<span class="co">##  [7] SNPRelate_1.28.0                  gdsfmt_1.30.0                    </span>
<span class="co">##  [9] snpStats_1.44.0                   Matrix_1.4-1                     </span>
<span class="co">## [11] survival_3.3-1                    SummarizedExperiment_1.24.0      </span>
<span class="co">## [13] Biobase_2.54.0                    GenomicRanges_1.46.1             </span>
<span class="co">## [15] GenomeInfoDb_1.30.1               IRanges_2.28.0                   </span>
<span class="co">## [17] S4Vectors_0.32.4                  BiocGenerics_0.40.0              </span>
<span class="co">## [19] MatrixGenerics_1.6.0              matrixStats_0.62.0               </span>
<span class="co">## [21] lassosum_0.4.5                   </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] highr_0.9                               </span>
<span class="co">##  [19] knitr_1.39                              </span>
<span class="co">##  [20] rstudioapi_0.13                         </span>
<span class="co">##  [21] robustbase_0.95-0                       </span>
<span class="co">##  [22] TTR_0.24.3                              </span>
<span class="co">##  [23] listenv_0.8.0                           </span>
<span class="co">##  [24] urltools_1.7.3                          </span>
<span class="co">##  [25] optparse_1.7.1                          </span>
<span class="co">##  [26] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [27] farver_2.1.0                            </span>
<span class="co">##  [28] bit64_4.0.5                             </span>
<span class="co">##  [29] rprojroot_2.0.3                         </span>
<span class="co">##  [30] parallelly_1.31.1                       </span>
<span class="co">##  [31] vctrs_0.4.1                             </span>
<span class="co">##  [32] generics_0.1.2                          </span>
<span class="co">##  [33] xfun_0.31                               </span>
<span class="co">##  [34] BiocFileCache_2.2.1                     </span>
<span class="co">##  [35] R6_2.5.1                                </span>
<span class="co">##  [36] rmdformats_1.0.3                        </span>
<span class="co">##  [37] bitops_1.0-7                            </span>
<span class="co">##  [38] cachem_1.0.6                            </span>
<span class="co">##  [39] DelayedArray_0.20.0                     </span>
<span class="co">##  [40] assertthat_0.2.1                        </span>
<span class="co">##  [41] gwasrapidd_0.99.12                      </span>
<span class="co">##  [42] BiocIO_1.4.0                            </span>
<span class="co">##  [43] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [44] scales_1.2.0                            </span>
<span class="co">##  [45] gtable_0.3.0                            </span>
<span class="co">##  [46] globals_0.15.0                          </span>
<span class="co">##  [47] rlang_1.0.2                             </span>
<span class="co">##  [48] systemfonts_1.0.4                       </span>
<span class="co">##  [49] splines_4.1.3                           </span>
<span class="co">##  [50] rsnps_0.5.0.0                           </span>
<span class="co">##  [51] sitadela_1.3.1                          </span>
<span class="co">##  [52] BiocManager_1.30.16                     </span>
<span class="co">##  [53] yaml_2.3.5                              </span>
<span class="co">##  [54] reshape2_1.4.4                          </span>
<span class="co">##  [55] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [56] quantmod_0.4.20                         </span>
<span class="co">##  [57] RMySQL_0.10.23                          </span>
<span class="co">##  [58] RBGL_1.70.0                             </span>
<span class="co">##  [59] tools_4.1.3                             </span>
<span class="co">##  [60] lava_1.6.10                             </span>
<span class="co">##  [61] bookdown_0.26                           </span>
<span class="co">##  [62] liftOver_1.18.0                         </span>
<span class="co">##  [63] ggplot2_3.3.5                           </span>
<span class="co">##  [64] ellipsis_0.3.2                          </span>
<span class="co">##  [65] kableExtra_1.3.4                        </span>
<span class="co">##  [66] jquerylib_0.1.4                         </span>
<span class="co">##  [67] Rcpp_1.0.8.3                            </span>
<span class="co">##  [68] rrBLUP_4.6.1                            </span>
<span class="co">##  [69] plyr_1.8.7                              </span>
<span class="co">##  [70] progress_1.2.2                          </span>
<span class="co">##  [71] zlibbioc_1.40.0                         </span>
<span class="co">##  [72] purrr_0.3.4                             </span>
<span class="co">##  [73] RCurl_1.98-1.6                          </span>
<span class="co">##  [74] prettyunits_1.1.1                       </span>
<span class="co">##  [75] cowplot_1.1.1                           </span>
<span class="co">##  [76] zoo_1.8-10                              </span>
<span class="co">##  [77] fs_1.5.2                                </span>
<span class="co">##  [78] crul_1.2.0                              </span>
<span class="co">##  [79] magrittr_2.0.3                          </span>
<span class="co">##  [80] data.table_1.14.2                       </span>
<span class="co">##  [81] openxlsx_4.2.5                          </span>
<span class="co">##  [82] triebeard_0.3.0                         </span>
<span class="co">##  [83] mvtnorm_1.1-3                           </span>
<span class="co">##  [84] survcomp_1.44.1                         </span>
<span class="co">##  [85] hms_1.1.1                               </span>
<span class="co">##  [86] evaluate_0.15                           </span>
<span class="co">##  [87] XML_3.99-0.9                            </span>
<span class="co">##  [88] RMTstat_0.3.1                           </span>
<span class="co">##  [89] shape_1.4.6                             </span>
<span class="co">##  [90] compiler_4.1.3                          </span>
<span class="co">##  [91] biomaRt_2.50.3                          </span>
<span class="co">##  [92] tibble_3.1.7                            </span>
<span class="co">##  [93] KernSmooth_2.23-20                      </span>
<span class="co">##  [94] crayon_1.5.1                            </span>
<span class="co">##  [95] R.oo_1.24.0                             </span>
<span class="co">##  [96] htmltools_0.5.2                         </span>
<span class="co">##  [97] pcaPP_2.0-1                             </span>
<span class="co">##  [98] tzdb_0.3.0                              </span>
<span class="co">##  [99] tidyr_1.2.0                             </span>
<span class="co">## [100] rrcov_1.7-0                             </span>
<span class="co">## [101] lubridate_1.8.0                         </span>
<span class="co">## [102] DBI_1.1.2                               </span>
<span class="co">## [103] SuppDists_1.1-9.7                       </span>
<span class="co">## [104] dbplyr_2.1.1                            </span>
<span class="co">## [105] MASS_7.3-56                             </span>
<span class="co">## [106] rappdirs_0.3.3                          </span>
<span class="co">## [107] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [108] getopt_1.20.3                           </span>
<span class="co">## [109] readr_2.1.2                             </span>
<span class="co">## [110] rmeta_3.0                               </span>
<span class="co">## [111] cli_3.3.0                               </span>
<span class="co">## [112] quadprog_1.5-8                          </span>
<span class="co">## [113] R.methodsS3_1.8.1                       </span>
<span class="co">## [114] parallel_4.1.3                          </span>
<span class="co">## [115] pkgconfig_2.0.3                         </span>
<span class="co">## [116] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [117] pkgdown_2.0.3.9000                      </span>
<span class="co">## [118] GenomicAlignments_1.30.0                </span>
<span class="co">## [119] signal_0.7-7                            </span>
<span class="co">## [120] xml2_1.3.3                              </span>
<span class="co">## [121] foreach_1.5.2                           </span>
<span class="co">## [122] svglite_2.1.0                           </span>
<span class="co">## [123] bslib_0.3.1                             </span>
<span class="co">## [124] webshot_0.5.3                           </span>
<span class="co">## [125] prodlim_2019.11.13                      </span>
<span class="co">## [126] rvest_1.0.2                             </span>
<span class="co">## [127] stringr_1.4.0                           </span>
<span class="co">## [128] VariantAnnotation_1.40.0                </span>
<span class="co">## [129] digest_0.6.29                           </span>
<span class="co">## [130] graph_1.72.0                            </span>
<span class="co">## [131] httpcode_0.3.0                          </span>
<span class="co">## [132] rmarkdown_2.14                          </span>
<span class="co">## [133] harmonicmeanp_3.0                       </span>
<span class="co">## [134] restfulr_0.0.13                         </span>
<span class="co">## [135] curl_4.3.2                              </span>
<span class="co">## [136] Rsamtools_2.10.0                        </span>
<span class="co">## [137] rjson_0.2.21                            </span>
<span class="co">## [138] tseries_0.10-51                         </span>
<span class="co">## [139] lifecycle_1.0.1                         </span>
<span class="co">## [140] jsonlite_1.8.0                          </span>
<span class="co">## [141] survivalROC_1.0.3                       </span>
<span class="co">## [142] desc_1.4.1                              </span>
<span class="co">## [143] viridisLite_0.4.0                       </span>
<span class="co">## [144] fansi_1.0.3                             </span>
<span class="co">## [145] pillar_1.7.0                            </span>
<span class="co">## [146] lattice_0.20-45                         </span>
<span class="co">## [147] DEoptimR_1.0-11                         </span>
<span class="co">## [148] KEGGREST_1.34.0                         </span>
<span class="co">## [149] fastmap_1.1.0                           </span>
<span class="co">## [150] httr_1.4.3                              </span>
<span class="co">## [151] GO.db_3.14.0                            </span>
<span class="co">## [152] xts_0.12.1                              </span>
<span class="co">## [153] glue_1.6.2                              </span>
<span class="co">## [154] zip_2.2.0                               </span>
<span class="co">## [155] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [156] png_0.1-7                               </span>
<span class="co">## [157] iterators_1.0.14                        </span>
<span class="co">## [158] pander_0.6.5                            </span>
<span class="co">## [159] glmnet_4.1-4                            </span>
<span class="co">## [160] bit_4.0.4                               </span>
<span class="co">## [161] stringi_1.7.6                           </span>
<span class="co">## [162] sass_0.4.1                              </span>
<span class="co">## [163] bootstrap_2019.6                        </span>
<span class="co">## [164] blob_1.2.3                              </span>
<span class="co">## [165] statgenGWAS_1.0.8                       </span>
<span class="co">## [166] textshaping_0.3.6                       </span>
<span class="co">## [167] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [168] memoise_2.0.1                           </span>
<span class="co">## [169] FMStable_0.1-2                          </span>
<span class="co">## [170] dplyr_1.0.8                             </span>
<span class="co">## [171] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
