<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Evaluation of external PRS or external GWAS â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Evaluation of external PRS or external GWAS">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Evaluation of external PRS or external GWAS</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-29</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/09-external.Rmd" class="external-link"><code>vignettes/09-external.Rmd</code></a></small>
      <div class="d-none name"><code>09-external.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Although the main PRISMA functionality has to do with <em>de novo</em> PRS derivation and evaluation, one can easily evaluate an externally derived PRS, for example a PRS retrieved from the <a href="https://www.pgscatalog.org/" class="external-link">EBI PGS Catalog</a>. This can be done in two ways:</p>
<ol style="list-style-type: decimal">
<li>Subsetting the <code>GWASExperiment</code> according to the SNPs in the retrieved PRS/PGS and running a PRISMA pipeline (<code><a href="../reference/prismaFuns.html">prisma()</a></code> or <code><a href="../reference/prismaFuns.html">prismaPipeline()</a></code>). This is the better way as new effects should be calculated suitable for our population. This pipeline should run quite fast, if the number of SNPs included in the external PRS are few.</li>
<li>Provided the retireved PRS directly in the <code><a href="../reference/prismaCvFuns.html">prsCrossValidate()</a></code> function. This will take the object as is (with own effects) and return some evaluation scores when the PRS is tested with our population and genotypes. The PRS retrieval functions included in the package (<code><a href="../reference/ebiQueryFuns.html">getPGSScores()</a></code>) will fetch data frames ready for usage with <code><a href="../reference/prismaCvFuns.html">prsCrossValidate()</a></code>. Quality control of the initial dataset may or may not be performed.</li>
</ol>
</div>
<div class="section level2">
<h2 id="testing-an-externally-retrieved-prs">Testing an externally retrieved PRS<a class="anchor" aria-label="anchor" href="#testing-an-externally-retrieved-prs"></a>
</h2>
<p>For the purposes of this vignette we are going to use the score file reported for <a href="https://www.pgscatalog.org/score/PGS000034/" class="external-link">PGS000034</a> with <em>Adult Body Mass Index</em> reported as trait. This score included with the package. We have also created a smaller toy dataset called <code>pgd</code> from the same source as the toy dataset but containing the common SNPs between PGS000034 and the source data that was used to create the toy dataset.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>,<span class="st">"extdata"</span>,<span class="st">"scores"</span><span class="op">)</span>
<span class="va">pgsId</span> <span class="op">&lt;-</span> <span class="st">"PGS000034"</span>
<span class="va">exPgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ebiQueryFuns.html">getPGSScores</a></span><span class="op">(</span>pgsId<span class="op">=</span><span class="va">pgsId</span>,base<span class="op">=</span><span class="va">base</span><span class="op">)</span>
<span class="co">## Scheduling 1 PGS API calls...</span>
<span class="co">##   Requesting 1 PGS Catalog API calls with PGS ID(s)...</span>
<span class="co">##     Call 1: PGS000034</span>
<span class="co">## Retrieving and enriching scores file(s) for PGS000034</span>
<span class="co">##   local file PGS000034.txt.gz</span>
<span class="co">##   processing PGS000034</span>
<span class="co">## Dealing with duplicates (if any)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">exPgs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">exPgs</span><span class="op">$</span><span class="va">variant_id</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exPgs</span><span class="op">)</span>
<span class="co">##            chromosome  position variant_id risk_allele reference_allele</span>
<span class="co">## rs1000940          17   5223976  rs1000940           G                A</span>
<span class="co">## rs10132280         14  24998019 rs10132280           C                A</span>
<span class="co">## rs1016287           2  59159129  rs1016287           T                C</span>
<span class="co">## rs10182181          2  25003800 rs10182181           G                A</span>
<span class="co">## rs10733682          9 128500735 rs10733682           A                G</span>
<span class="co">## rs10938397          4  44877284 rs10938397           G                A</span>
<span class="co">##            locus_name effect_weight       OR asm</span>
<span class="co">## rs1000940          NA         0.019 1.019182  nr</span>
<span class="co">## rs10132280         NA         0.023 1.023267  nr</span>
<span class="co">## rs1016287          NA         0.023 1.023267  nr</span>
<span class="co">## rs10182181         NA         0.031 1.031486  nr</span>
<span class="co">## rs10733682         NA         0.017 1.017145  nr</span>
<span class="co">## rs10938397         NA         0.040 1.040811  nr</span></code></pre></div>
<p>Load the <strong>pgd</strong> dataset and use the common SNPs with <code>exPgs</code>. They are already the same but we state the commands for demo purposes.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pgd</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">pgd</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"pgd.gds"</span><span class="op">)</span>

<span class="va">pgd</span>
<span class="co">## class: GWASExperiment </span>
<span class="co">## dim: 25 100 </span>
<span class="co">## metadata(5): genome backend filters gdsfile alleleOrder</span>
<span class="co">## assays class: SnpMatrix with 25 rows (SNPs) and 100 columns (samples)</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(25): rs657452 rs11583200 ... rs2033732 rs1928295</span>
<span class="co">## feature names(6): chromosome snp.name ... allele.1 allele.2</span>
<span class="co">## colnames(100): 12 21 ... 590 595</span>
<span class="co">## sample names(6): pedigree member ... sex affected</span>
<span class="co">## phenotype names(6): FID IID ... Age BMI</span>
<span class="co">## performed tests(0):</span>
<span class="co">## associated PRS(0):</span>

<span class="va">exPgs</span> <span class="op">&lt;-</span> <span class="va">exPgs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pgd</span><span class="op">)</span>,<span class="op">]</span></code></pre></div>
<p>Run a PRISMA full pipeline, only with GLM:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,<span class="va">d1</span><span class="op">)</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span>

<span class="fu"><a href="../reference/prismaFuns.html">prismaPipeline</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">pgd</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="st">"glm"</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace1</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"normal"</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>,
    outPath<span class="op">=</span><span class="va">outPath</span>
<span class="op">)</span></code></pre></div>
<p>A raw matrix of evaluation metrics:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaCvFuns.html">prsCrossValidate</a></span><span class="op">(</span><span class="va">exPgs</span>,<span class="va">pgd</span>,response<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,leaveOut<span class="op">=</span><span class="fl">0.2</span>,times<span class="op">=</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="testing-a-derived-prs-with-independent-genotypic-data">Testing a derived PRS with independent genotypic data<a class="anchor" aria-label="anchor" href="#testing-a-derived-prs-with-independent-genotypic-data"></a>
</h2>
<p>Often (actually always) we would like to test our derived PRS candidates to an independent population which could share similar characteristics, for example living in the same country and having similar origins. Sometimes it is also beneficial to test random PRSs with this population to see whether the original PRS does indeed have added value and it is not a random effect. The <code><a href="../reference/prismaCvFuns.html">applyPRS()</a></code> function can be used to accomplish the aforementioned processes.</p>
<p>In the example below, we do not run the PRS randomization process</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span>

<span class="va">wspace2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">prismaOut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prisma</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace2</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"normal"</span>,
    rc<span class="op">=</span><span class="fl">0.5</span>
<span class="op">)</span>

<span class="va">snpSelection</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaOutFuns.html">getPrsCandidates</a></span><span class="op">(</span><span class="va">prismaOut</span>,<span class="st">"statgen"</span>,<span class="fl">1</span><span class="op">)</span>

<span class="va">metrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaCvFuns.html">applyPRS</a></span><span class="op">(</span><span class="va">snpSelection</span>,<span class="va">toy</span>,response<span class="op">=</span><span class="va">response</span>,covariates<span class="op">=</span><span class="va">covariates</span><span class="op">)</span>

<span class="va">metrics</span><span class="op">$</span><span class="va">metrics</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">metrics</span><span class="op">$</span><span class="va">values</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] prisma_0.0.3                SNPRelate_1.28.0           </span>
<span class="co">##  [3] gdsfmt_1.30.0               snpStats_1.44.0            </span>
<span class="co">##  [5] Matrix_1.4-1                survival_3.3-1             </span>
<span class="co">##  [7] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        </span>
<span class="co">## [11] IRanges_2.28.0              S4Vectors_0.32.4           </span>
<span class="co">## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       </span>
<span class="co">## [15] matrixStats_0.62.0          lassosum_0.4.5             </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] knitr_1.39                              </span>
<span class="co">##  [19] rstudioapi_0.13                         </span>
<span class="co">##  [20] robustbase_0.95-0                       </span>
<span class="co">##  [21] TTR_0.24.3                              </span>
<span class="co">##  [22] listenv_0.8.0                           </span>
<span class="co">##  [23] optparse_1.7.1                          </span>
<span class="co">##  [24] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [25] bit64_4.0.5                             </span>
<span class="co">##  [26] rprojroot_2.0.3                         </span>
<span class="co">##  [27] parallelly_1.31.1                       </span>
<span class="co">##  [28] vctrs_0.4.1                             </span>
<span class="co">##  [29] generics_0.1.2                          </span>
<span class="co">##  [30] xfun_0.31                               </span>
<span class="co">##  [31] BiocFileCache_2.2.1                     </span>
<span class="co">##  [32] R6_2.5.1                                </span>
<span class="co">##  [33] rmdformats_1.0.3                        </span>
<span class="co">##  [34] bitops_1.0-7                            </span>
<span class="co">##  [35] cachem_1.0.6                            </span>
<span class="co">##  [36] DelayedArray_0.20.0                     </span>
<span class="co">##  [37] assertthat_0.2.1                        </span>
<span class="co">##  [38] gwasrapidd_0.99.12                      </span>
<span class="co">##  [39] BiocIO_1.4.0                            </span>
<span class="co">##  [40] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [41] scales_1.2.0                            </span>
<span class="co">##  [42] gtable_0.3.0                            </span>
<span class="co">##  [43] globals_0.15.0                          </span>
<span class="co">##  [44] rlang_1.0.2                             </span>
<span class="co">##  [45] systemfonts_1.0.4                       </span>
<span class="co">##  [46] splines_4.1.3                           </span>
<span class="co">##  [47] rtracklayer_1.54.0                      </span>
<span class="co">##  [48] rsnps_0.5.0.0                           </span>
<span class="co">##  [49] sitadela_1.5.1                          </span>
<span class="co">##  [50] BiocManager_1.30.16                     </span>
<span class="co">##  [51] yaml_2.3.5                              </span>
<span class="co">##  [52] reshape2_1.4.4                          </span>
<span class="co">##  [53] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [54] quantmod_0.4.20                         </span>
<span class="co">##  [55] RBGL_1.70.0                             </span>
<span class="co">##  [56] tools_4.1.3                             </span>
<span class="co">##  [57] lava_1.6.10                             </span>
<span class="co">##  [58] bookdown_0.26                           </span>
<span class="co">##  [59] liftOver_1.18.0                         </span>
<span class="co">##  [60] ggplot2_3.3.5                           </span>
<span class="co">##  [61] ellipsis_0.3.2                          </span>
<span class="co">##  [62] kableExtra_1.3.4                        </span>
<span class="co">##  [63] jquerylib_0.1.4                         </span>
<span class="co">##  [64] Rcpp_1.0.8.3                            </span>
<span class="co">##  [65] rrBLUP_4.6.1                            </span>
<span class="co">##  [66] plyr_1.8.7                              </span>
<span class="co">##  [67] progress_1.2.2                          </span>
<span class="co">##  [68] zlibbioc_1.40.0                         </span>
<span class="co">##  [69] purrr_0.3.4                             </span>
<span class="co">##  [70] RCurl_1.98-1.6                          </span>
<span class="co">##  [71] prettyunits_1.1.1                       </span>
<span class="co">##  [72] tidyjson_0.3.1                          </span>
<span class="co">##  [73] cowplot_1.1.1                           </span>
<span class="co">##  [74] zoo_1.8-10                              </span>
<span class="co">##  [75] fs_1.5.2                                </span>
<span class="co">##  [76] crul_1.2.0                              </span>
<span class="co">##  [77] magrittr_2.0.3                          </span>
<span class="co">##  [78] data.table_1.14.2                       </span>
<span class="co">##  [79] openxlsx_4.2.5                          </span>
<span class="co">##  [80] mvtnorm_1.1-3                           </span>
<span class="co">##  [81] survcomp_1.44.1                         </span>
<span class="co">##  [82] hms_1.1.1                               </span>
<span class="co">##  [83] evaluate_0.15                           </span>
<span class="co">##  [84] XML_3.99-0.9                            </span>
<span class="co">##  [85] RMTstat_0.3.1                           </span>
<span class="co">##  [86] shape_1.4.6                             </span>
<span class="co">##  [87] compiler_4.1.3                          </span>
<span class="co">##  [88] biomaRt_2.50.3                          </span>
<span class="co">##  [89] tibble_3.1.7                            </span>
<span class="co">##  [90] KernSmooth_2.23-20                      </span>
<span class="co">##  [91] crayon_1.5.1                            </span>
<span class="co">##  [92] R.oo_1.24.0                             </span>
<span class="co">##  [93] htmltools_0.5.2                         </span>
<span class="co">##  [94] pcaPP_2.0-1                             </span>
<span class="co">##  [95] tzdb_0.3.0                              </span>
<span class="co">##  [96] tidyr_1.2.0                             </span>
<span class="co">##  [97] rrcov_1.7-0                             </span>
<span class="co">##  [98] lubridate_1.8.0                         </span>
<span class="co">##  [99] DBI_1.1.2                               </span>
<span class="co">## [100] SuppDists_1.1-9.7                       </span>
<span class="co">## [101] dbplyr_2.1.1                            </span>
<span class="co">## [102] MASS_7.3-56                             </span>
<span class="co">## [103] rappdirs_0.3.3                          </span>
<span class="co">## [104] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [105] getopt_1.20.3                           </span>
<span class="co">## [106] readr_2.1.2                             </span>
<span class="co">## [107] rmeta_3.0                               </span>
<span class="co">## [108] cli_3.3.0                               </span>
<span class="co">## [109] quadprog_1.5-8                          </span>
<span class="co">## [110] R.methodsS3_1.8.1                       </span>
<span class="co">## [111] parallel_4.1.3                          </span>
<span class="co">## [112] pkgconfig_2.0.3                         </span>
<span class="co">## [113] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [114] pkgdown_2.0.3.9000                      </span>
<span class="co">## [115] GenomicAlignments_1.30.0                </span>
<span class="co">## [116] signal_0.7-7                            </span>
<span class="co">## [117] xml2_1.3.3                              </span>
<span class="co">## [118] foreach_1.5.2                           </span>
<span class="co">## [119] svglite_2.1.0                           </span>
<span class="co">## [120] bslib_0.3.1                             </span>
<span class="co">## [121] webshot_0.5.3                           </span>
<span class="co">## [122] XVector_0.34.0                          </span>
<span class="co">## [123] prodlim_2019.11.13                      </span>
<span class="co">## [124] rvest_1.0.2                             </span>
<span class="co">## [125] stringr_1.4.0                           </span>
<span class="co">## [126] VariantAnnotation_1.40.0                </span>
<span class="co">## [127] digest_0.6.29                           </span>
<span class="co">## [128] graph_1.72.0                            </span>
<span class="co">## [129] httpcode_0.3.0                          </span>
<span class="co">## [130] Biostrings_2.62.0                       </span>
<span class="co">## [131] rmarkdown_2.14                          </span>
<span class="co">## [132] harmonicmeanp_3.0                       </span>
<span class="co">## [133] restfulr_0.0.13                         </span>
<span class="co">## [134] curl_4.3.2                              </span>
<span class="co">## [135] Rsamtools_2.10.0                        </span>
<span class="co">## [136] rjson_0.2.21                            </span>
<span class="co">## [137] tseries_0.10-51                         </span>
<span class="co">## [138] lifecycle_1.0.1                         </span>
<span class="co">## [139] jsonlite_1.8.0                          </span>
<span class="co">## [140] survivalROC_1.0.3                       </span>
<span class="co">## [141] desc_1.4.1                              </span>
<span class="co">## [142] viridisLite_0.4.0                       </span>
<span class="co">## [143] BSgenome_1.62.0                         </span>
<span class="co">## [144] fansi_1.0.3                             </span>
<span class="co">## [145] pillar_1.7.0                            </span>
<span class="co">## [146] lattice_0.20-45                         </span>
<span class="co">## [147] DEoptimR_1.0-11                         </span>
<span class="co">## [148] KEGGREST_1.34.0                         </span>
<span class="co">## [149] fastmap_1.1.0                           </span>
<span class="co">## [150] httr_1.4.3                              </span>
<span class="co">## [151] GO.db_3.14.0                            </span>
<span class="co">## [152] xts_0.12.1                              </span>
<span class="co">## [153] glue_1.6.2                              </span>
<span class="co">## [154] zip_2.2.0                               </span>
<span class="co">## [155] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [156] png_0.1-7                               </span>
<span class="co">## [157] iterators_1.0.14                        </span>
<span class="co">## [158] pander_0.6.5                            </span>
<span class="co">## [159] glmnet_4.1-4                            </span>
<span class="co">## [160] bit_4.0.4                               </span>
<span class="co">## [161] stringi_1.7.6                           </span>
<span class="co">## [162] sass_0.4.1                              </span>
<span class="co">## [163] bootstrap_2019.6                        </span>
<span class="co">## [164] blob_1.2.3                              </span>
<span class="co">## [165] statgenGWAS_1.0.8                       </span>
<span class="co">## [166] textshaping_0.3.6                       </span>
<span class="co">## [167] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [168] memoise_2.0.1                           </span>
<span class="co">## [169] FMStable_0.1-2                          </span>
<span class="co">## [170] dplyr_1.0.8                             </span>
<span class="co">## [171] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
