<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="prisma">
<title>Evaluation of external PRS or external GWAS â€¢ prisma</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Evaluation of external PRS or external GWAS">
<meta property="og:description" content="prisma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Evaluation of external PRS or external GWAS</h1>
                        <h4 data-toc-skip class="author">Panagiotis Moulos</h4>
            
            <h4 data-toc-skip class="date">2022-06-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmoulos/prisma/blob/HEAD/vignettes/09-external.Rmd" class="external-link"><code>vignettes/09-external.Rmd</code></a></small>
      <div class="d-none name"><code>09-external.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Although the main PRISMA functionality has to do with <em>de novo</em> PRS derivation and evaluation, one can easily evaluate an externally derived PRS, for example a PRS retrieved from the <a href="https://www.pgscatalog.org/" class="external-link">EBI PGS Catalog</a>. This can be done in two ways:</p>
<ol style="list-style-type: decimal">
<li>Subsetting the <code>GWASExperiment</code> according to the SNPs in the retrieved PRS/PGS and running a PRISMA pipeline (<code><a href="../reference/prismaFuns.html">prisma()</a></code> or <code><a href="../reference/prismaFuns.html">prismaPipeline()</a></code>). This is the better way as new effects should be calculated suitable for our population. This pipeline should run quite fast, if the number of SNPs included in the external PRS are few.</li>
<li>Provided the retireved PRS directly in the <code><a href="../reference/prismaCvFuns.html">prsCrossValidate()</a></code> function. This will take the object as is (with own effects) and return some evaluation scores when the PRS is tested with our population and genotypes. The PRS retrieval functions included in the package (<code><a href="../reference/ebiQueryFuns.html">getPGSScores()</a></code>) will fetch data frames ready for usage with <code><a href="../reference/prismaCvFuns.html">prsCrossValidate()</a></code>. Quality control of the initial dataset may or may not be performed.</li>
</ol>
</div>
<div class="section level2">
<h2 id="testing-an-externally-retrieved-prs">Testing an externally retrieved PRS<a class="anchor" aria-label="anchor" href="#testing-an-externally-retrieved-prs"></a>
</h2>
<p>For the purposes of this vignette we are going to use the score file reported for <a href="https://www.pgscatalog.org/score/PGS000034/" class="external-link">PGS000034</a> with <em>Adult Body Mass Index</em> reported as trait. This score included with the package. We have also created a smaller toy dataset called <code>pgd</code> from the same source as the toy dataset but containing the common SNPs between PGS000034 and the source data that was used to create the toy dataset.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>,<span class="st">"extdata"</span>,<span class="st">"scores"</span><span class="op">)</span>
<span class="va">pgsId</span> <span class="op">&lt;-</span> <span class="st">"PGS000034"</span>
<span class="va">exPgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ebiQueryFuns.html">getPGSScores</a></span><span class="op">(</span>pgsId<span class="op">=</span><span class="va">pgsId</span>,base<span class="op">=</span><span class="va">base</span><span class="op">)</span>
<span class="co">## Scheduling 1 PGS API calls...</span>
<span class="co">##   Requesting 1 PGS Catalog API calls with PGS ID(s)...</span>
<span class="co">##     Call 1: PGS000034</span>
<span class="co">## Retrieving and enriching scores file(s) for PGS000034</span>
<span class="co">##   local file PGS000034.txt.gz</span>
<span class="co">##   processing PGS000034</span>
<span class="co">## Dealing with duplicates (if any)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">exPgs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">exPgs</span><span class="op">$</span><span class="va">variant_id</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exPgs</span><span class="op">)</span>
<span class="co">##            chromosome  position variant_id risk_allele reference_allele</span>
<span class="co">## rs1000940          17   5223976  rs1000940           G                A</span>
<span class="co">## rs10132280         14  24998019 rs10132280           C                A</span>
<span class="co">## rs1016287           2  59159129  rs1016287           T                C</span>
<span class="co">## rs10182181          2  25003800 rs10182181           G                A</span>
<span class="co">## rs10733682          9 128500735 rs10733682           A                G</span>
<span class="co">## rs10938397          4  44877284 rs10938397           G                A</span>
<span class="co">##            locus_name effect_weight       OR asm</span>
<span class="co">## rs1000940          NA         0.019 1.019182  nr</span>
<span class="co">## rs10132280         NA         0.023 1.023267  nr</span>
<span class="co">## rs1016287          NA         0.023 1.023267  nr</span>
<span class="co">## rs10182181         NA         0.031 1.031486  nr</span>
<span class="co">## rs10733682         NA         0.017 1.017145  nr</span>
<span class="co">## rs10938397         NA         0.040 1.040811  nr</span></code></pre></div>
<p>Load the <strong>pgd</strong> dataset and use the common SNPs with <code>exPgs</code>. They are already the same but we state the commands for demo purposes.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pgd</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">pgd</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"pgd.gds"</span><span class="op">)</span>

<span class="va">pgd</span>
<span class="co">## class: GWASExperiment </span>
<span class="co">## dim: 25 100 </span>
<span class="co">## metadata(5): genome backend filters gdsfile alleleOrder</span>
<span class="co">## assays class: SnpMatrix with 25 rows (SNPs) and 100 columns (samples)</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(25): rs657452 rs11583200 ... rs2033732 rs1928295</span>
<span class="co">## feature names(6): chromosome snp.name ... allele.1 allele.2</span>
<span class="co">## colnames(100): 12 21 ... 590 595</span>
<span class="co">## sample names(6): pedigree member ... sex affected</span>
<span class="co">## phenotype names(6): FID IID ... Age BMI</span>
<span class="co">## performed tests(0):</span>
<span class="co">## associated PRS(0):</span>

<span class="va">exPgs</span> <span class="op">&lt;-</span> <span class="va">exPgs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pgd</span><span class="op">)</span>,<span class="op">]</span></code></pre></div>
<p>Run a PRISMA full pipeline, only with GLM:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">wspace1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,<span class="va">d1</span><span class="op">)</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span>

<span class="fu"><a href="../reference/prismaFuns.html">prismaPipeline</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">pgd</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="st">"glm"</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace1</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"normal"</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>,
    outPath<span class="op">=</span><span class="va">outPath</span>
<span class="op">)</span></code></pre></div>
<p>A raw matrix of evaluation metrics:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaCvFuns.html">prsCrossValidate</a></span><span class="op">(</span><span class="va">exPgs</span>,<span class="va">pgd</span>,response<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,leaveOut<span class="op">=</span><span class="fl">0.2</span>,times<span class="op">=</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="testing-a-derived-prs-with-independent-genotypic-data">Testing a derived PRS with independent genotypic data<a class="anchor" aria-label="anchor" href="#testing-a-derived-prs-with-independent-genotypic-data"></a>
</h2>
<p>Often (actually always) we would like to test our derived PRS candidates to an independent population which could share similar characteristics, for example living in the same country and having similar origins. Sometimes it is also beneficial to test random PRSs with this population to see whether the original PRS does indeed have added value and it is not a random effect. The <code><a href="../reference/prismaCvFuns.html">applyPRS()</a></code> function can be used to accomplish the aforementioned processes.</p>
<p>In the example below, we do not run the PRS randomization process</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span>

<span class="co"># Traits</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span>
<span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span>

<span class="co"># Lose filters for the toy dataset</span>
<span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>
<span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span>
<span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span>

<span class="va">wspace2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">prismaOut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaFuns.html">prisma</a></span><span class="op">(</span>
    gwe<span class="op">=</span><span class="va">toy</span>,
    phenotype<span class="op">=</span><span class="va">response</span>,
    covariates<span class="op">=</span><span class="va">covariates</span>,
    pcs<span class="op">=</span><span class="cn">FALSE</span>,
    trainSize<span class="op">=</span><span class="fl">0.8</span>,
    niter<span class="op">=</span><span class="fl">4</span>,
    resolution<span class="op">=</span><span class="st">"frequency"</span>,
    minSnps<span class="op">=</span><span class="fl">2</span>,
    filters<span class="op">=</span><span class="va">filts</span>,
    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,
    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,
    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,
    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
    prsWorkspace<span class="op">=</span><span class="va">wspace2</span>,
    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,
    logging<span class="op">=</span><span class="st">"file"</span>,
    output<span class="op">=</span><span class="st">"normal"</span>,
    rc<span class="op">=</span><span class="fl">0.25</span>
<span class="op">)</span>
<span class="co">## </span>
<span class="co">## 2022-06-03 15:07:03: Data processing started...</span>
<span class="co">## </span>
<span class="co">## Input</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="co">## </span>
<span class="co">## Regression parameters</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Phenotypic response (phenotype)          : BMI</span>
<span class="co">## Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="co">## Principal Components in covariates (pcs) : No</span>
<span class="co">## Number of Principal Components (npcs)    : -</span>
<span class="co">## </span>
<span class="co">## Preprocessing options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Sample and genotype filtering options (filters)     :</span>
<span class="co">##   snpCallRate    : 0.98</span>
<span class="co">##   sampleCallRate : 0.95</span>
<span class="co">##   maf            : 0.05</span>
<span class="co">##   hwe            : 0.001</span>
<span class="co">##   heteroStat     : median</span>
<span class="co">##   heteroFac      : 3</span>
<span class="co">##   heteroHard     : NA</span>
<span class="co">##   pcaOut         : No</span>
<span class="co">##   pcaRobust      : hubert</span>
<span class="co">##   nPC            : NA</span>
<span class="co">##   LD             : 0.2</span>
<span class="co">##   IBD            : NA</span>
<span class="co">##   inbreed        : NA</span>
<span class="co">## Principal Components calculation method (pcaMethod) : snprel</span>
<span class="co">## Impute missing genotypes (imputeMissing)            : No</span>
<span class="co">## Missing genotype imputation method (imputeMethod)   : singlesplit</span>
<span class="co">## </span>
<span class="co">## GWA analysis options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Genome-Wide Association tests (gwaMethods)        : statgen</span>
<span class="co">## Association tests combination method (gwaCombine) : fishersimesmaxminharmonicwhitlockpandora</span>
<span class="co">## Association tests options                         :</span>
<span class="co">##   statgen :</span>
<span class="co">##     kinship : astle</span>
<span class="co">##     reml    : EMMA</span>
<span class="co">##     gls     : single</span>
<span class="co">## </span>
<span class="co">## PRS candidate extraction options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="co">## PRS algorithms options (prsMethods)    :</span>
<span class="co">##   lassosum :</span>
<span class="co">##     anc   : eur</span>
<span class="co">##     valid : auto</span>
<span class="co">##   prsice :</span>
<span class="co">##     clump_kb : 250</span>
<span class="co">##     clump_r2 : 0.1</span>
<span class="co">##     clump_p  : 0.75</span>
<span class="co">##     score    : avg</span>
<span class="co">##     perm     : 100</span>
<span class="co">##     seed     : 42</span>
<span class="co">## </span>
<span class="co">## PRS construction and evaluation options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Training set size (trainSize)                               : 80%</span>
<span class="co">## Number of PRS selection iterations (niter)                  : 4</span>
<span class="co">## PRS selection resolution method (resolution)                : frequency</span>
<span class="co">## PRS selection resolution step (step)                        : 1</span>
<span class="co">## Minimum SNP frequency in PRS (minFreq)                      : 2</span>
<span class="co">## Minimum number of SNPs in PRS (minSnps)                     : 2</span>
<span class="co">## Drop quantiles with same number of SNPs (dropSameQuantiles) : Yes</span>
<span class="co">## PRS aggregation method (aggregation)                        : intersectionunion</span>
<span class="co">## SNP effect summarization method (effectWeight)              : meanmedianweight</span>
<span class="co">## PRS evaluation framework (evalWith)                         : vanillaprscice</span>
<span class="co">## </span>
<span class="co">## PRS selection options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## PRS selection method (prsSelectMethod)           : maxima</span>
<span class="co">## PRS selection criterion (prsSelectCrit)          : prs_r2</span>
<span class="co">## PRS criterion summarization (prsSelectStat)      : mean</span>
<span class="co">## R^2 type if criterion includes R^2 (prsSelectR2) : adjusted</span>
<span class="co">## </span>
<span class="co">## Miscellaneous options</span>
<span class="co">## --------------------------------------------------------------------</span>
<span class="co">## Local workspace directory (prsWorkspace) : /tmp/Rtmpo0FmJg/file720331a48079</span>
<span class="co">## Cleanup level (cleanup)                  : intermediate</span>
<span class="co">## PRS pipeline output type (dnOutput)      : normal</span>
<span class="co">## Output level (output)                    : normal</span>
<span class="co">## Run ID (runId)                           : auto</span>
<span class="co">## Available cores fraction (rc)            : 25%</span>
<span class="co">## </span>
<span class="co">## ****************************************************************</span>
<span class="co">## Executing PRISMA pipeline with statgen</span>
<span class="co">## ****************************************************************</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 1. Discovery of de novo PRS candidates</span>
<span class="co">## ################################################################</span>
<span class="co">## prisma operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline</span>
<span class="co">## </span>
<span class="co">## Executing denovo pipeline iteration 1, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 119</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 126</span>
<span class="co">##   Hardy-Weinberg equilibrium: 3</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 25 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 25 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 21 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 21 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 25 associations found out of 564 markers</span>
<span class="co">##   Part 21: 21 associations found out of 316 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 46 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.118721512563089</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1/base_RDOGF7798M \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1/pheno_RDOGF7798M \</span>
<span class="co">##   --target /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1/plink_prsice_RDOGF7798M \</span>
<span class="co">##   --cov /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1/covar_RDOGF7798M \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_RDOGF7798M \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.237934</span>
<span class="co">## Full model R^2 is: 0.653614</span>
<span class="co">## Null model R^2 is: 0.41568</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_1/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 2, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 118</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 125</span>
<span class="co">##   Hardy-Weinberg equilibrium: 4</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 9 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 9 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 9 associations found out of 567 markers</span>
<span class="co">##   Part 21: 13 associations found out of 314 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 22 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.133238483673401</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2/base_HYUHQ8347O \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2/pheno_HYUHQ8347O \</span>
<span class="co">##   --target /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2/plink_prsice_HYUHQ8347O \</span>
<span class="co">##   --cov /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2/covar_HYUHQ8347O \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_HYUHQ8347O \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.182992</span>
<span class="co">## Full model R^2 is: 0.669374</span>
<span class="co">## Null model R^2 is: 0.486382</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_2/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 3, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 117</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 26 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 26 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 13 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 26 associations found out of 567 markers</span>
<span class="co">##   Part 21: 13 associations found out of 316 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 39 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.00237324085266332</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3/base_SZVFZ9835D \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3/pheno_SZVFZ9835D \</span>
<span class="co">##   --target /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3/plink_prsice_SZVFZ9835D \</span>
<span class="co">##   --cov /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3/covar_SZVFZ9835D \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_SZVFZ9835D \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0881702</span>
<span class="co">## Full model R^2 is: 0.714939</span>
<span class="co">## Null model R^2 is: 0.626768</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_3/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing denovo pipeline iteration 4, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Denovo pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## ----- Dataset partitioning -----</span>
<span class="co">## Partitioning dataset into 1 partitions by BMI</span>
<span class="co">## </span>
<span class="co">## ----- Training (base) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 2</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Test (target) QC -----</span>
<span class="co">## </span>
<span class="co">## Performing basic filtering</span>
<span class="co">##   Calculating SNP summaries for filtering</span>
<span class="co">##   SNP filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Minor Allele Frequency</span>
<span class="co">##     Hardy-Weinberg equilibrium</span>
<span class="co">##   Sample filters</span>
<span class="co">##     Call rate</span>
<span class="co">##     Heterozygosity</span>
<span class="co">##   Summarizing SNP filter results</span>
<span class="co">##   Summarizing sample filter results</span>
<span class="co">## </span>
<span class="co">## ----- Filtering report: -----</span>
<span class="co">## Filtered SNPs:</span>
<span class="co">##   SNP call rate             : 0</span>
<span class="co">##   Minor Allele Frequency    : 121</span>
<span class="co">##   Hardy-Weinberg equilibrium: 1</span>
<span class="co">## </span>
<span class="co">## Filtered samples:</span>
<span class="co">##   Sample call rate          : 0</span>
<span class="co">##   Heterozygosity            : 0</span>
<span class="co">##   Inbreed                   : 0</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## ----- Training (base) GWAS -----</span>
<span class="co">## </span>
<span class="co">## Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 18</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 24 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ========== Analyzing chromosome/part 21</span>
<span class="co">## </span>
<span class="co">## Preparing data for statgenGWAS...</span>
<span class="co">## Performing GWAS with statgenGWAS model</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Found 17 associations (uncorrected p-values).</span>
<span class="co">## </span>
<span class="co">## Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## Finished, the following putative associations were found per part:</span>
<span class="co">##   Part 18: 24 associations found out of 564 markers</span>
<span class="co">##   Part 21: 17 associations found out of 314 markers</span>
<span class="co">## </span>
<span class="co">## Overall, found 41 associations with (uncorrected) combined p-values).</span>
<span class="co">## </span>
<span class="co">## ----- PRS analysis with base and target -----</span>
<span class="co">## </span>
<span class="co">## Starting PRS analysis with lassosum, prsice</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with lassosum</span>
<span class="co">## lassosum operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4</span>
<span class="co">## Preparing lassosum run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4</span>
<span class="co">##   preparing summary statistics...</span>
<span class="co">##   preparing PLINK temporary files for target...</span>
<span class="co">## Preflight...</span>
<span class="co">##   copying linkage disequilibrium block files...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running lassosum PRS algorithm</span>
<span class="co">## Running lassosum validation and PRS calculation</span>
<span class="co">## 20 out of 20 samples kept in pheno.</span>
<span class="co">## 20 out of 20 samples kept in covar.</span>
<span class="co">## Done! R^2 is: 0.0764543645450464</span>
<span class="co">## </span>
<span class="co">## ========== Running PRS analysis with prsice</span>
<span class="co">## prsice operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4</span>
<span class="co">## Preparing PRSice run at workspace directory: /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4</span>
<span class="co">##   preparing summary statistics for the base file...</span>
<span class="co">##   preparing the covariates file...</span>
<span class="co">##   preparing the phenotype file...</span>
<span class="co">##   preparing PLINK temporary files - target...</span>
<span class="co">## Preflight...</span>
<span class="co">## Preparation done!</span>
<span class="co">## Running PRSice PRS algorithm in calculate mode</span>
<span class="co">## </span>
<span class="co">## Executing:</span>
<span class="co">## Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="co">##   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="co">##   --base /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4/base_RPDAE4453Z \</span>
<span class="co">##   --beta \</span>
<span class="co">##   --binary-target F \</span>
<span class="co">##   --pheno /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4/pheno_RPDAE4453Z \</span>
<span class="co">##   --target /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4/plink_prsice_RPDAE4453Z \</span>
<span class="co">##   --cov /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4/covar_RPDAE4453Z \</span>
<span class="co">##   --clump-kb 250 \</span>
<span class="co">##   --clump-r2 0.1 \</span>
<span class="co">##   --clump-p 0.75 \</span>
<span class="co">##   --score avg \</span>
<span class="co">##   --seed 42 \</span>
<span class="co">##   --out prsice_out_RPDAE4453Z \</span>
<span class="co">##   --print-snp \</span>
<span class="co">##   --perm 100</span>
<span class="co">## Done!</span>
<span class="co">## Adjusted PRSice R^2 is: 0.0812213</span>
<span class="co">## Full model R^2 is: 0.588261</span>
<span class="co">## Null model R^2 is: 0.507039</span>
<span class="co">## </span>
<span class="co">## Running PRS algorithms finished! Processing the results...</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline/YFPBSPZRCDT4480K_denovo_4/denovo_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/Rtmpo0FmJg/file720331a48079/statgen/baseline</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 2. Calculation of PRS evaluation metrics</span>
<span class="co">## ################################################################</span>
<span class="co">## Aggregation mode: intersect</span>
<span class="co">##   iteration 1 yields 52 markers</span>
<span class="co">##   iteration 2 yields 100 markers</span>
<span class="co">##   iteration 3 yields 1 markers</span>
<span class="co">##   iteration 4 yields 3 markers</span>
<span class="co">## Calculating frequencies</span>
<span class="co">## Averaging effects with mean</span>
<span class="co">## Constructing output</span>
<span class="co">## Done!</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 2 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9949696</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9949696</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## 1 SNPs not found in target data! Excluding...</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9949696</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs9949696</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/2</span>
<span class="co">## </span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Frequency 3 &lt;-----</span>
<span class="co">## ================================================================</span>
<span class="co">## prisma operations will be executed at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3</span>
<span class="co">## </span>
<span class="co">## Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_iter_1.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 1</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 1 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_iter_2.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 2</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 2 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_iter_3.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 3</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 3 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_iter_4.log</span>
<span class="co">## ================================================================</span>
<span class="co">## -----&gt; Evaluation pipeline iteration 4</span>
<span class="co">## ================================================================</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the null model</span>
<span class="co">## Testing full models</span>
<span class="co">##   testing PRS with SNPs from rs1740090 to rs7280065</span>
<span class="co">## </span>
<span class="co">## ----- Saving iteration 4 result to /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="co">## ================================================================</span>
<span class="co">## Cleaning up temporary program-specific intermediate files from /tmp/Rtmpo0FmJg/file720331a48079/statgen/selection/3</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 3. Selection of best PRS candidate markers</span>
<span class="co">## ################################################################</span>
<span class="co">## The optimal number of markers for PRS based on prs_r2 is 3 at prs_r2=0.09165</span>
<span class="co">## </span>
<span class="co">## ################################################################</span>
<span class="co">## 4. Wrap-up and graphics</span>
<span class="co">## ################################################################</span>
<span class="co">## sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="co">## Getting latest annotation on the fly for hg19 from refseq</span>
<span class="co">## Retrieving genome information for hg19 from refseq</span>
<span class="co">## Retrieving latest gene annotation for hg19 from refseq</span>
<span class="co">## Loading required namespace: RMySQL</span>
<span class="co">## Converting annotation to GenomicRanges object...</span>
<span class="co">## </span>
<span class="co">## Attaching package: 'Biostrings'</span>
<span class="co">## The following object is masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     strsplit</span>
<span class="co">## Getting DNA sequences...</span>
<span class="co">## Getting GC content...</span>
<span class="co">## </span>
<span class="co">## 2022-06-03 15:08:51: Data processing finished!</span>
<span class="co">## Total processing time: 01 minutes 48 seconds</span>

<span class="va">snpSelection</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaOutFuns.html">getPrsCandidates</a></span><span class="op">(</span><span class="va">prismaOut</span>,<span class="st">"statgen"</span>,<span class="fl">1</span><span class="op">)</span>

<span class="va">metrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prismaCvFuns.html">applyPRS</a></span><span class="op">(</span><span class="va">snpSelection</span>,<span class="va">toy</span>,response<span class="op">=</span><span class="va">response</span>,covariates<span class="op">=</span><span class="va">covariates</span><span class="op">)</span>
<span class="co">## All input PRS SNPs found in the input object! Proceeding...</span>
<span class="co">## </span>
<span class="co">## Regressing with GLM</span>
<span class="co">## Trait(s)    : BMI</span>
<span class="co">## Covariate(s): Case_Control, Gender, Age</span>
<span class="co">## Regression  : gaussian (predicted)</span>
<span class="co">## Use PCs     : No</span>
<span class="co">## Creating the reduced model</span>
<span class="co">## Creating the full model</span>

<span class="va">metrics</span><span class="op">$</span><span class="va">metrics</span>
<span class="co">##       reduced_r2          full_r2           prs_r2       prs_pvalue </span>
<span class="co">##     4.386799e-01     5.639940e-01     1.253141e-01     1.026894e-06 </span>
<span class="co">##     reduced_rmse        full_rmse      reduced_mae         full_mae </span>
<span class="co">##     3.551591e+00     3.130140e+00     2.829719e+00     2.457688e+00 </span>
<span class="co">## reduced_pred_cor    full_pred_cor  reduced_pred_r2     full_pred_r2 </span>
<span class="co">##     6.623292e-01     7.509954e-01     4.386799e-01     5.639940e-01 </span>
<span class="co">##      prs_pred_r2 </span>
<span class="co">##     1.253141e-01</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">metrics</span><span class="op">$</span><span class="va">values</span><span class="op">)</span>
<span class="co">##     raw_pheno red_pheno full_pheno       prs</span>
<span class="co">## 12   30.15677  31.57696   31.58775 1.6726284</span>
<span class="co">## 21   30.16859  31.33548   28.75451 3.0074684</span>
<span class="co">## 139  30.93989  31.14021   32.60563 0.9687017</span>
<span class="co">## 142  42.38751  31.62525   34.32095 0.2647751</span>
<span class="co">## 148  30.30177  31.72185   29.23190 3.0074684</span>
<span class="co">## 157  29.51856  24.07311   23.48451 1.6361218</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="r-session-info">R session info<a class="anchor" aria-label="anchor" href="#r-session-info"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.62.0                  </span>
<span class="co">##  [3] rtracklayer_1.54.0                Biostrings_2.62.0                </span>
<span class="co">##  [5] XVector_0.34.0                    prisma_0.0.1                     </span>
<span class="co">##  [7] SNPRelate_1.28.0                  gdsfmt_1.30.0                    </span>
<span class="co">##  [9] snpStats_1.44.0                   Matrix_1.4-1                     </span>
<span class="co">## [11] survival_3.3-1                    SummarizedExperiment_1.24.0      </span>
<span class="co">## [13] Biobase_2.54.0                    GenomicRanges_1.46.1             </span>
<span class="co">## [15] GenomeInfoDb_1.30.1               IRanges_2.28.0                   </span>
<span class="co">## [17] S4Vectors_0.32.4                  BiocGenerics_0.40.0              </span>
<span class="co">## [19] MatrixGenerics_1.6.0              matrixStats_0.62.0               </span>
<span class="co">## [21] lassosum_0.4.5                   </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                              </span>
<span class="co">##   [2] R.utils_2.11.0                          </span>
<span class="co">##   [3] tidyselect_1.1.2                        </span>
<span class="co">##   [4] gwascat_2.26.0                          </span>
<span class="co">##   [5] RSQLite_2.2.14                          </span>
<span class="co">##   [6] AnnotationDbi_1.56.2                    </span>
<span class="co">##   [7] grid_4.1.3                              </span>
<span class="co">##   [8] BiocParallel_1.28.3                     </span>
<span class="co">##   [9] munsell_0.5.0                           </span>
<span class="co">##  [10] codetools_0.2-18                        </span>
<span class="co">##  [11] ragg_1.2.2                              </span>
<span class="co">##  [12] future_1.25.0                           </span>
<span class="co">##  [13] quincunx_0.1.4                          </span>
<span class="co">##  [14] islasso_1.4.3                           </span>
<span class="co">##  [15] colorspace_2.0-3                        </span>
<span class="co">##  [16] filelock_1.0.2                          </span>
<span class="co">##  [17] OrganismDbi_1.36.0                      </span>
<span class="co">##  [18] knitr_1.39                              </span>
<span class="co">##  [19] rstudioapi_0.13                         </span>
<span class="co">##  [20] robustbase_0.95-0                       </span>
<span class="co">##  [21] TTR_0.24.3                              </span>
<span class="co">##  [22] listenv_0.8.0                           </span>
<span class="co">##  [23] optparse_1.7.1                          </span>
<span class="co">##  [24] GenomeInfoDbData_1.2.7                  </span>
<span class="co">##  [25] farver_2.1.0                            </span>
<span class="co">##  [26] bit64_4.0.5                             </span>
<span class="co">##  [27] rprojroot_2.0.3                         </span>
<span class="co">##  [28] parallelly_1.31.1                       </span>
<span class="co">##  [29] vctrs_0.4.1                             </span>
<span class="co">##  [30] generics_0.1.2                          </span>
<span class="co">##  [31] xfun_0.31                               </span>
<span class="co">##  [32] BiocFileCache_2.2.1                     </span>
<span class="co">##  [33] R6_2.5.1                                </span>
<span class="co">##  [34] rmdformats_1.0.3                        </span>
<span class="co">##  [35] bitops_1.0-7                            </span>
<span class="co">##  [36] cachem_1.0.6                            </span>
<span class="co">##  [37] DelayedArray_0.20.0                     </span>
<span class="co">##  [38] assertthat_0.2.1                        </span>
<span class="co">##  [39] gwasrapidd_0.99.12                      </span>
<span class="co">##  [40] BiocIO_1.4.0                            </span>
<span class="co">##  [41] Homo.sapiens_1.3.1                      </span>
<span class="co">##  [42] scales_1.2.0                            </span>
<span class="co">##  [43] gtable_0.3.0                            </span>
<span class="co">##  [44] globals_0.15.0                          </span>
<span class="co">##  [45] rlang_1.0.2                             </span>
<span class="co">##  [46] systemfonts_1.0.4                       </span>
<span class="co">##  [47] splines_4.1.3                           </span>
<span class="co">##  [48] rsnps_0.5.0.0                           </span>
<span class="co">##  [49] sitadela_1.3.1                          </span>
<span class="co">##  [50] BiocManager_1.30.16                     </span>
<span class="co">##  [51] yaml_2.3.5                              </span>
<span class="co">##  [52] reshape2_1.4.4                          </span>
<span class="co">##  [53] GenomicFeatures_1.46.5                  </span>
<span class="co">##  [54] quantmod_0.4.20                         </span>
<span class="co">##  [55] RMySQL_0.10.23                          </span>
<span class="co">##  [56] RBGL_1.70.0                             </span>
<span class="co">##  [57] tools_4.1.3                             </span>
<span class="co">##  [58] lava_1.6.10                             </span>
<span class="co">##  [59] bookdown_0.26                           </span>
<span class="co">##  [60] liftOver_1.18.0                         </span>
<span class="co">##  [61] ggplot2_3.3.5                           </span>
<span class="co">##  [62] ellipsis_0.3.2                          </span>
<span class="co">##  [63] kableExtra_1.3.4                        </span>
<span class="co">##  [64] jquerylib_0.1.4                         </span>
<span class="co">##  [65] Rcpp_1.0.8.3                            </span>
<span class="co">##  [66] rrBLUP_4.6.1                            </span>
<span class="co">##  [67] plyr_1.8.7                              </span>
<span class="co">##  [68] progress_1.2.2                          </span>
<span class="co">##  [69] zlibbioc_1.40.0                         </span>
<span class="co">##  [70] purrr_0.3.4                             </span>
<span class="co">##  [71] RCurl_1.98-1.6                          </span>
<span class="co">##  [72] prettyunits_1.1.1                       </span>
<span class="co">##  [73] tidyjson_0.3.1                          </span>
<span class="co">##  [74] cowplot_1.1.1                           </span>
<span class="co">##  [75] zoo_1.8-10                              </span>
<span class="co">##  [76] fs_1.5.2                                </span>
<span class="co">##  [77] crul_1.2.0                              </span>
<span class="co">##  [78] magrittr_2.0.3                          </span>
<span class="co">##  [79] data.table_1.14.2                       </span>
<span class="co">##  [80] openxlsx_4.2.5                          </span>
<span class="co">##  [81] mvtnorm_1.1-3                           </span>
<span class="co">##  [82] survcomp_1.44.1                         </span>
<span class="co">##  [83] hms_1.1.1                               </span>
<span class="co">##  [84] evaluate_0.15                           </span>
<span class="co">##  [85] XML_3.99-0.9                            </span>
<span class="co">##  [86] RMTstat_0.3.1                           </span>
<span class="co">##  [87] shape_1.4.6                             </span>
<span class="co">##  [88] compiler_4.1.3                          </span>
<span class="co">##  [89] biomaRt_2.50.3                          </span>
<span class="co">##  [90] tibble_3.1.7                            </span>
<span class="co">##  [91] KernSmooth_2.23-20                      </span>
<span class="co">##  [92] crayon_1.5.1                            </span>
<span class="co">##  [93] R.oo_1.24.0                             </span>
<span class="co">##  [94] htmltools_0.5.2                         </span>
<span class="co">##  [95] pcaPP_2.0-1                             </span>
<span class="co">##  [96] tzdb_0.3.0                              </span>
<span class="co">##  [97] tidyr_1.2.0                             </span>
<span class="co">##  [98] rrcov_1.7-0                             </span>
<span class="co">##  [99] lubridate_1.8.0                         </span>
<span class="co">## [100] DBI_1.1.2                               </span>
<span class="co">## [101] SuppDists_1.1-9.7                       </span>
<span class="co">## [102] dbplyr_2.1.1                            </span>
<span class="co">## [103] MASS_7.3-56                             </span>
<span class="co">## [104] rappdirs_0.3.3                          </span>
<span class="co">## [105] PhenotypeSimulator_0.3.4                </span>
<span class="co">## [106] getopt_1.20.3                           </span>
<span class="co">## [107] readr_2.1.2                             </span>
<span class="co">## [108] rmeta_3.0                               </span>
<span class="co">## [109] cli_3.3.0                               </span>
<span class="co">## [110] quadprog_1.5-8                          </span>
<span class="co">## [111] R.methodsS3_1.8.1                       </span>
<span class="co">## [112] parallel_4.1.3                          </span>
<span class="co">## [113] pkgconfig_2.0.3                         </span>
<span class="co">## [114] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 </span>
<span class="co">## [115] pkgdown_2.0.3.9000                      </span>
<span class="co">## [116] GenomicAlignments_1.30.0                </span>
<span class="co">## [117] signal_0.7-7                            </span>
<span class="co">## [118] xml2_1.3.3                              </span>
<span class="co">## [119] foreach_1.5.2                           </span>
<span class="co">## [120] svglite_2.1.0                           </span>
<span class="co">## [121] bslib_0.3.1                             </span>
<span class="co">## [122] webshot_0.5.3                           </span>
<span class="co">## [123] prodlim_2019.11.13                      </span>
<span class="co">## [124] rvest_1.0.2                             </span>
<span class="co">## [125] stringr_1.4.0                           </span>
<span class="co">## [126] VariantAnnotation_1.40.0                </span>
<span class="co">## [127] digest_0.6.29                           </span>
<span class="co">## [128] graph_1.72.0                            </span>
<span class="co">## [129] httpcode_0.3.0                          </span>
<span class="co">## [130] rmarkdown_2.14                          </span>
<span class="co">## [131] harmonicmeanp_3.0                       </span>
<span class="co">## [132] restfulr_0.0.13                         </span>
<span class="co">## [133] curl_4.3.2                              </span>
<span class="co">## [134] Rsamtools_2.10.0                        </span>
<span class="co">## [135] rjson_0.2.21                            </span>
<span class="co">## [136] tseries_0.10-51                         </span>
<span class="co">## [137] lifecycle_1.0.1                         </span>
<span class="co">## [138] jsonlite_1.8.0                          </span>
<span class="co">## [139] survivalROC_1.0.3                       </span>
<span class="co">## [140] desc_1.4.1                              </span>
<span class="co">## [141] viridisLite_0.4.0                       </span>
<span class="co">## [142] fansi_1.0.3                             </span>
<span class="co">## [143] pillar_1.7.0                            </span>
<span class="co">## [144] lattice_0.20-45                         </span>
<span class="co">## [145] DEoptimR_1.0-11                         </span>
<span class="co">## [146] KEGGREST_1.34.0                         </span>
<span class="co">## [147] fastmap_1.1.0                           </span>
<span class="co">## [148] httr_1.4.3                              </span>
<span class="co">## [149] GO.db_3.14.0                            </span>
<span class="co">## [150] xts_0.12.1                              </span>
<span class="co">## [151] glue_1.6.2                              </span>
<span class="co">## [152] zip_2.2.0                               </span>
<span class="co">## [153] SNPlocs.Hsapiens.dbSNP151.GRCh38_0.99.20</span>
<span class="co">## [154] png_0.1-7                               </span>
<span class="co">## [155] iterators_1.0.14                        </span>
<span class="co">## [156] pander_0.6.5                            </span>
<span class="co">## [157] glmnet_4.1-4                            </span>
<span class="co">## [158] bit_4.0.4                               </span>
<span class="co">## [159] stringi_1.7.6                           </span>
<span class="co">## [160] sass_0.4.1                              </span>
<span class="co">## [161] bootstrap_2019.6                        </span>
<span class="co">## [162] blob_1.2.3                              </span>
<span class="co">## [163] statgenGWAS_1.0.8                       </span>
<span class="co">## [164] textshaping_0.3.6                       </span>
<span class="co">## [165] org.Hs.eg.db_3.14.0                     </span>
<span class="co">## [166] memoise_2.0.1                           </span>
<span class="co">## [167] FMStable_0.1-2                          </span>
<span class="co">## [168] dplyr_1.0.8                             </span>
<span class="co">## [169] future.apply_1.9.0</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
