<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function may be used to adjust the SNP effects
    to derive the final PRS (e.g. with the total dataset)
    after the whole extraction and assessment process.
    The process is not obligatory."><title>PRS weight adjustment through final regression — adjustPrsWeights • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PRS weight adjustment through final regression — adjustPrsWeights"><meta property="og:description" content="This function may be used to adjust the SNP effects
    to derive the final PRS (e.g. with the total dataset)
    after the whole extraction and assessment process.
    The process is not obligatory."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>PRS weight adjustment through final regression</h1>
      
      <div class="d-none name"><code>adjustPrsWeights.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function may be used to adjust the SNP effects
    to derive the final PRS (e.g. with the total dataset)
    after the whole extraction and assessment process.
    The process is not obligatory.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">adjustPrsWeights</span><span class="op">(</span><span class="va">snpSelection</span>, <span class="va">gwe</span>, <span class="va">response</span>, 
        covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">FALSE</span>, family <span class="op">=</span> <span class="cn">NULL</span>,
        rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>snpSelection</dt>
<dd><p>A data frame with SNPs comprising
    a PRS, properly formatted (see 
    <code><a href="aggregatePrsMarkers.html">aggregatePrsMarkers</a>)</code>.</p></dd>
<dt>gwe</dt>
<dd><p>The <code>GWASExperiment</code> object used to
    derive the PRS.</p></dd>
<dt>response</dt>
<dd><p>A character with the phenotypic response
    used to derive the PRS.</p></dd>
<dt>covariates</dt>
<dd><p>A character vector with covariares
    used to derive the PRS.</p></dd>
<dt>pcs</dt>
<dd><p>Use the PCs present in <code>gwe</code>? Defaults
    to <code>FALSE</code>. If the PRS has been extracted with PCs
    as covariates, it should be <code>TRUE</code>.</p></dd>
<dt>family</dt>
<dd><p>Regression family, e.g. <code>"gaussian"</code>,
    see also <code><a href="gwaFuns.html">gwa</a></code>.</p></dd>
<dt>rc</dt>
<dd><p>The fraction of available cores to use for 
    parallel calculations. Default is <code>NULL</code> for no
    parallelization.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A new data frame with SNPs comprising the PRS with
    adjusted effects.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">prisma_out</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">prs</span> <span class="op">&lt;-</span> <span class="fu"><a href="prismaOutFuns.html">getPrsCandidates</a></span><span class="op">(</span><span class="va">prismaOut</span>,<span class="st">"statgen"</span>,<span class="fl">1</span><span class="op">)</span></span>
<span class="r-in">    <span class="co">#prs &lt;- prismaOut$results$statgen$candidates[[1]]</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">prsa</span> <span class="op">&lt;-</span> <span class="fu">adjustPrsWeights</span><span class="op">(</span><span class="va">prs</span>,<span class="va">toy</span>,<span class="st">"BMI"</span>,</span>
<span class="r-in">        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span>,<span class="st">"Case_Control"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender, Case_Control</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1442343 to rs6506142</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 3 associations (uncorrected p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">prs</span><span class="op">$</span><span class="va">effect_weight</span>,<span class="va">prsa</span><span class="op">$</span><span class="va">effect_weight</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] FALSE</span>
<span class="r-in">    <span class="co"># FALSE</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

