<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function performs PCA using three available methods
    to the genetic data included in the iput object in order
    to capture potential underlying population structure or
    outliers, and derive PCs to be used as covariates in
    downstream GWA. Two of the three methods can be used to
    perform robust PCA with the intention of better 
    capturing potential outliers."><title>Calculate PCA covariates for GWASExperiments — calcPcaCovar • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate PCA covariates for GWASExperiments — calcPcaCovar"><meta property="og:description" content="This function performs PCA using three available methods
    to the genetic data included in the iput object in order
    to capture potential underlying population structure or
    outliers, and derive PCs to be used as covariates in
    downstream GWA. Two of the three methods can be used to
    perform robust PCA with the intention of better 
    capturing potential outliers."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Calculate PCA covariates for GWASExperiments</h1>
      
      <div class="d-none name"><code>calcPcaCovar.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs PCA using three available methods
    to the genetic data included in the iput object in order
    to capture potential underlying population structure or
    outliers, and derive PCs to be used as covariates in
    downstream GWA. Two of the three methods can be used to
    perform robust PCA with the intention of better 
    capturing potential outliers.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">calcPcaCovar</span><span class="op">(</span><span class="va">obj</span>, ld <span class="op">=</span> <span class="fl">0.2</span>, method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, 
        <span class="st">"snprel"</span>, <span class="st">"grid"</span>, <span class="st">"hubert"</span><span class="op">)</span>, npc <span class="op">=</span> <span class="fl">0</span>, pval <span class="op">=</span> <span class="fl">0.05</span>, 
        rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>obj</dt>
<dd><p>A <code>GWASExperiment</code> object without 
    missing values.</p></dd>
<dt>ld</dt>
<dd><p>Linkage Disequilibrium cutoff (defaults to
    <code>0.2</code>) for SNPs to include in PCA. See also
    Details.</p></dd>
<dt>method</dt>
<dd><p>Method used to calculate PCs. One of
    <code>"auto"</code> (default), <code>"snprel"</code>, <code>"grid"</code>
    or <code>"hubert"</code>. See also Details.</p></dd>
<dt>npc</dt>
<dd><p>The number of PCs to attach to <code>obj</code>.
    When <code>0</code> (default), the Tracy-Widom test is used
    to determine the best number of PCs to include.</p></dd>
<dt>pval</dt>
<dd><p>p-value for the Tracy-Widom test. Defaults
    to <code>0.05</code>. PCs with p-value below this threshold
    are ignored.</p></dd>
<dt>rc</dt>
<dd><p>Fraction of available cores to use when
    <code>method="snprel"</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The input GWASExperiment object with attached PC
    covariates corresponding to the data hosted in the
    object. They can be retrieved with 
<code>pcaCovariates(obj).</code></p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>ld</code> argument, it is generally
    recommended in the literature to use LD-pruned SNPs
    for PCA in GWAS Experiments. This argument controls
    the threshold of LD for LD-pruning. It is based on
    recommendations from Reed at al., 2013.</p>    
<p>Regarding the <code>method</code> argument, when 
    <code>"auto"</code>, if PCs already exist in the object, LD
    pruning is redone following by robust PCA if the 
    attached filters in the object have robust PCA marked.
    If not, PCA with the package SNPRelate is performed.
    It should be noted that for larger datasets, robust
    PCA can take a lot of time, while much less for the
    optimized method in SNPRelate.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># Only 1 PC will be returned because of dataset size</span></span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu">calcPcaCovar</span><span class="op">(</span><span class="va">toy</span>,method<span class="op">=</span><span class="st">"snprel"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA to select covariates to include in associations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading GDS file /tmp/RtmplLIKfg/temp_libpath4f71185f9e1c/prisma/extdata/toy.gds...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing LD pruning...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP pruning based on LD:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 962</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sliding window: 500,000 basepairs, Inf SNPs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     |LD| threshold: 0.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     method: composite</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 18: 23.44%, 154/657</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 21: 21.28%, 73/343</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 227 markers are selected in total.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LD pruning finished! 227 SNPs will be used for PCA.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Principal Component Analysis (PCA) on genotypes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 773 SNPs (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 227</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of principal components: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PCA:    the sum of all selected genotypes (0,1,2) = 10268</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CPU capabilities: Double-Precision SSE2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fri Jun  3 14:28:17 2022    (internal increment: 13024)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fri Jun  3 14:28:17 2022    Begin (eigenvalues and eigenvectors)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fri Jun  3 14:28:17 2022    Done.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting number of PCs to include</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   with Tracy-Widom test for significant eigenvalues</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     selected 1 PCs</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

