<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function implements various genotype and sample
    filters based on properties of the genotypic data
    present in the input GWAS object. Most filters are
    losely based in the protocol by Reed et al., 2015
    (PubMed ID: 26343929)."><title>Aggregate PRS candidate SNPs from iterative extraction — filterGWAS • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Aggregate PRS candidate SNPs from iterative extraction — filterGWAS"><meta property="og:description" content="This function implements various genotype and sample
    filters based on properties of the genotypic data
    present in the input GWAS object. Most filters are
    losely based in the protocol by Reed et al., 2015
    (PubMed ID: 26343929)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Aggregate PRS candidate SNPs from iterative extraction</h1>
      
      <div class="d-none name"><code>filterGWAS.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function implements various genotype and sample
    filters based on properties of the genotypic data
    present in the input GWAS object. Most filters are
    losely based in the protocol by Reed et al., 2015
    (PubMed ID: 26343929).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">filterGWAS</span><span class="op">(</span><span class="va">obj</span>, filters <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>,
        imputeMissing <span class="op">=</span> <span class="cn">TRUE</span>, imputeMode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"single"</span>,<span class="st">"split"</span><span class="op">)</span>,
        withPlink <span class="op">=</span> <span class="cn">FALSE</span>, rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>obj</dt>
<dd><p>An object of class <code>GWASExperiment</code> holding
    the genotypic data to be filtered.</p></dd>
<dt>filters</dt>
<dd><p>A set of genotype filters, see also Details
    and <code><a href="getDefaults.html">getDefaults</a></code>.</p></dd>
<dt>imputeMissing</dt>
<dd><p>Impute the remaining missing values
    after filtering? An iterative imputation method based
    on the package snpStats is used, see also 
    <code><a href="imputeGWAS.html">imputeGWAS</a></code>.</p></dd>
<dt>imputeMode</dt>
<dd><p>Perform the imputation within the same
    dataset (<code>"single"</code>, default) or split the dataset
    into a part containing missing values and a part with
    no missing values (<code>"split"</code>), may happen in large
    dataset and could be more accurate.</p></dd>
<dt>withPlink</dt>
<dd><p>Perform basic filtering (not IBD or PCA)
    with PLINK instead of snpStats. Maybe faster and suitable
    for huge datasets but with less options. Defaults to 
    <code>FALSE</code>. Also, in this case, the <code>filterRecord</code>
    is not filled.</p></dd>
<dt>rc</dt>
<dd><p>The fraction of available cores to use for 
    parallel calculations. Default is <code>NULL</code> for no
    parallelization.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A filtered <code>GWASExperiment</code> object with or without
    missing values according to <code>imputeMissing</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>filters</code> argument, this is a list
    whose members denote SNP or sample-based filters and
    their values are filter cutoffs. A more detailed
    explanation of these filters can be found in Reed et 
    al., 2015 (PubMed ID: 26343929).</p>    
<p>The following filters are available:</p><ul><li><p><code>snpCallRate</code>: The SNPs call rate within
        a sample. Defaults to <code>0.98</code>.</p></li>
<li><p><code>sampleCallRate</code>: The sample call rate of
        SNPs accross samples. Defaults to <code>0.95</code>.</p></li>
<li><p><code>maf</code>: The Minimum Allele Frequency across
        samples, below which SNPs are filtered. Defaults to 
        <code>0.05</code>.</p></li>
<li><p><code>hwe</code>: The Hardy-Weinberg equilibrium 
        p-value. Defaults to <code>1e-6</code>.</p></li>
<li><p><code>heteroStat</code>: The statistic with which
        heterozygosity distribution center is defined. 
        Defaults to <code>"median"</code>.</p></li>
<li><p><code>heteroFac</code>: The factor with which the 
        scale (standard deviation, MAD) of the heterozygosity
        distribution is multiplied and added before and after
        the <code>heteroStat</code> to create a range of acceptable
        heterozygosity values, e.g. mean +/- 3*std. Defaults
        to <code>3</code>. If <code>heteroStat="mean"</code>, the scale
        is the standard deviation. If 
        <code>heteroStat="median"</code>, the scale is the MAD.</p></li>
<li><p><code>heteroHard</code>: A hard filter for 
        heterozygosity values. If set, <code>heteroStat</code> and
        <code>heteroFac</code> are ignored.</p></li>
<li><p><code>pcaOut</code>: Automatically remove samples 
        based on PCA filtering? Defaults to <code>TRUE</code> but
        should be avoided for datasets with a few samples.
        Also, it is applied if a robust PCA method is chosen.</p></li>
<li><p><code>pcaRobust</code>: The robust PCA method to be
        applied for sample exclustion. One of <code>"grid"</code>
        or <code>"hubert"</code> (default). See the package rrcov
        for more details.</p></li>
<li><p><code>nPC</code>: Fixed number of PCs to use for PCA
        filtering. If <code>NA</code> (default), they are 
        determined using the <code><a href="twTest.html">twTest</a>.</code></p></li>
<li><p><code>LD</code>: Linkage Disequilibrium filter 
        (calculated using package SNPRelate fuctions).
        Defaults to <code>0.2</code>.</p></li>
<li><p><code>IBD</code>: Identity By Descent filter 
        (calculated using package SNPRelate fuctions).
        Defaults to <code>0.2</code>.</p></li>
<li><p><code>inbreed</code>: Inbreed coefficient filter.
        Defaults to <code>0.1</code>.</p></li>
</ul><p>See also the output of <code>getDefaults("filters")</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,</span>
<span class="r-in">        <span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">    <span class="co"># Lose filters for the toy dataset</span></span>
<span class="r-in">    <span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span class="r-in">    <span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span></span>
<span class="r-in">    <span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span class="r-in">    <span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toyF</span> <span class="op">&lt;-</span> <span class="fu">filterGWAS</span><span class="op">(</span><span class="va">toy</span>,filters<span class="op">=</span><span class="va">filts</span>,</span>
<span class="r-in">        imputeMissing<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in">        </span>
<span class="r-in">    <span class="co"># With PLINK</span></span>
<span class="r-in">    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span>
<span class="r-in">        <span class="va">toyF</span> <span class="op">&lt;-</span> <span class="fu">filterGWAS</span><span class="op">(</span><span class="va">toy</span>,filters<span class="op">=</span><span class="va">filts</span>,</span>
<span class="r-in">            imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,withPlink<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Exporting temporary PLINK files for the calculations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmps2YAD0/file14c2c93352d.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmps2YAD0/file14c2c93352d.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmps2YAD0/file14c2c93352d.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --bfile /tmp/Rtmps2YAD0/file14c2c93352d \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out /tmp/Rtmps2YAD0/file14c7b010438 --het</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --bfile /tmp/Rtmps2YAD0/file14c2c93352d \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out /tmp/Rtmps2YAD0/file14c7b010438 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --make-bed \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --geno 0.02 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --maf 0.05 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --hwe 0.001 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --mind 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading PLINK files with snpStats framework</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 881 SNPs and 100 samples remaining</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

