<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="These functions perform the association tests to create 
    summary statistics for a GWA dataset prior to the 
    extraction of PRS candidates. They are part of the 
    overall PRISMA workflow (prisma) but they
    are designed to be also used independently just to look
    for univariate association cases, i.e. look for 
    associations for a single trait using various tests."><title>Functions to perform GWA tests with various algorithms — gwaFuns • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Functions to perform GWA tests with various algorithms — gwaFuns"><meta property="og:description" content="These functions perform the association tests to create 
    summary statistics for a GWA dataset prior to the 
    extraction of PRS candidates. They are part of the 
    overall PRISMA workflow (prisma) but they
    are designed to be also used independently just to look
    for univariate association cases, i.e. look for 
    associations for a single trait using various tests."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Functions to perform GWA tests with various algorithms</h1>
      
      <div class="d-none name"><code>gwaFuns.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>These functions perform the association tests to create 
    summary statistics for a GWA dataset prior to the 
    extraction of PRS candidates. They are part of the 
    overall PRISMA workflow (<code><a href="prismaFuns.html">prisma</a></code>) but they
    are designed to be also used independently just to look
    for univariate association cases, i.e. look for 
    associations for a single trait using various tests.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">gwa</span><span class="op">(</span><span class="va">obj</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">FALSE</span>,
        psig <span class="op">=</span> <span class="fl">0.05</span>, methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>, <span class="st">"rrblup"</span>, <span class="st">"statgen"</span>,
            <span class="st">"snptest"</span>, <span class="st">"plink"</span>, <span class="st">"lasso"</span><span class="op">)</span>,
        combine <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fisher"</span>, <span class="st">"simes"</span>, <span class="st">"max"</span>, <span class="st">"min"</span>, <span class="st">"harmonic"</span>,
            <span class="st">"whitlock"</span>,<span class="st">"pandora"</span><span class="op">)</span>, glmOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span>,
        rrblupOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"rrblup"</span><span class="op">)</span>, 
        statgenOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"statgen"</span><span class="op">)</span>,
        snptestOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span>,
        plinkOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span>, rc <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span>
    
    <span class="fu">gwaBlup</span><span class="op">(</span><span class="va">obj</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, 
        usepc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, <span class="st">"estim"</span>, <span class="st">"rrint"</span>, <span class="st">"fixed"</span>, <span class="st">"none"</span><span class="op">)</span>, 
        npcs <span class="op">=</span> <span class="cn">NULL</span>, psig <span class="op">=</span> <span class="fl">0.05</span>, rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
    
    <span class="fu">gwaGlm</span><span class="op">(</span><span class="va">obj</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">FALSE</span>,
        family <span class="op">=</span> <span class="cn">NULL</span>, psig <span class="op">=</span> <span class="fl">0.05</span>, penalized <span class="op">=</span> <span class="cn">FALSE</span>,
        size <span class="op">=</span> <span class="fl">1</span>, rc <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span>
        
    <span class="fu">gwaPlink</span><span class="op">(</span><span class="va">obj</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">TRUE</span>,
        psig <span class="op">=</span> <span class="fl">0.05</span>, seed <span class="op">=</span> <span class="fl">42</span>, cliopts <span class="op">=</span> <span class="cn">NULL</span>, wspace <span class="op">=</span> <span class="cn">NULL</span>, 
        rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
    
    <span class="fu">gwaStatgen</span><span class="op">(</span><span class="va">obj</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">TRUE</span>, 
        psig <span class="op">=</span> <span class="fl">0.05</span>, rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
    
    <span class="fu">gwaSnptest</span><span class="op">(</span><span class="va">obj</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">TRUE</span>,
        psig <span class="op">=</span> <span class="fl">0.05</span>, test <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"frequentist"</span>, <span class="st">"bayesian"</span><span class="op">)</span>, 
        model <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"additive"</span>, <span class="st">"dominant"</span>, <span class="st">"recessive"</span>, 
        <span class="st">"general"</span>, <span class="st">"heterozygote"</span><span class="op">)</span>, wspace <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>obj</dt>
<dd><p>an object of class <code>GWASExperiment</code> holding
    the genotypic and phenotypic data to be used for association
    tests.</p></dd>
<dt>response</dt>
<dd><p>The trait for which associations are 
    searched for. It should be a character of length 1 and 
    exist in the phenotypes of <code>obj</code>.</p></dd>
<dt>covariates</dt>
<dd><p>The covariates to include in association
    methods/regression models. It should be a character vector 
    of traits all existing in the phenotypes of <code>obj</code>.</p></dd>
<dt>pcs</dt>
<dd><p>Inlcude PCs in the association models? Defaults
    to <code>FALSE</code>.</p></dd>
<dt>psig</dt>
<dd><p>The significance level (p-value cutoff) for which
    an association between a SNP/marker and a phenotype 
    (<code>response</code>) is considered significant. Defaults to
    <code>0.05</code>.</p></dd>
<dt>methods</dt>
<dd><p>The GWA methods to apply on <code>obj</code> with
    respect to <code>response</code> and <code>covariates</code>. It can be
    one or more of <code>"glm"</code>, <code>"rrblup"</code>, <code>"statgen"</code>,
    <code>"snptest"</code>, <code>"plink"</code>, <code>"lasso"</code>. See also
    Details for explanation of each case.</p></dd>
<dt>combine</dt>
<dd><p>How (if) p-values should be combined to
    produce a summarized p-value from all GWA methods for
    each association test. Not all methods are yet available,
    see also Details.</p></dd>
<dt>glmOpts</dt>
<dd><p>A list of options for GLM regression 
    model fit. See also Details.</p></dd>
<dt>rrblupOpts</dt>
<dd><p>A list of options for ridge regression 
    model fit (rrBLUP). See also Details.</p></dd>
<dt>statgenOpts</dt>
<dd><p>A list of options for regression fit
    with statgenGWAS. See also Details.</p></dd>
<dt>snptestOpts</dt>
<dd><p>A list of options for modeling
    associations with SNPTEST. See also Details.</p></dd>
<dt>plinkOpts</dt>
<dd><p>A list of options for modeling
    associations with PLINK. See also Details.</p></dd>
<dt>usepc</dt>
<dd><p>What to do with PCA and PCs when using
    <code>gwaBlup</code> and rrBLUP model fitting. Defaults to
    <code>"auto"</code> and also <code>"rrint"</code> is preferred.
    See also Details.</p></dd>
<dt>npcs</dt>
<dd><p>Number of PCs to use with rrBLUP. Defaults
    to <code>NULL</code> for auto-determining what to do.</p></dd>
<dt>family</dt>
<dd><p>The model family for GLM, e.g. 
    <code>"gaussian"</code> for classic regression with continuous
    response variable or <code>"binomial"</code> for logistic
    regression. By default (<code>NULL</code>), it is guessed from
    the form of the <code>response</code> variable.</p></dd>
<dt>penalized</dt>
<dd><p>Use penalized (LASSO) regression for
    GLM. Defaults to <code>FALSE</code>.</p></dd>
<dt>size</dt>
<dd><p>The number of SNPs to include in a GLM 
    model, usually utilized with <code>penalized=TRUE</code>.
    Defaults to <code>1</code> for models with 1 SNP as 
    covariate.</p></dd>
<dt>wspace</dt>
<dd><p>A path to create workspaces and temporary
    files for SNPTEST and PLINK which operate outside R.</p></dd>
<dt>test</dt>
<dd><p>One of supported test types for SNPTEST. It
    can be <code>"frequentist"</code> (default) or 
    <code>"bayesian"</code>. See also SNPTEST documentation.</p></dd>
<dt>seed</dt>
<dd><p>A random seed for PLINK resampling 
    operations. Defaults to <code>42</code>.</p></dd>
<dt>cliopts</dt>
<dd><p>A string with additional command-line
    parameters to pass to the PLINK executable. Defaults
    to <code>NULL</code>.</p></dd>
<dt>model</dt>
<dd><p>One of the supported genotype modeling
    types for SNPTEST. It can be one of <code>"additive"</code>
    (default), <code>"dominant"</code>, <code>"recessive"</code>,
    <code>"general"</code> or <code>"heterozygote"</code>. See also
    SNPTEST documentation.</p></dd>
<dt>rc</dt>
<dd><p>The fraction of available cores to use for 
    parallel calculations. Default is <code>NULL</code> for no
    parallelization.</p></dd>
<dt>...</dt>
<dd><p>Additional arguments passed to regression
    functions.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Depending on the function used, either a data frame
    with summary statistics (always with at least SNP
    name, effects/ORs and p-values)</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>methods</code> argument, the following
    briefly explain how the associations are modeled in
    each case:</p><ul><li><p><code>glm</code>: In this case, a SNP is checked
        for association with the <code>response</code> trait
        using a General Linear Regression model of the
        form: <code>response ~ covariates + [SNP]</code> or
        <code>response ~ covariates + pcs + [SNP]</code>
        depending on whether PCs are added as covariates
        or not. <code>[SNP]</code> is simply the recorded genotype
        (<code>0</code> for homozygous minor allele, <code>1</code>
        for heterozygous allele and <code>2</code> for homozygous
        major allele). The p-value and the regression
        coefficient (effect/OR) of the [SNP] is recorded
        and used to determine if the SNP is associated
        with the covariate or not. The function can be
        used independently by directly using <code>gwaGlm</code>.
        In this case, a data frame of summary statistics
        is returned instead of the input 
        <code>GWASExperiment</code>.</p></li>
<li><p><code>rrblup</code>: In this case, the 
        <code>GWAS</code> function from the package
        rrBLUP is used to model the associations, the
        effects and the significance of each SNP using
        Ridge Regression. The function can be used
        independently by directly using <code>gwaBlup</code>.
        In this case, a data frame of ridge regression 
        summary statistics is returned instead of the 
        input <code>GWASExperiment</code>.</p></li>
<li><p><code>statgen</code>: In this case, the
        <code>runSingleTraitGwas</code> function from
        the statgenGWAS package is used to model the 
        associations, the effects and the significance 
        of each SNP using also linear regression. The 
        function can be used independently by directly 
        using <code>gwaStatgen</code>. In this case, a data 
        frame of summary statistics is returned instead 
        of the input <code>GWASExperiment</code>.</p></li>
<li><p><code>snptest</code>: In this case, the SNPTEST
        software is used to check for associations
        and summary statistics. It should be noted that
        SNPTEST operates with at least 100 samples. The
        function can be used directly as the previous 
        ones by <code>gwaSnptest</code>.</p></li>
<li><p><code>plink</code>: In this case, the PLINK
        software is used to check for associations and
        return summary statistics. The function can be used
        directly as the previous ones by <code>gwaPlink</code>.</p></li>
</ul><p>Regarding the <code>combine</code> argument, this is used to
    control how the p-values returned by the usage of many
    association tests will be combined to produce a 
    summarized p-value. Currently <code>"whitlock"</code>, 
    <code>"pandora"</code> and <code>"harmonic"</code> are placeholders
    since the estimation of p-value weights are work in
    progress. For the rest, please see Moulos and Hatzis,
    2015 (PMID: 25452340) and Fanidis and Moulos, 2021 
    (PMID: 32778872).</p>    
<p>Regarding the <code>glmOpts</code> argument, this is a named
    list with members:</p><ul><li><p><code>family</code>: The regression family, which
        defaults to <code>NULL</code> for auto-detection.</p></li>
<li><p><code>size</code>: The number of sequential SNPs to
        include in a lasso regression model (currently in
        testing mode).</p></li>
</ul><p>Regarding the <code>rrblupOpts</code> argument, this is a 
    named list with members:</p><ul><li><p><code>pcblup</code>: How should the PCs included
        in ridge regression be calculated? Defaults to
        <code>"auto"</code>, see the main function arguments
        for the rest of the options.</p></li>
<li><p><code>npcs</code>: The number of PCs to include.</p></li>
</ul><p>Regarding the <code>statgenOpts</code> argument, this is a 
    named list with members:</p><ul><li><p><code>kinship</code>: The kinship matrix estimation
        method, to be included in the statgenGWAS model
        fitting. Defaults to <code>"astle"</code>, see the
        statgenGWAS package for other options.</p></li>
<li><p><code>reml</code>: The method for calculated the
        Restricted Maximum Likelihood in the statgenGWAS
        model fitting. Defaults to <code>"EMMA"</code>, see the
        statgenGWAS package for other options.</p></li>
<li><p><code>gls</code>: The method used to estimate the
        SNP effects, <code>"signle"</code> (default) for using
        a single kinship matrix, or <code>"multi"</code> for 
        using chromosome specific kinship matrices. See
        the statgenGWAS package for further explanations.</p></li>
</ul><p>Regarding the <code>snptestOpts</code> argument, this is a 
    named list with members:</p><ul><li><p><code>test</code>: The type of test to be used (see
        also main arguments above). Defaults to 
        <code>"frequentist"</code>. See also SNPTEST documentation.</p></li>
<li><p><code>model</code>: The genotype effect estimation
        mode, defaults to <code>"additive"</code>, See also main
        arguments and SNPTEST documentation.</p></li>
<li><p><code>workspace</code>: The path to SNPTEST
        temporary workspace. Defaults to <code>NULL</code> for
        using <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a>. It should be provided
        for maximum control.</code></p></li>
</ul><p>Regarding the <code>plinkOpts</code> argument, this is a
    named list with members:</p><ul><li><p><code>effect</code>: How should the effects be
        modeled for each SNP (similar to the <code>model</code>
        of SNPTEST). Please see PLINK documentation for a
        full list of options.</p></li>
<li><p><code>seed</code>: A seed for PLINK randomized 
        operations such as resampling and permuting.
        Defaults to <code>42</code>.</p></li>
<li><p><code>cliopts</code>: Further options to be passed
        to <code>plink</code> command, as these would be given
        to the command line. Please note that these should
        be switched and options accepting numbers or srings
        but _not_ input/output files.</p></li>
</ul><p>Regarding the <code>usepc</code> argument of <code>gwaBlup</code>,
    this can be one of <code>"auto"</code>, <code>"estim"</code>,
    <code>"rrint"</code>, <code>"fixed"</code> or <code>"none"</code>. It
    controls how rrBLUP models use the PCs in the generic
    ridge regression model. The following are available:</p><ul><li><p><code>"auto"</code>: When in automated mode, if
        the input object contains calculated PCs to be
        used as covariates, these are used, otherwise
        the optimal number is estimate (switch to 
        <code>"estim"</code>.)</p></li>
<li><p><code>"estim"</code>: Automatic iterative estimation.</p></li>
<li><p><code>"rrint"</code>: In this case, the number of PCs
        to include are taken from existing PCs in the input
        <code>GWASExperiment</code> object, or from the <code>npcs</code>
        argument if there are no PCs in the input. Then,
        this number is used internally to the <code>GWAS</code>
        rrBLUP function to include PCs in the model. Although
        not the default, it seems that this is the more
        stable method and it is recommended to be used.</p></li>
<li><p><code>"fixed"</code>: This uses <code>npcs</code> only
        (in a similar way as <code>"auto"</code>.)</p></li>
<li><p><code>"none"</code>: Do not include PCs in the
        model.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># Only 1 PC will be returned because of dataset size</span></span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="calcPcaCovar.html">calcPcaCovar</a></span><span class="op">(</span><span class="va">toy</span>,method<span class="op">=</span><span class="st">"snprel"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA to select covariates to include in associations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing LD pruning...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP pruning based on LD:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 962</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sliding window: 500,000 basepairs, Inf SNPs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     |LD| threshold: 0.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     method: composite</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 18: 23.44%, 154/657</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 21: 21.28%, 73/343</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 227 markers are selected in total.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LD pruning finished! 227 SNPs will be used for PCA.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Principal Component Analysis (PCA) on genotypes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 773 SNPs (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 227</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of principal components: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PCA:    the sum of all selected genotypes (0,1,2) = 10268</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CPU capabilities: Double-Precision SSE2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mon Jun  6 20:06:52 2022    (internal increment: 13024)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mon Jun  6 20:06:52 2022    Begin (eigenvalues and eigenvectors)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mon Jun  6 20:06:52 2022    Done.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting number of PCs to include</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   with Tracy-Widom test for significant eigenvalues</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     selected 1 PCs</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toyG</span> <span class="op">&lt;-</span> <span class="fu">gwa</span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,</span>
<span class="r-in">        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 54 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 54 associations found out of 657 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 35 associations found out of 343 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 89 associations with (uncorrected) combined p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">glmSumstat</span> <span class="op">&lt;-</span> <span class="fu">gwaGlm</span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 42 associations (uncorrected p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">rrSumstat</span> <span class="op">&lt;-</span> <span class="fu">gwaBlup</span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,usepc<span class="op">=</span><span class="st">"rrint"</span>,npcs<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing genotypes for rrBLUP...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with rrBLUP model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 55 associations (uncorrected p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">sgSumstat</span> <span class="op">&lt;-</span> <span class="fu">gwaStatgen</span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 42 associations (uncorrected p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span>
<span class="r-in">        <span class="va">snpSumstat</span> <span class="op">&lt;-</span> <span class="fu">gwaSnptest</span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">            covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snptest operations will be executed at /tmp/Rtmpxo9Dha/snptest_AZQIL3254L</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing SNPTEST association run at workspace directory: /tmp/Rtmpxo9Dha/snptest_AZQIL3254L</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  writing SNPTEST sample file in /tmp/Rtmpxo9Dha/snptest_AZQIL3254L</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   writing PLINK files in /tmp/Rtmpxo9Dha/snptest_AZQIL3254L</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpxo9Dha/snptest_AZQIL3254L/plink_snptest_DPKHE3458Z.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpxo9Dha/snptest_AZQIL3254L/plink_snptest_DPKHE3458Z.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpxo9Dha/snptest_AZQIL3254L/plink_snptest_DPKHE3458Z.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with SNPTEST</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> /home/panos/.local/share/R/snptest-2.5.4/snptest \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   -data /tmp/Rtmpxo9Dha/snptest_AZQIL3254L/plink_snptest_DPKHE3458Z.bed \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     /tmp/Rtmpxo9Dha/snptest_AZQIL3254L/pheno.sample \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   -frequentist 1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   -method score \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   -pheno BMI \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   -cov_all \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   -o /tmp/Rtmpxo9Dha/snptest_AZQIL3254L/snptest.out</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 38 associations (uncorrected p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span>
<span class="r-in">        <span class="va">plSumstat</span> <span class="op">&lt;-</span> <span class="fu">gwaPlink</span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">            covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> plink operations will be executed at /tmp/Rtmpxo9Dha/plink_PIICY8094S</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PLINK association run at workspace directory: /tmp/Rtmpxo9Dha/plink_PIICY8094S</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   writing the PLINK phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   writing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   writing PLINK genotype files in /tmp/Rtmpxo9Dha/plink_PIICY8094S</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpxo9Dha/plink_PIICY8094S/plink_plink_QJCXD6659L.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpxo9Dha/plink_PIICY8094S/plink_plink_QJCXD6659L.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpxo9Dha/plink_PIICY8094S/plink_plink_QJCXD6659L.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with PLINK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> /home/panos/.local/share/R/plink-1.90/plink \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --bfile /tmp/Rtmpxo9Dha/plink_PIICY8094S/plink_plink_QJCXD6659L \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpxo9Dha/plink_PIICY8094S/pheno_QJCXD6659L \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --allow-no-sex \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --ci 0.95 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out /tmp/Rtmpxo9Dha/plink_PIICY8094S/out_QJCXD6659L \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --covar /tmp/Rtmpxo9Dha/plink_PIICY8094S/covar_QJCXD6659L \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --linear perm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 43 associations (uncorrected p-values).</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

