<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='These functions either convert a GWASExperiment to 
    another object or file required for functionalities 
    in the supporting packages, or convert between human
    genome coordinate systems. Finally, to facilitate this
    conversion process when required, a functionality to
    "guess" the human genome version is provided.'><title>Functions to convert GWASExperiment to other classes — gweConvertFuns • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Functions to convert GWASExperiment to other classes — gweConvertFuns"><meta property="og:description" content='These functions either convert a GWASExperiment to 
    another object or file required for functionalities 
    in the supporting packages, or convert between human
    genome coordinate systems. Finally, to facilitate this
    conversion process when required, a functionality to
    "guess" the human genome version is provided.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Functions to convert GWASExperiment to other classes</h1>
      
      <div class="d-none name"><code>gweConvertFuns.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>These functions either convert a GWASExperiment to 
    another object or file required for functionalities 
    in the supporting packages, or convert between human
    genome coordinate systems. Finally, to facilitate this
    conversion process when required, a functionality to
    "guess" the human genome version is provided.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">GWASExperiment2gData</span><span class="op">(</span><span class="va">obj</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>, pcs <span class="op">=</span> <span class="cn">FALSE</span>,
        reverse <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
    
    <span class="fu">GWASExperiment2GDS</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>
    
    <span class="fu">GWASExperimentLiftOver</span><span class="op">(</span><span class="va">obj</span>, <span class="va">from</span>, <span class="va">to</span><span class="op">)</span>
    
    <span class="fu">guessHumanGenomeVersion</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>obj</dt>
<dd><p>a <code>GWASExperiment</code> object.</p></dd>
<dt>covariates</dt>
<dd><p>the covariates to include in the
    <code>gData</code> object (package statgenGWAS) including the
    response phenotype.</p></dd>
<dt>pcs</dt>
<dd><p>include the PCs present in <code>obj</code> in the
    <code>gData</code> object? Defaults to <code>FALSE</code>.</p></dd>
<dt>reverse</dt>
<dd><p>reverse the genotypes in the construction
    of a <code>gData</code> object? Defaults to <code>FALSE</code>.</p></dd>
<dt>from</dt>
<dd><p>the source genome version, one of 
    <code>"hg19"</code> or <code>"hg38"</code>.</p></dd>
<dt>to</dt>
<dd><p>the target genome version, one of 
    <code>"hg19"</code> or <code>"hg38"</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>For <code>GWASExperiment2gData</code>, a <code>gData</code> object
    that can be used with functions from the statgenGWAS 
    package.
    For <code>GWASExperiment2GDS</code>, a GDS file is written to
    the location specified by <code>gdsfile(obj).</code>    For <code>GWASExperimentLiftOver</code>, the same 
    GWASExperiment object is returned, with changed SNP 
    positions from one genome version to the other. Some
    SNPs may be dropped because of inability of the liftOver
    algorithm to map between versions (rare).
    For <code>guessHumanGenomeVersion</code>, a string denoting
    the guessed human genome version (<code>"hg19"</code> or 
<code>"hg38"</code>).</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>guessHumanGenomeVersion</code> function,
    it queries dbSNP for 5 random SNPs present in <code>obj</code>
    using the package rsnps. Depending on the result, the
    genome version is guessed.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># There are some non-numeric phenotypes - ignore</span></span>
<span class="r-in">    <span class="va">gd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu">GWASExperiment2gData</span><span class="op">(</span><span class="va">toy</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu">GWASExperiment2GDS</span><span class="op">(</span><span class="va">toy</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmps2YAD0/file14c681a9e70.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmps2YAD0/file14c681a9e70.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmps2YAD0/file14c681a9e70.bed (SNP-major mode)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Start file conversion from PLINK BED to SNP GDS ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BED file: '/tmp/Rtmps2YAD0/file14c681a9e70.bed'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         SNP-major mode (Sample X SNP), 24.4K</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     FAM file: '/tmp/Rtmps2YAD0/file14c681a9e70.fam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BIM file: '/tmp/Rtmps2YAD0/file14c681a9e70.bim'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wed Jun 29 19:38:12 2022     (store sample id, snp id, position, and chromosome)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     start writing: 100 samples, 1000 SNPs ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wed Jun 29 19:38:12 2022 	Done.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Optimize the access efficiency ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clean up the fragments of GDS file:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     open the file '/tmp/Rtmps2YAD0/file14c2393c7e1' (34.6K)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of fragments: 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     save to '/tmp/Rtmps2YAD0/file14c2393c7e1.tmp'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rename '/tmp/Rtmps2YAD0/file14c2393c7e1.tmp' (34.3K, reduced: 312B)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of fragments: 24</span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span><span class="op">)</span> <span class="co"># TRUE</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">genome</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="co"># hg19</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "hg19"</span>
<span class="r-in">    <span class="va">toy2</span> <span class="op">&lt;-</span> <span class="fu">GWASExperimentLiftOver</span><span class="op">(</span><span class="va">toy</span>,from<span class="op">=</span><span class="st">"hg19"</span>,</span>
<span class="r-in">        to<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Lifting over features</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   lifting chr18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   lifting chr21</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="fu">guessHumanGenomeVersion</span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="co"># hg19</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting info about the following rsIDs: rs238146, rs273746, rs13048320, rs4798085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "hg19"</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

