<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function creates a GWASExperiment object
    from PLINK BED/BIM/FAM files using functions from the
    package snpStats. In the future, importing with the
    package bigSNP will be supported."><title>Create GWASExperiments from PLINK files — importGWAS • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create GWASExperiments from PLINK files — importGWAS"><meta property="og:description" content="This function creates a GWASExperiment object
    from PLINK BED/BIM/FAM files using functions from the
    package snpStats. In the future, importing with the
    package bigSNP will be supported."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Create GWASExperiments from PLINK files</h1>
      
      <div class="d-none name"><code>importGWAS.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function creates a <code>GWASExperiment</code> object
    from PLINK BED/BIM/FAM files using functions from the
    package snpStats. In the future, importing with the
    package bigSNP will be supported.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">importGWAS</span><span class="op">(</span><span class="va">input</span>, phenos <span class="op">=</span> <span class="cn">NULL</span>, backend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"snpStats"</span>,
        <span class="st">"bigsnpr"</span><span class="op">)</span>, selection <span class="op">=</span> <span class="cn">NULL</span>, genome <span class="op">=</span> <span class="cn">NA_character_</span>,
        alleleOrder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"plink"</span>, <span class="st">"reverse"</span><span class="op">)</span>, writeGds <span class="op">=</span> <span class="cn">TRUE</span>,
        gdsfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">backend</span> <span class="op">==</span> <span class="st">"snpStats"</span>, <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>, <span class="cn">NA</span><span class="op">)</span>,
        gdsOverwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>input</dt>
<dd><p>A list with the full path to a PLINK file 
    triplet (BED/BIM/FAM) or a string with the triplet 
    basename which will be suffixed with bed/bim/fam. See
    als Details.</p></dd>
<dt>phenos</dt>
<dd><p>A data frame or a file with the dataset
    phenotypes. It must have the same number of rows as the
    samples present in the PLINK triplet.</p></dd>
<dt>backend</dt>
<dd><p>the R backend to use for object management.
    See Details.</p></dd>
<dt>selection</dt>
<dd><p>A list with members named <code>snps</code>
    and <code>samples</code> to be used as the related 
    <code>select.snps</code> and <code>select.subjects</code> arguments
    of the <code><a href="https://rdrr.io/pkg/snpStats/man/read.plink.html" class="external-link">read.plink</a></code> function from snpStats.</p></dd>
<dt>genome</dt>
<dd><p>The reference genome which the SNPs are
    mapped to. Typically one of <code>"hg19"</code> or 
    <code>"hg38"</code> if this information is available, otherwise
    <code>NA</code>.</p></dd>
<dt>alleleOrder</dt>
<dd><p>Should the alleles be reversed? If yes
    choose <code>"reverse"</code>.</p></dd>
<dt>writeGds</dt>
<dd><p>Write a GDS supporting file while 
    importing PLINK files? In some cases such as reading
    files for external imputation is not needed. Defaults 
    to <code>TRUE</code>.</p></dd>
<dt>gdsfile</dt>
<dd><p>A GDS file supported by the package
    SNPRelate is necessary for some operations like
    SNPRelate PCA and LD. Where should this be written?
    It is adivsed to explicitly provide this path.</p></dd>
<dt>gdsOverwrite</dt>
<dd><p>overwrite the provided GDS file if
    exists? Defaults to <code>TRUE</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A GWASExperiment object with the contents of the PLINK
    files triplet and optionally, additional associated
    phenotypes.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>input</code> argument, when this is a list,
    the latter must have named members with names <code>bed</code>,
    <code>bim</code>, <code>fam</code>. When this is a single string, it
    must point to a full path where the triplet is found and
    must not end with any of <code>bim</code>, <code>bed</code>, 
    <code>fam</code> as these are appended to the base.</p>    
<p>Regarding the <code>backend</code> argument, current PRISMA
    design goals are to support both importing and processing
    data with the R package snpStats and the R package 
    bigSNP. Currently, only snpStats is supported.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="va">fam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.fam"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">bim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.bim"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">bed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/sample.bed"</span>,package<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>fam<span class="op">=</span><span class="va">fam</span>,bim<span class="op">=</span><span class="va">bim</span>,bed<span class="op">=</span><span class="va">bed</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu">snpStats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snpStats/man/read.plink.html" class="external-link">read.plink</a></span><span class="op">(</span><span class="va">input</span><span class="op">$</span><span class="va">bed</span>,<span class="va">input</span><span class="op">$</span><span class="va">bim</span>,<span class="va">input</span><span class="op">$</span><span class="va">fam</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># Without phenotypes</span></span>
<span class="r-in">    <span class="va">gwe</span> <span class="op">&lt;-</span> <span class="fu">importGWAS</span><span class="op">(</span><span class="va">input</span>,backend<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading PLINK files with snpStats framework</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading PLINK files with SNPRelate framework and storing output tp /tmp/Rtmps2YAD0/file14c7c0de2bb</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Start file conversion from PLINK BED to SNP GDS ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BED file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bed'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         SNP-major mode (Sample X SNP), 603 bytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     FAM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.fam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BIM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bim'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wed Jun 29 19:38:24 2022     (store sample id, snp id, position, and chromosome)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     start writing: 120 samples, 20 SNPs ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wed Jun 29 19:38:24 2022 	Done.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Optimize the access efficiency ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clean up the fragments of GDS file:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     open the file '/tmp/Rtmps2YAD0/file14c7c0de2bb' (5.1K)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of fragments: 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     save to '/tmp/Rtmps2YAD0/file14c7c0de2bb.tmp'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rename '/tmp/Rtmps2YAD0/file14c7c0de2bb.tmp' (4.8K, reduced: 312B)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of fragments: 24</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># With phenotypes</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">pseudopheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span class="r-in">        case_control<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sample</span><span class="op">$</span><span class="va">fam</span><span class="op">)</span>,replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">        other_pheno<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"drug"</span>,<span class="st">"nodrug"</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sample</span><span class="op">$</span><span class="va">fam</span><span class="op">)</span>,replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">        cont<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sample</span><span class="op">$</span><span class="va">fam</span><span class="op">)</span><span class="op">)</span>,<span class="fl">3</span><span class="op">)</span>,</span>
<span class="r-in">        row.names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sample</span><span class="op">$</span><span class="va">fam</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in">    <span class="va">gwe</span> <span class="op">&lt;-</span> <span class="fu">importGWAS</span><span class="op">(</span><span class="va">input</span>,phenos<span class="op">=</span><span class="va">pseudopheno</span>,backend<span class="op">=</span><span class="st">"snpStats"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading PLINK files with snpStats framework</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading PLINK files with SNPRelate framework and storing output tp /tmp/Rtmps2YAD0/file14c6a724626</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Start file conversion from PLINK BED to SNP GDS ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BED file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bed'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         SNP-major mode (Sample X SNP), 603 bytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     FAM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.fam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BIM file: '/home/panos/R/x86_64-pc-linux-gnu-library/4.1/snpStats/extdata/sample.bim'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wed Jun 29 19:38:24 2022     (store sample id, snp id, position, and chromosome)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     start writing: 120 samples, 20 SNPs ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wed Jun 29 19:38:24 2022 	Done.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Optimize the access efficiency ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clean up the fragments of GDS file:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     open the file '/tmp/Rtmps2YAD0/file14c6a724626' (5.1K)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of fragments: 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     save to '/tmp/Rtmps2YAD0/file14c6a724626.tmp'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rename '/tmp/Rtmps2YAD0/file14c6a724626.tmp' (4.8K, reduced: 312B)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of fragments: 24</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

