<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="These functions list the associated analyses with a
    GWASExperiment, i.e. GWA tests for association summary
    statistics, or PRS weights from PRS extraction analysis.
    A GWASExperiment may have multiple GWA and PRS analysis
    performed on it, such as GWA with multiple or no PCs as
    covariates. Each analysis is attached to the main
    GWASExperiment to avoid repetition of large genotypic
    data."><title>Functions to list analyses in a GWASExperiment — listAttrFuns • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Functions to list analyses in a GWASExperiment — listAttrFuns"><meta property="og:description" content="These functions list the associated analyses with a
    GWASExperiment, i.e. GWA tests for association summary
    statistics, or PRS weights from PRS extraction analysis.
    A GWASExperiment may have multiple GWA and PRS analysis
    performed on it, such as GWA with multiple or no PCs as
    covariates. Each analysis is attached to the main
    GWASExperiment to avoid repetition of large genotypic
    data."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Functions to list analyses in a GWASExperiment</h1>
      
      <div class="d-none name"><code>listAttrFuns.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>These functions list the associated analyses with a
    GWASExperiment, i.e. GWA tests for association summary
    statistics, or PRS weights from PRS extraction analysis.
    A GWASExperiment may have multiple GWA and PRS analysis
    performed on it, such as GWA with multiple or no PCs as
    covariates. Each analysis is attached to the main
    GWASExperiment to avoid repetition of large genotypic
    data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">listGwa</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>
    <span class="fu">listPrs</span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>obj</dt>
<dd><p>a <code>GWASExperiment</code> object.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>These functions only print information on screen.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># Only 1 PC will be returned because of dataset size</span></span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="calcPcaCovar.html">calcPcaCovar</a></span><span class="op">(</span><span class="va">toy</span>,method<span class="op">=</span><span class="st">"snprel"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA to select covariates to include in associations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading GDS file /tmp/RtmplLIKfg/temp_libpath4f71185f9e1c/prisma/extdata/toy.gds...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing LD pruning...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP pruning based on LD:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 962</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sliding window: 500,000 basepairs, Inf SNPs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     |LD| threshold: 0.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     method: composite</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 18: 23.44%, 154/657</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 21: 21.28%, 73/343</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 227 markers are selected in total.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LD pruning finished! 227 SNPs will be used for PCA.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Principal Component Analysis (PCA) on genotypes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 773 SNPs (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 227</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of principal components: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PCA:    the sum of all selected genotypes (0,1,2) = 10268</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CPU capabilities: Double-Precision SSE2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fri Jun  3 14:30:10 2022    (internal increment: 13024)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fri Jun  3 14:30:10 2022    Begin (eigenvalues and eigenvectors)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Fri Jun  3 14:30:10 2022    Done.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting number of PCs to include</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   with Tracy-Widom test for significant eigenvalues</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     selected 1 PCs</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,</span>
<span class="r-in">        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 54 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 54 associations found out of 657 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 35 associations found out of 343 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 89 associations with (uncorrected) combined p-values).</span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,</span>
<span class="r-in">        pcs<span class="op">=</span><span class="cn">FALSE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 25 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 25 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 52 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 16 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 16 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 52 associations found out of 657 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 35 associations found out of 343 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 87 associations with (uncorrected) combined p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="fu">listGwa</span><span class="op">(</span><span class="va">toy</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      Putative formula: BMI ~ Age + Gender + PC1 + [SNP]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      Putative formula: BMI ~ Age + Gender + [SNP]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Note: The PC variables have putative names based on the number of PCs included in the model.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For example, instead of PC1, PC2, PC3 the actual PCs may be PC1, PC3, PC4</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">prsiceOpts</span> <span class="op">&lt;-</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">prsiceOpts</span><span class="op">$</span><span class="va">perm</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span class="r-in">    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">        <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="prsFuns.html">runPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">            covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,</span>
<span class="r-in">            pcs<span class="op">=</span><span class="cn">TRUE</span>,prsiceOpts<span class="op">=</span><span class="va">prsiceOpts</span><span class="op">)</span></span>
<span class="r-in">        <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="prsFuns.html">runPRS</a></span><span class="op">(</span>base<span class="op">=</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">            covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,</span>
<span class="r-in">            pcs<span class="op">=</span><span class="cn">FALSE</span>,prsiceOpts<span class="op">=</span><span class="va">prsiceOpts</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">}</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/lassosum_QYJSJ6461I</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/lassosum_QYJSJ6461I</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpwic9pB/lassosum_QYJSJ6461I/plink_lassosum_target_VHSGZ1399P.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpwic9pB/lassosum_QYJSJ6461I/plink_lassosum_target_VHSGZ1399P.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpwic9pB/lassosum_QYJSJ6461I/plink_lassosum_target_VHSGZ1399P.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with reference panel...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats, reference panel, and test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Splitting genome by LD blocks ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum with s=1...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating polygenic scores ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.756356747467697</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/prsice_HWUPU5469K</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/prsice_HWUPU5469K</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpwic9pB/prsice_HWUPU5469K/plink_prsice_YJTRP3784U.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpwic9pB/prsice_HWUPU5469K/plink_prsice_YJTRP3784U.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpwic9pB/prsice_HWUPU5469K/plink_prsice_YJTRP3784U.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/prsice_HWUPU5469K/base_YJTRP3784U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/prsice_HWUPU5469K/pheno_YJTRP3784U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/prsice_HWUPU5469K/plink_prsice_YJTRP3784U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/prsice_HWUPU5469K/covar_YJTRP3784U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_YJTRP3784U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.708854</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.867266</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.158412</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/lassosum_EWZRO6903A</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/lassosum_EWZRO6903A</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpwic9pB/lassosum_EWZRO6903A/plink_lassosum_target_JVIAQ8582L.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpwic9pB/lassosum_EWZRO6903A/plink_lassosum_target_JVIAQ8582L.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpwic9pB/lassosum_EWZRO6903A/plink_lassosum_target_JVIAQ8582L.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with reference panel...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats, reference panel, and test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Splitting genome by LD blocks ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum with s=1...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating polygenic scores ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.757420227733504</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/prsice_SAKPN7219U</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/prsice_SAKPN7219U</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpwic9pB/prsice_SAKPN7219U/plink_prsice_WEHUS1327G.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpwic9pB/prsice_SAKPN7219U/plink_prsice_WEHUS1327G.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpwic9pB/prsice_SAKPN7219U/plink_prsice_WEHUS1327G.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/prsice_SAKPN7219U/base_WEHUS1327G \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/prsice_SAKPN7219U/pheno_WEHUS1327G \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/prsice_SAKPN7219U/plink_prsice_WEHUS1327G \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/prsice_SAKPN7219U/covar_WEHUS1327G \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_WEHUS1327G \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.702364</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.857239</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.154874</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-in">        </span>
<span class="r-in">    <span class="fu">listPrs</span><span class="op">(</span><span class="va">toy</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      Putative formula: BMI ~ Age + Gender + PC1 + [PRS]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 -&gt; Phenotype: BMI | Covariates: Age, Gender | PCs: -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      Putative formula: BMI ~ Age + Gender + [PRS]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Note: The PC variables have putative names based on the number of PCs included in the model.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For example, instead of PC1, PC2, PC3 the actual PCs may be PC1, PC3, PC4</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

