<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function attempts to merge two GWASExperiments with
    various goals, such as merging similar populations with
    common phenotypic and/or anthropometric data to increase
    statistical power. See Details for a more rigorous 
    explanation of the merging process."><title>Merge GWASExperiments from different populations — mergeGWAS • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Merge GWASExperiments from different populations — mergeGWAS"><meta property="og:description" content="This function attempts to merge two GWASExperiments with
    various goals, such as merging similar populations with
    common phenotypic and/or anthropometric data to increase
    statistical power. See Details for a more rigorous 
    explanation of the merging process."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Merge GWASExperiments from different populations</h1>
      
      <div class="d-none name"><code>mergeGWAS.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function attempts to merge two GWASExperiments with
    various goals, such as merging similar populations with
    common phenotypic and/or anthropometric data to increase
    statistical power. See Details for a more rigorous 
    explanation of the merging process.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">mergeGWAS</span><span class="op">(</span><span class="va">gwe1</span>, <span class="va">gwe2</span>, output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"common"</span>, <span class="st">"all"</span><span class="op">)</span>, 
        mafResolve <span class="op">=</span> <span class="cn">NULL</span>, gdsfile <span class="op">=</span> <span class="cn">NULL</span>, writegds <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>gwe1</dt>
<dd><p>A <code>GWASExperiment</code> object from the first
    dataset.</p></dd>
<dt>gwe2</dt>
<dd><p>A <code>GWASExperiment</code> object from the second
    dataset.</p></dd>
<dt>output</dt>
<dd><p>One of <code>"common"</code> (default) to operate
    on common SNPs between <code>gwe1</code> and <code>gwe2</code> or 
    <code>"all"</code> to operate on all SNPs. SNPs unique to each
    dataset are returned with missing genotypes in the
    merged set.</p></dd>
<dt>mafResolve</dt>
<dd><p>How would alleles that cannot be 
    resolved just by strand fliping be resolved? Based on
    the frequency in one of the two datasets (<code>1</code> or
    <code>2</code>) or from online resources (dbSNP, population
    studies etc.). The later is the default (<code>NULL</code>).</p></dd>
<dt>gdsfile</dt>
<dd><p>path to the new GDS file. If <code>NULL</code>, a 
    temporary file will be written (unadvised if 
    <code>writegds=TRUE</code>!).</p></dd>
<dt>writegds</dt>
<dd><p>write the new GDS file? Defaults to 
    <code>FALSE</code>. It can be written anytime with the
    <code><a href="gweConvertFuns.html">GWASExperiment2GDS</a></code> function.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A new GWASExperiment object with merged genotypes,
    samples and phenotypes.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The merging of <code>GWASExperiment</code> objects is performed
    based on the following steps and assumptions. Firstly, the
    two objects are checked for genome version compatibility.
    If they are not compatible the process does not start.
    You should consider using the 
    <code><a href="gweConvertFuns.html">GWASExperimentLiftOver</a></code> function that can
    potentially render the two datasets compatible. Also,
    both datasets should have a common reference SNP naming.
    It is not possible to merge a dataset where the markers
    are named e.g. by Illumina IDs and a dataset where the
    markers are named with dbSNP (rs) IDs.</p>    
<p>If the two datasets are compatible in terms of genome 
    version and coordinates, the common SNPs are then
    identified between the two datasets. As these must
    point to the same information (location, minor and major
    alleles, same strand), the process then checks these
    attributes and tries to resolve any discrepancies.</p>    
<p>The first check is whether non identical SNPs in terms
    of alleles can be solved with strand-fliping. Those that
    cannot be resolved with strand-fliping may not point to
    the same risk allele. This is resolved either by using
    one dataset as a reference (<code>mafResolve=1 or 2</code>)
    or even better, by querying online resources (e.g.
    Ensembl and dbSNP) to identify the true minor and major
    alleles based on summarized frequencies from population
    studies (there should be an internet connection for 
    this). SNPs whose alleles cannot be resolved by any
    means (strand-flip, online MAF resolve) are completely
    dropped.</p>    
<p>If the common SNPs between the two objects continue not
    being identical, it probably means that there are 
    location/position mismatches potentially derived from
    older annotations. These can only be resolved online.
    If after this there are any remaining non-resolvable
    SNPs, they are dropped. Then, the genotypes of common
    SNPs are merged. The genotypes of the alleles that were 
    flipped are also switched.</p>    
<p>Finally, the common annotation elements and genotypes
    are combined. If <code>output=="common"</code> the process
    stops. If <code>output="all"</code>, the unique elements of
    each dataset are also combined with the common. The
    genotypes of SNPs unique to each dataset are set to
    missing.</p>    
<p>With respect to the allele resolving of the common
    SNPs that cannot be resolved with strand-flipping 
    online:</p><ol><li><p>SNPs are queried with biomaRt to resolve 
        allele frequencies. Genome version is not important 
        at this point as only alleles are used and IDs.</p></li>
<li><p>If minor allele is available from Ensembl, 
        then the flip decision is based on it. If not (e.g. 
        outdated/merged SNPs, dbSNP is queried with package
        rsnps and fill the Ensembl result.</p></li>
<li><p>Some SNPs may have been removed from dbSNP. 
        Nothing can be done. Flip for these will be based 
        on the majority of the rest.</p></li>
<li><p>Some found SNPs may not have information on 
        MAF and minor alleles. dbSNP is requeried with 
        rsnps and filling missing info is retried. Again, 
        some will not be resolved and will be flipped 
        according to majority.</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toy1</span> <span class="op">&lt;-</span> <span class="va">toy</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">800</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span></span>
<span class="r-in">    <span class="va">toy2</span> <span class="op">&lt;-</span> <span class="va">toy</span><span class="op">[</span><span class="fl">201</span><span class="op">:</span><span class="fl">1000</span>,<span class="fl">51</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toym</span> <span class="op">&lt;-</span> <span class="fu">mergeGWAS</span><span class="op">(</span><span class="va">toy1</span>,<span class="va">toy2</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now trying to merge features (markers, SNPs)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Found 600 common SNPs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Checking common SNPs for reversed strands or alleles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Found 0 SNPs with no mathcing alleles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Found 0 SNPs fixable by strand-flip</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Checking for non-resolvable alleles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Merging features</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now trying to merge samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now trying to construct unified genotypes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Merging genotypes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">toym</span><span class="op">)</span> <span class="co"># 600 100</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 600 100</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toym</span> <span class="op">&lt;-</span> <span class="fu">mergeGWAS</span><span class="op">(</span><span class="va">toy1</span>,<span class="va">toy2</span>,output<span class="op">=</span><span class="st">"all"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now trying to merge features (markers, SNPs)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Found 600 common SNPs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Checking common SNPs for reversed strands or alleles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Found 0 SNPs with no mathcing alleles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Found 0 SNPs fixable by strand-flip</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Checking for non-resolvable alleles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Retrieving SNPs specific to each dataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Merging features</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now trying to merge samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now trying to construct unified genotypes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Merging genotypes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> coercing object of mode  numeric  to SnpMatrix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> object has no names - using numeric order for row/column names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> coercing object of mode  numeric  to SnpMatrix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> object has no names - using numeric order for row/column names</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">toym</span><span class="op">)</span> <span class="co"># 1000 100</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1000  100</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

