<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="These functions comprises the main PRISMA workhorse as 
    they perform all the steps required to extract a PRS 
    potentially chatacterizing a phenotypic trait, given a 
    set of samples, their genetic polymorphims and the 
    phenotypes to participate in the subsequent GWAS 
    associations in a GWASExperiment object. Out of these,
    the prisma function combines all three and
    performs:
Input checking and subsequent management
        according to the input details
Iterative calls to prsPipeline for
        each SNP effect estimation method and is the main 
        prisma iterator for extracting multiple PRSs to be
        later combined.
Call to prsSelection which evaluates
        the output of prsPipeline and selects the
        best PRS candidates for each SNP effect estimation
        method.
Wrap up of the data to feed the later PRISMA
        report and basic evaluation graphics.
Output of an object containing all the results 
        and input to the PRISMA reporting function.

    
prsPipeline is the iterator of the PRISMA 
    workflow. It will produce a list of GWASExperiment
    objects or their summaries, up to the filling of the
    prsbetas slot of the GWASExperiment.    
prsSelection accepts as input the output of
    prsPipeline and evaluate and select the best
    PRS candidates according to the input arguments.    
prismaPipeline seves as a wrapper of 
    prisma and makes sure that the pipeline will
    resume if crashed at any point as well as write
    checkpoints and create the report."><title>Main prisma PRS extraction and validation pipelines — prismaFuns • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Main prisma PRS extraction and validation pipelines — prismaFuns"><meta property="og:description" content="These functions comprises the main PRISMA workhorse as 
    they perform all the steps required to extract a PRS 
    potentially chatacterizing a phenotypic trait, given a 
    set of samples, their genetic polymorphims and the 
    phenotypes to participate in the subsequent GWAS 
    associations in a GWASExperiment object. Out of these,
    the prisma function combines all three and
    performs:
Input checking and subsequent management
        according to the input details
Iterative calls to prsPipeline for
        each SNP effect estimation method and is the main 
        prisma iterator for extracting multiple PRSs to be
        later combined.
Call to prsSelection which evaluates
        the output of prsPipeline and selects the
        best PRS candidates for each SNP effect estimation
        method.
Wrap up of the data to feed the later PRISMA
        report and basic evaluation graphics.
Output of an object containing all the results 
        and input to the PRISMA reporting function.

    
prsPipeline is the iterator of the PRISMA 
    workflow. It will produce a list of GWASExperiment
    objects or their summaries, up to the filling of the
    prsbetas slot of the GWASExperiment.    
prsSelection accepts as input the output of
    prsPipeline and evaluate and select the best
    PRS candidates according to the input arguments.    
prismaPipeline seves as a wrapper of 
    prisma and makes sure that the pipeline will
    resume if crashed at any point as well as write
    checkpoints and create the report."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Main prisma PRS extraction and validation pipelines</h1>
      
      <div class="d-none name"><code>prismaFuns.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>These functions comprises the main PRISMA workhorse as 
    they perform all the steps required to extract a PRS 
    potentially chatacterizing a phenotypic trait, given a 
    set of samples, their genetic polymorphims and the 
    phenotypes to participate in the subsequent GWAS 
    associations in a GWASExperiment object. Out of these,
    the <code>prisma</code> function combines all three and
    performs:</p><ul><li><p>Input checking and subsequent management
        according to the input details</p></li>
<li><p>Iterative calls to <code>prsPipeline</code> for
        each SNP effect estimation method and is the main 
        prisma iterator for extracting multiple PRSs to be
        later combined.</p></li>
<li><p>Call to <code>prsSelection</code> which evaluates
        the output of <code>prsPipeline</code> and selects the
        best PRS candidates for each SNP effect estimation
        method.</p></li>
<li><p>Wrap up of the data to feed the later PRISMA
        report and basic evaluation graphics.</p></li>
<li><p>Output of an object containing all the results 
        and input to the PRISMA reporting function.</p></li>
</ul><p><code>prsPipeline</code> is the iterator of the PRISMA 
    workflow. It will produce a list of GWASExperiment
    objects or their summaries, up to the filling of the
    <code>prsbetas</code> slot of the GWASExperiment.</p>    
<p><code>prsSelection</code> accepts as input the output of
    <code>prsPipeline</code> and evaluate and select the best
    PRS candidates according to the input arguments.</p>    
<p><code>prismaPipeline</code> seves as a wrapper of 
    <code>prisma</code> and makes sure that the pipeline will
    resume if crashed at any point as well as write
    checkpoints and create the report.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">prisma</span><span class="op">(</span>
        <span class="va">gwe</span>,
        <span class="va">phenotype</span>,
        <span class="va">covariates</span>,
        pcs <span class="op">=</span> <span class="cn">FALSE</span>,
        npcs <span class="op">=</span> <span class="fl">0</span>,
        trainSize <span class="op">=</span> <span class="fl">0.8</span>,
        niter <span class="op">=</span> <span class="fl">10</span>,
        resolution <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"frequency"</span>, <span class="st">"quantile"</span><span class="op">)</span>,
        step <span class="op">=</span> <span class="kw">if</span> <span class="op">(</span><span class="va">resolution</span> <span class="op">==</span> <span class="st">"frequency"</span><span class="op">)</span> <span class="fl">1</span> <span class="kw">else</span> 
            <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>, <span class="fl">0.8</span>, <span class="fl">0.9</span>, <span class="fl">0.95</span>, <span class="fl">0.99</span><span class="op">)</span>,
        minFreq <span class="op">=</span> <span class="fl">2</span>,
        minSnps <span class="op">=</span> <span class="fl">5</span>,
        dropSameQuantiles <span class="op">=</span> <span class="cn">TRUE</span>,
        aggregation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"intersection"</span>, <span class="st">"union"</span><span class="op">)</span>,
        effectWeight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mean"</span>, <span class="st">"median"</span>, <span class="st">"weight"</span><span class="op">)</span>,
        prsSelectMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"maxima"</span>, <span class="st">"elbow"</span><span class="op">)</span>,
        prsSelectCrit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prs_r2"</span>, <span class="st">"prs_pvalue"</span>, <span class="st">"prs_aic"</span><span class="op">)</span>,
        prsSelectStat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mean"</span>, <span class="st">"median"</span>, <span class="st">"none"</span><span class="op">)</span>,
        prsSelectR2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"adjusted"</span>, <span class="st">"raw"</span><span class="op">)</span>,
        sigTest <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ttest"</span>, <span class="st">"wilcoxon"</span>, <span class="st">"empirical"</span><span class="op">)</span>,
        filters <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>,
        pcaMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, <span class="st">"snprel"</span>, <span class="st">"grid"</span>, <span class="st">"hubert"</span><span class="op">)</span>,
        imputeMissing <span class="op">=</span> <span class="cn">FALSE</span>,
        imputeMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"single"</span>, <span class="st">"split"</span><span class="op">)</span>,
        gwaMethods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>, <span class="st">"rrblup"</span>, <span class="st">"statgen"</span>, <span class="st">"snptest"</span>,
            <span class="st">"plink"</span><span class="op">)</span>,
        gwaCombine <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fisher"</span>, <span class="st">"simes"</span>, <span class="st">"max"</span>, <span class="st">"min"</span>, <span class="st">"harmonic"</span>, 
            <span class="st">"whitlock"</span>, <span class="st">"pandora"</span><span class="op">)</span>,
        glmOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span>,
        rrblupOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"rrblup"</span><span class="op">)</span>,
        statgenOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"statgen"</span><span class="op">)</span>,
        snptestOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span>,
        plinkOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span>,
        prsMethods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lassosum"</span>,<span class="st">"prsice"</span><span class="op">)</span>,
        lassosumOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"lassosum"</span><span class="op">)</span>,
        prsiceOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span>,
        prsWorkspace <span class="op">=</span> <span class="cn">NULL</span>,
        cleanup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"intermediate"</span>, <span class="st">"all"</span><span class="op">)</span>,
        logging <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"screen"</span>, <span class="st">"file"</span><span class="op">)</span>,
        dnOutput <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gwaslist"</span>, <span class="st">"summaries"</span><span class="op">)</span>,
        output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"full"</span><span class="op">)</span>,
        continue <span class="op">=</span> <span class="cn">FALSE</span>,
        useDenovoWorkspace <span class="op">=</span> <span class="cn">NULL</span>,
        runId <span class="op">=</span> <span class="cn">NULL</span>,
        evalOnSplit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"original"</span>, <span class="st">"new"</span><span class="op">)</span>,
        evalWith <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"vanilla"</span>, <span class="st">"prscice"</span><span class="op">)</span>,
        sitadelaDb <span class="op">=</span> <span class="cn">NULL</span>,
        rc <span class="op">=</span> <span class="cn">NULL</span>
    <span class="op">)</span>
    
    <span class="fu">prsPipeline</span><span class="op">(</span>
        <span class="va">gwe</span>,
        <span class="va">phenotype</span>,
        <span class="va">covariates</span>,
        pcs <span class="op">=</span> <span class="cn">FALSE</span>,
        npcs <span class="op">=</span> <span class="fl">0</span>,
        snpSelection <span class="op">=</span> <span class="cn">NULL</span>,
        trainSize <span class="op">=</span> <span class="fl">0.8</span>,
        niter <span class="op">=</span> <span class="fl">10</span>,
        filters <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>,
        pcaMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, <span class="st">"snprel"</span>, <span class="st">"grid"</span>, <span class="st">"hubert"</span><span class="op">)</span>,
        imputeMissing <span class="op">=</span> <span class="cn">FALSE</span>,
        imputeMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"single"</span>, <span class="st">"split"</span><span class="op">)</span>,
        gwaMethods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>, <span class="st">"rrblup"</span>, <span class="st">"statgen"</span>, <span class="st">"snptest"</span>,
            <span class="st">"plink"</span><span class="op">)</span>,
        gwaCombine <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fisher"</span>, <span class="st">"simes"</span>, <span class="st">"max"</span>, <span class="st">"min"</span>, <span class="st">"harmonic"</span>, 
            <span class="st">"whitlock"</span>, <span class="st">"pandora"</span><span class="op">)</span>,
        glmOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span>,
        rrblupOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"rrblup"</span><span class="op">)</span>,
        statgenOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"statgen"</span><span class="op">)</span>,
        snptestOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span>,
        plinkOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span>,
        prsMethods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lassosum"</span>,<span class="st">"prsice"</span><span class="op">)</span>,
        lassosumOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"lassosum"</span><span class="op">)</span>,
        prsiceOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span>,
        prsWorkspace <span class="op">=</span> <span class="cn">NULL</span>,
        cleanup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"intermediate"</span>, <span class="st">"all"</span><span class="op">)</span>,
        logging <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"screen"</span>, <span class="st">"file"</span><span class="op">)</span>,
        output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gwaslist"</span>, <span class="st">"summaries"</span><span class="op">)</span>,
        continue <span class="op">=</span> <span class="cn">FALSE</span>,
        useDenovoWorkspace <span class="op">=</span> <span class="cn">NULL</span>,
        runId <span class="op">=</span> <span class="cn">NULL</span>,
        evalWith <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"vanilla"</span>, <span class="st">"prscice"</span><span class="op">)</span>,
        rc <span class="op">=</span> <span class="cn">NULL</span>
    <span class="op">)</span>
    
    <span class="fu">prsSelection</span><span class="op">(</span>
        <span class="va">dnList</span>,
        <span class="va">gwe</span>,
        <span class="va">phenotype</span>,
        <span class="va">covariates</span>,
        pcs <span class="op">=</span> <span class="cn">FALSE</span>,
        npcs <span class="op">=</span> <span class="fl">0</span>,
        trainSize <span class="op">=</span> <span class="fl">0.8</span>,
        niter <span class="op">=</span> <span class="fl">10</span>,
        resolution <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"frequency"</span>, <span class="st">"quantile"</span><span class="op">)</span>,
        step <span class="op">=</span> <span class="kw">if</span> <span class="op">(</span><span class="va">resolution</span> <span class="op">==</span> <span class="st">"frequency"</span><span class="op">)</span> <span class="fl">1</span> <span class="kw">else</span> 
            <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>, <span class="fl">0.8</span>, <span class="fl">0.9</span>, <span class="fl">0.95</span>, <span class="fl">0.99</span><span class="op">)</span>,
        minFreq <span class="op">=</span> <span class="fl">2</span>,
        minSnps <span class="op">=</span> <span class="fl">5</span>,
        dropSameQuantiles <span class="op">=</span> <span class="cn">TRUE</span>,
        aggregation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"intersection"</span>, <span class="st">"union"</span><span class="op">)</span>,
        effectWeight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mean"</span>, <span class="st">"median"</span>, <span class="st">"weight"</span><span class="op">)</span>,
        filters <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span>,
        pcaMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, <span class="st">"snprel"</span>, <span class="st">"grid"</span>, <span class="st">"hubert"</span><span class="op">)</span>,
        imputeMissing <span class="op">=</span> <span class="cn">FALSE</span>,
        imputeMethod <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"single"</span>, <span class="st">"split"</span><span class="op">)</span>,
        gwaMethods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>, <span class="st">"rrblup"</span>, <span class="st">"statgen"</span>, <span class="st">"snptest"</span>,
            <span class="st">"plink"</span><span class="op">)</span>, <span class="co"># lasso later</span>
        gwaCombine <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fisher"</span>, <span class="st">"simes"</span>, <span class="st">"max"</span>, <span class="st">"min"</span>, <span class="st">"harmonic"</span>, 
            <span class="st">"whitlock"</span>, <span class="st">"pandora"</span><span class="op">)</span>,
        glmOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"glm"</span><span class="op">)</span>,
        rrblupOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"rrblup"</span><span class="op">)</span>,
        statgenOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"statgen"</span><span class="op">)</span>,
        snptestOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"snptest"</span><span class="op">)</span>,
        plinkOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span>,
        prsMethods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lassosum"</span>,<span class="st">"prsice"</span><span class="op">)</span>,
        lassosumOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"lassosum"</span><span class="op">)</span>,
        prsiceOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span>,
        prsWorkspace <span class="op">=</span> <span class="cn">NULL</span>,
        cleanup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"intermediate"</span>, <span class="st">"all"</span><span class="op">)</span>,
        logging <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"screen"</span>, <span class="st">"file"</span><span class="op">)</span>,
        continue <span class="op">=</span> <span class="cn">FALSE</span>,
        useDenovoWorkspace <span class="op">=</span> <span class="cn">NULL</span>,
        runId <span class="op">=</span> <span class="cn">NULL</span>,
        evalWith <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"vanilla"</span>, <span class="st">"prscice"</span><span class="op">)</span>,
        output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"summary"</span>, <span class="st">"full"</span><span class="op">)</span>,
        rc <span class="op">=</span> <span class="cn">NULL</span>
    <span class="op">)</span>
    
    <span class="fu">prismaPipeline</span><span class="op">(</span><span class="va">...</span>, retry <span class="op">=</span> <span class="fl">10</span>, outPath <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>gwe</dt>
<dd><p>A GWASExperiment object with GWA tests performed.
    In other words, <code><a href="gwaFuns.html">gwa</a></code> must have been applied.</p></dd>
<dt>dnList</dt>
<dd><p>A list resulting from a <em>de novo</em> PRS 
    extraction pipeline. See also Details.</p></dd>
<dt>phenotype</dt>
<dd><p>The trait for which a PRS is sought to be
    extracted. It should be a character of length 1 and exist
    in the phenotypes of <code>gwe</code>.</p></dd>
<dt>covariates</dt>
<dd><p>The covariates to include in regression
    models. It should be a character vector of traits all 
    existing in the phenotypes of <code>gwe</code>.</p></dd>
<dt>pcs</dt>
<dd><p>Inlcude PCs in the regression model? Defaults
    to <code>FALSE</code>.</p></dd>
<dt>npcs</dt>
<dd><p>The number of PCs to use if <code>pcs=TRUE</code>.
    If <code>npcs=0</code>, the PCs available in <code>gwe</code> will
    be used.</p></dd>
<dt>trainSize</dt>
<dd><p>The fraction of the training set used
    for summary statistics calculation in each iteration. 
    See also Details. Defaults to <code>0.8</code>.</p></dd>
<dt>niter</dt>
<dd><p>Number of PRS extraction iterations. 
    Defaults to <code>100</code>.</p></dd>
<dt>resolution</dt>
<dd><p>How to resolve the best number of
    SNPs in a PRS candidate, either by validating a PRS
    with as many SNPs as the number of appearances at each
    frequency cutoff (<code>frequency</code>, default) or as many
    SNPs as those included at different quantile cutoffs of 
    the frequency of appearance distribution 
    (<code>"quantile"</code>). See also Details.</p></dd>
<dt>step</dt>
<dd><p>The number of steps or step increment for
    <code>resolution</code>. Defaults to <code>1</code> for 
    <code>resolution="frequency"</code> or a set of predefined
    quantiles when <code>resolution="quantile"</code>. See also
    Details.</p></dd>
<dt>minFreq</dt>
<dd><p>The minimum accepted frequency of 
    appearance of a SNP in <code>niter</code> iterations to be 
    considered as member of a PRS candidate. Defaults to 
    <code>2</code>.</p></dd>
<dt>minSnps</dt>
<dd><p>The minimum number of SNPs included in
    a PRS candidate. Defaults to <code>5</code>.</p></dd>
<dt>dropSameQuantiles</dt>
<dd><p>When 
    <code>resolution="quantile"</code> drop quantiles with the 
    same SNPs? Defaults to <code>TRUE</code>. See also Details.</p></dd>
<dt>aggregation</dt>
<dd><p>How should the SNPs from different 
    PRS algorithms should be aggregated at the end of the 
    iterations in order to create the distribution of 
    appearances. One of <code>"intersection"</code> (default)
    or <code>"union"</code>. See also Details.</p></dd>
<dt>effectWeight</dt>
<dd><p>How to summarize SNP effects 
    (regression coefficients) across the <code>niter</code>
    iterations. One of <code>"mean"</code> (default),
    <code>"mean"</code> or <code>"weight"</code>. See also Details.</p></dd>
<dt>prsSelectMethod</dt>
<dd><p>How to select the number of SNPs
    for the PRS after <code>niter</code> iterations. One of
    <code>"maxima"</code> (default) or <code>"elbow"</code>. See also
    Details.</p></dd>
<dt>prsSelectCrit</dt>
<dd><p>One of <code>"prs_r2"</code> (default), 
    <code>"prs_pvalue"</code> or <code>"prs_aic"</code>. Determines what
    criterion is used for PRS selection. See also Details.</p></dd>
<dt>prsSelectStat</dt>
<dd><p>One of <code>"mean"</code> (default),
    <code>"median"</code> or <code>"none"</code>. Determines which
    calculated statistic (mean, median) is used to summarize
    the <code>prsSelectCrit</code> in the evaluation metrics and
    when applicable (e.g. not applicable in p-values).</p></dd>
<dt>prsSelectR2</dt>
<dd><p>One of <code>"adjusted"</code> (default) or 
    <code>"raw"</code>. The first one uses the PRS contribution to
    the evaluation regression model and the second one the 
    R^2 of the full model.</p></dd>
<dt>sigTest</dt>
<dd><p>How to assess statistical significance
    between initial PRS R^2 calculations and the respective
    R^2 (or other <code>prsSelectCrit</code> metrics) from the 
    prisma aggreagated result. One or of <code>"ttest"</code> 
    (default), <code>"wilcoxon"</code> or <code>"empirical"</code>. See
    also Details.</p></dd>
<dt>filters</dt>
<dd><p>Filters to be applied to the <code>gwe</code>,
    during the workflow executed in each prisma iteration.
    For details, see the <code><a href="filterGWAS.html">filterGWAS</a></code> function.</p></dd>
<dt>pcaMethod</dt>
<dd><p>Which method to use for PCA. It can be
    one of <code>"auto"</code> (default), <code>"snprel"</code> for the
    parallel PCA method implemented in the package SNPRelate, 
    <code>"grid"</code> or <code>"hubert"</code> for robust (but very 
    slow) PCA. See also the <code><a href="gwaFuns.html">gwa</a></code> function.</p></dd>
<dt>imputeMissing</dt>
<dd><p>Impute missing values in each
    prisma iteration? Defaults to <code>TRUE</code>, see also
    Details.</p></dd>
<dt>imputeMethod</dt>
<dd><p>One of <code>"single"</code> (default) or
    <code>"split"</code>. See the <code><a href="imputeGWAS.html">imputeGWAS</a></code> for
    details.</p></dd>
<dt>gwaMethods</dt>
<dd><p>The GWAS SNP effect estimation 
    method(s) to use in order to estimate the effects that
    will be used for PRS weights. It can be one or more from
    <code>"glm"</code>, <code>"rrblup"</code>, <code>"snptest"</code>,
    <code>"statgen"</code> and <code>"plink"</code>. See the 
    <code><a href="gwaFuns.html">gwa</a></code> function for more details.</p></dd>
<dt>gwaCombine</dt>
<dd><p>How to combine SNP p-values from more
    than one GWAS effect estimation methods. Currently not
    used.</p></dd>
<dt>glmOpts</dt>
<dd><p>Options list specific to GLM effect 
    estimation. See the <code><a href="gwaFuns.html">gwa</a></code> function for 
    more details.</p></dd>
<dt>rrblupOpts</dt>
<dd><p>Options list specific to rrBLUP 
    effect estimation. See the <code><a href="gwaFuns.html">gwa</a></code> function 
    for more details.</p></dd>
<dt>statgenOpts</dt>
<dd><p>Options list specific to statgenGWAS
    effect estimation. See the <code><a href="gwaFuns.html">gwa</a></code> function 
    for more details.</p></dd>
<dt>snptestOpts</dt>
<dd><p>Options list specific to SNPTEST
    effect estimation. See the <code><a href="gwaFuns.html">gwa</a></code> function 
    for more details.</p></dd>
<dt>plinkOpts</dt>
<dd><p>Options list specific to PLINK
    effect estimation. See the <code><a href="gwaFuns.html">gwa</a></code> function 
    for more details.</p></dd>
<dt>prsMethods</dt>
<dd><p>One or more PRS extraction algorithms
    to be included in the prisma workflow. Can be 
    <code>"prcise"</code>, <code>"lassosum"</code> or both.</p></dd>
<dt>lassosumOpts</dt>
<dd><p>Options list specific to lassosum
    PRS extraction. See the <code><a href="prsFuns.html">runPRS</a></code> function
    for more details.</p></dd>
<dt>prsiceOpts</dt>
<dd><p>Options list specific to PRSice2
    PRS extraction. See the <code><a href="prsFuns.html">runPRS</a></code> function
    for more details.</p></dd>
<dt>prsWorkspace</dt>
<dd><p>A local directory to be used for
    prisma operations. Defaults to <code>NULL</code> for
    auto-generation. See also Details.</p></dd>
<dt>cleanup</dt>
<dd><p>Cleanup level after the finish of the
    prisma pipeline. <code>"none"</code> (default) leaves all
    intermediate output intact. <code>"intermediate"</code> 
    deletes the files produced in effect estimation as
    well as BED, BIM and FAM files required by PRSice2
    and lassosum linkage files. <code>"all"</code> deletes
    the entire <code>prsWorkspace</code>.</p></dd>
<dt>logging</dt>
<dd><p>One of <code>"screen"</code> (default) or
    <code>"file"</code>. Where to write log messages. If prisma
    runs in parallel, it is <code>"file"</code> all the time.</p></dd>
<dt>dnOutput</dt>
<dd><p>The output of <code>prsPipeline</code> called
    from this function. Can be <code>"gwaslist"</code> (default) 
    for a list of GWASExperiments or <code>"summaries"</code> for
    a summary required for next steps of the prisma pipelie.
    The latter has obviously a smaller size.</p></dd>
<dt>output</dt>
<dd><p>The level of output of this function. It
    can be <code>"normal"</code> for a list with the necessary
    information to create the prisma report or <code>"full"</code>
    which also contains the itermediate output of 
    <code>prsPipeline</code> called within prisma.</p></dd>
<dt>continue</dt>
<dd><p>Continue a crashed pipeline? Defaults
    to <code>FALSE</code>, see Details.</p></dd>
<dt>useDenovoWorkspace</dt>
<dd><p>Validate the prisma combined
    PRS in the initial dataset splits or generate new ones.
    Defaults to <code>NULL</code>, see also Details.</p></dd>
<dt>runId</dt>
<dd><p>A unique run id for the run. If not given,
    it will be auto-generated.</p></dd>
<dt>evalOnSplit</dt>
<dd><p>Which data split to use to evaluate the
    PRS. <code>"original"</code> (default) uses the original split 
    and SNP effect calculations, while <code>"new"</code> creates
    a new split taking much more time.</p></dd>
<dt>evalWith</dt>
<dd><p>How to perform the PRS evaluation 
    regressions. It can be <code>"vanilla"</code> (default) to do
    it with R or <code>"prsice"</code> to do it with PRSice2. The
    latter takes much more time as there is intermediate
    writing of PLINK files.</p></dd>
<dt>sitadelaDb</dt>
<dd><p>Either <code>NULL</code> or a sitadela SQLite 
    database with at least human RefSeq annotations for hg19
    and hg38. It is used to add gene names to PRS SNP loci.</p></dd>
<dt>snpSelection</dt>
<dd><p>A data frame with candidate SNPs, the
    output, or part of the output of 
    <code><a href="aggregatePrsMarkers.html">aggregatePrsMarkers</a></code>, or one of the list 
    members of the output of <code><a href="selectPrs.html">selectPrs</a></code>.</p></dd>
<dt>rc</dt>
<dd><p>The fraction of available cores to use for 
    parallel calculations. Default is <code>NULL</code> for no
    parallelization.</p></dd>
<dt>...</dt>
<dd><p>Arguments passed to <code>prisma</code> from the
    <code>prismaPipeline</code> wrapper..</p></dd>
<dt>retry</dt>
<dd><p>The number of pipeline relaunches in case
    of crash for any reason. Defaults to <code>10</code>.</p></dd>
<dt>outPath</dt>
<dd><p>The path to the directory where 
    checkpoint files and the report will be written. It
    can be different from <code>prsWorkspace</code> and
    defaults to <code>NULL</code> for auto-generation.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list with the members <code>params</code> and <code>results</code>.
    Both are lists.
    The <code>params</code> member has the following contents:</p><ul><li><p><code>call</code> which contains the function call.</p></li>
<li><p><code>args</code> which contains a list of all
        arguments used, default and user-specific, 
        irrespective of <code>call</code>.</p></li>
</ul><p>The <code>results</code> member is a named list with length as
    the vector <code>gweMethods</code> and each sublist corresponds
    to results for each SNP effect calculation method. Each 
    one of them has the following contents:</p><ul><li><p><code>candidates</code>, a named list of data frames
        containing the SNPs within each PRS candidate. The
        names correspond to the number of SNPs in the PRS
        candidates and the candidates are determined based
        on <code>prsSelectMethod</code>.</p></li>
<li><p><code>iterations</code> which is <code>NULL</code> or
        the output of each prisma iteration (multiple
        outputs of <code>prsSelection</code>) if 
        <code>output="full"</code>.</p></li>
<li><p><code>reportData</code> which is an internal 
        structure containing PRS evaluation metrics and
        ggplot2 objects used to make the prisma report.</p></li>
</ul><p>This object can be saved in and .rda file and later used
    with cross-validation metrics and lookup in the GWAS 
    catalog (<code><a href="ebiQueryFuns.html">prismaLookup</a></code> function). These 
    three elements are required to build a prisma report
    (<code><a href="prismaOutFuns.html">prismaReport</a></code> function).</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>dnList</code> argument, this is a list
    of either <code>GWASExperiment</code> objects or a list of
    lists, each summarizing a <code>GWASExperiment</code> object
    to save space/memory as not all the elements of a
    <code>GWASExperiment</code> object are required for downstream
    analyses. The content of <code>dnList</code> is determined by
    the <code>output</code> argument of <code>prsPipeline</code>. When
    the latter is <code>"summaries"</code>, <code>dnList</code> sublists
    have the following members:</p><ul><li><p>baseIndex: the sample indices that were used
        for the extraction of summary statistics via
        <code>gwa</code> (training).</p></li>
<li><p>targetIndex: the sample indices that were
        used for PRS extraction and validation in the
        supported pipelines (testing).</p></li>
<li><p>effects: the effects (regression coefficients
        or odds ratios) for each SNP (summary statistics).</p></li>
<li><p>prsbetas: a subset of regression coefficients
        and/or adjusted ones in the case of lassosum, which
        comprise the PRS weights.</p></li>
<li><p>npcs: the number of principal components 
        included in the regression model.</p></li>
<li><p>pr2: a data frame with metrics returned from
        PRSice2, such as PRS p-value and R^2.</p></li>
</ul><p><code>dnList</code> can essentially be the outcome only of a
    <em>de novo</em> PRS extraction pipeline.</p>    
<p>Regarding the <code>trainSize</code> argument, this controls
    the fraction of the input GWASExperiment object (i.e.
    the fraction of the input dataset) that will be used
    for the calculation of summary statistics using the
    <code>link{gwa}</code> function (usually). These summary
    statistics are then used with the rest of the samples
    (fraction 1 - <code>trainSize</code>) for PRS extraction,
    that is a input to the supported PRS algorithms 
    (PRSice2, lassosum). Therefore, in each prisma
    iteration (out of <code>niter</code>), 
    <code>trainSize*ncol(gwe)</code> samples are used to calculate
    summary statistics and <code>(1-trainSize)*ncol(gwe)</code>
    are used as targets in the PRS extrction process.</p>    
<p>Regarding the <code>dropSameQuantiles</code> argument, when
    <code>resolution="quantile"</code>, it is possible that the
    number of SNPs at different quantiles of the frequency
    of appearance distribution is the same. By default,
    the PRS is not evaluated at the highest quantile with
    the same number of SNPs as the next lower one
    (<code>dropSameQuantiles=FALSE</code>). For example, if
    the number of PRS SNPs at Q=0.5 and Q=0.75 is 1000,
    then there will be no evaluation at Q=0.75. To
    force such an evaluation, set 
    <code>dropSameQuantiles=TRUE</code>.</p>    
<p>Regarding the <code>"aggregation"</code> argument, this
    controls how should PRS SNP candidates from different
    PRS algorithms should be consolidated. When
    <code>"intersection"</code>, SNPs included in the PRS
    suggested by all algorithms synthesize the PRS
    candidate set. When <code>"union"</code>, it is the union
    of SNPs present in all PRSs. The final SNP set
    is then used to create a distribution of appearances
    (frequency of each SNP across <code>"niter"</code>
    iterations) which is the basis for later selection.</p>    
<p>Regarding the <code>effectWeight</code> argument, this
    controls how the SNP effects in PRS candidates are
    summarized over <code>niter</code> iterations to produce the
    final effects for the calculation of weighted PRS. When
    <code>"mean"</code> (default), the final effect for a SNP is
    the mean off all its effects across <code>niter</code>
    iterations, when it appears. When <code>"median"</code>, it's
    the respective median while when <code>"weight"</code>,
    the SNP effects are weighted by 1 minus the recorded
    PRS R^2 so that the "best" runs can be given more
    value.</p>    
<p>Regarding the <code>prsSelectMethod</code> argument, this
    controls how the final PRS is selected after the end
    of <code>niter</code> iterations and based on the requested
    evaluation metrics (typically the PRS R^2). When
    <code>"maxima"</code>(default), a distribution of PRS R^2
    values is created with the PRS R^2 values (or rarer, 
    other chosen metric) derived from <code>niter</code> 
    iterations. This distribution (if R^2-related) may
    further be adjusted according to the number of SNPs
    comrpising in the PRS, according to the 
    <code>prsSelectR2</code> argument. Then, from this 
    distribution, local maxima are calculated. The local
    maximum with the highest R^2 is considered the best
    candidate and provides the number of SNPs. The other 
    maxima are returned also as further candidates. When
    <code>"elbow"</code> the, elbow point method is used for
    determining the optimal number of SNPs. The elbow method
    is a heuristic used in determining the number of 
    features in a data set with respect to some optimization
    criterion. In this case the criterion is the value of
    PRS R^2 and the method consists of plotting the PRS
    explained variation (as depicted by R^2) as a function 
    of the number of SNPs in the PRS.</p>    
<p>Regarding the <code>prsSelectCrit</code> argument, this
    controls the criterion used to select the final PRS
    after the end of the iterative process and according
    to the evaluation metrics. When <code>"prs_r2"</code> 
    (default), the decision making plots and the 
    <code>prsSelectMethod</code> operate on the PRS R^2 calculated
    in each PRS extraction iteration. When 
    <code>"prs_pvalue"</code>, the -log_10(PRS p-valye) is used
    and when <code>"prs_aic"</code>, the AIC (Akaike's Information
    Criterion) is used.</p>    
<p>Regarding the <code>sigTest</code> argument, this controls
    how (any) statistical significance is assessed in the
    comparison of the initial PRS R^2 calculations (as this
    is returned by PRSice2) and the PRS R^2 of the prisma
    combined approach. Essentially, the two PRS R^2 (or
    full model R^2 as determined by <code>prsSelectCrit</code>)
    distributions are compared using a t-test (default),
    a Wilcoxon rank test or an empirical test. In the latter
    case, the 'p-value' is the number of times that the
    R^2 of the prisma combination method is smaller than
    the initial R^2 divided by the number of iterations.
    The outcome is presented in the prisma report.</p>    
<p>Regarding the <code>imputeMissing</code> argument, it is
    practically not used with a pre-processed dataset which
    should not contain missing values. If the initial 
    dataset has missing values, imputation will be performed
    in each step, slowing down the procedure.</p>    
<p>Regarding the <code>prsWorkspace</code> argument, this
    controls where the prisma intermediate files and
    results as well as configuration files will be stored.
    If <code>NULL</code> (not recommended!), a temporary
    directory is used. This directory hosts:</p><ul><li><p>a JSON file with run parameters called
        <code>params.json</code>.</p></li>
<li><p>one subdirectory for each iteration which 
        contains intermediate files used e.g. by SNPTEST,
        PLINK, PRSice2 etc. These directories are named
        based on the <code>runId</code> and the iteration number.</p></li>
<li><p>log file for each iteration, named similarly
        as the run subdirectory.</p></li>
</ul><p>Regarding the <code>continue</code> argument, this can be
    set to <code>TRUE</code> to continue a crashed pipeline, 
    given the initial <code>gwe</code> argument and the prisma
    workspace. If the workspace is not found, a new run
    will be started. Generally, as the prisma pipeline
    combines several often disjoint components (including
    several system commands), it is possible that something
    can go wrong. We have done our best to modularize the
    pipeline which can be restarted from various 
    automatically detected checkpoints.</p>    
<p>Regarding the <code>useDenovoWorkspace</code> argument, if
    <code>useDenovoWorkspace</code> is a valid prisma workspace or
    a list output from <code>prsPipeline</code>, the PRS validation
    is performed using the original data split. If 
    <code>NULL</code>, then a completely new split is generated.
    The later can take very long time, so it is advised to 
    use an existing split if available. This argument is 
    available in the <code>prsPipeline</code> and 
    <code>prsSelection</code> functions when used directly. When
    executing the complete pipeline through <code>prisma</code>,
    this is controlled by the <code>evalOnSplit</code> argument.
    When the latter is <code>evalOnSplit="original"</code>, the 
    workspace generated in <code>prisma</code> initiation is used. 
    When <code>evalOnSplit="new"</code>, the initial split is
    ignored and a new split is created and evaluated,
    taking much more time as the SNP effect calculation
    takes place again.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Lose filters for the toy dataset</span></span>
<span class="r-in"><span class="va">filts</span> <span class="op">&lt;-</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"filters"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">filts</span><span class="op">$</span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span class="r-in"><span class="va">filts</span><span class="op">$</span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-3</span></span>
<span class="r-in"><span class="va">filts</span><span class="op">$</span><span class="va">pcaOut</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span class="r-in"><span class="va">filts</span><span class="op">$</span><span class="va">inbreed</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Workspace</span></span>
<span class="r-in"><span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">wspace1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d1</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Traits</span></span>
<span class="r-in"><span class="va">response</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span></span>
<span class="r-in"><span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Case_Control"</span>,<span class="st">"Gender"</span>,<span class="st">"Age"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Complete 4-iteration PRS pipeline - ~1 min to run with 4 cores - no PCA</span></span>
<span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">prismaOut</span> <span class="op">&lt;-</span> <span class="fu">prisma</span><span class="op">(</span></span>
<span class="r-in">    gwe<span class="op">=</span><span class="va">toy</span>,</span>
<span class="r-in">    phenotype<span class="op">=</span><span class="va">response</span>,</span>
<span class="r-in">    covariates<span class="op">=</span><span class="va">covariates</span>,</span>
<span class="r-in">    pcs<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    trainSize<span class="op">=</span><span class="fl">0.8</span>,</span>
<span class="r-in">    niter<span class="op">=</span><span class="fl">4</span>,</span>
<span class="r-in">    resolution<span class="op">=</span><span class="st">"frequency"</span>,</span>
<span class="r-in">    minSnps<span class="op">=</span><span class="fl">2</span>,</span>
<span class="r-in">    filters<span class="op">=</span><span class="va">filts</span>,</span>
<span class="r-in">    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,</span>
<span class="r-in">    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    gwaMethods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span>,</span>
<span class="r-in">    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,</span>
<span class="r-in">    prsWorkspace<span class="op">=</span><span class="va">wspace1</span>,</span>
<span class="r-in">    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,</span>
<span class="r-in">    logging<span class="op">=</span><span class="st">"file"</span>,</span>
<span class="r-in">    output<span class="op">=</span><span class="st">"normal"</span><span class="co">#,</span></span>
<span class="r-in">    <span class="co">#rc=0.25</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Below, individual wrapper steps implemented in prisma as whole</span></span>
<span class="r-in"><span class="co"># Output only of the discovery PRS pipeline with statgenGWAS</span></span>
<span class="r-in"><span class="va">d2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">wspace2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d2</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dnList</span> <span class="op">&lt;-</span> <span class="fu">prsPipeline</span><span class="op">(</span></span>
<span class="r-in">    gwe<span class="op">=</span><span class="va">toy</span>,</span>
<span class="r-in">    phenotype<span class="op">=</span><span class="va">response</span>,</span>
<span class="r-in">    covariates<span class="op">=</span><span class="va">covariates</span>,</span>
<span class="r-in">    pcs<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    trainSize<span class="op">=</span><span class="fl">0.8</span>,</span>
<span class="r-in">    niter<span class="op">=</span><span class="fl">4</span>,</span>
<span class="r-in">    filters<span class="op">=</span><span class="va">filts</span>,</span>
<span class="r-in">    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,</span>
<span class="r-in">    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,</span>
<span class="r-in">    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,</span>
<span class="r-in">    prsWorkspace<span class="op">=</span><span class="va">wspace2</span>,</span>
<span class="r-in">    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,</span>
<span class="r-in">    logging<span class="op">=</span><span class="st">"file"</span><span class="co">#,</span></span>
<span class="r-in">    <span class="co">#rc=0.25</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># After, calculate the evaluation metrics</span></span>
<span class="r-in"><span class="va">d3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">wspace3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="va">d3</span><span class="op">)</span></span>
<span class="r-in"><span class="va">evalMetrics</span> <span class="op">&lt;-</span> <span class="fu">prsSelection</span><span class="op">(</span></span>
<span class="r-in">    dnList<span class="op">=</span><span class="va">dnList</span>,</span>
<span class="r-in">    gwe<span class="op">=</span><span class="va">toy</span>,</span>
<span class="r-in">    phenotype<span class="op">=</span><span class="va">response</span>,</span>
<span class="r-in">    covariates<span class="op">=</span><span class="va">covariates</span>,</span>
<span class="r-in">    pcs<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    trainSize<span class="op">=</span><span class="fl">0.8</span>,</span>
<span class="r-in">    niter<span class="op">=</span><span class="fl">4</span>,</span>
<span class="r-in">    resolution<span class="op">=</span><span class="st">"frequency"</span>,</span>
<span class="r-in">    minSnps<span class="op">=</span><span class="fl">2</span>,</span>
<span class="r-in">    filters<span class="op">=</span><span class="va">filts</span>,</span>
<span class="r-in">    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,</span>
<span class="r-in">    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    gwaMethods<span class="op">=</span><span class="st">"statgen"</span>,</span>
<span class="r-in">    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,</span>
<span class="r-in">    prsWorkspace<span class="op">=</span><span class="va">wspace3</span>,</span>
<span class="r-in">    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,</span>
<span class="r-in">    logging<span class="op">=</span><span class="st">"file"</span>,</span>
<span class="r-in">    output<span class="op">=</span><span class="st">"summary"</span>,</span>
<span class="r-in">    useDenovoWorkspace<span class="op">=</span><span class="va">dnList</span><span class="co">#,</span></span>
<span class="r-in">    <span class="co">#rc=0.25</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">candidates</span> <span class="op">&lt;-</span> <span class="fu"><a href="selectPrs.html">selectPrs</a></span><span class="op">(</span></span>
<span class="r-in">    metrics<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">metrics</span>,</span>
<span class="r-in">    snpSelection<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">pgs</span>,</span>
<span class="r-in">    gwe<span class="op">=</span><span class="va">toy</span>,</span>
<span class="r-in">    base<span class="op">=</span><span class="va">evalMetrics</span><span class="op">$</span><span class="va">baseline</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:31:01: Data processing started...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression parameters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Phenotypic response (phenotype)          : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Principal Components in covariates (pcs) : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of Principal Components (npcs)    : -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preprocessing options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample and genotype filtering options (filters)     :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   snpCallRate    : 0.98</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   sampleCallRate : 0.95</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   maf            : 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   hwe            : 0.001</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroStat     : median</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroFac      : 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroHard     : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pcaOut         : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pcaRobust      : hubert</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   nPC            : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   LD             : 0.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   IBD            : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   inbreed        : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Principal Components calculation method (pcaMethod) : snprel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Impute missing genotypes (imputeMissing)            : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Missing genotype imputation method (imputeMethod)   : singlesplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> GWA analysis options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Genome-Wide Association tests (gwaMethods)        : glm, statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Association tests combination method (gwaCombine) : fishersimesmaxminharmonicwhitlockpandora</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Association tests options                         : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   glm : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     family : -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     size   : 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   statgen : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     kinship : astle</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     reml    : EMMA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     gls     : single</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS candidate extraction options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS algorithms options (prsMethods)    : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   lassosum : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anc   : eur</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     valid : auto</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   prsice : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_kb : 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_r2 : 0.1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_p  : 0.75</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     score    : avg</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     perm     : 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     seed     : 42</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS construction and evaluation options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Training set size (trainSize)                               : 80%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of PRS selection iterations (niter)                  : 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection resolution method (resolution)                : frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection resolution step (step)                        : 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Minimum SNP frequency in PRS (minFreq)                      : 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Minimum number of SNPs in PRS (minSnps)                     : 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Drop quantiles with same number of SNPs (dropSameQuantiles) : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS aggregation method (aggregation)                        : intersectionunion</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> SNP effect summarization method (effectWeight)              : meanmedianweight</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS evaluation framework (evalWith)                         : vanillaprscice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection method (prsSelectMethod)           : maxima</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection criterion (prsSelectCrit)          : prs_r2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS criterion summarization (prsSelectStat)      : mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> R^2 type if criterion includes R^2 (prsSelectR2) : adjusted</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Miscellaneous options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Local workspace directory (prsWorkspace) : /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleanup level (cleanup)                  : intermediate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS pipeline output type (dnOutput)      : normal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Output level (output)                    : normal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Run ID (runId)                           : auto</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Available cores fraction (rc)            : single core</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing PRISMA pipeline with glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. Discovery of de novo PRS candidates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 122</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 133</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 11 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 11 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 24 associations found out of 561 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 11 associations found out of 313 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.00433642030136858</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1/base_AKJAD3604X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1/pheno_AKJAD3604X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1/plink_prsice_AKJAD3604X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1/covar_AKJAD3604X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_AKJAD3604X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0880657</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.611383</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.523317</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_1/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 120</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 20 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 20 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 24 associations found out of 567 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 20 associations found out of 311 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 44 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.52053653038099</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2/base_FPZBY3337F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2/pheno_FPZBY3337F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2/plink_prsice_FPZBY3337F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2/covar_FPZBY3337F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_FPZBY3337F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.204038</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.873717</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.66968</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_2/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 121</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 21 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 21 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 14 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 21 associations found out of 562 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 14 associations found out of 315 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0400010719796725</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3/base_IQHZE4132U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3/pheno_IQHZE4132U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3/plink_prsice_IQHZE4132U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3/covar_IQHZE4132U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_IQHZE4132U \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.022937</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.463183</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.440246</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_3/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 124</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 117</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 22 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 22 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 9 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 9 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 22 associations found out of 562 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 9 associations found out of 312 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 31 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0904002403440418</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4/base_WIDOQ1376X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4/pheno_WIDOQ1376X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4/plink_prsice_WIDOQ1376X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4/covar_WIDOQ1376X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_WIDOQ1376X \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0534201</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.653311</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.59989</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline/JUMSPCEZUPG2807Q_denovo_4/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. Calculation of PRS evaluation metrics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation mode: intersect</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 1 yields 2 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 2 yields 47 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 3 yields 3 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 4 yields 61 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating frequencies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Averaging effects with mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing output</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 2 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 3 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs8093690</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/glm/selection/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. Selection of best PRS candidate markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The optimal number of markers for PRS based on prs_r2 is 4 at prs_r2=0.10411</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. Wrap-up and graphics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting latest annotation on the fly for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving genome information for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving latest gene annotation for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: RMySQL</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 1 imported as numeric</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 2 imported as numeric</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting annotation to GenomicRanges object...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biostrings’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     strsplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting DNA sequences...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting GC content...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing PRISMA pipeline with statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. Discovery of de novo PRS candidates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 116</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 134</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 17 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 17 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 13 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 17 associations found out of 568 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 13 associations found out of 316 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 30 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0345300904223554</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1/base_LMEOS4099F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1/pheno_LMEOS4099F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1/plink_prsice_LMEOS4099F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1/covar_LMEOS4099F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_LMEOS4099F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0384586</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.600874</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.562416</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_1/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 118</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 129</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 32 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 32 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 14 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 32 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 14 associations found out of 313 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 46 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.154268723191176</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2/base_WAZVG6475F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2/pheno_WAZVG6475F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2/plink_prsice_WAZVG6475F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2/covar_WAZVG6475F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_WAZVG6475F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.097563</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.490894</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.393331</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_2/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 123</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 129</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 27 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 27 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 14 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 27 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 14 associations found out of 312 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 41 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.236351467360557</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3/base_XVUPR2986P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3/pheno_XVUPR2986P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3/plink_prsice_XVUPR2986P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3/covar_XVUPR2986P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_XVUPR2986P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.104648</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.604003</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.499356</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_3/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 121</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 135</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 13 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 24 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 13 associations found out of 314 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 37 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0771703588765321</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4/base_QDNRW9625K \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4/pheno_QDNRW9625K \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4/plink_prsice_QDNRW9625K \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4/covar_QDNRW9625K \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_QDNRW9625K \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.174849</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.701351</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.526502</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline/JUMSPCEZUPG2807Q_denovo_4/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. Calculation of PRS evaluation metrics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation mode: intersect</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 1 yields 2 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 2 yields 44 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 3 yields 98 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 4 yields 10 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating frequencies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Averaging effects with mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing output</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 2 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 SNPs not found in target data! Excluding...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9977668</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 3 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9303974</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9303974</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9303974</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9303974</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file50523848a345/statgen/selection/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. Selection of best PRS candidate markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The optimal number of markers for PRS based on prs_r2 is 5 at prs_r2=0.11178</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. Wrap-up and graphics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting latest annotation on the fly for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving genome information for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving latest gene annotation for hg19 from refseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 1 imported as numeric</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 2 imported as numeric</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting annotation to GenomicRanges object...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting DNA sequences...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting GC content...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:34:41: Data processing finished!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total processing time: 03 minutes 40 seconds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input pipeline list (dnList) : a list of PRS pipeline summaries</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input dataset object (gwe)   : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression parameters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Phenotypic response (phenotype)          : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Principal Components in covariates (pcs) : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of Principal Components (npcs)    : -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preprocessing options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample and genotype filtering options (filters)     :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   snpCallRate    : 0.98</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   sampleCallRate : 0.95</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   maf            : 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   hwe            : 0.001</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroStat     : median</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroFac      : 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroHard     : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pcaOut         : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pcaRobust      : hubert</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   nPC            : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   LD             : 0.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   IBD            : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   inbreed        : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Principal Components calculation method (pcaMethod) : snprel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Impute missing genotypes (imputeMissing)            : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Missing genotype imputation method (imputeMethod)   : single</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> GWA analysis options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Genome-Wide Association tests (gwaMethods)        : statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Association tests combination method (gwaCombine) : fisher</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Association tests options                         : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   statgen : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     kinship : astle</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     reml    : EMMA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     gls     : single</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS candidate extraction options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS algorithms options (prsMethods)    : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   lassosum : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anc   : eur</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     valid : auto</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   prsice : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_kb : 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_r2 : 0.1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_p  : 0.75</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     score    : avg</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     perm     : 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     seed     : 42</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS construction and evaluation options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Training set size (trainSize)              : 80%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of PRS selection iterations (niter) : 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS evaluation framework (evalWith)        : vanilla</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Miscellaneous options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Local workspace directory (prsWorkspace) : /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleanup level (cleanup)                  : intermediate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Output type (output)                     : gwaslist</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Run ID (runId)                           : XCLQPXOSPSO9247L</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Available cores fraction (rc)            : single core</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 1, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 131</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 25 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 25 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 10 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 10 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 25 associations found out of 564 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 10 associations found out of 316 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.014729117709171</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1/base_JVXIR2543F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1/pheno_JVXIR2543F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1/plink_prsice_JVXIR2543F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1/covar_JVXIR2543F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_JVXIR2543F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.075073</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.513838</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.438765</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_1/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 2, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 120</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 125</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 30 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 30 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 12 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 12 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 30 associations found out of 566 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 12 associations found out of 314 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 42 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.189424254816086</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2/base_ISNSU0521Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2/pheno_ISNSU0521Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2/plink_prsice_ISNSU0521Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2/covar_ISNSU0521Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_ISNSU0521Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.204691</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.644458</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.439766</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_2/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 3, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 115</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 121</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 26 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 26 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 12 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 12 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 26 associations found out of 567 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 12 associations found out of 317 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 38 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0803632054297046</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3/base_MSANV2643F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3/pheno_MSANV2643F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3/plink_prsice_MSANV2643F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3/covar_MSANV2643F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_MSANV2643F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.152777</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.724196</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.571419</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_3/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 4, log is at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 117</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 134</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 14 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 24 associations found out of 569 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 14 associations found out of 312 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 38 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.00882964763212319</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4/base_AJXPV2017Q \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4/pheno_AJXPV2017Q \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4/plink_prsice_AJXPV2017Q \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4/covar_AJXPV2017Q \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_AJXPV2017Q \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0364975</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.355984</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.319487</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126/XCLQPXOSPSO9247L_denovo_4/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB//tmp/Rtmpwic9pB/file5052170bc126</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation mode: intersect</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 1 yields 4 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 2 yields 68 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 3 yields 2 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 4 yields 4 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating frequencies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Averaging effects with mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing output</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression parameters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(args$phenotype):</span> object 'response' not found</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Complete pipeline with report and checkpoints</span></span>
<span class="r-in"><span class="co"># Complete 4-iteration PRS pipeline - ~1 min to run with 4 cores - no PCA</span></span>
<span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">outPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">d4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">wspace4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,<span class="va">d4</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">prismaPipeline</span><span class="op">(</span></span>
<span class="r-in">    gwe<span class="op">=</span><span class="va">toy</span>,</span>
<span class="r-in">    phenotype<span class="op">=</span><span class="va">response</span>,</span>
<span class="r-in">    covariates<span class="op">=</span><span class="va">covariates</span>,</span>
<span class="r-in">    pcs<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    trainSize<span class="op">=</span><span class="fl">0.8</span>,</span>
<span class="r-in">    niter<span class="op">=</span><span class="fl">4</span>,</span>
<span class="r-in">    resolution<span class="op">=</span><span class="st">"frequency"</span>,</span>
<span class="r-in">    minSnps<span class="op">=</span><span class="fl">2</span>,</span>
<span class="r-in">    filters<span class="op">=</span><span class="va">filts</span>,</span>
<span class="r-in">    pcaMethod<span class="op">=</span><span class="st">"snprel"</span>,</span>
<span class="r-in">    imputeMissing<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">    gwaMethods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span>,</span>
<span class="r-in">    prsiceOpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clump_p<span class="op">=</span><span class="fl">0.75</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,</span>
<span class="r-in">    prsWorkspace<span class="op">=</span><span class="va">wspace4</span>,</span>
<span class="r-in">    cleanup<span class="op">=</span><span class="st">"intermediate"</span>,</span>
<span class="r-in">    logging<span class="op">=</span><span class="st">"file"</span>,</span>
<span class="r-in">    output<span class="op">=</span><span class="st">"normal"</span>,</span>
<span class="r-in">    <span class="co">#rc=0.25,</span></span>
<span class="r-in">    outPath<span class="op">=</span><span class="va">outPath</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,<span class="st">"prismaOut.RData"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="co"># TRUE</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This is PRISMA 0.0.1 PRS builder</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:35:53 - Try 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:35:53: Data processing started...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input object (gwe) : a GWASExperiment object with 1000 markers (SNPs) and 100 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression parameters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Phenotypic response (phenotype)          : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression covariates (covariates)       : Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Principal Components in covariates (pcs) : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of Principal Components (npcs)    : -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preprocessing options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample and genotype filtering options (filters)     :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   snpCallRate    : 0.98</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   sampleCallRate : 0.95</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   maf            : 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   hwe            : 0.001</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroStat     : median</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroFac      : 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   heteroHard     : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pcaOut         : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pcaRobust      : hubert</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   nPC            : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   LD             : 0.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   IBD            : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   inbreed        : NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Principal Components calculation method (pcaMethod) : snprel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Impute missing genotypes (imputeMissing)            : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Missing genotype imputation method (imputeMethod)   : singlesplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> GWA analysis options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Genome-Wide Association tests (gwaMethods)        : glm, statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Association tests combination method (gwaCombine) : fishersimesmaxminharmonicwhitlockpandora</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Association tests options                         : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   glm : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     family : -</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     size   : 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   statgen : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     kinship : astle</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     reml    : EMMA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     gls     : single</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS candidate extraction options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS extraction algorithms (prsMethods) : lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS algorithms options (prsMethods)    : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   lassosum : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anc   : eur</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     valid : auto</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   prsice : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_kb : 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_r2 : 0.1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clump_p  : 0.75</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     score    : avg</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     perm     : 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     seed     : 42</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS construction and evaluation options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Training set size (trainSize)                               : 80%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of PRS selection iterations (niter)                  : 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection resolution method (resolution)                : frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection resolution step (step)                        : 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Minimum SNP frequency in PRS (minFreq)                      : 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Minimum number of SNPs in PRS (minSnps)                     : 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Drop quantiles with same number of SNPs (dropSameQuantiles) : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS aggregation method (aggregation)                        : intersectionunion</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> SNP effect summarization method (effectWeight)              : meanmedianweight</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS evaluation framework (evalWith)                         : vanillaprscice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection method (prsSelectMethod)           : maxima</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS selection criterion (prsSelectCrit)          : prs_r2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS criterion summarization (prsSelectStat)      : mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> R^2 type if criterion includes R^2 (prsSelectR2) : adjusted</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Miscellaneous options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Local workspace directory (prsWorkspace) : /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleanup level (cleanup)                  : intermediate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PRS pipeline output type (dnOutput)      : normal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Output level (output)                    : normal</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Run ID (runId)                           : auto</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Available cores fraction (rc)            : single core</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing PRISMA pipeline with glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. Discovery of de novo PRS candidates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 1, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 118</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 128</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 22 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 22 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 15 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 15 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 22 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 15 associations found out of 316 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 37 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.00454763100621826</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1/base_UUOHQ4284D \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1/pheno_UUOHQ4284D \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1/plink_prsice_UUOHQ4284D \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1/covar_UUOHQ4284D \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_UUOHQ4284D \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0800997</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.550064</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.469964</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_1/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 2, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 118</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 24 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 21 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 21 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 24 associations found out of 566 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 21 associations found out of 314 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 45 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.104679411827688</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2/base_GTKRN1473Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2/pheno_GTKRN1473Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2/plink_prsice_GTKRN1473Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2/covar_GTKRN1473Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_GTKRN1473Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0962855</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.603342</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.507056</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_2/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 3, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 120</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 118</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 22 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 22 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 13 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 22 associations found out of 564 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 13 associations found out of 315 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.00205692052718663</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3/base_TBJCU2021O \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3/pheno_TBJCU2021O \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3/plink_prsice_TBJCU2021O \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3/covar_TBJCU2021O \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_TBJCU2021O \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0822372</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.769317</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.68708</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_3/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 4, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 120</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 128</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 19 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 19 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 18 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 19 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 18 associations found out of 315 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 37 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.00764704864422903</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4/base_GPKSB7920W \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4/pheno_GPKSB7920W \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4/plink_prsice_GPKSB7920W \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4/covar_GPKSB7920W \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_GPKSB7920W \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0540151</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.634166</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.580151</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline/ETQQBQRMODZ4103Z_denovo_4/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. Calculation of PRS evaluation metrics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation mode: intersect</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 1 yields 4 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 2 yields 65 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 3 yields 8 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 4 yields 61 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating frequencies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Averaging effects with mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing output</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 2 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 SNPs not found in target data! Excluding...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9949696</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9949696</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9949696</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9949696</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 3 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3/3_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 4 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs7280065</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs7280065</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs7280065</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs16943205 to rs7280065</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4/4_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/glm/selection/4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. Selection of best PRS candidate markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The optimal number of markers for PRS based on prs_r2 is 3 at prs_r2=0.04812</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. Wrap-up and graphics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting latest annotation on the fly for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving genome information for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving latest gene annotation for hg19 from refseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 1 imported as numeric</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 2 imported as numeric</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting annotation to GenomicRanges object...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting DNA sequences...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting GC content...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing PRISMA pipeline with statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ****************************************************************</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. Discovery of de novo PRS candidates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 1, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 120</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 120</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 15 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 15 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 14 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 14 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 15 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 14 associations found out of 313 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 29 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 out of 17 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 out of 17 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0216081429776474</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1/base_IPQEJ9475C \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1/pheno_IPQEJ9475C \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1/plink_prsice_IPQEJ9475C \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1/covar_IPQEJ9475C \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_IPQEJ9475C \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.332271</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.817882</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.485611</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_1/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 2, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 126</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 28 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 28 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 13 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 13 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 28 associations found out of 566 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 13 associations found out of 314 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 41 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.110477819737347</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2/base_BHQOG6084H \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2/pheno_BHQOG6084H \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2/plink_prsice_BHQOG6084H \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2/covar_BHQOG6084H \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_BHQOG6084H \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0987099</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.685136</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.586426</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_2/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 3, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 116</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 125</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 26 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 26 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 10 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 10 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 26 associations found out of 568 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 10 associations found out of 316 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 36 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0184869548478244</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3/base_IWVRP9947M \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3/pheno_IWVRP9947M \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3/plink_prsice_IWVRP9947M \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3/covar_IWVRP9947M \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_IWVRP9947M \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0551661</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.720956</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.66579</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_3/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing denovo pipeline iteration 4, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Denovo pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Dataset partitioning -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 119</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Test (target) QC -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing basic filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Calculating SNP summaries for filtering</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Minor Allele Frequency</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Hardy-Weinberg equilibrium</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample filters</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Call rate</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Heterozygosity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing SNP filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Summarizing sample filter results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Filtering report: -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered SNPs:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   SNP call rate             : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Minor Allele Frequency    : 129</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Hardy-Weinberg equilibrium: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered samples:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Sample call rate          : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Heterozygosity            : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Inbreed                   : 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Training (base) GWAS -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 1 methods: statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 25 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 25 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 8 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 8 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 25 associations found out of 565 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 8 associations found out of 315 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 33 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- PRS analysis with base and target -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20 out of 20 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.0321168599104062</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4/base_NSNNX5893N \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4/pheno_NSNNX5893N \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4/plink_prsice_NSNNX5893N \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4/covar_NSNNX5893N \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 0.75 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_NSNNX5893N \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.0755859</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.569995</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.494409</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline/ETQQBQRMODZ4103Z_denovo_4/denovo_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/baseline</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. Calculation of PRS evaluation metrics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation mode: intersect</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 1 yields 1 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 2 yields 17 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 3 yields 74 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   iteration 4 yields 6 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating frequencies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Averaging effects with mean</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing output</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Frequency 2 &lt;-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prisma operations will be executed at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 1, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_iter_1.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 1 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_evaluate_1/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 2, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_iter_2.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 2 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_evaluate_2/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 3, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_iter_3.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 3 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_evaluate_3/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing evaluation pipeline iteration 4, log is at /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_iter_4.log</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Evaluation pipeline iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the null model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing full models</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing PRS with SNPs from rs1740090 to rs9635852</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Saving iteration 4 result to /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2/2_evaluate_4/evaluate_result.RData-----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cleaning up temporary program-specific intermediate files from /tmp/Rtmpwic9pB/file50526e1966ad//tmp/Rtmpwic9pB/file5052327f4a/statgen/selection/2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. Selection of best PRS candidate markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The optimal number of markers for PRS based on prs_r2 is 14 at prs_r2=0.15788</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. Wrap-up and graphics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> sitadela database found at /home/panos/.local/share/R/sitadela directory</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting latest annotation on the fly for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving genome information for hg19 from refseq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving latest gene annotation for hg19 from refseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 1 imported as numeric</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Unsigned INTEGER in col 2 imported as numeric</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting annotation to GenomicRanges object...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting DNA sequences...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting GC content...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:39:31: Data processing finished!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total processing time: 03 minutes 37 seconds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:39:34 - Cross-validation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running cross-validations for glm with 10 leave-out cases across 1 PRS candidates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validating PRS candidate with 3 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 5% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 10% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 15% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 20% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 25% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 30% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 35% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 40% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 45% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 50% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running cross-validations for statgen with 10 leave-out cases across 1 PRS candidates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validating PRS candidate with 14 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 5% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 10% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 15% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 20% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 25% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 30% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 35% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 40% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 45% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ----- Leave-out percentage 50% -----</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regressing with GLM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Case_Control, Gender, Age</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : No</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 7</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 8</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Cross-validation iteration 10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ================================================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Partitioning dataset into 1 partitions by BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the reduced model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating the full model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! Use summarizeCvMetrics() and plotCvMetrics() to summarize and inspect the results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ################################################################</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:39:52 - GWAS Catalog lookup</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Querying for glm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Querying GWAS Catalog with the largest PRS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -----&gt; Querying for statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Querying GWAS Catalog with the largest PRS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-06-03 14:39:54 - Report</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> processing file: prisma_report.Rmd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |.                                                                     |   1%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    inline R code fragments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A project name here  |                                                                              |.                                                                     |   2%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> label: knitr_init (with options) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> List of 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ echo : logi FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ cache: logi FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..                                                                    |   3%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..                                                                    |   4%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'pander'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:htmltools':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     p</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'grid'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:Biostrings':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pattern</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Summarizing cross-validation statistics for PRS with 3 SNPs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Summarizing cross-validation statistics for PRS with 14 SNPs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...                                                                   |   4%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....                                                                  |   5%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....                                                                  |   6%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....                                                                 |   7%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |......                                                                |   8%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |......                                                                |   9%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......                                                               |  10%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......                                                               |  11%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |........                                                              |  11%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........                                                             |  12%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........                                                             |  13%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........                                                            |  14%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........                                                            |  15%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...........                                                           |  16%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............                                                          |  17%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............                                                          |  18%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............                                                         |  18%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..............                                                        |  19%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..............                                                        |  20%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............                                                       |  21%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............                                                       |  22%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |................                                                      |  23%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................                                                     |  24%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................                                                     |  25%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................                                                    |  25%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................                                                    |  26%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...................                                                   |  27%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................                                                  |  28%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................                                                  |  29%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....................                                                 |  30%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....................                                                 |  31%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |......................                                                |  32%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......................                                               |  32%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......................                                               |  33%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |........................                                              |  34%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........................                                             |  35%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........................                                             |  36%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........................                                            |  37%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........................                                            |  38%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...........................                                           |  39%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............................                                          |  39%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............................                                          |  40%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............................                                         |  41%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............................                                         |  42%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..............................                                        |  43%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............................                                       |  44%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............................                                       |  45%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |................................                                      |  46%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................................                                     |  46%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................................                                     |  47%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................                                    |  48%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................                                    |  49%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...................................                                   |  50%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................................                                  |  51%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................................                                  |  52%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....................................                                 |  53%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....................................                                 |  54%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |......................................                                |  54%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......................................                               |  55%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......................................                               |  56%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |........................................                              |  57%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........................................                             |  58%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........................................                             |  59%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........................................                            |  60%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........................................                            |  61%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...........................................                           |  61%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............................................                          |  62%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............................................                          |  63%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............................................                         |  64%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............................................                         |  65%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..............................................                        |  66%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............................................                       |  67%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............................................                       |  68%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |................................................                      |  68%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................................................                     |  69%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................................................                     |  70%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................................                    |  71%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................................                    |  72%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...................................................                   |  73%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................................................                  |  74%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................................................                  |  75%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....................................................                 |  75%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.....................................................                 |  76%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |......................................................                |  77%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......................................................               |  78%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.......................................................               |  79%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |........................................................              |  80%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |........................................................              |  81%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.........................................................             |  82%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........................................................            |  82%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..........................................................            |  83%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...........................................................           |  84%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............................................................          |  85%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |............................................................          |  86%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............................................................         |  87%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.............................................................         |  88%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..............................................................        |  89%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............................................................       |  89%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...............................................................       |  90%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |................................................................      |  91%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |................................................................      |  92%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |.................................................................     |  93%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................................................    |  94%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................................................    |  95%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |...................................................................   |  96%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................................................................  |  96%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |....................................................................  |  97%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................................................... |  98%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |..................................................................... |  99%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |......................................................................| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> output file: prisma_report.knit.md</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> /usr/bin/pandoc +RTS -K512m -RTS prisma_report.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output /tmp/Rtmpwic9pB/file50526e1966ad/index.html --variable 'robobook:true' --variable 'lightbox:true' --variable 'gallery:false' --variable 'cards:true' --variable bs3=TRUE --standalone --section-divs --table-of-contents --toc-depth 2 --template /home/panos/R/x86_64-pc-linux-gnu-library/4.1/rmdformats/templates/template.html --highlight-style kate --variable theme=bootstrap --include-in-header /tmp/Rtmpwic9pB/rmarkdown-str50526c830ef8.html </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Output created: /tmp/Rtmpwic9pB/file50526e1966ad/index.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

