<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="These functions perform the PRS extraction given one
    or two (base and target) GWASExperiment objects.
    lassosumPRS is a wrapper around the lassosum
    PRS extraction algorithm while prsicePRS is a
    wrapper around the PRSice2 PRS extraction algorithm.
    The runPRS is a wrapper around the two others
    and makes sure that the proper objects are retured."><title>Functions to extract PRS candidates with various algorithms — prsFuns • prisma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Functions to extract PRS candidates with various algorithms — prsFuns"><meta property="og:description" content="These functions perform the PRS extraction given one
    or two (base and target) GWASExperiment objects.
    lassosumPRS is a wrapper around the lassosum
    PRS extraction algorithm while prsicePRS is a
    wrapper around the PRSice2 PRS extraction algorithm.
    The runPRS is a wrapper around the two others
    and makes sure that the proper objects are retured."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">prisma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01-introduction.html">Introduction to the prisma package</a>
    <a class="dropdown-item" href="../articles/02-installation.html">PRISMA installation</a>
    <a class="dropdown-item" href="../articles/03-gwasexperiment.html">The GWASExperiment class</a>
    <a class="dropdown-item" href="../articles/04-inoutfilter.html">Reading, writing and filtering files and objects</a>
    <a class="dropdown-item" href="../articles/05-imputemerge.html">Imputation of missing genotypes, genotype extension and dataset merging</a>
    <a class="dropdown-item" href="../articles/06-gwasfuns.html">Single trait association tests and summary statistics</a>
    <a class="dropdown-item" href="../articles/07-prsfuns.html">Derivation of PRS candidates and evaluation</a>
    <a class="dropdown-item" href="../articles/08-pipelines.html">Complete PRISMA pipelines</a>
    <a class="dropdown-item" href="../articles/09-external.html">Evaluation of external PRS or external GWAS</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pmoulos/prisma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Functions to extract PRS candidates with various algorithms</h1>
      
      <div class="d-none name"><code>prsFuns.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>These functions perform the PRS extraction given one
    or two (base and target) <code>GWASExperiment</code> objects.
    <code>lassosumPRS</code> is a wrapper around the lassosum
    PRS extraction algorithm while <code>prsicePRS</code> is a
    wrapper around the PRSice2 PRS extraction algorithm.
    The <code>runPRS</code> is a wrapper around the two others
    and makes sure that the proper objects are retured.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">lassosumPRS</span><span class="op">(</span><span class="va">base</span>, target <span class="op">=</span> <span class="va">base</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>,
        pcs <span class="op">=</span> <span class="cn">FALSE</span>, wspace <span class="op">=</span> <span class="cn">NULL</span>, anc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"eur"</span>, <span class="st">"asn"</span>, <span class="st">"afr"</span><span class="op">)</span>,
        valid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, <span class="st">"stack"</span>, <span class="st">"split"</span><span class="op">)</span>, rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
    
    <span class="fu">PRS</span><span class="op">(</span><span class="va">gwe</span>, <span class="va">snpSelection</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"avg"</span>, <span class="st">"sum"</span>, <span class="st">"std"</span><span class="op">)</span><span class="op">)</span>
    
    <span class="fu">prsicePRS</span><span class="op">(</span><span class="va">base</span>, target <span class="op">=</span> <span class="va">base</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>,
        pcs <span class="op">=</span> <span class="cn">FALSE</span>, mode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"calculate"</span>, <span class="st">"apply"</span><span class="op">)</span>, wspace <span class="op">=</span> <span class="cn">NULL</span>,
        clump_kb <span class="op">=</span> <span class="fl">250</span>, clump_r2 <span class="op">=</span> <span class="fl">0.1</span>, clump_p <span class="op">=</span> <span class="fl">1</span>,
        score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"avg"</span>, <span class="st">"sum"</span>, <span class="st">"std"</span>, <span class="st">"con-std"</span><span class="op">)</span>, perm <span class="op">=</span> <span class="fl">10000</span>,
        seed <span class="op">=</span> <span class="fl">42</span>, rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
    
    <span class="fu">runPRS</span><span class="op">(</span><span class="va">base</span>, target <span class="op">=</span> <span class="va">base</span>, <span class="va">response</span>, covariates <span class="op">=</span> <span class="cn">NULL</span>,
        pcs <span class="op">=</span> <span class="cn">FALSE</span>, methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lassosum"</span>, <span class="st">"prsice"</span><span class="op">)</span>,
        lassosumOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"lassosum"</span><span class="op">)</span>,
        prsiceOpts <span class="op">=</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span>, wspace <span class="op">=</span> <span class="cn">NULL</span>, 
        rc <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>base</dt>
<dd><p>A <code>GWASExperiment</code> object with GWA
    tests performed and containing summary statistics apart
    from genotypes.</p></dd>
<dt>target</dt>
<dd><p>A <code>GWASExperiment</code> object with or
    without GWA tests, they are ignored. The PRS is 
    calculated based on the target dataset.</p></dd>
<dt>response</dt>
<dd><p>The trait for which a PRS is sought to be
    extracted. It should be a character of length 1 and exist
    in the phenotypes of <code>base</code> and <code>target</code>.</p></dd>
<dt>covariates</dt>
<dd><p>The covariates to include in regression
    models. It should be a character vector of traits all 
    existing in the phenotypes of <code>base</code> and 
    <code>target</code>.</p></dd>
<dt>pcs</dt>
<dd><p>Inlcude PCs in the regression model? Defaults
    to <code>FALSE</code>.</p></dd>
<dt>mode</dt>
<dd><p>The mode to run PRSice2. One of calculating and
    extracting a PRS (<code>"calculate"</code>, default) or applying
    a PRS given a GWASExperiment with existing <code>prsbetas</code>
    (<code>"apply"</code>).</p></dd>
<dt>wspace</dt>
<dd><p>A path to a workspace directory for 
    PRSice. If <code>NULL</code>, a temporary folder is used.</p></dd>
<dt>anc</dt>
<dd><p>One of the precalculated LD structures
    on lassosum, to be taken into account while extracting
    PRS. It can be <code>"eur"</code> (default) for European
    ancestry, <code>"asn"</code> for Asian or <code>"afr"</code> for
    African.</p></dd>
<dt>valid</dt>
<dd><p>Validation mode for lassosum PRS extraction. 
    One of <code>"auto"</code> (default), <code>"split"</code> or 
    <code>"stack"</code>. See also the lassosum package.</p></dd>
<dt>gwe</dt>
<dd><p>A codeGWASExperiment object with the
    slot <code>effects</code> and/or <code>prsbetas</code> filled.</p></dd>
<dt>snpSelection</dt>
<dd><p>A data frame with candidate SNPs, the
    output, or part of the output of 
    <code><a href="aggregatePrsMarkers.html">aggregatePrsMarkers</a></code>, or one of the list 
    members of the output of <code><a href="selectPrs.html">selectPrs</a></code>.</p></dd>
<dt>type</dt>
<dd><p>Same as <code>score</code> in <code>prsicePRS</code>
    function (first three cases).</p></dd>
<dt>clump_kb</dt>
<dd><p>The genomic interval (in kbs) within 
    which PRSice2 performs clumping. Defaults to 
    <code>250</code>.</p></dd>
<dt>clump_r2</dt>
<dd><p>The R^2 threshold to keep SNPs after
    clumping in PRSice2. Defaults to <code>0.1</code>.</p></dd>
<dt>clump_p</dt>
<dd><p>The p-value threshold for SNPs to 
    include in clumping. Defaults to <code>1</code>.</p></dd>
<dt>score</dt>
<dd><p>How should PRSice2 calculate the PRS?
    One of <code>"avg"</code> (default), <code>"sum"</code>, 
    <code>"std"</code>, <code>"con-std"</code>. See also the manual
    of PRSice2 and Details.</p></dd>
<dt>perm</dt>
<dd><p>How many permutations should PRSice2
    perform? Defaults to 10000.</p></dd>
<dt>seed</dt>
<dd><p>Seed for PRSice2 permutations, for 
    reproducibility. Defaults to <code>42</code>.</p></dd>
<dt>methods</dt>
<dd><p>The methods to use for PRS extraction,
    one or more of <code>"prsice"</code> and <code>"lassosum"</code>.</p></dd>
<dt>lassosumOpts</dt>
<dd><p>List of options passed to lassosum
    (<code>lassosumPRS</code>). See also Details.</p></dd>
<dt>prsiceOpts</dt>
<dd><p>List of options passed to PRSice2
    (<code>prsicePRS</code>). See also Details.</p></dd>
<dt>rc</dt>
<dd><p>The fraction of available cores to use for 
    parallel calculations. Default is <code>NULL</code> for no
    parallelization.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>For <code>lassosumPRS</code> and <code>prsicePRS</code>, a data
    frame with the SNPs comprising the extracted PRS,
    compatible with the score files returned by the PGS
    Catalog and the candidates returned by the complete
<code><a href="prismaFuns.html">prisma</a></code> pipeline.
    For <code>runPRS</code>, the input <code>target</code>    GWASExperiment object with the <code>prsbetas</code>    slots filled. Essentially the effect weights or
    ORs for the SNPs comprising the PRS.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Regarding the <code>score</code> argument, it controls how
    PRSice2 calculates the PRS and creates regression models
    as well as how it reports it. Although typically it 
    should be the same as the <code>prsiceOpts$score</code>
    argument in the main <code><a href="prismaFuns.html">prisma</a></code> pipeline
    (and it is when <code><a href="prismaFuns.html">prisma</a></code> is used), it is
    not obligatory as PRISMA recalculates the final PRS
    before reporting and validating. However, if not the
    same, the actual score distribution will differ much
    between PRSice2 results and PRISMA results.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Panagiotis Moulos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">toy</span>,package<span class="op">=</span><span class="st">"prisma"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="GWASExperiment-class.html">gdsfile</a></span><span class="op">(</span><span class="va">toy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"prisma"</span>,<span class="st">"extdata"</span>,<span class="st">"toy.gds"</span><span class="op">)</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="co"># Only 1 PC will be returned because of dataset size</span></span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="calcPcaCovar.html">calcPcaCovar</a></span><span class="op">(</span><span class="va">toy</span>,method<span class="op">=</span><span class="st">"snprel"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA to select covariates to include in associations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading GDS file /tmp/RtmpezHjqb/temp_libpath372f48f1c2bc/prisma/extdata/toy.gds...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing LD pruning...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP pruning based on LD:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 38 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 962</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sliding window: 500,000 basepairs, Inf SNPs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     |LD| threshold: 0.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     method: composite</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 18: 23.44%, 154/657</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Chromosome 21: 21.28%, 73/343</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 227 markers are selected in total.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LD pruning finished! 227 SNPs will be used for PCA.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing PCA...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Principal Component Analysis (PCA) on genotypes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 773 SNPs (non-autosomes or non-selection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of samples: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of SNPs: 227</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     using 1 thread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     # of principal components: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PCA:    the sum of all selected genotypes (0,1,2) = 10268</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CPU capabilities: Double-Precision SSE2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mon Jun  6 20:20:29 2022    (internal increment: 13024)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 1s</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mon Jun  6 20:20:30 2022    Begin (eigenvalues and eigenvectors)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Mon Jun  6 20:20:30 2022    Done.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Selecting number of PCs to include</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   with Tracy-Widom test for significant eigenvalues</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     selected 1 PCs</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu"><a href="gwaFuns.html">gwa</a></span><span class="op">(</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,</span>
<span class="r-in">        pcs<span class="op">=</span><span class="cn">TRUE</span>,psig<span class="op">=</span><span class="fl">0.05</span>,methods<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"glm"</span>,<span class="st">"statgen"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting GWA analysis in 2 chunks with 2 methods: glm, statgen</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 18</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs1009238 to rs1568327</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 24 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 54 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Analyzing chromosome/part 21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with GLM over 1 chunks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Regression  : gaussian (predicted)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   testing SNPs from rs2824974 to rs2828316</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for statgenGWAS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing GWAS with statgenGWAS model</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Trait(s)    : BMI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Covariate(s): Age, Gender</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use PCs     : Yes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found 18 associations (uncorrected p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Combining p-values from multi-GWAS using fisher method</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 35 associations with (uncorrected) combined p-values).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished, the following putative associations were found per part:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 18: 54 associations found out of 657 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Part 21: 35 associations found out of 343 markers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Overall, found 89 associations with (uncorrected) combined p-values).</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">prsiceOpts</span> <span class="op">&lt;-</span> <span class="fu"><a href="getDefaults.html">getDefaults</a></span><span class="op">(</span><span class="st">"prsice"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">prsiceOpts</span><span class="op">$</span><span class="va">perm</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span class="r-in">    </span>
<span class="r-in">    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span>
<span class="r-in">        <span class="va">toy</span> <span class="op">&lt;-</span> <span class="fu">runPRS</span><span class="op">(</span>base<span class="op">=</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">            covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span class="r-in">            prsiceOpts<span class="op">=</span><span class="va">prsiceOpts</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting PRS analysis with lassosum, prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with lassosum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpxo9Dha/lassosum_WSJPC7148G</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpxo9Dha/lassosum_WSJPC7148G</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpxo9Dha/lassosum_WSJPC7148G/plink_lassosum_target_EIRID5993K.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpxo9Dha/lassosum_WSJPC7148G/plink_lassosum_target_EIRID5993K.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpxo9Dha/lassosum_WSJPC7148G/plink_lassosum_target_EIRID5993K.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with reference panel...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats, reference panel, and test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Splitting genome by LD blocks ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum with s=1...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating polygenic scores ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.756356747467697</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========== Running PRS analysis with prsice</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpxo9Dha/prsice_FKVJN3606M</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpxo9Dha/prsice_FKVJN3606M</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/plink_prsice_WOGET5330Y.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/plink_prsice_WOGET5330Y.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/plink_prsice_WOGET5330Y.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/base_WOGET5330Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/pheno_WOGET5330Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/plink_prsice_WOGET5330Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpxo9Dha/prsice_FKVJN3606M/covar_WOGET5330Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_WOGET5330Y \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.708854</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.867266</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.158412</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRS algorithms finished! Processing the results...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="va">lprs</span> <span class="op">&lt;-</span> <span class="fu">lassosumPRS</span><span class="op">(</span>base<span class="op">=</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">        covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> lassosum operations will be executed at /tmp/Rtmpxo9Dha/lassosum_OAORP9897W</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing lassosum run at workspace directory: /tmp/Rtmpxo9Dha/lassosum_OAORP9897W</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files for target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpxo9Dha/lassosum_OAORP9897W/plink_lassosum_target_VDWTZ8527W.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpxo9Dha/lassosum_OAORP9897W/plink_lassosum_target_VDWTZ8527W.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpxo9Dha/lassosum_OAORP9897W/plink_lassosum_target_VDWTZ8527W.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   copying linkage disequilibrium block files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum PRS algorithm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with reference panel...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats with test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Coordinating summary stats, reference panel, and test data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Splitting genome by LD blocks ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> s =  0.9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running lassosum with s=1...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating polygenic scores ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running lassosum validation and PRS calculation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in pheno.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 100 out of 100 samples kept in covar.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done! R^2 is: 0.756356747467697</span>
<span class="r-in">    </span>
<span class="r-in">    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"PRSice_linux"</span><span class="op">)</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span>
<span class="r-in">        <span class="va">pprs</span> <span class="op">&lt;-</span> <span class="fu">prsicePRS</span><span class="op">(</span>base<span class="op">=</span><span class="va">toy</span>,response<span class="op">=</span><span class="st">"BMI"</span>,</span>
<span class="r-in">            covariates<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Age"</span>,<span class="st">"Gender"</span><span class="op">)</span>,pcs<span class="op">=</span><span class="cn">TRUE</span>,perm<span class="op">=</span><span class="fl">100</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> prsice operations will be executed at /tmp/Rtmpxo9Dha/prsice_AYASI0629U</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing PRSice run at workspace directory: /tmp/Rtmpxo9Dha/prsice_AYASI0629U</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing summary statistics for the base file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the covariates file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing the phenotype file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   preparing PLINK temporary files - target...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preflight...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing FAM file to /tmp/Rtmpxo9Dha/prsice_AYASI0629U/plink_prsice_RNNVO2324P.fam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing extended MAP file to /tmp/Rtmpxo9Dha/prsice_AYASI0629U/plink_prsice_RNNVO2324P.bim </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Writing BED file to /tmp/Rtmpxo9Dha/prsice_AYASI0629U/plink_prsice_RNNVO2324P.bed (SNP-major mode)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparation done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running PRSice PRS algorithm in calculate mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Executing:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript /home/panos/.local/share/R/prsice-2.3.3/PRSice.R \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --prsice /home/panos/.local/share/R/prsice-2.3.3/PRSice_linux \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --base /tmp/Rtmpxo9Dha/prsice_AYASI0629U/base_RNNVO2324P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --beta \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --binary-target F \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --pheno /tmp/Rtmpxo9Dha/prsice_AYASI0629U/pheno_RNNVO2324P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --target /tmp/Rtmpxo9Dha/prsice_AYASI0629U/plink_prsice_RNNVO2324P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --cov /tmp/Rtmpxo9Dha/prsice_AYASI0629U/covar_RNNVO2324P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-kb 250 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-r2 0.1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --clump-p 1 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --score avg \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --seed 42 \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --out prsice_out_RNNVO2324P \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --print-snp \</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   --perm 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adjusted PRSice R^2 is: 0.708854</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Full model R^2 is: 0.867266</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Null model R^2 is: 0.158412</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Panagiotis Moulos.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

